---
date: "`r Sys.Date()`"
output: html_document
---

# Trajectory analysis

Trajectory analysis is a powerful tool in single-cell RNA-seq allowing us to track how cells transition between different states over time, rather than simply classifying them into discrete clusters. This approach is particularly useful for studying dynamic processes such as cellular differentiation, cell cycle progression, or disease progression. One of the key advantages of trajectory analysis is the ability to identify genes or (in the case of long read data) isoforms whose expression patterns are correlated with pseudotime

In our example dataset we are interested in looking at genes or isofroms that change as a function of pseudotime as this will highlight imporat features that control stem cell to neuronal differentiating into neural cells. In the bellow code block is the `perform_trajectory_analysis` function which uses monocle3 [@trapnell2014] to learn the trajectory and order the cells by pseudotime. We also provide an optional mode to to identify gene or isoforms that change with pseudotime. [^09_trajectory-1]

[^09_trajectory-1]: Itâ€™s important to note that analyzing features in relation to pseudotime can be computationally intensive Users can choose to plot the trajectory without the added computational load of ordering features by pseudotime, making the analysis more flexible. The parameter `quantile` can be used to specify the number of features to test (0.75 means your testing the top 25% of features)

    .

```{r traj_fucntion, include=FALSE}

perform_trajectory_analysis <- function(seurat_obj, assay, prefix, cluster_name="harm_cluster", 
                                        root_cells = NULL, cores = 8, order_features_by_pseudotime = FALSE, quantile = 0.50) {
  library(monocle3)
  library(SeuratWrappers)
  
  # Step 1: Subset and Remove RNA Assay
  DefaultAssay(seurat_obj) <- assay
  seurat_obj <- JoinLayers(seurat_obj)
  
  # Step 2: Convert to Monocle cell_data_set object
  cds <- as.cell_data_set(seurat_obj)
  
  # Add gene_short_name to feature data (gene metadata)
  fData(cds)$gene_short_name <- rownames(fData(cds))
  
  # Estimate size factors
  cds <- estimate_size_factors(cds)
  
  # Step 3: Assign cluster info and UMAP coordinates
  # Partitions
  recreate_partition <- as.factor(rep(1, length(colnames(seurat_obj))))
  names(recreate_partition) <- colnames(seurat_obj)
  cds@clusters$UMAP$partitions <- recreate_partition
  
  # Clustering information from Seurat
  cds@clusters$UMAP$clusters <- seurat_obj@meta.data[["orig.ident"]]
  
  # UMAP embeddings from Seurat
  cds@int_colData@listData$reducedDims$UMAP <- seurat_obj@reductions$umap.harm@cell.embeddings
  
  # Plot before trajectory
  cluster_before_trajectory <- plot_cells(cds, color_cells_by = 'orig.ident') +
    theme(legend.position = "right")
  
  # Save plot
  print(cluster_before_trajectory)

  
  # Step 4: Learn trajectory graph
  cds <- learn_graph(cds, use_partition = FALSE, close_loop = FALSE)
  
  # Plot learned trajectory
  plot_trajectory <- plot_cells(cds, color_cells_by = cluster_name) +
    theme(legend.position = "right")
  
  print(plot_trajectory)
  
  # Step 5: Order cells in pseudotime
  cds <- order_cells(cds, reduction_method = 'UMAP', root_cells = root_cells)
  
  # Plot pseudotime
  p_pseudotime <- plot_cells(cds, color_cells_by = 'pseudotime') +
    theme(legend.position = "right")
  
  
  print(p_pseudotime)
  
  # Save pseudotime to Seurat object
  seurat_obj$pseudotime <- pseudotime(cds)
  
  if (order_features_by_pseudotime == TRUE) {
  # Step 6: Finding genes that change as a function of pseudotime
  threshold <- quantile(rowData(cds)$mean_expression, quantile) # 0.75 means Top 25% genes
  cds_subset <- cds[rowData(cds)$mean_expression > threshold, ]
  deg_results <- graph_test(cds_subset, neighbor_graph = 'principal_graph', cores = cores)
  
  # Filter for significant genes
  MI_sig <- deg_results %>%
    filter(q_value < 0.01, morans_I > 0.5, status == 'OK') %>%
    arrange(q_value)
  
  # Save significant genes
  write.csv(MI_sig, paste0(prefix, "_sig_genes_pseudotime.csv"))
  
  # Plot top genes
  top_genes <- MI_sig %>% 
    arrange(desc(morans_I)) %>% 
    head(16)
  
  pdf(file = paste0(prefix, "_top_genes_pseudotime.pdf"), width = 10, height = 10)
  print(FeaturePlot(seurat_obj, features = top_genes$gene_short_name, reduction ="umap.harm"))
  dev.off()
  
  
  ##### plot pseduotime
  ####plot genes based on pseudo time expresstion 
  my_genes <- row.names(subset(fData(cds), gene_short_name %in% top_genes$gene_short_name)) 
  #my_genes <- row.names(subset(fData(cds), gene_short_name %in% c("DLK1","GRIA2","GRIN2B", "VIM", "PKM", "GAD1", "DCX", "POU5F1", "HOPX"))) 
  
  cds_subset <- cds[my_genes,]
  
  monocle_plot <- monocle3::plot_genes_in_pseudotime(cds_subset = cds_subset, min_expr = 0.5, cell_size = 0.25,
                                     nrow = NULL, ncol = 1, panel_order = NULL,
                                     label_by_short_name = TRUE, color_cells_by = cluster_name)
  
  pdf(file = paste0(prefix, "monocle_top_genes_pseudotime.pdf"), width = 10, height = 10)
  print(monocle_plot)
  dev.off()
  
    }
  
  # Save the Seurat object with pseudotime info
  #saveRDS(seurat_obj, file = paste0(prefix, "_seurat_with_pseudotime.rds"))

}


```

```{r traj_genes, include=FALSE}
perform_trajectory_analysis(seurat_obj = obj, "RNA", prefix = "Multi-sample_tutorial",
                            cluster_name="sctype_db",  root_cells = NULL, cores = 8, order_features_by_pseudotime = FALSE)

```

```{r traj_iso, include=FALSE}
perform_trajectory_analysis(seurat_obj = obj, "iso", prefix = "iso_Multi-sample_tutorial",
                            cluster_name="sctype_db",  root_cells = NULL, cores = 8, order_features_by_pseudotime = FALSE)

```
