<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Add isoform counts to Seurat object | FLAMES Single-Cell RNA-seq Tutorial</title>
  <meta name="description" content="Chapter 4 Add isoform counts to Seurat object | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Add isoform counts to Seurat object | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Add isoform counts to Seurat object | FLAMES Single-Cell RNA-seq Tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer Postdoctoral Research Fellow in Long Reads and Single-Cell Transcriptomics University of Melbourne" />


<meta name="date" content="2025-04-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"/>
<link rel="next" href="finding-marker-genes-and-isoforms.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single cell RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualizing-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualizing OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>9</b> <strong>Multisample analysis</strong></a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>9.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="9.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>9.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>9.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multi-sample-integration"><i class="fa fa-check"></i><b>9.2</b> Multi-Sample integration</a></li>
<li class="chapter" data-level="9.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>9.3</b> Add isoform counts</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Trajectory analysis</strong></a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i><b>10.1</b> Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="10.2" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i><b>10.2</b> Multisample QC function</a></li>
<li class="chapter" data-level="10.3" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-count-matrix"><i class="fa fa-check"></i><b>10.3</b> Convert Oarfish files to count matrix</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FLAMES Single-Cell RNA-seq Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="add-isoform-counts-to-seurat-object" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> <strong>Add isoform counts to Seurat object</strong><a href="add-isoform-counts-to-seurat-object.html#add-isoform-counts-to-seurat-object" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<hr />
<div id="create-a-seurat-object-with-isoform-expression-data" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Create a Seurat object with isoform expression data<a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have 309 high-quality cells for our downstream analysis, let’s incorporate isoform-level information into our Seurat object. The first step is to read in the outputs from FLAMES, which provides isoform-level counts from Oarfish. We’ll create a new Seurat object for this data, specifically for the isoform counts, and add it as a new assay to our existing object. This setup will allow us to explore both gene-level and isoform-level counts within a single Seurat object, giving us a more comprehensive view of our data.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="add-isoform-counts-to-seurat-object.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co">#This function reads in Oarfish count files and creates a csv file of count data. The function also appends the gene symbol to the ENSTID</span></span>
<span id="cb54-2"><a href="add-isoform-counts-to-seurat-object.html#cb54-2" aria-hidden="true" tabindex="-1"></a>process_oarfish_files_to_counts_matrix <span class="ot">&lt;-</span> <span class="cf">function</span>(flames_output_folder, sample_name, output_dir) {</span>
<span id="cb54-3"><a href="add-isoform-counts-to-seurat-object.html#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="add-isoform-counts-to-seurat-object.html#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read in the resource table (transcript_id, gene_id, gene_symbol)</span></span>
<span id="cb54-5"><a href="add-isoform-counts-to-seurat-object.html#cb54-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define the file paths based on the sample name</span></span>
<span id="cb54-6"><a href="add-isoform-counts-to-seurat-object.html#cb54-6" aria-hidden="true" tabindex="-1"></a>  count_matrix_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(flames_output_folder, <span class="fu">paste0</span>(sample_name, <span class="st">&quot;.count.mtx&quot;</span>))</span>
<span id="cb54-7"><a href="add-isoform-counts-to-seurat-object.html#cb54-7" aria-hidden="true" tabindex="-1"></a>  barcodes_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(flames_output_folder, <span class="fu">paste0</span>(sample_name, <span class="st">&quot;.barcodes.txt&quot;</span>))</span>
<span id="cb54-8"><a href="add-isoform-counts-to-seurat-object.html#cb54-8" aria-hidden="true" tabindex="-1"></a>  features_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(flames_output_folder, <span class="fu">paste0</span>(sample_name, <span class="st">&quot;.features.txt&quot;</span>))</span>
<span id="cb54-9"><a href="add-isoform-counts-to-seurat-object.html#cb54-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-10"><a href="add-isoform-counts-to-seurat-object.html#cb54-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read the data</span></span>
<span id="cb54-11"><a href="add-isoform-counts-to-seurat-object.html#cb54-11" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">readMM</span>(count_matrix_path)</span>
<span id="cb54-12"><a href="add-isoform-counts-to-seurat-object.html#cb54-12" aria-hidden="true" tabindex="-1"></a>  barcodes <span class="ot">&lt;-</span> <span class="fu">readLines</span>(barcodes_path)</span>
<span id="cb54-13"><a href="add-isoform-counts-to-seurat-object.html#cb54-13" aria-hidden="true" tabindex="-1"></a>  features <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(features_path, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-14"><a href="add-isoform-counts-to-seurat-object.html#cb54-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-15"><a href="add-isoform-counts-to-seurat-object.html#cb54-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Transpose the matrix for Seurat compatibility</span></span>
<span id="cb54-16"><a href="add-isoform-counts-to-seurat-object.html#cb54-16" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">t</span>(counts)</span>
<span id="cb54-17"><a href="add-isoform-counts-to-seurat-object.html#cb54-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-18"><a href="add-isoform-counts-to-seurat-object.html#cb54-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set row and column names</span></span>
<span id="cb54-19"><a href="add-isoform-counts-to-seurat-object.html#cb54-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(counts) <span class="ot">&lt;-</span> features<span class="sc">$</span>V1</span>
<span id="cb54-20"><a href="add-isoform-counts-to-seurat-object.html#cb54-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(counts) <span class="ot">&lt;-</span> barcodes</span>
<span id="cb54-21"><a href="add-isoform-counts-to-seurat-object.html#cb54-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-22"><a href="add-isoform-counts-to-seurat-object.html#cb54-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert to a data frame</span></span>
<span id="cb54-23"><a href="add-isoform-counts-to-seurat-object.html#cb54-23" aria-hidden="true" tabindex="-1"></a>  counts_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts)</span>
<span id="cb54-24"><a href="add-isoform-counts-to-seurat-object.html#cb54-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-25"><a href="add-isoform-counts-to-seurat-object.html#cb54-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add transcript_id as the first column</span></span>
<span id="cb54-26"><a href="add-isoform-counts-to-seurat-object.html#cb54-26" aria-hidden="true" tabindex="-1"></a>  counts_df<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_df)</span>
<span id="cb54-27"><a href="add-isoform-counts-to-seurat-object.html#cb54-27" aria-hidden="true" tabindex="-1"></a>  counts_df <span class="ot">&lt;-</span> counts_df[, <span class="fu">c</span>(<span class="fu">ncol</span>(counts_df), <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(counts_df)<span class="sc">-</span><span class="dv">1</span>))]</span>
<span id="cb54-28"><a href="add-isoform-counts-to-seurat-object.html#cb54-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-29"><a href="add-isoform-counts-to-seurat-object.html#cb54-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge with the resource table to add gene symbols</span></span>
<span id="cb54-30"><a href="add-isoform-counts-to-seurat-object.html#cb54-30" aria-hidden="true" tabindex="-1"></a>  df_genesymbol <span class="ot">&lt;-</span> counts_df <span class="sc">%&gt;%</span></span>
<span id="cb54-31"><a href="add-isoform-counts-to-seurat-object.html#cb54-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(isoform_gene_dict, <span class="at">by =</span> <span class="st">&quot;transcript_id&quot;</span>)</span>
<span id="cb54-32"><a href="add-isoform-counts-to-seurat-object.html#cb54-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-33"><a href="add-isoform-counts-to-seurat-object.html#cb54-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove the gene_id column and reorder the columns</span></span>
<span id="cb54-34"><a href="add-isoform-counts-to-seurat-object.html#cb54-34" aria-hidden="true" tabindex="-1"></a>  df_genesymbol<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb54-35"><a href="add-isoform-counts-to-seurat-object.html#cb54-35" aria-hidden="true" tabindex="-1"></a>  df_genesymbol <span class="ot">&lt;-</span> df_genesymbol[, <span class="fu">c</span>(<span class="fu">ncol</span>(df_genesymbol), <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(df_genesymbol)<span class="sc">-</span><span class="dv">1</span>))]</span>
<span id="cb54-36"><a href="add-isoform-counts-to-seurat-object.html#cb54-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-37"><a href="add-isoform-counts-to-seurat-object.html#cb54-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Update row names to include gene symbol instead of transcript_id</span></span>
<span id="cb54-38"><a href="add-isoform-counts-to-seurat-object.html#cb54-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(df_genesymbol) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df_genesymbol<span class="sc">$</span>transcript_id, <span class="st">&quot;_&quot;</span>, df_genesymbol<span class="sc">$</span>gene_symbol)</span>
<span id="cb54-39"><a href="add-isoform-counts-to-seurat-object.html#cb54-39" aria-hidden="true" tabindex="-1"></a>  df_genesymbol<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb54-40"><a href="add-isoform-counts-to-seurat-object.html#cb54-40" aria-hidden="true" tabindex="-1"></a>  df_genesymbol<span class="sc">$</span>gene_symbol <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb54-41"><a href="add-isoform-counts-to-seurat-object.html#cb54-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-42"><a href="add-isoform-counts-to-seurat-object.html#cb54-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Write the output to a CSV file</span></span>
<span id="cb54-43"><a href="add-isoform-counts-to-seurat-object.html#cb54-43" aria-hidden="true" tabindex="-1"></a>  output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="fu">paste0</span>(<span class="st">&quot;gene_symbol_&quot;</span>, sample_name, <span class="st">&quot;_counts.csv&quot;</span>))</span>
<span id="cb54-44"><a href="add-isoform-counts-to-seurat-object.html#cb54-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(df_genesymbol, output_path)</span>
<span id="cb54-45"><a href="add-isoform-counts-to-seurat-object.html#cb54-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-46"><a href="add-isoform-counts-to-seurat-object.html#cb54-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Processed sample:&quot;</span>, sample_name, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Output saved to:&quot;</span>, output_path, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb54-47"><a href="add-isoform-counts-to-seurat-object.html#cb54-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-48"><a href="add-isoform-counts-to-seurat-object.html#cb54-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df_genesymbol)</span>
<span id="cb54-49"><a href="add-isoform-counts-to-seurat-object.html#cb54-49" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb54-50"><a href="add-isoform-counts-to-seurat-object.html#cb54-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-51"><a href="add-isoform-counts-to-seurat-object.html#cb54-51" aria-hidden="true" tabindex="-1"></a>oarfish_counts <span class="ot">&lt;-</span> <span class="fu">process_oarfish_files_to_counts_matrix</span>(</span>
<span id="cb54-52"><a href="add-isoform-counts-to-seurat-object.html#cb54-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">flames_output_folder =</span> <span class="st">&quot;/data/scratch/users/yairp/FLAMES_Day55/outs/&quot;</span>,</span>
<span id="cb54-53"><a href="add-isoform-counts-to-seurat-object.html#cb54-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample_name =</span> <span class="st">&quot;oarfish&quot;</span>,</span>
<span id="cb54-54"><a href="add-isoform-counts-to-seurat-object.html#cb54-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_dir =</span> <span class="st">&quot;./output_files/counts/&quot;</span></span>
<span id="cb54-55"><a href="add-isoform-counts-to-seurat-object.html#cb54-55" aria-hidden="true" tabindex="-1"></a>    )</span></code></pre></div>
</details>
<pre><code>## Processed sample: oarfish 
## Output saved to: ./output_files/counts//gene_symbol_oarfish_counts.csv</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="add-isoform-counts-to-seurat-object.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the Seurat object with iso counts</span></span>
<span id="cb56-2"><a href="add-isoform-counts-to-seurat-object.html#cb56-2" aria-hidden="true" tabindex="-1"></a>iso_seurat_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> oarfish_counts, <span class="at">project =</span> sample_id)</span>
<span id="cb56-3"><a href="add-isoform-counts-to-seurat-object.html#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="add-isoform-counts-to-seurat-object.html#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(iso_seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>), <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;QC plots (isoform level) BEFORE Filtering&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/process_oarfish_file-1.png" width="672" /></p>
</div>
<div id="filter-the-new-seurat-object-based-on-gene-level-information" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Filter the new Seurat object based on gene level information<a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have now created a Seurat object with isoform-level count data. Users can take this object and follow similar processing steps as at the gene level <a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-sources-of-unwanted-noise-from-the-single-cell-dataset">3</a>, which might include filtering out low-quality cells and removing ambient RNA counts. In our case, we evaluated these steps and found they don’t add much value as low quality cells are already removed at the gene level and we can leverage this information rather than processing the isoform counts in the same way. Additionally, tools like DeconX and SoupX haven’t been tested on isoform count data, so we decided to exclude this analysis. Instead, we believe the most valuable approach is to append the isoform data to our gene-level Seurat object, resulting in two assays: one for gene-level and one for isoform-level data. If users have short-read data, they could similarly add it as a third assay.</p>
<p>Since we’ve already filtered out low-quality cells based on our gene-level data, we’ll use these high-quality cells to filter our isoform data.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="add-isoform-counts-to-seurat-object.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="do">#### isoform assay </span></span>
<span id="cb57-2"><a href="add-isoform-counts-to-seurat-object.html#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="do">## filter the data so iso and gene cells match</span></span>
<span id="cb57-3"><a href="add-isoform-counts-to-seurat-object.html#cb57-3" aria-hidden="true" tabindex="-1"></a>oarfish_iso_matched_gene <span class="ot">&lt;-</span> <span class="fu">subset</span>(iso_seurat_obj, <span class="at">cells =</span>filt_seurat_object<span class="sc">@</span>graphs[[<span class="st">&quot;RNA_nn&quot;</span>]]<span class="sc">@</span>Dimnames[[<span class="dv">1</span>]])</span>
<span id="cb57-4"><a href="add-isoform-counts-to-seurat-object.html#cb57-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-5"><a href="add-isoform-counts-to-seurat-object.html#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Rejoin data sets after integration</span></span>
<span id="cb57-6"><a href="add-isoform-counts-to-seurat-object.html#cb57-6" aria-hidden="true" tabindex="-1"></a>oarfish_iso_matched_gene <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(oarfish_iso_matched_gene)</span>
<span id="cb57-7"><a href="add-isoform-counts-to-seurat-object.html#cb57-7" aria-hidden="true" tabindex="-1"></a>counts_table_iso <span class="ot">&lt;-</span> oarfish_iso_matched_gene[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">$</span>counts</span>
<span id="cb57-8"><a href="add-isoform-counts-to-seurat-object.html#cb57-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-9"><a href="add-isoform-counts-to-seurat-object.html#cb57-9" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(counts_table_iso) <span class="ot">-&gt;</span> df_iso</span>
<span id="cb57-10"><a href="add-isoform-counts-to-seurat-object.html#cb57-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-11"><a href="add-isoform-counts-to-seurat-object.html#cb57-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows where the sum is 0</span></span>
<span id="cb57-12"><a href="add-isoform-counts-to-seurat-object.html#cb57-12" aria-hidden="true" tabindex="-1"></a>df_iso <span class="ot">&lt;-</span> df_iso[<span class="fu">rowSums</span>(df_iso) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb57-13"><a href="add-isoform-counts-to-seurat-object.html#cb57-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-14"><a href="add-isoform-counts-to-seurat-object.html#cb57-14" aria-hidden="true" tabindex="-1"></a>filt_seurat_object[[<span class="st">&quot;iso&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">CreateAssay5Object</span>(<span class="at">counts =</span> df_iso)</span>
<span id="cb57-15"><a href="add-isoform-counts-to-seurat-object.html#cb57-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-16"><a href="add-isoform-counts-to-seurat-object.html#cb57-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize the new assay data</span></span>
<span id="cb57-17"><a href="add-isoform-counts-to-seurat-object.html#cb57-17" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(filt_seurat_object, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>)</span>
<span id="cb57-18"><a href="add-isoform-counts-to-seurat-object.html#cb57-18" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(filt_seurat_object, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>)</span>
<span id="cb57-19"><a href="add-isoform-counts-to-seurat-object.html#cb57-19" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(filt_seurat_object, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>)</span>
<span id="cb57-20"><a href="add-isoform-counts-to-seurat-object.html#cb57-20" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(filt_seurat_object, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">reduction.name =</span> <span class="st">&quot;pca_iso&quot;</span>)</span>
<span id="cb57-21"><a href="add-isoform-counts-to-seurat-object.html#cb57-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-22"><a href="add-isoform-counts-to-seurat-object.html#cb57-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Run UMAP</span></span>
<span id="cb57-23"><a href="add-isoform-counts-to-seurat-object.html#cb57-23" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(filt_seurat_object, <span class="at">reduction =</span> <span class="st">&quot;pca_iso&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>,</span>
<span id="cb57-24"><a href="add-isoform-counts-to-seurat-object.html#cb57-24" aria-hidden="true" tabindex="-1"></a>                             <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">reduction.name =</span> <span class="st">&quot;umap_iso&quot;</span>)</span>
<span id="cb57-25"><a href="add-isoform-counts-to-seurat-object.html#cb57-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-26"><a href="add-isoform-counts-to-seurat-object.html#cb57-26" aria-hidden="true" tabindex="-1"></a><span class="co">#check to see that we have two assays</span></span>
<span id="cb57-27"><a href="add-isoform-counts-to-seurat-object.html#cb57-27" aria-hidden="true" tabindex="-1"></a>filt_seurat_object</span></code></pre></div>
</details>
<pre><code>## An object of class Seurat 
## 93557 features across 309 samples within 3 assays 
## Active assay: RNA (14906 features, 2000 variable features)
##  3 layers present: counts, data, scale.data
##  2 other assays present: joined, iso
##  4 dimensional reductions calculated: pca, umap, pca_iso, umap_iso</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="add-isoform-counts-to-seurat-object.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># visualize the UMAP for Gene and isoform</span></span>
<span id="cb59-2"><a href="add-isoform-counts-to-seurat-object.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(filt_seurat_object, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;UMAP gene level clustering&quot;</span>) <span class="sc">|</span></span>
<span id="cb59-3"><a href="add-isoform-counts-to-seurat-object.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(filt_seurat_object, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">reduction =</span> <span class="st">&quot;umap_iso&quot;</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;UMAP isoform level clustering&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/isoform_assay-1.png" width="768" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="add-isoform-counts-to-seurat-object.html#cb60-1" aria-hidden="true" tabindex="-1"></a>seu_obj <span class="ot">&lt;-</span> filt_seurat_object</span>
<span id="cb60-2"><a href="add-isoform-counts-to-seurat-object.html#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seu_obj, <span class="at">file =</span> <span class="st">&quot;./output_files/seu_objects/Day55_tutorial_gene_and_isoform_seurat.rds&quot;</span>)</span></code></pre></div>
</details>
</div>
<div id="add-the-isoform-assay-to-the-seurat-object" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Add the isoform assay to the Seurat object<a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Great! We now have an object containing both assays, so we can start by plotting some of our favorite genes and isoforms. This setup gives us the flexibility to visualize gene expression on isoform UMAPs and vice versa, allowing us to integrate and explore the expression of both gene and isoform expression within single cells on the same dimensional reduction.</p>
<p>Let’s begin by plotting VIM and TBR1. VIM is a marker of progenitor cells, while TBR1 is a marker of deep layer neurons.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="add-isoform-counts-to-seurat-object.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="do">####### test plotting on iso assays ##</span></span>
<span id="cb61-2"><a href="add-isoform-counts-to-seurat-object.html#cb61-2" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">rownames</span>(filt_seurat_object<span class="sc">@</span>assays<span class="sc">$</span>iso<span class="sc">@</span>features)</span>
<span id="cb61-3"><a href="add-isoform-counts-to-seurat-object.html#cb61-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-4"><a href="add-isoform-counts-to-seurat-object.html#cb61-4" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="st">&quot;VIM&quot;</span></span>
<span id="cb61-5"><a href="add-isoform-counts-to-seurat-object.html#cb61-5" aria-hidden="true" tabindex="-1"></a>plot_features_list <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">&quot;(^|-|</span><span class="sc">\\</span><span class="st">b)&quot;</span>, gene, <span class="st">&quot;($|</span><span class="sc">\\</span><span class="st">b)&quot;</span>), features, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb61-6"><a href="add-isoform-counts-to-seurat-object.html#cb61-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-7"><a href="add-isoform-counts-to-seurat-object.html#cb61-7" aria-hidden="true" tabindex="-1"></a><span class="co">#We can now plot our favorite gene and all its corresponding isoforms on either the gene or isoform UMAPs. To switch between them, simply change the reduction argument to the desired UMAP (either the gene or isoform reduction)</span></span>
<span id="cb61-8"><a href="add-isoform-counts-to-seurat-object.html#cb61-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-9"><a href="add-isoform-counts-to-seurat-object.html#cb61-9" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">features =</span> plot_features_list, <span class="at">reduction =</span> <span class="st">&quot;umap_iso&quot;</span>) </span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ploting_features-1.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="add-isoform-counts-to-seurat-object.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">features =</span> gene, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ploting_features-2.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="add-isoform-counts-to-seurat-object.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lets also plot the isoform data on the gene level clsuters.</span></span>
<span id="cb63-2"><a href="add-isoform-counts-to-seurat-object.html#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">features =</span> plot_features_list, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>) <span class="sc">+</span></span>
<span id="cb63-3"><a href="add-isoform-counts-to-seurat-object.html#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb63-4"><a href="add-isoform-counts-to-seurat-object.html#cb63-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&#39;VIM isoform expression plotted on gene level clusters&#39;</span>,</span>
<span id="cb63-5"><a href="add-isoform-counts-to-seurat-object.html#cb63-5" aria-hidden="true" tabindex="-1"></a>    <span class="co">#caption = &#39;made with patchwork&#39;,</span></span>
<span id="cb63-6"><a href="add-isoform-counts-to-seurat-object.html#cb63-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">theme =</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span>
<span id="cb63-7"><a href="add-isoform-counts-to-seurat-object.html#cb63-7" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ploting_features-3.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="add-isoform-counts-to-seurat-object.html#cb64-1" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="st">&quot;TBR1&quot;</span></span>
<span id="cb64-2"><a href="add-isoform-counts-to-seurat-object.html#cb64-2" aria-hidden="true" tabindex="-1"></a>plot_features_list <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">&quot;(^|-|</span><span class="sc">\\</span><span class="st">b)&quot;</span>, gene, <span class="st">&quot;($|</span><span class="sc">\\</span><span class="st">b)&quot;</span>), features, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb64-3"><a href="add-isoform-counts-to-seurat-object.html#cb64-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-4"><a href="add-isoform-counts-to-seurat-object.html#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="co">#We can now plot our favorite gene and all its corresponding isoforms on either the gene or isoform UMAPs. To switch between them, simply change the reduction argument to the desired UMAP (either the gene or isoform reduction)</span></span>
<span id="cb64-5"><a href="add-isoform-counts-to-seurat-object.html#cb64-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-6"><a href="add-isoform-counts-to-seurat-object.html#cb64-6" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">features =</span> plot_features_list, <span class="at">reduction =</span> <span class="st">&quot;umap_iso&quot;</span>) </span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ploting_features-4.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="add-isoform-counts-to-seurat-object.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">features =</span> gene, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ploting_features-5.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="add-isoform-counts-to-seurat-object.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lets also plot the isoform data on the gene level clsuters.</span></span>
<span id="cb66-2"><a href="add-isoform-counts-to-seurat-object.html#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">features =</span> plot_features_list, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>) <span class="sc">+</span></span>
<span id="cb66-3"><a href="add-isoform-counts-to-seurat-object.html#cb66-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb66-4"><a href="add-isoform-counts-to-seurat-object.html#cb66-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&#39;TBR1 isoform expresstion plotted on gene level clusters&#39;</span>,</span>
<span id="cb66-5"><a href="add-isoform-counts-to-seurat-object.html#cb66-5" aria-hidden="true" tabindex="-1"></a>    <span class="co">#caption = &#39;made with patchwork&#39;,</span></span>
<span id="cb66-6"><a href="add-isoform-counts-to-seurat-object.html#cb66-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">theme =</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span>
<span id="cb66-7"><a href="add-isoform-counts-to-seurat-object.html#cb66-7" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ploting_features-6.png" width="672" /></p>
<p>We observe that the expression of VIM and TBR1 are is localized to two distinct clusters, which aligns with our expectations. This suggests that some cells remain in the progenitor/radial glial stage, while others have already begun differentiating into neurons. Furthermore, our data reveals isoform-level expression for each of our genes, including TBR1, which shows distinct isoform expression patterns across different cell populations. To plot genes and expressed isoforms from that gene simply change the ‘gene’ value to your favorite gene and run the code chunk above.</p>
<p>Now we have a count matrix with two assays, gene and isoform. There are many avenues for analysis. We will begin with finding DE genes and isoforms. Then we identify cell types and perform trajectory analysis. finally we look into interesting isoforms unique to each cluster</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="finding-marker-genes-and-isoforms.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/04-Adding-Isoform-Information-to-the-Seurat-object.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/04-Adding-Isoform-Information-to-the-Seurat-object.Rmd",
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
