<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Appendix | FLAMES Single-Cell RNA-seq Tutorial</title>
  <meta name="description" content="Appendix | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Appendix | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Appendix | FLAMES Single-Cell RNA-seq Tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer Postdoctoral Research Fellow in Long Reads and Single-Cell Transcriptomics University of Melbourne" />


<meta name="date" content="2025-02-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="conclusion.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single cell RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualizing-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualizing OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>9</b> Multisample analysis</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>9.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="9.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>9.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>9.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multi-sample-integration"><i class="fa fa-check"></i><b>9.2</b> Multi-Sample integration</a></li>
<li class="chapter" data-level="9.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>9.3</b> Add isoform counts</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Trajectory analysis</strong></a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i><b>10.1</b> Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="10.2" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i><b>10.2</b> Multisample QC function</a></li>
<li class="chapter" data-level="10.3" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-count-matrix"><i class="fa fa-check"></i><b>10.3</b> Convert Oarfish files to count matrix</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FLAMES Single-Cell RNA-seq Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="appendix" class="section level1 unnumbered hasAnchor">
<h1><strong>Appendix</strong><a href="appendix.html#appendix" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="calculating-the-ambient-rna-profile" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Calculating the ambient RNA profile<a href="appendix.html#calculating-the-ambient-rna-profile" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>By calculating the ambient RNA profile, we can perform two important analyses to improve data quality: (1) removing empty droplets and (2) normalizing for background RNA contamination. Chapter 3 of the tutorial outlines both processes. However, the first step is to calculate the background RNA profile by quantifying gene expression for a list of barcodes known to represent background droplets rather than true cells.</p>
<p>When running FLAMES, one of the outputs is the empty_bc_list.csv file, which contains 2,000 barcodes identified as background, not associated with cells. Users can manually add additional background barcodes if needed (for advanced users). For a detailed understanding of how this list is determined, refer to the BLAZE publication <span class="citation">(<a href="references.html#ref-you2023a" role="doc-biblioref">You et al., 2023</a>)</span>.</p>
<p>To incorporate the background barcode list, users should run FLAMES as they did previously, but include the background barcode list as the barcode_file parameter. There is no need to identify or quantify isoforms so users can set the following parameters to false in the config file</p>
<div class="line-block"><code>"do_isoform_identification": false,</code><br />
<code>"bambu_isoform_identification": false,</code><br />
<code>"do_read_realignment": false,</code><br />
<code>"do_transcript_quantification": false</code><br />
</div>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="appendix.html#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run FLAMES using background barcode list genrated in the first run. </span></span>
<span id="cb144-2"><a href="appendix.html#cb144-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-3"><a href="appendix.html#cb144-3" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">sc_long_pipeline</span>(<span class="at">fastq=</span>fastq, <span class="at">outdir=</span>output, <span class="at">annot=</span>GTF, <span class="at">genome_bookdfa=</span>genome, </span>
<span id="cb144-4"><a href="appendix.html#cb144-4" aria-hidden="true" tabindex="-1"></a>                         <span class="co">#minimap=minimap_dir, k8=k8,</span></span>
<span id="cb144-5"><a href="appendix.html#cb144-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">config_file=</span>config_file,</span>
<span id="cb144-6"><a href="appendix.html#cb144-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">expect_cell_number=</span><span class="dv">1000</span>,  <span class="at">barcodes_file=</span><span class="st">&#39;path/to/emtpy_bc_list.csv&#39;</span>)</span></code></pre></div>
</details>
</div>
<div id="multisample-qc-function" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Multisample QC function<a href="appendix.html#multisample-qc-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="appendix.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform QC for each sample </span></span>
<span id="cb145-2"><a href="appendix.html#cb145-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-3"><a href="appendix.html#cb145-3" aria-hidden="true" tabindex="-1"></a>perform_qc_filtering <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">count.matrix=</span>C1_STC, <span class="at">min.features =</span> <span class="dv">5000</span> ,</span>
<span id="cb145-4"><a href="appendix.html#cb145-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">max.features =</span> <span class="dv">10000</span>, <span class="at">max.counts =</span> <span class="dv">100000</span>,</span>
<span id="cb145-5"><a href="appendix.html#cb145-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">min.counts =</span> <span class="dv">3000</span>, <span class="at">npc =</span> <span class="dv">15</span>, <span class="at">cluster_res =</span> <span class="fl">0.9</span>,</span>
<span id="cb145-6"><a href="appendix.html#cb145-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">fig_name =</span> <span class="st">&#39;1&#39;</span>, <span class="at">project =</span> <span class="st">&quot;2&quot;</span>,</span>
<span id="cb145-7"><a href="appendix.html#cb145-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">MT =</span> <span class="dv">10</span>, <span class="at">doublet_rate =</span> <span class="fl">0.039</span>) {</span>
<span id="cb145-8"><a href="appendix.html#cb145-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-9"><a href="appendix.html#cb145-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Function to calculate min.features and max.features if not provided</span></span>
<span id="cb145-10"><a href="appendix.html#cb145-10" aria-hidden="true" tabindex="-1"></a>  calculate_feature_range <span class="ot">&lt;-</span> <span class="cf">function</span>(nFeature_RNA) {</span>
<span id="cb145-11"><a href="appendix.html#cb145-11" aria-hidden="true" tabindex="-1"></a>    min_feature <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(nFeature_RNA) <span class="sc">-</span> (<span class="fl">1.5</span> <span class="sc">*</span> <span class="fu">sd</span>(nFeature_RNA)))</span>
<span id="cb145-12"><a href="appendix.html#cb145-12" aria-hidden="true" tabindex="-1"></a>    max_feature <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(nFeature_RNA) <span class="sc">+</span> (<span class="fl">1.5</span> <span class="sc">*</span> <span class="fu">sd</span>(nFeature_RNA)))</span>
<span id="cb145-13"><a href="appendix.html#cb145-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">min_feature =</span> min_feature, <span class="at">max_feature =</span> max_feature))</span>
<span id="cb145-14"><a href="appendix.html#cb145-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb145-15"><a href="appendix.html#cb145-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-16"><a href="appendix.html#cb145-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If min.features and max.features not provided, calculate them</span></span>
<span id="cb145-17"><a href="appendix.html#cb145-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(min.features) <span class="sc">||</span> <span class="fu">is.null</span>(max.features)) {</span>
<span id="cb145-18"><a href="appendix.html#cb145-18" aria-hidden="true" tabindex="-1"></a>    seurat_obj_org <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> count.matrix, <span class="at">project =</span> project)</span>
<span id="cb145-19"><a href="appendix.html#cb145-19" aria-hidden="true" tabindex="-1"></a>    feature_range <span class="ot">&lt;-</span> <span class="fu">calculate_feature_range</span>(seurat_obj_org<span class="sc">$</span>nFeature_RNA)</span>
<span id="cb145-20"><a href="appendix.html#cb145-20" aria-hidden="true" tabindex="-1"></a>    min.features <span class="ot">&lt;-</span> feature_range<span class="sc">$</span>min_feature</span>
<span id="cb145-21"><a href="appendix.html#cb145-21" aria-hidden="true" tabindex="-1"></a>    max.features <span class="ot">&lt;-</span> feature_range<span class="sc">$</span>max_feature</span>
<span id="cb145-22"><a href="appendix.html#cb145-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb145-23"><a href="appendix.html#cb145-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-24"><a href="appendix.html#cb145-24" aria-hidden="true" tabindex="-1"></a>  rst_figures <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb145-25"><a href="appendix.html#cb145-25" aria-hidden="true" tabindex="-1"></a>  rst_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb145-26"><a href="appendix.html#cb145-26" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb145-27"><a href="appendix.html#cb145-27" aria-hidden="true" tabindex="-1"></a>  <span class="do">##### remove features expressed in less than 1% of cells ##### </span></span>
<span id="cb145-28"><a href="appendix.html#cb145-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate the percentage of cells expressing each gene</span></span>
<span id="cb145-29"><a href="appendix.html#cb145-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#gene_percent_expression &lt;- base::rowMeans(count.matrix &gt; 0) * 100</span></span>
<span id="cb145-30"><a href="appendix.html#cb145-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-31"><a href="appendix.html#cb145-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select genes expressed in at least 1% of cells</span></span>
<span id="cb145-32"><a href="appendix.html#cb145-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#genes_filter &lt;- names(gene_percent_expression[gene_percent_expression &gt; 1])</span></span>
<span id="cb145-33"><a href="appendix.html#cb145-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-34"><a href="appendix.html#cb145-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter counts</span></span>
<span id="cb145-35"><a href="appendix.html#cb145-35" aria-hidden="true" tabindex="-1"></a>  <span class="co">#counts_sub &lt;- count.matrix[genes_filter, ]</span></span>
<span id="cb145-36"><a href="appendix.html#cb145-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-37"><a href="appendix.html#cb145-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Record the number of features removed</span></span>
<span id="cb145-38"><a href="appendix.html#cb145-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#removed_features &lt;- dim(count.matrix)[1] - length(genes_filter)</span></span>
<span id="cb145-39"><a href="appendix.html#cb145-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-40"><a href="appendix.html#cb145-40" aria-hidden="true" tabindex="-1"></a>  <span class="do">#######</span></span>
<span id="cb145-41"><a href="appendix.html#cb145-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialize Seurat object</span></span>
<span id="cb145-42"><a href="appendix.html#cb145-42" aria-hidden="true" tabindex="-1"></a>  seurat_object <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> count.matrix, </span>
<span id="cb145-43"><a href="appendix.html#cb145-43" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">project =</span> project)</span>
<span id="cb145-44"><a href="appendix.html#cb145-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-45"><a href="appendix.html#cb145-45" aria-hidden="true" tabindex="-1"></a>  plot_scatter1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(seurat_object,</span>
<span id="cb145-46"><a href="appendix.html#cb145-46" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb145-47"><a href="appendix.html#cb145-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>) <span class="sc">+</span> </span>
<span id="cb145-48"><a href="appendix.html#cb145-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">NoLegend</span>() <span class="sc">+</span> </span>
<span id="cb145-49"><a href="appendix.html#cb145-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Association between reads and </span><span class="sc">\n</span><span class="st">unique genes per cell BEFORE filtering&quot;</span>)</span>
<span id="cb145-50"><a href="appendix.html#cb145-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-51"><a href="appendix.html#cb145-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(plot_scatter1)</span>
<span id="cb145-52"><a href="appendix.html#cb145-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-53"><a href="appendix.html#cb145-53" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add mitochondrial percentage</span></span>
<span id="cb145-54"><a href="appendix.html#cb145-54" aria-hidden="true" tabindex="-1"></a>  seurat_object[[<span class="st">&quot;joined&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(seurat_object[[<span class="st">&quot;RNA&quot;</span>]])</span>
<span id="cb145-55"><a href="appendix.html#cb145-55" aria-hidden="true" tabindex="-1"></a>  seurat_object[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(seurat_object, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb145-56"><a href="appendix.html#cb145-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-57"><a href="appendix.html#cb145-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot violin plots</span></span>
<span id="cb145-58"><a href="appendix.html#cb145-58" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(seurat_object, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>))</span>
<span id="cb145-59"><a href="appendix.html#cb145-59" aria-hidden="true" tabindex="-1"></a>  p1 <span class="sc">+</span>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;QC plots (gene level) BEFORE Filtering&quot;</span>)</span>
<span id="cb145-60"><a href="appendix.html#cb145-60" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-61"><a href="appendix.html#cb145-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(p1)</span>
<span id="cb145-62"><a href="appendix.html#cb145-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-63"><a href="appendix.html#cb145-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove low quality cells</span></span>
<span id="cb145-64"><a href="appendix.html#cb145-64" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">subset</span>(seurat_object,</span>
<span id="cb145-65"><a href="appendix.html#cb145-65" aria-hidden="true" tabindex="-1"></a>                              <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> min.features <span class="sc">&amp;</span> nFeature_RNA <span class="sc">&lt;</span> max.features <span class="sc">&amp;</span> percent.mt <span class="sc">&lt;</span> MT <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> max.counts <span class="sc">&amp;</span> nCount_RNA <span class="sc">&gt;</span> min.counts) </span>
<span id="cb145-66"><a href="appendix.html#cb145-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-67"><a href="appendix.html#cb145-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot quality metrics after filtering</span></span>
<span id="cb145-68"><a href="appendix.html#cb145-68" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(filt_seurat_object, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>))</span>
<span id="cb145-69"><a href="appendix.html#cb145-69" aria-hidden="true" tabindex="-1"></a>  p2 <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;QC metrics gene level AFTER Filtering&quot;</span>)</span>
<span id="cb145-70"><a href="appendix.html#cb145-70" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-71"><a href="appendix.html#cb145-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(p2)</span>
<span id="cb145-72"><a href="appendix.html#cb145-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-73"><a href="appendix.html#cb145-73" aria-hidden="true" tabindex="-1"></a>  <span class="do">#######</span></span>
<span id="cb145-74"><a href="appendix.html#cb145-74" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Normalize data</span></span>
<span id="cb145-75"><a href="appendix.html#cb145-75" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(filt_seurat_object,</span>
<span id="cb145-76"><a href="appendix.html#cb145-76" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>,</span>
<span id="cb145-77"><a href="appendix.html#cb145-77" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">scale.factor =</span> <span class="dv">10000</span>)</span>
<span id="cb145-78"><a href="appendix.html#cb145-78" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-79"><a href="appendix.html#cb145-79" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify highly variable features</span></span>
<span id="cb145-80"><a href="appendix.html#cb145-80" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(filt_seurat_object,</span>
<span id="cb145-81"><a href="appendix.html#cb145-81" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>,</span>
<span id="cb145-82"><a href="appendix.html#cb145-82" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb145-83"><a href="appendix.html#cb145-83" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-84"><a href="appendix.html#cb145-84" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply linear transformation</span></span>
<span id="cb145-85"><a href="appendix.html#cb145-85" aria-hidden="true" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(filt_seurat_object)</span>
<span id="cb145-86"><a href="appendix.html#cb145-86" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(filt_seurat_object, <span class="at">features =</span> all_genes)</span>
<span id="cb145-87"><a href="appendix.html#cb145-87" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-88"><a href="appendix.html#cb145-88" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform PCA</span></span>
<span id="cb145-89"><a href="appendix.html#cb145-89" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(filt_seurat_object,</span>
<span id="cb145-90"><a href="appendix.html#cb145-90" aria-hidden="true" tabindex="-1"></a>                              <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> filt_seurat_object))</span>
<span id="cb145-91"><a href="appendix.html#cb145-91" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-92"><a href="appendix.html#cb145-92" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Visualize PCA</span></span>
<span id="cb145-93"><a href="appendix.html#cb145-93" aria-hidden="true" tabindex="-1"></a>  rst_figures <span class="ot">&lt;-</span> <span class="fu">append</span>(rst_figures, <span class="fu">ElbowPlot</span>(filt_seurat_object))</span>
<span id="cb145-94"><a href="appendix.html#cb145-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-95"><a href="appendix.html#cb145-95" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Cluster cells</span></span>
<span id="cb145-96"><a href="appendix.html#cb145-96" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(filt_seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>npc)</span>
<span id="cb145-97"><a href="appendix.html#cb145-97" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(filt_seurat_object, <span class="at">resolution =</span> cluster_res)</span>
<span id="cb145-98"><a href="appendix.html#cb145-98" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-99"><a href="appendix.html#cb145-99" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform UMAP</span></span>
<span id="cb145-100"><a href="appendix.html#cb145-100" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(filt_seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>npc)</span>
<span id="cb145-101"><a href="appendix.html#cb145-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-102"><a href="appendix.html#cb145-102" aria-hidden="true" tabindex="-1"></a>  <span class="do">### Filter out doublets (remember to modify doublet rate if samples have variable target cells)</span></span>
<span id="cb145-103"><a href="appendix.html#cb145-103" aria-hidden="true" tabindex="-1"></a>  <span class="do">## pK Identification (no ground-truth) ---------------------------------------------------------------------------------------</span></span>
<span id="cb145-104"><a href="appendix.html#cb145-104" aria-hidden="true" tabindex="-1"></a>  sweep.res.list_pbmc <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(filt_seurat_object, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span>
<span id="cb145-105"><a href="appendix.html#cb145-105" aria-hidden="true" tabindex="-1"></a>  sweep.stats_pbmc <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_pbmc, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb145-106"><a href="appendix.html#cb145-106" aria-hidden="true" tabindex="-1"></a>  bcmvn_pbmc <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_pbmc)</span>
<span id="cb145-107"><a href="appendix.html#cb145-107" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-108"><a href="appendix.html#cb145-108" aria-hidden="true" tabindex="-1"></a>  pK <span class="ot">&lt;-</span> bcmvn_pbmc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(BCmetric <span class="sc">==</span> <span class="fu">max</span>(BCmetric)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(pK) </span>
<span id="cb145-109"><a href="appendix.html#cb145-109" aria-hidden="true" tabindex="-1"></a>  pK <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(pK[[<span class="dv">1</span>]]))</span>
<span id="cb145-110"><a href="appendix.html#cb145-110" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-111"><a href="appendix.html#cb145-111" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Homotypic Doublet Proportion Estimate -------------------------------------------------------------------------------------</span></span>
<span id="cb145-112"><a href="appendix.html#cb145-112" aria-hidden="true" tabindex="-1"></a>  annotations <span class="ot">&lt;-</span> filt_seurat_object<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb145-113"><a href="appendix.html#cb145-113" aria-hidden="true" tabindex="-1"></a>  homotypic.prop <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations)</span>
<span id="cb145-114"><a href="appendix.html#cb145-114" aria-hidden="true" tabindex="-1"></a>  nExp_poi <span class="ot">&lt;-</span> <span class="fu">round</span>(doublet_rate <span class="sc">*</span> <span class="fu">nrow</span>(filt_seurat_object<span class="sc">@</span>meta.data))</span>
<span id="cb145-115"><a href="appendix.html#cb145-115" aria-hidden="true" tabindex="-1"></a>  nExp_poi.adj <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> homotypic.prop))</span>
<span id="cb145-116"><a href="appendix.html#cb145-116" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-117"><a href="appendix.html#cb145-117" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run doubletFinder </span></span>
<span id="cb145-118"><a href="appendix.html#cb145-118" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(filt_seurat_object,</span>
<span id="cb145-119"><a href="appendix.html#cb145-119" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,</span>
<span id="cb145-120"><a href="appendix.html#cb145-120" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">pN =</span> <span class="fl">0.25</span>,</span>
<span id="cb145-121"><a href="appendix.html#cb145-121" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">pK =</span> pK,</span>
<span id="cb145-122"><a href="appendix.html#cb145-122" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">nExp =</span> nExp_poi.adj,</span>
<span id="cb145-123"><a href="appendix.html#cb145-123" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>,</span>
<span id="cb145-124"><a href="appendix.html#cb145-124" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">sct =</span> <span class="cn">FALSE</span>)</span>
<span id="cb145-125"><a href="appendix.html#cb145-125" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-126"><a href="appendix.html#cb145-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(filt_seurat_object<span class="sc">@</span>meta.data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;DF.classifications_.*$&quot;</span>, <span class="st">&quot;DF.classifications&quot;</span>, <span class="fu">colnames</span>(filt_seurat_object<span class="sc">@</span>meta.data))</span>
<span id="cb145-127"><a href="appendix.html#cb145-127" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-128"><a href="appendix.html#cb145-128" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Summary doublets</span></span>
<span id="cb145-129"><a href="appendix.html#cb145-129" aria-hidden="true" tabindex="-1"></a>  statsDoublets <span class="ot">&lt;-</span> filt_seurat_object<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb145-130"><a href="appendix.html#cb145-130" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(DF.classifications) <span class="sc">%&gt;%</span></span>
<span id="cb145-131"><a href="appendix.html#cb145-131" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">Median_nCount_RNA =</span> <span class="fu">median</span>(nCount_RNA),</span>
<span id="cb145-132"><a href="appendix.html#cb145-132" aria-hidden="true" tabindex="-1"></a>                     <span class="at">Median_nFeature_RNA =</span> <span class="fu">median</span>(nFeature_RNA), <span class="at">Count =</span> <span class="fu">n</span>())</span>
<span id="cb145-133"><a href="appendix.html#cb145-133" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-134"><a href="appendix.html#cb145-134" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Visualize doublets</span></span>
<span id="cb145-135"><a href="appendix.html#cb145-135" aria-hidden="true" tabindex="-1"></a>  doublets <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(filt_seurat_object,</span>
<span id="cb145-136"><a href="appendix.html#cb145-136" aria-hidden="true" tabindex="-1"></a>                      <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>,</span>
<span id="cb145-137"><a href="appendix.html#cb145-137" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group.by =</span> <span class="st">&quot;DF.classifications&quot;</span>)</span>
<span id="cb145-138"><a href="appendix.html#cb145-138" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-139"><a href="appendix.html#cb145-139" aria-hidden="true" tabindex="-1"></a>  <span class="do">### i want to save the seurat object with doublets listed </span></span>
<span id="cb145-140"><a href="appendix.html#cb145-140" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object_doublets <span class="ot">&lt;-</span> filt_seurat_object</span>
<span id="cb145-141"><a href="appendix.html#cb145-141" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-142"><a href="appendix.html#cb145-142" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">subset</span>(filt_seurat_object, <span class="at">subset =</span> DF.classifications <span class="sc">==</span> <span class="st">&#39;Singlet&#39;</span>)</span>
<span id="cb145-143"><a href="appendix.html#cb145-143" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-144"><a href="appendix.html#cb145-144" aria-hidden="true" tabindex="-1"></a><span class="co"># figures</span></span>
<span id="cb145-145"><a href="appendix.html#cb145-145" aria-hidden="true" tabindex="-1"></a>  ggplot_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb145-146"><a href="appendix.html#cb145-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ElbowPlot</span>(filt_seurat_object) <span class="sc">+</span></span>
<span id="cb145-147"><a href="appendix.html#cb145-147" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&#39;SD explained by each PC&#39;</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb145-148"><a href="appendix.html#cb145-148" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-149"><a href="appendix.html#cb145-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeatureScatter</span>(filt_seurat_object, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb145-150"><a href="appendix.html#cb145-150" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span> </span>
<span id="cb145-151"><a href="appendix.html#cb145-151" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Association between reads and </span><span class="sc">\n</span><span class="st">unique genes per cell AFTER filtering&quot;</span>),</span>
<span id="cb145-152"><a href="appendix.html#cb145-152" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-153"><a href="appendix.html#cb145-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DimPlot</span>(filt_seurat_object, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>) <span class="sc">+</span> </span>
<span id="cb145-154"><a href="appendix.html#cb145-154" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">&quot;Cluster </span><span class="sc">\n</span><span class="st">(from PCA)&quot;</span>, <span class="at">title =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span> </span>
<span id="cb145-155"><a href="appendix.html#cb145-155" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb145-156"><a href="appendix.html#cb145-156" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-157"><a href="appendix.html#cb145-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">features =</span> <span class="st">&#39;nCount_RNA&#39;</span>) <span class="sc">+</span> </span>
<span id="cb145-158"><a href="appendix.html#cb145-158" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">&quot;UMI count&quot;</span>, <span class="at">title =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span> </span>
<span id="cb145-159"><a href="appendix.html#cb145-159" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb145-160"><a href="appendix.html#cb145-160" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-161"><a href="appendix.html#cb145-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">features =</span> <span class="st">&#39;nFeature_RNA&#39;</span>) <span class="sc">+</span> </span>
<span id="cb145-162"><a href="appendix.html#cb145-162" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color =</span> <span class="fu">str_wrap</span>(<span class="st">&quot;Feature count (gene)&quot;</span>, <span class="dv">15</span>), <span class="at">title =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span> </span>
<span id="cb145-163"><a href="appendix.html#cb145-163" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb145-164"><a href="appendix.html#cb145-164" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-165"><a href="appendix.html#cb145-165" aria-hidden="true" tabindex="-1"></a>  p2</span>
<span id="cb145-166"><a href="appendix.html#cb145-166" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb145-167"><a href="appendix.html#cb145-167" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-168"><a href="appendix.html#cb145-168" aria-hidden="true" tabindex="-1"></a>   combined_plots <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(<span class="at">plotlist =</span> ggplot_list, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb145-169"><a href="appendix.html#cb145-169" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-170"><a href="appendix.html#cb145-170" aria-hidden="true" tabindex="-1"></a>   <span class="fu">plot</span>(combined_plots)</span>
<span id="cb145-171"><a href="appendix.html#cb145-171" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb145-172"><a href="appendix.html#cb145-172" aria-hidden="true" tabindex="-1"></a>   <span class="fu">plot</span>(<span class="fu">DimPlot</span>(filt_seurat_object_doublets, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>,</span>
<span id="cb145-173"><a href="appendix.html#cb145-173" aria-hidden="true" tabindex="-1"></a>                 <span class="at">group.by =</span> <span class="st">&quot;DF.classifications&quot;</span>))</span>
<span id="cb145-174"><a href="appendix.html#cb145-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-175"><a href="appendix.html#cb145-175" aria-hidden="true" tabindex="-1"></a>     tbl_sts1 <span class="ot">&lt;-</span> <span class="fu">tableGrob</span>(statsDoublets)</span>
<span id="cb145-176"><a href="appendix.html#cb145-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.newpage</span>()</span>
<span id="cb145-177"><a href="appendix.html#cb145-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.draw</span>(tbl_sts1)</span>
<span id="cb145-178"><a href="appendix.html#cb145-178" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-179"><a href="appendix.html#cb145-179" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-180"><a href="appendix.html#cb145-180" aria-hidden="true" tabindex="-1"></a><span class="co"># summary stats  </span></span>
<span id="cb145-181"><a href="appendix.html#cb145-181" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb145-182"><a href="appendix.html#cb145-182" aria-hidden="true" tabindex="-1"></a> stats_sumary <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="st">&quot;Sample ID&quot;</span> <span class="ot">=</span> project,</span>
<span id="cb145-183"><a href="appendix.html#cb145-183" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Cells_before_filter&quot;</span> <span class="ot">=</span> <span class="fu">dim</span>(seurat_object)[<span class="dv">2</span>],</span>
<span id="cb145-184"><a href="appendix.html#cb145-184" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Cells_after_filter&quot;</span> <span class="ot">=</span> <span class="fu">dim</span>(filt_seurat_object)[<span class="dv">2</span>],</span>
<span id="cb145-185"><a href="appendix.html#cb145-185" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Feature per Cell before filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nFeature_RNA),</span>
<span id="cb145-186"><a href="appendix.html#cb145-186" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Reads per Gene before filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb145-187"><a href="appendix.html#cb145-187" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Feature per Cell&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(filt_seurat_object<span class="sc">$</span>nFeature_RNA),</span>
<span id="cb145-188"><a href="appendix.html#cb145-188" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Reads per Gene&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(filt_seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb145-189"><a href="appendix.html#cb145-189" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Max Features&quot;</span> <span class="ot">=</span> max.features,</span>
<span id="cb145-190"><a href="appendix.html#cb145-190" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Min Features&quot;</span> <span class="ot">=</span> min.features,</span>
<span id="cb145-191"><a href="appendix.html#cb145-191" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Min Counts&quot;</span> <span class="ot">=</span> min.counts,</span>
<span id="cb145-192"><a href="appendix.html#cb145-192" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Max Counts&quot;</span> <span class="ot">=</span> max.counts,</span>
<span id="cb145-193"><a href="appendix.html#cb145-193" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;MT Percentage&quot;</span> <span class="ot">=</span> MT,</span>
<span id="cb145-194"><a href="appendix.html#cb145-194" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;NPCs&quot;</span> <span class="ot">=</span> npc,</span>
<span id="cb145-195"><a href="appendix.html#cb145-195" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Percent MT before Filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(seurat_object<span class="sc">@</span>meta.data[[<span class="st">&quot;percent.mt&quot;</span>]]),</span>
<span id="cb145-196"><a href="appendix.html#cb145-196" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Percent MT after Filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(filt_seurat_object<span class="sc">@</span>meta.data[[<span class="st">&quot;percent.mt&quot;</span>]])</span>
<span id="cb145-197"><a href="appendix.html#cb145-197" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb145-198"><a href="appendix.html#cb145-198" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb145-199"><a href="appendix.html#cb145-199" aria-hidden="true" tabindex="-1"></a>  tbl_sts2 <span class="ot">&lt;-</span> <span class="fu">tableGrob</span>(stats_sumary)</span>
<span id="cb145-200"><a href="appendix.html#cb145-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.newpage</span>()</span>
<span id="cb145-201"><a href="appendix.html#cb145-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.draw</span>(tbl_sts2)</span>
<span id="cb145-202"><a href="appendix.html#cb145-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-203"><a href="appendix.html#cb145-203" aria-hidden="true" tabindex="-1"></a>       <span class="fu">list</span>(filt_seurat_object, </span>
<span id="cb145-204"><a href="appendix.html#cb145-204" aria-hidden="true" tabindex="-1"></a>            statsDoublets, </span>
<span id="cb145-205"><a href="appendix.html#cb145-205" aria-hidden="true" tabindex="-1"></a>            stats_sumary, </span>
<span id="cb145-206"><a href="appendix.html#cb145-206" aria-hidden="true" tabindex="-1"></a>            filt_seurat_object_doublets)</span>
<span id="cb145-207"><a href="appendix.html#cb145-207" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="convert-oarfish-files-to-count-matrix" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Convert Oarfish files to count matrix<a href="appendix.html#convert-oarfish-files-to-count-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This is a function to take all oarfish files from FLAMES multisample output folder and convert the files into a gene count matrix that contains gene symbol ids instead of ENSGIDs. To use this fucntion ensure that the <code>isoform_gene_dict.csv</code> file has been genrated as decibed in <a href="#creating-resource-files"><strong>??</strong></a></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="appendix.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="do">### readSeurat### read in oarfish count files and add them to seurat objects </span><span class="al">###</span></span>
<span id="cb146-2"><a href="appendix.html#cb146-2" aria-hidden="true" tabindex="-1"></a>process_oarfish_files_to_counts_matrix <span class="ot">&lt;-</span> <span class="cf">function</span>(sample_name, resource_table_path, output_dir, input_dir) {</span>
<span id="cb146-3"><a href="appendix.html#cb146-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-4"><a href="appendix.html#cb146-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load required libraries</span></span>
<span id="cb146-5"><a href="appendix.html#cb146-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(Matrix)</span>
<span id="cb146-6"><a href="appendix.html#cb146-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(dplyr)</span>
<span id="cb146-7"><a href="appendix.html#cb146-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-8"><a href="appendix.html#cb146-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read in the resource table (transcript_id, gene_id, gene_symbol)</span></span>
<span id="cb146-9"><a href="appendix.html#cb146-9" aria-hidden="true" tabindex="-1"></a>  combined_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(resource_table_path)</span>
<span id="cb146-10"><a href="appendix.html#cb146-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-11"><a href="appendix.html#cb146-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define the file paths based on the sample name</span></span>
<span id="cb146-12"><a href="appendix.html#cb146-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct file paths</span></span>
<span id="cb146-13"><a href="appendix.html#cb146-13" aria-hidden="true" tabindex="-1"></a>  count_matrix_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="fu">paste0</span>(sample_name, <span class="st">&quot;.count.mtx&quot;</span>))</span>
<span id="cb146-14"><a href="appendix.html#cb146-14" aria-hidden="true" tabindex="-1"></a>  barcodes_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="fu">paste0</span>(sample_name, <span class="st">&quot;.barcodes.txt&quot;</span>))</span>
<span id="cb146-15"><a href="appendix.html#cb146-15" aria-hidden="true" tabindex="-1"></a>  features_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="fu">paste0</span>(sample_name, <span class="st">&quot;.features.txt&quot;</span>))</span>
<span id="cb146-16"><a href="appendix.html#cb146-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-17"><a href="appendix.html#cb146-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read the data</span></span>
<span id="cb146-18"><a href="appendix.html#cb146-18" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">readMM</span>(count_matrix_path)</span>
<span id="cb146-19"><a href="appendix.html#cb146-19" aria-hidden="true" tabindex="-1"></a>  barcodes <span class="ot">&lt;-</span> <span class="fu">readLines</span>(barcodes_path)</span>
<span id="cb146-20"><a href="appendix.html#cb146-20" aria-hidden="true" tabindex="-1"></a>  features <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(features_path, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb146-21"><a href="appendix.html#cb146-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-22"><a href="appendix.html#cb146-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Transpose the matrix if needed (for Seurat compatibility)</span></span>
<span id="cb146-23"><a href="appendix.html#cb146-23" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">t</span>(counts)</span>
<span id="cb146-24"><a href="appendix.html#cb146-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-25"><a href="appendix.html#cb146-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set row and column names</span></span>
<span id="cb146-26"><a href="appendix.html#cb146-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(counts) <span class="ot">&lt;-</span> features<span class="sc">$</span>V1</span>
<span id="cb146-27"><a href="appendix.html#cb146-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(counts) <span class="ot">&lt;-</span> barcodes</span>
<span id="cb146-28"><a href="appendix.html#cb146-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-29"><a href="appendix.html#cb146-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert to a data frame</span></span>
<span id="cb146-30"><a href="appendix.html#cb146-30" aria-hidden="true" tabindex="-1"></a>  counts_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts)</span>
<span id="cb146-31"><a href="appendix.html#cb146-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-32"><a href="appendix.html#cb146-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add transcript_id as the first column</span></span>
<span id="cb146-33"><a href="appendix.html#cb146-33" aria-hidden="true" tabindex="-1"></a>  counts_df<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts_df)</span>
<span id="cb146-34"><a href="appendix.html#cb146-34" aria-hidden="true" tabindex="-1"></a>  counts_df <span class="ot">&lt;-</span> counts_df[, <span class="fu">c</span>(<span class="fu">ncol</span>(counts_df), <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(counts_df)<span class="sc">-</span><span class="dv">1</span>))]</span>
<span id="cb146-35"><a href="appendix.html#cb146-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-36"><a href="appendix.html#cb146-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge with the resource table to add gene symbols</span></span>
<span id="cb146-37"><a href="appendix.html#cb146-37" aria-hidden="true" tabindex="-1"></a>  df_genesymbol <span class="ot">&lt;-</span> counts_df <span class="sc">%&gt;%</span></span>
<span id="cb146-38"><a href="appendix.html#cb146-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(combined_data, <span class="at">by =</span> <span class="st">&quot;transcript_id&quot;</span>)</span>
<span id="cb146-39"><a href="appendix.html#cb146-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-40"><a href="appendix.html#cb146-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove the gene_id column and reorder the columns</span></span>
<span id="cb146-41"><a href="appendix.html#cb146-41" aria-hidden="true" tabindex="-1"></a>  df_genesymbol<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb146-42"><a href="appendix.html#cb146-42" aria-hidden="true" tabindex="-1"></a>  df_genesymbol <span class="ot">&lt;-</span> df_genesymbol[, <span class="fu">c</span>(<span class="fu">ncol</span>(df_genesymbol), <span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(df_genesymbol)<span class="sc">-</span><span class="dv">1</span>))]</span>
<span id="cb146-43"><a href="appendix.html#cb146-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-44"><a href="appendix.html#cb146-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Update row names to include gene symbol instead of transcript_id</span></span>
<span id="cb146-45"><a href="appendix.html#cb146-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(df_genesymbol) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df_genesymbol<span class="sc">$</span>transcript_id, <span class="st">&quot;_&quot;</span>, df_genesymbol<span class="sc">$</span>gene_symbol)</span>
<span id="cb146-46"><a href="appendix.html#cb146-46" aria-hidden="true" tabindex="-1"></a>  df_genesymbol<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb146-47"><a href="appendix.html#cb146-47" aria-hidden="true" tabindex="-1"></a>  df_genesymbol<span class="sc">$</span>gene_symbol <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb146-48"><a href="appendix.html#cb146-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-49"><a href="appendix.html#cb146-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Write the output to a CSV file</span></span>
<span id="cb146-50"><a href="appendix.html#cb146-50" aria-hidden="true" tabindex="-1"></a>  output_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="fu">paste0</span>(<span class="st">&quot;gene_symbol_oarfish_&quot;</span>, sample_name, <span class="st">&quot;_counts.csv&quot;</span>))</span>
<span id="cb146-51"><a href="appendix.html#cb146-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fwrite</span>(df_genesymbol, output_path, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb146-52"><a href="appendix.html#cb146-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb146-53"><a href="appendix.html#cb146-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Processed sample:&quot;</span>, sample_name, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Output saved to:&quot;</span>, output_path, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb146-54"><a href="appendix.html#cb146-54" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb146-55"><a href="appendix.html#cb146-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-56"><a href="appendix.html#cb146-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage:</span></span>
<span id="cb146-57"><a href="appendix.html#cb146-57" aria-hidden="true" tabindex="-1"></a><span class="co"># List of sample names</span></span>
<span id="cb146-58"><a href="appendix.html#cb146-58" aria-hidden="true" tabindex="-1"></a>sample_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;C1_STC&quot;</span>)</span>
<span id="cb146-59"><a href="appendix.html#cb146-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-60"><a href="appendix.html#cb146-60" aria-hidden="true" tabindex="-1"></a><span class="co">#create a resource </span></span>
<span id="cb146-61"><a href="appendix.html#cb146-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the helper function defined in code block above to create a dictionary containing corresponding gene information for each isoform</span></span>
<span id="cb146-62"><a href="appendix.html#cb146-62" aria-hidden="true" tabindex="-1"></a><span class="co"># This may take a few minutes </span></span>
<span id="cb146-63"><a href="appendix.html#cb146-63" aria-hidden="true" tabindex="-1"></a><span class="co"># The FLAMES ref can be found in your selected output folder after running the Flames pipeline. </span></span>
<span id="cb146-64"><a href="appendix.html#cb146-64" aria-hidden="true" tabindex="-1"></a>FLAMES_gtf_file <span class="ot">&lt;-</span> <span class="st">&quot;./data/muti_sample/isoform_annotated.gtf&quot;</span> <span class="co">#ensure file is unzipped</span></span>
<span id="cb146-65"><a href="appendix.html#cb146-65" aria-hidden="true" tabindex="-1"></a>reference_gtf_file <span class="ot">&lt;-</span> <span class="st">&quot;./data/gencode.v47.annotation.gtf&quot;</span> <span class="co"># ensure file is unzipped</span></span>
<span id="cb146-66"><a href="appendix.html#cb146-66" aria-hidden="true" tabindex="-1"></a>output_file <span class="ot">&lt;-</span> <span class="st">&quot;multi_isoform_gene_dict.csv&quot;</span></span>
<span id="cb146-67"><a href="appendix.html#cb146-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-68"><a href="appendix.html#cb146-68" aria-hidden="true" tabindex="-1"></a>isoform_gene_dict <span class="ot">&lt;-</span> <span class="fu">make_isoform_gene_symbol_dict</span>(FLAMES_gtf_file,</span>
<span id="cb146-69"><a href="appendix.html#cb146-69" aria-hidden="true" tabindex="-1"></a>                                                   reference_gtf_file,</span>
<span id="cb146-70"><a href="appendix.html#cb146-70" aria-hidden="true" tabindex="-1"></a>                                                   output_file)</span>
<span id="cb146-71"><a href="appendix.html#cb146-71" aria-hidden="true" tabindex="-1"></a><span class="co"># run this in the output FLAMES dir</span></span>
<span id="cb146-72"><a href="appendix.html#cb146-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-73"><a href="appendix.html#cb146-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each sample and process the count files</span></span>
<span id="cb146-74"><a href="appendix.html#cb146-74" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (sample <span class="cf">in</span> sample_names) {</span>
<span id="cb146-75"><a href="appendix.html#cb146-75" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use tryCatch to handle errors</span></span>
<span id="cb146-76"><a href="appendix.html#cb146-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb146-77"><a href="appendix.html#cb146-77" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Call the function to process the sample</span></span>
<span id="cb146-78"><a href="appendix.html#cb146-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">process_oarfish_files_to_counts_matrix</span>(</span>
<span id="cb146-79"><a href="appendix.html#cb146-79" aria-hidden="true" tabindex="-1"></a>      <span class="at">sample_name =</span> sample,</span>
<span id="cb146-80"><a href="appendix.html#cb146-80" aria-hidden="true" tabindex="-1"></a>      <span class="at">resource_table_path =</span> <span class="st">&quot;./output_files/ref_files/multi_isoform_gene_dict.csv&quot;</span>,</span>
<span id="cb146-81"><a href="appendix.html#cb146-81" aria-hidden="true" tabindex="-1"></a>      <span class="at">output_dir =</span> <span class="st">&quot;./output_files/mutli_sample/oarfish_counts&quot;</span>,</span>
<span id="cb146-82"><a href="appendix.html#cb146-82" aria-hidden="true" tabindex="-1"></a>      <span class="at">input_dir =</span> <span class="st">&quot;./output_files/mutli_sample/oarfish_counts&quot;</span></span>
<span id="cb146-83"><a href="appendix.html#cb146-83" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb146-84"><a href="appendix.html#cb146-84" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb146-85"><a href="appendix.html#cb146-85" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Print the error message and exit the loop</span></span>
<span id="cb146-86"><a href="appendix.html#cb146-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;Error processing sample:&quot;</span>, sample, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Error message:&quot;</span>, e<span class="sc">$</span>message, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Exiting loop.</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb146-87"><a href="appendix.html#cb146-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(e)</span>
<span id="cb146-88"><a href="appendix.html#cb146-88" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb146-89"><a href="appendix.html#cb146-89" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="conclusion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/12_Appendix.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/12_Appendix.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
