<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Citations and Session info | FLAMES Single-Cell RNA-seq Tutorial</title>
  <meta name="description" content="Chapter 6 Citations and Session info | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Citations and Session info | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Citations and Session info | FLAMES Single-Cell RNA-seq Tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer" />


<meta name="date" content="2025-01-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="finding-differentially-expressed-genes-and-isoforms.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single sample RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> Setup</a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> Removing sources of unwanted noise from the single cell dataset</a>
<ul>
<li class="chapter" data-level="3.0.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.0.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.1</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.2</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> Add isoform counts to Seurat object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> Finding differentially expressed genes and isoforms</a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#de-genes-and-isofroms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isofroms based on annotaed cell types.</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.3.1</b> Volcano plots</a></li>
<li class="chapter" data-level="5.3.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#gene-set-enrichemt"><i class="fa fa-check"></i><b>5.3.2</b> Gene Set enrichemt</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#finding-isofrms-of-interest"><i class="fa fa-check"></i><b>5.4</b> Finding isofrms of interest</a></li>
<li class="chapter" data-level="5.5" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#section"><i class="fa fa-check"></i><b>5.5</b> </a></li>
<li class="chapter" data-level="5.6" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#plotting-with-ditoplot"><i class="fa fa-check"></i><b>5.6</b> plotting with ditoplot</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="citations-and-session-info.html"><a href="citations-and-session-info.html"><i class="fa fa-check"></i><b>6</b> Citations and Session info</a>
<ul>
<li class="chapter" data-level="6.1" data-path="citations-and-session-info.html"><a href="citations-and-session-info.html#citations"><i class="fa fa-check"></i><b>6.1</b> Citations</a></li>
<li class="chapter" data-level="6.2" data-path="citations-and-session-info.html"><a href="citations-and-session-info.html#session-info"><i class="fa fa-check"></i><b>6.2</b> Session info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FLAMES Single-Cell RNA-seq Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="citations-and-session-info" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Citations and Session info<a href="citations-and-session-info.html#citations-and-session-info" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="citations" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Citations<a href="citations-and-session-info.html#citations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><span class="citation">[@tian2021]</span></p>
</div>
<div id="session-info" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Session info<a href="citations-and-session-info.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>## R version 4.4.0 (2024-04-24)
## Platform: x86_64-pc-linux-gnu
## Running under: Red Hat Enterprise Linux 9.4 (Plow)
## 
## Matrix products: default
## BLAS/LAPACK: FlexiBLAS OPENBLAS;  LAPACK version 3.10.1
## 
## locale:
##  [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C               LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
##  [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8    LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: Australia/Melbourne
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] EnhancedVolcano_1.24.0      ggrepel_0.9.6               HGNChelper_0.8.14           data.tree_1.1.0            
##  [5] igraph_2.1.1                ggraph_2.2.1                lubridate_1.9.3             forcats_1.0.0              
##  [9] dplyr_1.1.4                 purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
## [13] tibble_3.2.1                ggplot2_3.5.1               tidyverse_2.0.0             patchwork_1.3.0            
## [17] cowplot_1.1.3               stringr_1.5.1               DoubletFinder_2.0.4         celda_1.20.0               
## [21] Matrix_1.7-1                BiocParallel_1.40.0         data.table_1.16.2           gridExtra_2.3              
## [25] DropletUtils_1.26.0         SingleCellExperiment_1.28.0 SummarizedExperiment_1.36.0 Biobase_2.66.0             
## [29] MatrixGenerics_1.18.0       matrixStats_1.4.1           Seurat_5.1.0                SeuratObject_5.0.2         
## [33] sp_2.1-4                    rtracklayer_1.64.0          GenomicRanges_1.58.0        GenomeInfoDb_1.42.0        
## [37] IRanges_2.40.0              S4Vectors_0.44.0            BiocGenerics_0.52.0        
## 
## loaded via a namespace (and not attached):
##   [1] spatstat.sparse_3.1-0     bitops_1.0-9              httr_1.4.7                RColorBrewer_1.1-3       
##   [5] doParallel_1.0.17         tools_4.4.0               sctransform_0.4.1         utf8_1.2.4               
##   [9] R6_2.5.1                  HDF5Array_1.34.0          lazyeval_0.2.2            uwot_0.2.2               
##  [13] rhdf5filters_1.18.0       withr_3.0.2               progressr_0.15.0          cli_3.6.3                
##  [17] spatstat.explore_3.3-1    fastDummies_1.7.4         enrichR_3.2               labeling_0.4.3           
##  [21] sass_0.4.9                spatstat.data_3.1-2       ggridges_0.5.6            pbapply_1.7-2            
##  [25] Rsamtools_2.20.0          R.utils_2.12.3            parallelly_1.38.0         WriteXLS_6.7.0           
##  [29] limma_3.62.1              rstudioapi_0.15.0         generics_0.1.3            BiocIO_1.16.0            
##  [33] combinat_0.0-8            ica_1.0-3                 spatstat.random_3.3-1     zip_2.3.0                
##  [37] ggbeeswarm_0.7.2          fansi_1.0.6               abind_1.4-8               R.methodsS3_1.8.2        
##  [41] lifecycle_1.0.4           yaml_2.3.10               edgeR_4.4.0               rhdf5_2.50.0             
##  [45] SparseArray_1.6.0         Rtsne_0.17                promises_1.3.0            dqrng_0.4.1              
##  [49] crayon_1.5.3              miniUI_0.1.1.1            lattice_0.22-6            beachmat_2.22.0          
##  [53] pillar_1.9.0              knitr_1.48                rjson_0.2.23              boot_1.3-31              
##  [57] future.apply_1.11.3       codetools_0.2-20          leiden_0.4.3.1            glue_1.8.0               
##  [61] spatstat.univar_3.0-0     vctrs_0.6.5               png_0.1-8                 spam_2.10-0              
##  [65] gtable_0.3.6              cachem_1.1.0              openxlsx_4.2.6.1          xfun_0.49                
##  [69] S4Arrays_1.6.0            mime_0.12                 tidygraph_1.3.1           RcppEigen_0.3.4.0.2      
##  [73] survival_3.7-0            iterators_1.0.14          statmod_1.5.0             fitdistrplus_1.2-1       
##  [77] ROCR_1.0-11               nlme_3.1-166              RcppAnnoy_0.0.22          bslib_0.8.0              
##  [81] irlba_2.3.5.1             vipor_0.4.7               KernSmooth_2.23-24        splitstackshape_1.4.8    
##  [85] colorspace_2.1-1          ggrastr_1.0.2             tidyselect_1.2.1          compiler_4.4.0           
##  [89] curl_6.0.0                DelayedArray_0.32.0       plotly_4.10.4             bookdown_0.41            
##  [93] scales_1.3.0              lmtest_0.9-40             digest_0.6.37             goftest_1.2-3            
##  [97] presto_1.0.0              spatstat.utils_3.1-1      minqa_1.2.8               rmarkdown_2.29           
## [101] XVector_0.46.0            htmltools_0.5.8.1         pkgconfig_2.0.3           lme4_1.1-35.5            
## [105] sparseMatrixStats_1.18.0  highr_0.11                fastmap_1.2.0             rlang_1.1.4              
## [109] htmlwidgets_1.6.4         UCSC.utils_1.2.0          shiny_1.9.1               DelayedMatrixStats_1.28.0
## [113] farver_2.1.2              jquerylib_0.1.4           zoo_1.8-12                jsonlite_1.8.9           
## [117] R.oo_1.27.0               RCurl_1.98-1.16           magrittr_2.0.3            scuttle_1.16.0           
## [121] GenomeInfoDbData_1.2.13   dotCall64_1.2             Rhdf5lib_1.28.0           munsell_0.5.1            
## [125] Rcpp_1.0.13-1             viridis_0.6.5             reticulate_1.39.0         stringi_1.8.4            
## [129] MCMCprecision_0.4.0       zlibbioc_1.52.0           MASS_7.3-61               plyr_1.8.9               
## [133] parallel_4.4.0            listenv_0.9.1             deldir_2.0-4              graphlayouts_1.2.0       
## [137] Biostrings_2.72.1         splines_4.4.0             tensor_1.5                hms_1.1.3                
## [141] locfit_1.5-9.10           spatstat.geom_3.3-2       RcppHNSW_0.6.0            reshape2_1.4.4           
## [145] servr_0.32                XML_3.99-0.17             evaluate_1.0.1            tweenr_2.0.3             
## [149] nloptr_2.1.1              tzdb_0.4.0                foreach_1.5.2             httpuv_1.6.15            
## [153] RANN_2.6.2                polyclip_1.10-7           future_1.34.0             scattermore_1.2          
## [157] ggforce_0.4.2             xtable_1.8-4              monocle3_1.3.7            restfulr_0.0.15          
## [161] RSpectra_0.16-2           later_1.3.2               viridisLite_0.4.2         memoise_2.0.1            
## [165] beeswarm_0.4.0            GenomicAlignments_1.40.0  cluster_2.1.6             timechange_0.3.0         
## [169] globals_0.16.3</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="finding-differentially-expressed-genes-and-isoforms.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/07_Citations_and_Session_info.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/07_Citations_and_Session_info.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
