<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Conclusion | FLAMES Single-Cell RNA-seq Tutorial</title>
  <meta name="description" content="Chapter 9 Conclusion | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Conclusion | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Conclusion | FLAMES Single-Cell RNA-seq Tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer" />


<meta name="date" content="2025-01-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="novel-isofroms.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single sample RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> Setup</a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> Removing sources of unwanted noise from the single cell dataset</a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> Add isoform counts to Seurat object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> Finding differentially expressed genes and isoforms</a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#de-genes-and-isofroms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isofroms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> Exploring isoforms of interest</a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> Isoform Classification</a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isofroms.html"><a href="novel-isofroms.html"><i class="fa fa-check"></i><b>8</b> Novel isofroms</a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isofroms.html"><a href="novel-isofroms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>9</b> Conclusion</a>
<ul>
<li class="chapter" data-level="9.1" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i><b>9.1</b> Session info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FLAMES Single-Cell RNA-seq Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="conclusion" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Conclusion<a href="conclusion.html#conclusion" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this tutorial, we have demonstrated an end-to-end workflow for analysing and visualising long-read single cell data generated by FLAMES <span class="citation">(<a href="#ref-tian2021" role="doc-biblioref">Tian et al., 2021</a>)</span>. We have shown how to preprocess the data to retain high-quality cells, load gene and isoform counts into a Seurat object, identify differentially expressed genes and isoforms across cell types, visualize isoform expression and structure, and identify novel isoforms that may impact protein function.</p>
<p>While this tutorial covers the core aspects of single-cell long-read data analysis, several additional avenues remain unexplored. Further analyses could include enrichment analysis of differentially expressed transcription factors or splicing regulators, motif analysis to investigate regulatory element binding, and alternative splicing events that may impact cellular functions. Integrating these analyses can provide a more comprehensive understanding of cellular diversity and regulatory mechanisms.</p>
<p>Ultimately, this tutorial, in combination with long-read tools such as FLAMES, offers a powerful toolkit for exploring and analyzing isoform-level dynamics in single cells. This workflow can be applied to other experimental contexts to identify candidate isoforms involved in development or disease pathogenesis advancing our understanding of complex cellular processes.</p>
<p><strong>note: need to add in multiple mode. not sure if this goes in in a separate tutorial or chapter. This would include.</strong></p>
<ol style="list-style-type: decimal">
<li><p><strong>Data integration</strong></p></li>
<li><p><strong>trajectory analysis and isofroms that change with pseudotime</strong></p></li>
<li><p><strong>DTU using isoform switch analyzer</strong></p></li>
</ol>
<div id="session-info" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Session info<a href="conclusion.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>## R version 4.4.0 (2024-04-24)
## Platform: x86_64-pc-linux-gnu
## Running under: Red Hat Enterprise Linux 9.4 (Plow)
## 
## Matrix products: default
## BLAS/LAPACK: FlexiBLAS OPENBLAS;  LAPACK version 3.10.1
## 
## locale:
##  [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C               LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8     LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
##  [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: Australia/Melbourne
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] Gviz_1.50.0                       BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.74.0                   BiocIO_1.16.0                     GenomicFeatures_1.58.0           
##  [6] AnnotationDbi_1.68.0              ORFik_1.26.1                      GenomicAlignments_1.40.0          Rsamtools_2.20.0                  Biostrings_2.72.1                
## [11] XVector_0.46.0                    biomaRt_2.62.0                    EnhancedVolcano_1.24.0            ggrepel_0.9.6                     dittoSeq_1.16.0                  
## [16] gprofiler2_0.2.3                  HGNChelper_0.8.14                 data.tree_1.1.0                   igraph_2.1.1                      ggraph_2.2.1                     
## [21] lubridate_1.9.3                   forcats_1.0.0                     dplyr_1.1.4                       purrr_1.0.2                       readr_2.1.5                      
## [26] tidyr_1.3.1                       tibble_3.2.1                      ggplot2_3.5.1                     tidyverse_2.0.0                   patchwork_1.3.0                  
## [31] cowplot_1.1.3                     stringr_1.5.1                     DoubletFinder_2.0.4               celda_1.20.0                      Matrix_1.7-1                     
## [36] BiocParallel_1.40.0               data.table_1.16.2                 gridExtra_2.3                     DropletUtils_1.26.0               SingleCellExperiment_1.28.0      
## [41] SummarizedExperiment_1.36.0       Biobase_2.66.0                    MatrixGenerics_1.18.0             matrixStats_1.4.1                 Seurat_5.1.0                     
## [46] SeuratObject_5.0.2                sp_2.1-4                          rtracklayer_1.64.0                GenomicRanges_1.58.0              GenomeInfoDb_1.42.0              
## [51] IRanges_2.40.0                    S4Vectors_0.44.0                  BiocGenerics_0.52.0              
## 
## loaded via a namespace (and not attached):
##   [1] dichromat_2.0-0.1         R.methodsS3_1.8.2         progress_1.2.3            nnet_7.3-19               goftest_1.2-3             biomartr_1.0.7           
##   [7] HDF5Array_1.34.0          vctrs_0.6.5               spatstat.random_3.3-1     digest_0.6.37             png_0.1-8                 deldir_2.0-4             
##  [13] parallelly_1.38.0         combinat_0.0-8            MASS_7.3-61               reshape2_1.4.4            httpuv_1.6.15             foreach_1.5.2            
##  [19] withr_3.0.2               ggrastr_1.0.2             xfun_0.49                 survival_3.7-0            memoise_2.0.1             ggbeeswarm_0.7.2         
##  [25] zoo_1.8-12                pbapply_1.7-2             R.oo_1.27.0               Formula_1.2-5             prettyunits_1.2.0         KEGGREST_1.46.0          
##  [31] promises_1.3.0            httr_1.4.7                restfulr_0.0.15           globals_0.16.3            fitdistrplus_1.2-1        rhdf5filters_1.18.0      
##  [37] fstcore_0.9.18            rhdf5_2.50.0              rstudioapi_0.15.0         UCSC.utils_1.2.0          miniUI_0.1.1.1            generics_0.1.3           
##  [43] base64enc_0.1-3           curl_6.0.0                zlibbioc_1.52.0           polyclip_1.10-7           GenomeInfoDbData_1.2.13   SparseArray_1.6.0        
##  [49] RcppEigen_0.3.4.0.2       xtable_1.8-4              doParallel_1.0.17         evaluate_1.0.1            S4Arrays_1.6.0            BiocFileCache_2.14.0     
##  [55] hms_1.1.3                 bookdown_0.41             irlba_2.3.5.1             colorspace_2.1-1          filelock_1.0.3            ROCR_1.0-11              
##  [61] reticulate_1.39.0         spatstat.data_3.1-2       magrittr_2.0.3            lmtest_0.9-40             later_1.3.2               viridis_0.6.5            
##  [67] lattice_0.22-6            spatstat.geom_3.3-2       future.apply_1.11.3       scattermore_1.2           XML_3.99-0.17             scuttle_1.16.0           
##  [73] RcppAnnoy_0.0.22          Hmisc_5.2-0               pillar_1.9.0              nlme_3.1-166              iterators_1.0.14          compiler_4.4.0           
##  [79] beachmat_2.22.0           RSpectra_0.16-2           stringi_1.8.4             tensor_1.5                MCMCprecision_0.4.0       plyr_1.8.9               
##  [85] crayon_1.5.3              abind_1.4-8               locfit_1.5-9.10           graphlayouts_1.2.0        bit_4.5.0                 codetools_0.2-20         
##  [91] bslib_0.8.0               biovizBase_1.54.0         plotly_4.10.4             mime_0.12                 splines_4.4.0             Rcpp_1.0.13-1            
##  [97] fastDummies_1.7.4         fst_0.9.8                 dbplyr_2.5.0              sparseMatrixStats_1.18.0  interp_1.1-6              knitr_1.48               
## [103] blob_1.2.4                utf8_1.2.4                AnnotationFilter_1.30.0   WriteXLS_6.7.0            checkmate_2.3.2           listenv_0.9.1            
## [109] DelayedMatrixStats_1.28.0 openxlsx_4.2.6.1          statmod_1.5.0             tzdb_0.4.0                tweenr_2.0.3              pkgconfig_2.0.3          
## [115] pheatmap_1.0.12           tools_4.4.0               cachem_1.1.0              RSQLite_2.3.7             viridisLite_0.4.2         DBI_1.2.3                
## [121] splitstackshape_1.4.8     fastmap_1.2.0             rmarkdown_2.29            scales_1.3.0              ica_1.0-3                 sass_0.4.9               
## [127] VariantAnnotation_1.52.0  dotCall64_1.2             rpart_4.1.23              RANN_2.6.2                farver_2.1.2              tidygraph_1.3.1          
## [133] yaml_2.3.10               latticeExtra_0.6-30       foreign_0.8-87            cli_3.6.3                 txdbmaker_1.0.1           leiden_0.4.3.1           
## [139] lifecycle_1.0.4           uwot_0.2.2                backports_1.5.0           presto_1.0.0              timechange_0.3.0          gtable_0.3.6             
## [145] rjson_0.2.23              ggridges_0.5.6            progressr_0.15.0          parallel_4.4.0            limma_3.62.1              jsonlite_1.8.9           
## [151] edgeR_4.4.0               RcppHNSW_0.6.0            bitops_1.0-9              bit64_4.5.2               Rtsne_0.17                spatstat.utils_3.1-1     
## [157] zip_2.3.0                 jquerylib_0.1.4           highr_0.11                dqrng_0.4.1               enrichR_3.2               spatstat.univar_3.0-0    
## [163] R.utils_2.12.3            lazyeval_0.2.2            shiny_1.9.1               htmltools_0.5.8.1         sctransform_0.4.1         rappdirs_0.3.3           
## [169] ensembldb_2.28.0          glue_1.8.0                spam_2.10-0               httr2_0.2.3               RCurl_1.98-1.16           jpeg_0.1-10              
## [175] R6_2.5.1                  DESeq2_1.46.0             labeling_0.4.3            cluster_2.1.6             Rhdf5lib_1.28.0           ProtGenerics_1.38.0      
## [181] DelayedArray_0.32.0       tidyselect_1.2.1          vipor_0.4.7               htmlTable_2.4.3           ggforce_0.4.2             xml2_1.3.5               
## [187] future_1.34.0             munsell_0.5.1             KernSmooth_2.23-24        htmlwidgets_1.6.4         RColorBrewer_1.1-3        rlang_1.1.4              
## [193] spatstat.sparse_3.1-0     spatstat.explore_3.3-1    fansi_1.0.6               beeswarm_0.4.0</code></pre>

<div id="refs" class="references csl-bib-body hanging-indent" line-spacing="2">
<div class="csl-entry">
Chen, Y., Sim, A., Wan, Y. K., Yeo, K., Lee, J. J. X., Ling, M. H., Love, M. I., &amp; Göke, J. (2023). Context-aware transcript quantification from long-read RNA-seq data with Bambu. <em>Nature Methods</em>, <em>20</em>(8), 1187–1195. <a href="https://doi.org/10.1038/s41592-023-01908-w">https://doi.org/10.1038/s41592-023-01908-w</a>
</div>
<div class="csl-entry">
Hahne, F., &amp; Ivanek, R. (2016). <em>Visualizing genomic data using gviz and bioconductor</em> (pp. 335–351). Springer New York. <a href="https://doi.org/10.1007/978-1-4939-3578-9_16">https://doi.org/10.1007/978-1-4939-3578-9_16</a>
</div>
<div class="csl-entry">
Kovaka, S., Zimin, A. V., Pertea, G. M., Razaghi, R., Salzberg, S. L., &amp; Pertea, M. (2019). Transcriptome assembly from long-read RNA-seq alignments with StringTie2. <em>Genome Biology</em>, <em>20</em>(1). <a href="https://doi.org/10.1186/s13059-019-1910-1">https://doi.org/10.1186/s13059-019-1910-1</a>
</div>
<div class="csl-entry">
Pardo-Palacios, F. J., Arzalluz-Luque, A., Kondratova, L., Salguero, P., Mestre-Tomás, J., Amorín, R., Estevan-Morió, E., Liu, T., Nanni, A., McIntyre, L., Tseng, E., &amp; Conesa, A. (2024). SQANTI3: curation of long-read transcriptomes for accurate identification of known and novel isoforms. <em>Nature Methods</em>, <em>21</em>(5), 793–797. <a href="https://doi.org/10.1038/s41592-024-02229-2">https://doi.org/10.1038/s41592-024-02229-2</a>
</div>
<div class="csl-entry">
Smedley, D., Haider, S., Ballester, B., Holland, R., London, D., Thorisson, G., &amp; Kasprzyk, A. (2009). BioMart <span></span> biological queries made easy. <em>BMC Genomics</em>, <em>10</em>(1). <a href="https://doi.org/10.1186/1471-2164-10-22">https://doi.org/10.1186/1471-2164-10-22</a>
</div>
<div class="csl-entry">
Tian, L., Jabbari, J. S., Thijssen, R., Gouil, Q., Amarasinghe, S. L., Voogd, O., Kariyawasam, H., Du, M. R. M., Schuster, J., Wang, C., Su, S., Dong, X., Law, C. W., Lucattini, A., Prawer, Y. D. J., Collar-Fernández, C., Chung, J. D., Naim, T., Chan, A., … Ritchie, M. E. (2021). Comprehensive characterization of single-cell full-length isoforms in human and mouse with long-read sequencing. <em>Genome Biology</em>, <em>22</em>(1). <a href="https://doi.org/10.1186/s13059-021-02525-6">https://doi.org/10.1186/s13059-021-02525-6</a>
</div>
<div class="csl-entry">
Tjeldnes, H., Labun, K., Torres Cleuren, Y., Chyżyńska, K., Świrski, M., &amp; Valen, E. (2021). ORFik: a comprehensive R toolkit for the analysis of translation. <em>BMC Bioinformatics</em>, <em>22</em>(1). <a href="https://doi.org/10.1186/s12859-021-04254-w">https://doi.org/10.1186/s12859-021-04254-w</a>
</div>
<div class="csl-entry">
Wan, C. Y., Davis, J., Chauhan, M., Gleeson, J., Prawer, Y. J., De Paoli-Iseppi, R., Wells, C., Choi, J., &amp; Clark, M. (2024). IsoVis <span></span> a webserver for visualization and annotation of alternative RNA isoforms. <em>Nucleic Acids Research</em>, <em>52</em>(W1), W341–W347. <a href="https://doi.org/10.1093/nar/gkae343">https://doi.org/10.1093/nar/gkae343</a>
</div>
<div class="csl-entry">
You, Y., Prawer, Y. D. J., De Paoli-Iseppi, R., Hunt, C. P. J., Parish, C. L., Shim, H., &amp; Clark, M. B. (2023). Identification of cell barcodes from long-read single-cell RNA-seq with BLAZE. <em>Genome Biology</em>, <em>24</em>(1). <a href="https://doi.org/10.1186/s13059-023-02907-y">https://doi.org/10.1186/s13059-023-02907-y</a>
</div>
</div>
</div>
</div>








<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" line-spacing="2">
<div id="ref-tian2021" class="csl-entry">
Tian, L., Jabbari, J. S., Thijssen, R., Gouil, Q., Amarasinghe, S. L., Voogd, O., Kariyawasam, H., Du, M. R. M., Schuster, J., Wang, C., Su, S., Dong, X., Law, C. W., Lucattini, A., Prawer, Y. D. J., Collar-Fernández, C., Chung, J. D., Naim, T., Chan, A., … Ritchie, M. E. (2021). Comprehensive characterization of single-cell full-length isoforms in human and mouse with long-read sequencing. <em>Genome Biology</em>, <em>22</em>(1). <a href="https://doi.org/10.1186/s13059-021-02525-6">https://doi.org/10.1186/s13059-021-02525-6</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="novel-isofroms.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/08_Citations_and_Session_info.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/08_Citations_and_Session_info.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
