<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 33 Create the stacked bar plot with canonical isoforms at the bottom | figure_genration_for_flames_v2</title>
  <meta name="description" content="Chapter 33 Create the stacked bar plot with canonical isoforms at the bottom | figure_genration_for_flames_v2" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 33 Create the stacked bar plot with canonical isoforms at the bottom | figure_genration_for_flames_v2" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 33 Create the stacked bar plot with canonical isoforms at the bottom | figure_genration_for_flames_v2" />
  
  
  

<meta name="author" content="Sefi" />


<meta name="date" content="2025-03-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="reshape-the-data-for-the-stacked-bar-plot.html"/>
<link rel="next" href="now-we-can-do-this-for-a-list-of-genes.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single cell RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualizing-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualizing OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>9</b> Multisample analysis</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>9.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="9.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>9.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>9.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multi-sample-integration"><i class="fa fa-check"></i><b>9.2</b> Multi-Sample integration</a></li>
<li class="chapter" data-level="9.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>9.3</b> Add isoform counts</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Trajectory analysis</strong></a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i><b>10.1</b> Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="10.2" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i><b>10.2</b> Multisample QC function</a></li>
<li class="chapter" data-level="10.3" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-count-matrix"><i class="fa fa-check"></i><b>10.3</b> Convert Oarfish files to count matrix</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="chapter" data-level="11" data-path="compute-total-proportions-across-all-data-no-per-cell-type-grouping.html"><a href="compute-total-proportions-across-all-data-no-per-cell-type-grouping.html"><i class="fa fa-check"></i><b>11</b> Compute total proportions across all data (no per-cell-type grouping)</a></li>
<li class="chapter" data-level="12" data-path="define-structural-category-order.html"><a href="define-structural-category-order.html"><i class="fa fa-check"></i><b>12</b> Define structural category order</a></li>
<li class="chapter" data-level="13" data-path="define-a-custom-color-palette.html"><a href="define-a-custom-color-palette.html"><i class="fa fa-check"></i><b>13</b> Define a custom color palette</a></li>
<li class="chapter" data-level="14" data-path="plot-total-proportion.html"><a href="plot-total-proportion.html"><i class="fa fa-check"></i><b>14</b> Plot total proportion</a></li>
<li class="chapter" data-level="15" data-path="summarise-expression-by-structural-categories.html"><a href="summarise-expression-by-structural-categories.html"><i class="fa fa-check"></i><b>15</b> Summarise expression by structural categories</a></li>
<li class="chapter" data-level="16" data-path="set-the-order-of-structural-categories.html"><a href="set-the-order-of-structural-categories.html"><i class="fa fa-check"></i><b>16</b> Set the order of structural categories</a></li>
<li class="chapter" data-level="17" data-path="reorder-the-factor-levels.html"><a href="reorder-the-factor-levels.html"><i class="fa fa-check"></i><b>17</b> Reorder the factor levels</a></li>
<li class="chapter" data-level="18" data-path="plot.html"><a href="plot.html"><i class="fa fa-check"></i><b>18</b> Plot</a></li>
<li class="chapter" data-level="19" data-path="filter-select-relevant-columns-and-categorize-structural-categories.html"><a href="filter-select-relevant-columns-and-categorize-structural-categories.html"><i class="fa fa-check"></i><b>19</b> Filter, select relevant columns, and categorize structural categories</a></li>
<li class="chapter" data-level="20" data-path="summarize-expression-values-grouped-by-cell-type-structure-and-coding.html"><a href="summarize-expression-values-grouped-by-cell-type-structure-and-coding.html"><i class="fa fa-check"></i><b>20</b> Summarize expression values grouped by cell type, structure, and coding</a></li>
<li class="chapter" data-level="21" data-path="define-factor-levels-for-plotting.html"><a href="define-factor-levels-for-plotting.html"><i class="fa fa-check"></i><b>21</b> Define factor levels for plotting</a></li>
<li class="chapter" data-level="22" data-path="plot-with-alpha-for-coding-status.html"><a href="plot-with-alpha-for-coding-status.html"><i class="fa fa-check"></i><b>22</b> Plot with <code>alpha</code> for coding status</a></li>
<li class="chapter" data-level="23" data-path="filter-and-prepare-data.html"><a href="filter-and-prepare-data.html"><i class="fa fa-check"></i><b>23</b> Filter and prepare data</a></li>
<li class="chapter" data-level="24" data-path="set-factor-levels.html"><a href="set-factor-levels.html"><i class="fa fa-check"></i><b>24</b> Set factor levels</a></li>
<li class="chapter" data-level="25" data-path="plot-1.html"><a href="plot-1.html"><i class="fa fa-check"></i><b>25</b> Plot</a></li>
<li class="chapter" data-level="26" data-path="summarize-data-across-all-cell-types.html"><a href="summarize-data-across-all-cell-types.html"><i class="fa fa-check"></i><b>26</b> Summarize data across all cell types</a></li>
<li class="chapter" data-level="27" data-path="assuming-your-coding-variable-is-defined-as-follows.html"><a href="assuming-your-coding-variable-is-defined-as-follows.html"><i class="fa fa-check"></i><b>27</b> Assuming your coding variable is defined as follows</a></li>
<li class="chapter" data-level="28" data-path="example-coding-variable-values-coding-and-non-coding.html"><a href="example-coding-variable-values-coding-and-non-coding.html"><i class="fa fa-check"></i><b>28</b> Example coding variable values: “coding” and “non-coding”</a></li>
<li class="chapter" data-level="29" data-path="calculate-the-total-and-canonical-isoforms-by-gene-and-cell-type.html"><a href="calculate-the-total-and-canonical-isoforms-by-gene-and-cell-type.html"><i class="fa fa-check"></i><b>29</b> Calculate the total and canonical isoforms by gene and cell type</a></li>
<li class="chapter" data-level="30" data-path="handle-cases-where-total_isoforms-0-to-avoid-division-by-zero.html"><a href="handle-cases-where-total_isoforms-0-to-avoid-division-by-zero.html"><i class="fa fa-check"></i><b>30</b> Handle cases where total_isoforms == 0 to avoid division by zero</a></li>
<li class="chapter" data-level="31" data-path="calculate-non-canonical-proportions.html"><a href="calculate-non-canonical-proportions.html"><i class="fa fa-check"></i><b>31</b> Calculate non-canonical proportions</a></li>
<li class="chapter" data-level="32" data-path="reshape-the-data-for-the-stacked-bar-plot.html"><a href="reshape-the-data-for-the-stacked-bar-plot.html"><i class="fa fa-check"></i><b>32</b> Reshape the data for the stacked bar plot</a></li>
<li class="chapter" data-level="33" data-path="create-the-stacked-bar-plot-with-canonical-isoforms-at-the-bottom.html"><a href="create-the-stacked-bar-plot-with-canonical-isoforms-at-the-bottom.html"><i class="fa fa-check"></i><b>33</b> Create the stacked bar plot with canonical isoforms at the bottom</a></li>
<li class="chapter" data-level="34" data-path="now-we-can-do-this-for-a-list-of-genes.html"><a href="now-we-can-do-this-for-a-list-of-genes.html"><i class="fa fa-check"></i><b>34</b> Now we can do this for a list of genes</a></li>
<li class="chapter" data-level="35" data-path="reshape-the-data-for-the-stacked-bar-plot-1.html"><a href="reshape-the-data-for-the-stacked-bar-plot-1.html"><i class="fa fa-check"></i><b>35</b> Reshape the data for the stacked bar plot</a></li>
<li class="chapter" data-level="36" data-path="plot-2.html"><a href="plot-2.html"><i class="fa fa-check"></i><b>36</b> Plot</a></li>
<li class="chapter" data-level="37" data-path="filter-the-data-for-the-selected-genes-and-summarize-by-isoform-and-cell-type.html"><a href="filter-the-data-for-the-selected-genes-and-summarize-by-isoform-and-cell-type.html"><i class="fa fa-check"></i><b>37</b> Filter the data for the selected genes and summarize by isoform and cell type</a></li>
<li class="chapter" data-level="38" data-path="now-for-each-gene-and-cell-type-calculate-the-total-expression-for-that-gene.html"><a href="now-for-each-gene-and-cell-type-calculate-the-total-expression-for-that-gene.html"><i class="fa fa-check"></i><b>38</b> Now, for each gene and cell type, calculate the total expression for that gene</a></li>
<li class="chapter" data-level="39" data-path="calculate-the-proportion-of-each-isoforms-expression-relative-to-the-total-gene-expression-in-each-cell-type.html"><a href="calculate-the-proportion-of-each-isoforms-expression-relative-to-the-total-gene-expression-in-each-cell-type.html"><i class="fa fa-check"></i><b>39</b> Calculate the proportion of each isoform’s expression relative to the total gene expression in each cell type</a></li>
<li class="chapter" data-level="40" data-path="reshape-the-data-for-the-stacked-bar-plot-2.html"><a href="reshape-the-data-for-the-stacked-bar-plot-2.html"><i class="fa fa-check"></i><b>40</b> Reshape the data for the stacked bar plot</a></li>
<li class="chapter" data-level="41" data-path="plot-3.html"><a href="plot-3.html"><i class="fa fa-check"></i><b>41</b> Plot</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">figure_genration_for_flames_v2</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="create-the-stacked-bar-plot-with-canonical-isoforms-at-the-bottom" class="section level1 hasAnchor" number="33">
<h1><span class="header-section-number">Chapter 33</span> Create the stacked bar plot with canonical isoforms at the bottom<a href="create-the-stacked-bar-plot-with-canonical-isoforms-at-the-bottom.html#create-the-stacked-bar-plot-with-canonical-isoforms-at-the-bottom" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>p1 &lt;- ggplot(proportion_data_long, aes(x = cell_type, y = proportion, fill = factor(isoform_type, levels = c(“non_canonical_proportion”, “canonical_proportion”)))) +
geom_bar(stat = “identity”, color = “black”, size = 0.3) + # Stacked bars with black outline
geom_text(aes(label = paste0(round(proportion * 100, 1), “%”)),
position = position_stack(vjust = 0.5), color = “white”, size = 3.5) + # Percentage labels in bar
scale_fill_manual(values = c(“canonical_proportion” = “grey”,
“non_canonical_proportion” = “black”),
name = “Isoform Type”, # Legend title
labels = c(“canonical_proportion” = “Canonical Isoforms”,
“non_canonical_proportion” = “Non-Canonical Isoforms”)) + # Legend labels
labs(title = “Proportion of Canonical vs Non-Canonical expression per Cell Type”,
x = “Cell Type”,
y = “Proportion”) +
theme_minimal(base_size = 14) + # Base font size
theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 12, face = “bold”), # Rotate and bold x-axis labels
axis.text.y = element_text(size = 12), # Y-axis label size
plot.title = element_text(hjust = 0.5, size = 16, face = “bold”), # Center and bold title
legend.position = “right”, # Show legend on the right
panel.grid.major = element_line(size = 0.2), # Subtle grid lines
panel.grid.minor = element_blank()) # Remove minor grid lines</p>
<p>p1</p>
<div id="section-2" class="section level20" number="33.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1">
<p class="heading"><span class="header-section-number">33.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="reshape-the-data-for-the-stacked-bar-plot.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="now-we-can-do-this-for-a-list-of-genes.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/plots_for_flames_v2_paper.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/plots_for_flames_v2_paper.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
