<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Exploring isoforms of interest | FLAMES Single-Cell RNA-seq Tutorial</title>
  <meta name="description" content="Chapter 6 Exploring isoforms of interest | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Exploring isoforms of interest | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Exploring isoforms of interest | FLAMES Single-Cell RNA-seq Tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer" />


<meta name="date" content="2025-01-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="finding-differentially-expressed-genes-and-isoforms.html"/>
<link rel="next" href="novel-isofroms---working.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single sample RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> Setup</a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> Removing sources of unwanted noise from the single cell dataset</a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> Add isoform counts to Seurat object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> Finding differentially expressed genes and isoforms</a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#de-genes-and-isofroms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isofroms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> Exploring isoforms of interest</a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html"><i class="fa fa-check"></i><b>7</b> Novel isofroms - Working</a>
<ul>
<li class="chapter" data-level="7.1" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>7.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="7.2" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#add-analysis"><i class="fa fa-check"></i><b>7.2</b> add analysis</a></li>
<li class="chapter" data-level="7.3" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#isoform-proportion-analysis"><i class="fa fa-check"></i><b>7.3</b> <strong>Isoform Proportion Analysis</strong></a></li>
<li class="chapter" data-level="7.4" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#calculate-the-proportion-of-novel-isoforms-relative-to-total-gene-expression."><i class="fa fa-check"></i><b>7.4</b> Calculate the proportion of novel isoforms relative to total gene expression.</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>8</b> Session info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FLAMES Single-Cell RNA-seq Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exploring-isoforms-of-interest" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Exploring isoforms of interest<a href="exploring-isoforms-of-interest.html#exploring-isoforms-of-interest" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>One of the most powerful aspects of long-read single-cell sequencing is its ability to profile isoform-specific information at single-cell resolution. This capability opens up numerous avenues for analysis. In our lab we are interested in exploring the role of RNA isofroms in neuronal differentiation. There are many examples in the literate of isofroms regurgitating this process. We will cover some very general analysis with this focus in mind. For users of long-read single-cell tools who are interested in exploring additional biological questions, we encourage you to share your suggestions and we can expand this section to include analytical approaches that may cover a wider range of applications.</p>
<div id="isoforms-expressed-per-gene" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Isoforms expressed per gene<a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>With long-read single-cell data, we have the ability to analyze all the isoforms expressed by a given gene. In our data we can see that most genes express more than one isoform.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="exploring-isoforms-of-interest.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lets aggeragte the expresstion data by cell type </span></span>
<span id="cb123-2"><a href="exploring-isoforms-of-interest.html#cb123-2" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">AggregateExpression</span>(</span>
<span id="cb123-3"><a href="exploring-isoforms-of-interest.html#cb123-3" aria-hidden="true" tabindex="-1"></a>  seu_obj, </span>
<span id="cb123-4"><a href="exploring-isoforms-of-interest.html#cb123-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">assays =</span> <span class="st">&quot;iso&quot;</span>, </span>
<span id="cb123-5"><a href="exploring-isoforms-of-interest.html#cb123-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">return.seurat =</span> <span class="cn">FALSE</span>,</span>
<span id="cb123-6"><a href="exploring-isoforms-of-interest.html#cb123-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span></span>
<span id="cb123-7"><a href="exploring-isoforms-of-interest.html#cb123-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb123-8"><a href="exploring-isoforms-of-interest.html#cb123-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-9"><a href="exploring-isoforms-of-interest.html#cb123-9" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(counts) <span class="ot">-&gt;</span> df</span>
<span id="cb123-10"><a href="exploring-isoforms-of-interest.html#cb123-10" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(df) <span class="ot">-&gt;</span> df<span class="sc">$</span>gene</span>
<span id="cb123-11"><a href="exploring-isoforms-of-interest.html#cb123-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-12"><a href="exploring-isoforms-of-interest.html#cb123-12" aria-hidden="true" tabindex="-1"></a><span class="co">#split transcript ids into gene and transcript id</span></span>
<span id="cb123-13"><a href="exploring-isoforms-of-interest.html#cb123-13" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">separate</span>(gene, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;transcript_id&quot;</span>, <span class="st">&quot;gene_id&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;-&quot;</span>,  <span class="at">extra =</span> <span class="st">&quot;merge&quot;</span>) </span>
<span id="cb123-14"><a href="exploring-isoforms-of-interest.html#cb123-14" aria-hidden="true" tabindex="-1"></a><span class="co">#df$transcript_id &lt;- sub(&quot;\\..*&quot;, &quot;&quot;, df$transcript_id)</span></span>
<span id="cb123-15"><a href="exploring-isoforms-of-interest.html#cb123-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-16"><a href="exploring-isoforms-of-interest.html#cb123-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Count the number of isoforms per gene</span></span>
<span id="cb123-17"><a href="exploring-isoforms-of-interest.html#cb123-17" aria-hidden="true" tabindex="-1"></a>isoform_count_per_gene <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb123-18"><a href="exploring-isoforms-of-interest.html#cb123-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb123-19"><a href="exploring-isoforms-of-interest.html#cb123-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_isoforms =</span> <span class="fu">n_distinct</span>(transcript_id))</span>
<span id="cb123-20"><a href="exploring-isoforms-of-interest.html#cb123-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-21"><a href="exploring-isoforms-of-interest.html#cb123-21" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. count isforms per category </span></span>
<span id="cb123-22"><a href="exploring-isoforms-of-interest.html#cb123-22" aria-hidden="true" tabindex="-1"></a>isoform_count_per_gene <span class="ot">&lt;-</span> isoform_count_per_gene <span class="sc">%&gt;%</span></span>
<span id="cb123-23"><a href="exploring-isoforms-of-interest.html#cb123-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">isoform_category =</span> <span class="fu">case_when</span>(</span>
<span id="cb123-24"><a href="exploring-isoforms-of-interest.html#cb123-24" aria-hidden="true" tabindex="-1"></a>    n_isoforms <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">&quot;1&quot;</span>,</span>
<span id="cb123-25"><a href="exploring-isoforms-of-interest.html#cb123-25" aria-hidden="true" tabindex="-1"></a>    n_isoforms <span class="sc">&gt;=</span> <span class="dv">2</span> <span class="sc">&amp;</span> n_isoforms <span class="sc">&lt;=</span> <span class="dv">3</span> <span class="sc">~</span> <span class="st">&quot;2-3&quot;</span>,</span>
<span id="cb123-26"><a href="exploring-isoforms-of-interest.html#cb123-26" aria-hidden="true" tabindex="-1"></a>    n_isoforms <span class="sc">&gt;=</span> <span class="dv">4</span> <span class="sc">&amp;</span> n_isoforms <span class="sc">&lt;=</span> <span class="dv">5</span> <span class="sc">~</span> <span class="st">&quot;4-5&quot;</span>,</span>
<span id="cb123-27"><a href="exploring-isoforms-of-interest.html#cb123-27" aria-hidden="true" tabindex="-1"></a>    n_isoforms <span class="sc">&gt;=</span> <span class="dv">6</span> <span class="sc">~</span> <span class="st">&quot;6+&quot;</span></span>
<span id="cb123-28"><a href="exploring-isoforms-of-interest.html#cb123-28" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb123-29"><a href="exploring-isoforms-of-interest.html#cb123-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-30"><a href="exploring-isoforms-of-interest.html#cb123-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-31"><a href="exploring-isoforms-of-interest.html#cb123-31" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. Calculate the percentage of genes in each bin</span></span>
<span id="cb123-32"><a href="exploring-isoforms-of-interest.html#cb123-32" aria-hidden="true" tabindex="-1"></a>isoform_count_summary <span class="ot">&lt;-</span> isoform_count_per_gene <span class="sc">%&gt;%</span></span>
<span id="cb123-33"><a href="exploring-isoforms-of-interest.html#cb123-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(isoform_category) <span class="sc">%&gt;%</span></span>
<span id="cb123-34"><a href="exploring-isoforms-of-interest.html#cb123-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> (n <span class="sc">/</span> <span class="fu">sum</span>(n)) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb123-35"><a href="exploring-isoforms-of-interest.html#cb123-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-36"><a href="exploring-isoforms-of-interest.html#cb123-36" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(isoform_count_summary, <span class="fu">aes</span>(<span class="at">x =</span> isoform_category, <span class="at">y =</span> percent)) <span class="sc">+</span></span>
<span id="cb123-37"><a href="exploring-isoforms-of-interest.html#cb123-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;lightblue&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb123-38"><a href="exploring-isoforms-of-interest.html#cb123-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Number of Isoforms per Gene&quot;</span>, </span>
<span id="cb123-39"><a href="exploring-isoforms-of-interest.html#cb123-39" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">&quot;Isoforms per Gene&quot;</span>, </span>
<span id="cb123-40"><a href="exploring-isoforms-of-interest.html#cb123-40" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Genes, %&quot;</span>) <span class="sc">+</span></span>
<span id="cb123-41"><a href="exploring-isoforms-of-interest.html#cb123-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb123-42"><a href="exploring-isoforms-of-interest.html#cb123-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>, </span>
<span id="cb123-43"><a href="exploring-isoforms-of-interest.html#cb123-43" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
</div>
<div id="top-10-genes-with-most-isoforms" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Top 10 Genes with Most Isoforms<a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can see that about 45% of genes express a single isoform, however there are some genes like MIR9-1HG that have alot of unique isofroms, 97 in fact. The top 10 genes with the most isofroms are listed bellow.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="exploring-isoforms-of-interest.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes ranked by the number of transcript isoforms detected across all samples </span></span>
<span id="cb124-2"><a href="exploring-isoforms-of-interest.html#cb124-2" aria-hidden="true" tabindex="-1"></a>gene_transcript_counts <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb124-3"><a href="exploring-isoforms-of-interest.html#cb124-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb124-4"><a href="exploring-isoforms-of-interest.html#cb124-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">unique_transcripts =</span> <span class="fu">n_distinct</span>(transcript_id)) <span class="sc">%&gt;%</span></span>
<span id="cb124-5"><a href="exploring-isoforms-of-interest.html#cb124-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(unique_transcripts))</span>
<span id="cb124-6"><a href="exploring-isoforms-of-interest.html#cb124-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-7"><a href="exploring-isoforms-of-interest.html#cb124-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the top 10 genes based on unique transcript counts</span></span>
<span id="cb124-8"><a href="exploring-isoforms-of-interest.html#cb124-8" aria-hidden="true" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> gene_transcript_counts <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">10</span>, unique_transcripts)</span>
<span id="cb124-9"><a href="exploring-isoforms-of-interest.html#cb124-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-10"><a href="exploring-isoforms-of-interest.html#cb124-10" aria-hidden="true" tabindex="-1"></a>top10</span></code></pre></div>
</details>
<pre><code>## # A tibble: 10 × 2
##    gene_id         unique_transcripts
##    &lt;chr&gt;                        &lt;int&gt;
##  1 MIR9-1HG                        97
##  2 GAS5                            69
##  3 NUTM2A-AS1                      50
##  4 SNHG1                           50
##  5 FRG1HP                          43
##  6 TMEM161B-DT                     43
##  7 SNHG29                          42
##  8 ENSG00000300022                 41
##  9 FAM66A                          39
## 10 SNHG14                          38</code></pre>
<p>We can also plot unique transcripts per gene on a log scale showing that the number of isofroms per gene varies alot across out data.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="exploring-isoforms-of-interest.html#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot ranked genes by unique &quot;BambuTx&quot; transcript count</span></span>
<span id="cb126-2"><a href="exploring-isoforms-of-interest.html#cb126-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(gene_transcript_counts, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">rank</span>(<span class="fu">desc</span>(unique_transcripts)), <span class="at">y =</span> unique_transcripts)) <span class="sc">+</span></span>
<span id="cb126-3"><a href="exploring-isoforms-of-interest.html#cb126-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">&quot;darkblue&quot;</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span>  <span class="co"># Points for each gene</span></span>
<span id="cb126-4"><a href="exploring-isoforms-of-interest.html#cb126-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-5"><a href="exploring-isoforms-of-interest.html#cb126-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Log scale for both axes</span></span>
<span id="cb126-6"><a href="exploring-isoforms-of-interest.html#cb126-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb126-7"><a href="exploring-isoforms-of-interest.html#cb126-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb126-8"><a href="exploring-isoforms-of-interest.html#cb126-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb126-9"><a href="exploring-isoforms-of-interest.html#cb126-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Title and labels</span></span>
<span id="cb126-10"><a href="exploring-isoforms-of-interest.html#cb126-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb126-11"><a href="exploring-isoforms-of-interest.html#cb126-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Unique Transcripts per Gene&quot;</span>,</span>
<span id="cb126-12"><a href="exploring-isoforms-of-interest.html#cb126-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Rank (log scale)&quot;</span>,</span>
<span id="cb126-13"><a href="exploring-isoforms-of-interest.html#cb126-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;# Transcripts (log scale)&quot;</span></span>
<span id="cb126-14"><a href="exploring-isoforms-of-interest.html#cb126-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb126-15"><a href="exploring-isoforms-of-interest.html#cb126-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb126-16"><a href="exploring-isoforms-of-interest.html#cb126-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Highlight and label the top 10 genes with gray background and black border around the text</span></span>
<span id="cb126-17"><a href="exploring-isoforms-of-interest.html#cb126-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(</span>
<span id="cb126-18"><a href="exploring-isoforms-of-interest.html#cb126-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> gene_transcript_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_id <span class="sc">%in%</span> top10<span class="sc">$</span>gene_id),</span>
<span id="cb126-19"><a href="exploring-isoforms-of-interest.html#cb126-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">label =</span> gene_id),</span>
<span id="cb126-20"><a href="exploring-isoforms-of-interest.html#cb126-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>,          <span class="co"># Gray background for the label</span></span>
<span id="cb126-21"><a href="exploring-isoforms-of-interest.html#cb126-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>,         <span class="co"># Black text color</span></span>
<span id="cb126-22"><a href="exploring-isoforms-of-interest.html#cb126-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.size =</span> <span class="fl">0.25</span>,       <span class="co"># Border thickness around the label</span></span>
<span id="cb126-23"><a href="exploring-isoforms-of-interest.html#cb126-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.r =</span> <span class="fu">unit</span>(<span class="fl">0.15</span>, <span class="st">&quot;lines&quot;</span>),  <span class="co"># Border radius (rounded corners)</span></span>
<span id="cb126-24"><a href="exploring-isoforms-of-interest.html#cb126-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb126-25"><a href="exploring-isoforms-of-interest.html#cb126-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">box.padding =</span> <span class="fl">0.2</span>,</span>
<span id="cb126-26"><a href="exploring-isoforms-of-interest.html#cb126-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">max.overlaps =</span> <span class="dv">14</span> </span>
<span id="cb126-27"><a href="exploring-isoforms-of-interest.html#cb126-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb126-28"><a href="exploring-isoforms-of-interest.html#cb126-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb126-29"><a href="exploring-isoforms-of-interest.html#cb126-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Minimal theme and additional styling</span></span>
<span id="cb126-30"><a href="exploring-isoforms-of-interest.html#cb126-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb126-31"><a href="exploring-isoforms-of-interest.html#cb126-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb126-32"><a href="exploring-isoforms-of-interest.html#cb126-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>), <span class="co"># Centered title</span></span>
<span id="cb126-33"><a href="exploring-isoforms-of-interest.html#cb126-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>),             <span class="co"># Black axis tick labels</span></span>
<span id="cb126-34"><a href="exploring-isoforms-of-interest.html#cb126-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>),                       <span class="co"># Black axis titles</span></span>
<span id="cb126-35"><a href="exploring-isoforms-of-interest.html#cb126-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb126-36"><a href="exploring-isoforms-of-interest.html#cb126-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">linewidth =</span> <span class="dv">1</span>)  <span class="co"># Black border around the graph</span></span>
<span id="cb126-37"><a href="exploring-isoforms-of-interest.html#cb126-37" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot_unique_isoforms-1.png" width="672" /></p>
</div>
<div id="exploring-macf1-isoforms" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Exploring MACF1 isoforms<a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are interested in isoforms that regulate neuronal differentiation, we can look at some genes of interest. Lets look at gene <em>MACF1</em><strong>.</strong> This gene is know to…. [ref]. The gene seems to play some important role in neural migration which is not fully understood yet. First lets try and visualize the expression of these isofroms on a UMAP to see if we can uncover anything interesting. <em>MACF1</em> has 35 expressed isofroms so lets only plot the top 12 most highly expressed.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="exploring-isoforms-of-interest.html#cb127-1" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">rownames</span>(filt_seurat_object<span class="sc">@</span>assays<span class="sc">$</span>iso<span class="sc">@</span>features)</span>
<span id="cb127-2"><a href="exploring-isoforms-of-interest.html#cb127-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-3"><a href="exploring-isoforms-of-interest.html#cb127-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the gene of interest</span></span>
<span id="cb127-4"><a href="exploring-isoforms-of-interest.html#cb127-4" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="st">&quot;MACF1&quot;</span></span>
<span id="cb127-5"><a href="exploring-isoforms-of-interest.html#cb127-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-6"><a href="exploring-isoforms-of-interest.html#cb127-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Access the data matrix for the &#39;iso&#39; assay</span></span>
<span id="cb127-7"><a href="exploring-isoforms-of-interest.html#cb127-7" aria-hidden="true" tabindex="-1"></a>expression_matrix <span class="ot">&lt;-</span> <span class="fu">GetAssayData</span>(filt_seurat_object, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">slot =</span> <span class="st">&quot;data&quot;</span>)</span>
<span id="cb127-8"><a href="exploring-isoforms-of-interest.html#cb127-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-9"><a href="exploring-isoforms-of-interest.html#cb127-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter features containing the gene name</span></span>
<span id="cb127-10"><a href="exploring-isoforms-of-interest.html#cb127-10" aria-hidden="true" tabindex="-1"></a>matching_features <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">&quot;(^|-|</span><span class="sc">\\</span><span class="st">b)&quot;</span>, gene, <span class="st">&quot;($|</span><span class="sc">\\</span><span class="st">b)&quot;</span>), <span class="fu">rownames</span>(expression_matrix), <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb127-11"><a href="exploring-isoforms-of-interest.html#cb127-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-12"><a href="exploring-isoforms-of-interest.html#cb127-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the expression matrix to include only the matching features</span></span>
<span id="cb127-13"><a href="exploring-isoforms-of-interest.html#cb127-13" aria-hidden="true" tabindex="-1"></a>subset_expression <span class="ot">&lt;-</span> expression_matrix[matching_features, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb127-14"><a href="exploring-isoforms-of-interest.html#cb127-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-15"><a href="exploring-isoforms-of-interest.html#cb127-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the total expression for each matching feature</span></span>
<span id="cb127-16"><a href="exploring-isoforms-of-interest.html#cb127-16" aria-hidden="true" tabindex="-1"></a>total_expression <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">rowSums</span>(subset_expression)</span>
<span id="cb127-17"><a href="exploring-isoforms-of-interest.html#cb127-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-18"><a href="exploring-isoforms-of-interest.html#cb127-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank features by average expression</span></span>
<span id="cb127-19"><a href="exploring-isoforms-of-interest.html#cb127-19" aria-hidden="true" tabindex="-1"></a>top_features <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(total_expression, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))</span>
<span id="cb127-20"><a href="exploring-isoforms-of-interest.html#cb127-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-21"><a href="exploring-isoforms-of-interest.html#cb127-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the ranked features (optional)</span></span>
<span id="cb127-22"><a href="exploring-isoforms-of-interest.html#cb127-22" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">data.frame</span>(<span class="at">Feature =</span> top_features, <span class="at">Expression =</span> total_expression[top_features]))</span></code></pre></div>
</details>
<pre><code>##                                         Feature Expression
## ENST00000361689.7-MACF1 ENST00000361689.7-MACF1 143.862407
## ENST00000289893.8-MACF1 ENST00000289893.8-MACF1  62.133601
## ENST00000372925.6-MACF1 ENST00000372925.6-MACF1  45.932634
## ENST00000372915.8-MACF1 ENST00000372915.8-MACF1  45.461522
## ENST00000567887.5-MACF1 ENST00000567887.5-MACF1  45.092944
## ENST00000686657.1-MACF1 ENST00000686657.1-MACF1  44.609476
## ENST00000564288.6-MACF1 ENST00000564288.6-MACF1  40.689721
## ENST00000524432.5-MACF1 ENST00000524432.5-MACF1  27.310250
## ENST00000686687.1-MACF1 ENST00000686687.1-MACF1  19.939477
## ENST00000497964.1-MACF1 ENST00000497964.1-MACF1  17.945311
## ENST00000530275.3-MACF1 ENST00000530275.3-MACF1  16.728830
## ENST00000691623.1-MACF1 ENST00000691623.1-MACF1  14.286232
## ENST00000602528.2-MACF1 ENST00000602528.2-MACF1  13.942571
## ENST00000687271.1-MACF1 ENST00000687271.1-MACF1  12.934634
## ENST00000476350.1-MACF1 ENST00000476350.1-MACF1  12.210176
## ENST00000496804.5-MACF1 ENST00000496804.5-MACF1  12.181141
## ENST00000686067.1-MACF1 ENST00000686067.1-MACF1  10.238947
## ENST00000690080.1-MACF1 ENST00000690080.1-MACF1   8.279051
## ENST00000497807.1-MACF1 ENST00000497807.1-MACF1   8.021857
## ENST00000693209.1-MACF1 ENST00000693209.1-MACF1   7.429056
## ENST00000693392.1-MACF1 ENST00000693392.1-MACF1   7.317640
## ENST00000472385.2-MACF1 ENST00000472385.2-MACF1   7.075353
## ENST00000602421.5-MACF1 ENST00000602421.5-MACF1   6.698620
## ENST00000686260.1-MACF1 ENST00000686260.1-MACF1   6.297806
## ENST00000688426.1-MACF1 ENST00000688426.1-MACF1   6.232290
## ENST00000687997.1-MACF1 ENST00000687997.1-MACF1   6.056226
## ENST00000690939.1-MACF1 ENST00000690939.1-MACF1   5.445340
## ENST00000683517.1-MACF1 ENST00000683517.1-MACF1   3.526815
## ENST00000442046.5-MACF1 ENST00000442046.5-MACF1   3.018726
## ENST00000689911.1-MACF1 ENST00000689911.1-MACF1   2.657853
## ENST00000484793.5-MACF1 ENST00000484793.5-MACF1   2.181164
## ENST00000467673.5-MACF1 ENST00000467673.5-MACF1   2.131094
## ENST00000686941.1-MACF1 ENST00000686941.1-MACF1   1.058887
## ENST00000672812.1-MACF1 ENST00000672812.1-MACF1   1.055652
## ENST00000528611.1-MACF1 ENST00000528611.1-MACF1   0.401423</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="exploring-isoforms-of-interest.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">repr.plot.width=</span><span class="dv">12</span>, <span class="at">repr.plot.height=</span><span class="dv">12</span>)</span>
<span id="cb129-2"><a href="exploring-isoforms-of-interest.html#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the top 16 features in descending order of their average expression</span></span>
<span id="cb129-3"><a href="exploring-isoforms-of-interest.html#cb129-3" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(</span>
<span id="cb129-4"><a href="exploring-isoforms-of-interest.html#cb129-4" aria-hidden="true" tabindex="-1"></a>  filt_seurat_object,</span>
<span id="cb129-5"><a href="exploring-isoforms-of-interest.html#cb129-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">head</span>(top_features, <span class="dv">12</span>),</span>
<span id="cb129-6"><a href="exploring-isoforms-of-interest.html#cb129-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb129-7"><a href="exploring-isoforms-of-interest.html#cb129-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">order =</span> <span class="cn">TRUE</span>, <span class="co"># Ensures higher-expressing cells are plotted on top</span></span>
<span id="cb129-8"><a href="exploring-isoforms-of-interest.html#cb129-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pt.size =</span> <span class="dv">1</span>)</span>
<span id="cb129-9"><a href="exploring-isoforms-of-interest.html#cb129-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-10"><a href="exploring-isoforms-of-interest.html#cb129-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust title size for each plot</span></span>
<span id="cb129-11"><a href="exploring-isoforms-of-interest.html#cb129-11" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(plots, <span class="cf">function</span>(plot) {</span>
<span id="cb129-12"><a href="exploring-isoforms-of-interest.html#cb129-12" aria-hidden="true" tabindex="-1"></a>  plot <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>))</span>
<span id="cb129-13"><a href="exploring-isoforms-of-interest.html#cb129-13" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb129-14"><a href="exploring-isoforms-of-interest.html#cb129-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-15"><a href="exploring-isoforms-of-interest.html#cb129-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the adjusted plots</span></span>
<span id="cb129-16"><a href="exploring-isoforms-of-interest.html#cb129-16" aria-hidden="true" tabindex="-1"></a><span class="fu">CombinePlots</span>(<span class="at">plots =</span> plots, <span class="at">ncol =</span> <span class="dv">3</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plotting%20MACF1-1.png" width="960" /></p>
</div>
<div id="expression-of-mcaf1-isoforms-across-cell-types" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Expression of MCAF1 isoforms Across Cell Types<a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lets look at isofroms ENST00000564288.6 ENST00000361689.7 , ENST00000289893.8 and ENST00000524432.5 in some more detail and plot the normalized expression of these isoforms across each cell type. We can see that the expression of ENST00000524432.5shows a cell type specific profile.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="exploring-isoforms-of-interest.html#cb130-1" aria-hidden="true" tabindex="-1"></a>features_MACF1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ENST00000564288.6-MACF1&quot;</span>, <span class="co"># cononical</span></span>
<span id="cb130-2"><a href="exploring-isoforms-of-interest.html#cb130-2" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;ENST00000361689.7-MACF1&quot;</span>, <span class="co"># most cell types</span></span>
<span id="cb130-3"><a href="exploring-isoforms-of-interest.html#cb130-3" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;ENST00000289893.8-MACF1&quot;</span>, <span class="co"># msot cell types</span></span>
<span id="cb130-4"><a href="exploring-isoforms-of-interest.html#cb130-4" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;ENST00000524432.5-MACF1&quot;</span>) <span class="co"># radial glia</span></span>
<span id="cb130-5"><a href="exploring-isoforms-of-interest.html#cb130-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-6"><a href="exploring-isoforms-of-interest.html#cb130-6" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seu_obj, <span class="at">features =</span> features_MACF1, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/vln%20MACF1-1.png" width="960" />
We can also show this with A dotplot.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="exploring-isoforms-of-interest.html#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoDotPlot</span>(seu_obj, <span class="at">vars =</span> features_MACF1, <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/dot%20MACF1-1.png" width="672" /></p>
<p>lets look at our DE results comparing Glutamatergic neurons and Radial glia cells that we calcualted in the previous chapter and filter for significant MCAF1 isoforms. If we plot these features on a Volcano plotwe see isoform <strong>ENST00000524432.5</strong> is enriched in radial glia cells. In fact its enrichment compared to Glutamatergic neurons is pretty high with a Log2fold change of 4.16.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="exploring-isoforms-of-interest.html#cb132-1" aria-hidden="true" tabindex="-1"></a>glu_RG_iso <span class="sc">%&gt;%</span></span>
<span id="cb132-2"><a href="exploring-isoforms-of-interest.html#cb132-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;isoform&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb132-3"><a href="exploring-isoforms-of-interest.html#cb132-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;MACF1&quot;</span>, isoform)) <span class="sc">%&gt;%</span></span>
<span id="cb132-4"><a href="exploring-isoforms-of-interest.html#cb132-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.5</span>) </span></code></pre></div>
</details>
<pre><code>##                   isoform        p_val avg_log2FC pct.1 pct.2    p_val_adj
## 1 ENST00000524432.5-MACF1 1.025842e-10  -4.163237 0.011 0.384 6.539231e-06</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="exploring-isoforms-of-interest.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(glu_RG_iso, <span class="at">lab=</span><span class="fu">rownames</span>(glu_RG_iso),</span>
<span id="cb134-2"><a href="exploring-isoforms-of-interest.html#cb134-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">x=</span><span class="st">&#39;avg_log2FC&#39;</span>, <span class="at">y=</span><span class="st">&#39;p_val_adj&#39;</span>,</span>
<span id="cb134-3"><a href="exploring-isoforms-of-interest.html#cb134-3" aria-hidden="true" tabindex="-1"></a>                <span class="co">#selectLab= &quot;VIM&quot;,</span></span>
<span id="cb134-4"><a href="exploring-isoforms-of-interest.html#cb134-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">pCutoff=</span><span class="fl">0.05</span>, <span class="at">FCcutoff=</span><span class="dv">2</span>,</span>
<span id="cb134-5"><a href="exploring-isoforms-of-interest.html#cb134-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">selectLab =</span> <span class="st">&quot;ENST00000524432.5-MACF1&quot;</span>,</span>
<span id="cb134-6"><a href="exploring-isoforms-of-interest.html#cb134-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">boxedLabels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb134-7"><a href="exploring-isoforms-of-interest.html#cb134-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb134-8"><a href="exploring-isoforms-of-interest.html#cb134-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;Volcano Plot showing ENST00000524432.5-MACF1 isoforms </span><span class="sc">\n</span><span class="st"> Glutamatergic Neurons vs Radial glial Cells&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/DE_MACF1-1.png" width="672" /></p>
</div>
<div id="visualization-of-isoform-structures" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Visualization of Isoform Structures<a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we know some MCAF1 isoforms expression is significantly different in these two cell populations it may be of interest to visualize the isoform structures. This analysis will help us explore the similarities and differences between our isoforms of interest.</p>
<p>There are many visualization options available to us and many of these are available in R. In fact FLAMES has its own visualization function <code>FLAMES::plot_isoform_reduced_dim</code> . This function is designed to work on single cell experiment object and not Seurat object. Although it is possible to switch between these formats, for the purpose of this tutorial we want to keep file conversations to a minimum to keep the analysis simple.</p>
<p>We instead recommend using IsoViz <span class="citation">(<a href="#ref-wan2024" role="doc-biblioref">Wan et al., 2024</a>)</span>, which was developed in the Clark Lab. The tool is a web application specifically designed for visualizing isoform structures. This visualization can provide valuable insights into the potential functions of different isoforms.</p>
<p>First lets prepare the count data that we will load into Isoviz<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="exploring-isoforms-of-interest.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co">#extract some isoform expression data to visualize in isoviz </span></span>
<span id="cb135-2"><a href="exploring-isoforms-of-interest.html#cb135-2" aria-hidden="true" tabindex="-1"></a><span class="co">#use pseudobulk counts we from above  </span></span>
<span id="cb135-3"><a href="exploring-isoforms-of-interest.html#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(df) <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb135-4"><a href="exploring-isoforms-of-interest.html#cb135-4" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb135-5"><a href="exploring-isoforms-of-interest.html#cb135-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-6"><a href="exploring-isoforms-of-interest.html#cb135-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="st">&quot;output_files/psedubulk_exp.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
<p>To use IsoViz click on the following link <a href="https://isomix.org/isovis/" class="uri">https://isomix.org/isovis/</a> and uplaod the <code>isofrom_annotated.gtf</code> file located in the FLAMES output dir and <code>psedubulk_exp.csv</code> generated above For more detail on how to use Isoviz click the ‘IsoViz tutorial’ button or read the publication.</p>
<p>Embedded bellow is the figure generated by Isoviz. When Viewing this data in the web broswer users will have more functionality. (eg looking a protein domains or clicking on ENST ids and being linked to ensemble). Here we and visualizing our 4 isofroms of interest. We can see MCAF1 is a very complex gene with many exons, a variety of alternative transcription start sites and ORFs.</p>
<p>Our pseudobulk expression analysis clearly demonstrates that <strong>ENST00000524432.5</strong> is predominantly expressed in radial glia cells. This isoform is particularly interesting, as it is significantly shorter than others and lacks many of the protein domains present in <strong>ENST00000564288.6</strong>, the canonical isoform. Notably, all four isoforms exhibit different transcription start sites (TSS), suggesting that TSS variation may be linked to cell-type-specific expression or distinct protein functions.</p>
<p>Additionally, <strong>ENST00000289893.8</strong> shows comparably high expression levels across most cell types. However, a deeper exploration of this isoform reveals that it does not produce a functional protein. This is evident from examining the <strong>Ensembl</strong> data, accessible via the hyperlink, where we can see that no open reading frames (ORFs) are associated with this transcript.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="exploring-isoforms-of-interest.html#cb136-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/IsoVis_ENSG00000127603.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:isoviz-plot"></span>
<img src="images/IsoVis_ENSG00000127603.png" alt="IsoViz visualization of 4 MACF1 isoforms." width="1600px" />
<p class="caption">
Figure 6.1: IsoViz visualization of 4 MACF1 isoforms.
</p>
</div>
<p>Structural visualization aids in identifying critical variations such as alternative splicing events, unique protein-coding regions, and functional domains and severs as a powerful tool to infer isoform function. There are lots of additional analysis that could be performed to further explore the function of our <em>MCAF1</em> isofroms, these include domain enrichment analysis and protein folding to name a few.</p>
<p><strong>Idea to add Splice Variant Impact</strong><br />
Analyze how alternative splicing affects protein structure and function using tools like <strong>PROSITE</strong> or structural prediction algorithms.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" line-spacing="2">
<div id="ref-wan2024" class="csl-entry">
Wan, C. Y., Davis, J., Chauhan, M., Gleeson, J., Prawer, Y. J., De Paoli-Iseppi, R., Wells, C., Choi, J., &amp; Clark, M. (2024). IsoVis <span></span> a webserver for visualization and annotation of alternative RNA isoforms. <em>Nucleic Acids Research</em>, <em>52</em>(W1), W341–W347. <a href="https://doi.org/10.1093/nar/gkae343">https://doi.org/10.1093/nar/gkae343</a>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>Please be careful when interpreting pseudubulk expression data. Although the data can give you some indication of relative expression across cell types, this numbers can be affected by the number of cells in each cluster.<a href="exploring-isoforms-of-interest.html#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="finding-differentially-expressed-genes-and-isoforms.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="novel-isofroms---working.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/06_Finding-isoforms-of-interest.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/06_Finding-isoforms-of-interest.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
