<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Finding marker genes and isoforms | Long-read Single-Cell RNA-seq analysis tutorial</title>
  <meta name="description" content="Chapter 5 Finding marker genes and isoforms | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Finding marker genes and isoforms | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Finding marker genes and isoforms | Long-read Single-Cell RNA-seq analysis tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer Postdoctoral Research Fellow in Long Reads and Single-Cell Transcriptomics University of Melbourne" />


<meta name="date" content="2025-10-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="add-isoform-counts-to-seurat-object.html"/>
<link rel="next" href="exploring-isoforms-of-interest.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualizing-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualizing OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>9</b> <strong>Multisample analysis</strong></a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>9.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="9.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>9.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>9.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multi-sample-integration"><i class="fa fa-check"></i><b>9.2</b> Multi-Sample integration</a></li>
<li class="chapter" data-level="9.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>9.3</b> Add isoform counts</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Trajectory analysis</strong></a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i><b>10.1</b> Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="10.2" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i><b>10.2</b> Multisample QC function</a></li>
<li class="chapter" data-level="10.3" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-count-matrix"><i class="fa fa-check"></i><b>10.3</b> Convert Oarfish files to count matrix</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="finding-marker-genes-and-isoforms" class="section level1 hasAnchor" number="5" style="gray">
<h1><span class="header-section-number">Chapter 5</span> <strong>Finding marker genes and isoforms</strong><a href="finding-marker-genes-and-isoforms.html#finding-marker-genes-and-isoforms" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<hr />
<div id="differentially-expressed-genes-by-cluster-identity" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Differentially expressed genes by cluster identity<a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First we can look at marker genes for each cluster. This will help us identify which genes are DE in each cluster and indicate the identity of each cluster. We will also look at DE isoforms using the same methodology.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="finding-marker-genes-and-isoforms.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Find markers for all clusters using the &quot;RNA&quot; and &quot;iso&quot; assay</span></span>
<span id="cb67-2"><a href="finding-marker-genes-and-isoforms.html#cb67-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-3"><a href="finding-marker-genes-and-isoforms.html#cb67-3" aria-hidden="true" tabindex="-1"></a>all_markers_gene_cluster <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(seu_obj, <span class="at">assay =</span> <span class="st">&quot;RNA&quot;</span>, <span class="at">do.print =</span> <span class="cn">TRUE</span>,</span>
<span id="cb67-4"><a href="finding-marker-genes-and-isoforms.html#cb67-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>, <span class="at">min.pct =</span> <span class="fl">0.20</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb67-5"><a href="finding-marker-genes-and-isoforms.html#cb67-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-6"><a href="finding-marker-genes-and-isoforms.html#cb67-6" aria-hidden="true" tabindex="-1"></a>all_markers_iso_cluster <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(seu_obj, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">do.print =</span> <span class="cn">TRUE</span>,</span>
<span id="cb67-7"><a href="finding-marker-genes-and-isoforms.html#cb67-7" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>, <span class="at">min.pct =</span> <span class="fl">0.20</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb67-8"><a href="finding-marker-genes-and-isoforms.html#cb67-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-9"><a href="finding-marker-genes-and-isoforms.html#cb67-9" aria-hidden="true" tabindex="-1"></a><span class="co">#save the list of DE genes </span></span>
<span id="cb67-10"><a href="finding-marker-genes-and-isoforms.html#cb67-10" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(all_markers_gene_cluster, <span class="st">&quot;./output_files/DE/all_markers_one_gene.csv&quot;</span>)</span>
<span id="cb67-11"><a href="finding-marker-genes-and-isoforms.html#cb67-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(all_markers_iso_cluster, <span class="st">&quot;./output_files/DE/all_markers_one_iso.csv&quot;</span>)</span></code></pre></div>
</details>
</div>
<div id="identifying-cell-types" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Identifying cell types<a href="finding-marker-genes-and-isoforms.html#identifying-cell-types" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Based on these differentially expressed (DE) genes, we can identify the cell types present in our sample. This process is often complex and requires prior knowledge of cell markers as well as an understanding of the cell types expected in the sample. An alternative approach is to use automated cell type identification tools. In this tutorial, we will use <strong>scType</strong> [ref]. However, it is important to note that the accuracy of automated tools varies and depends heavily on the reference database they utilize. Therefore, it is recommended to use a combination of methods to cross-validate cell type identification and ensure robust results.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="finding-marker-genes-and-isoforms.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load libraries from sctype</span></span>
<span id="cb68-2"><a href="finding-marker-genes-and-isoforms.html#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">&quot;ggraph&quot;</span>,<span class="st">&quot;igraph&quot;</span>,<span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;data.tree&quot;</span>), library, <span class="at">character.only =</span> T))</span>
<span id="cb68-3"><a href="finding-marker-genes-and-isoforms.html#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">&quot;dplyr&quot;</span>,<span class="st">&quot;Seurat&quot;</span>,<span class="st">&quot;HGNChelper&quot;</span>), library, <span class="at">character.only =</span> T))</span>
<span id="cb68-4"><a href="finding-marker-genes-and-isoforms.html#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="finding-marker-genes-and-isoforms.html#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="co"># load gene set preparation function</span></span>
<span id="cb68-6"><a href="finding-marker-genes-and-isoforms.html#cb68-6" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R&quot;</span>)</span>
<span id="cb68-7"><a href="finding-marker-genes-and-isoforms.html#cb68-7" aria-hidden="true" tabindex="-1"></a><span class="co"># load cell type annotation function</span></span>
<span id="cb68-8"><a href="finding-marker-genes-and-isoforms.html#cb68-8" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R&quot;</span>)</span>
<span id="cb68-9"><a href="finding-marker-genes-and-isoforms.html#cb68-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-10"><a href="finding-marker-genes-and-isoforms.html#cb68-10" aria-hidden="true" tabindex="-1"></a><span class="do">####</span></span>
<span id="cb68-11"><a href="finding-marker-genes-and-isoforms.html#cb68-11" aria-hidden="true" tabindex="-1"></a><span class="co"># define functions</span></span>
<span id="cb68-12"><a href="finding-marker-genes-and-isoforms.html#cb68-12" aria-hidden="true" tabindex="-1"></a>perform_sctype_analysis <span class="ot">&lt;-</span> <span class="cf">function</span>(seurat_obj, db_, tissue, <span class="at">gs_removal_list =</span> <span class="fu">c</span>(), </span>
<span id="cb68-13"><a href="finding-marker-genes-and-isoforms.html#cb68-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">metadat_col_prefix =</span> <span class="st">&quot;db_prefix&quot;</span>, <span class="at">figure_prefix =</span><span class="st">&quot;fig_name&quot;</span>, <span class="at">cluster_res =</span> <span class="st">&quot;RNA_snn_res.0.9&quot;</span>, <span class="at">output_file =</span> <span class="st">&quot;&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>) {</span>
<span id="cb68-14"><a href="finding-marker-genes-and-isoforms.html#cb68-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-15"><a href="finding-marker-genes-and-isoforms.html#cb68-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Prepare gene sets</span></span>
<span id="cb68-16"><a href="finding-marker-genes-and-isoforms.html#cb68-16" aria-hidden="true" tabindex="-1"></a>  gs_list <span class="ot">&lt;-</span> <span class="fu">gene_sets_prepare</span>(db_, tissue)</span>
<span id="cb68-17"><a href="finding-marker-genes-and-isoforms.html#cb68-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-18"><a href="finding-marker-genes-and-isoforms.html#cb68-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove specified gene sets</span></span>
<span id="cb68-19"><a href="finding-marker-genes-and-isoforms.html#cb68-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (gs <span class="cf">in</span> gs_removal_list) {</span>
<span id="cb68-20"><a href="finding-marker-genes-and-isoforms.html#cb68-20" aria-hidden="true" tabindex="-1"></a>    gs_list[[<span class="st">&quot;gs_positive&quot;</span>]][[gs]] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb68-21"><a href="finding-marker-genes-and-isoforms.html#cb68-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb68-22"><a href="finding-marker-genes-and-isoforms.html#cb68-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-23"><a href="finding-marker-genes-and-isoforms.html#cb68-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate sctype scores</span></span>
<span id="cb68-24"><a href="finding-marker-genes-and-isoforms.html#cb68-24" aria-hidden="true" tabindex="-1"></a>  es.max <span class="ot">&lt;-</span> <span class="fu">sctype_score</span>(<span class="at">scRNAseqData =</span> seurat_obj<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">$</span>scale.data, <span class="at">scaled =</span> <span class="cn">TRUE</span>, </span>
<span id="cb68-25"><a href="finding-marker-genes-and-isoforms.html#cb68-25" aria-hidden="true" tabindex="-1"></a>                         <span class="at">gs =</span> gs_list<span class="sc">$</span>gs_positive, <span class="at">gs2 =</span> gs_list<span class="sc">$</span>gs_negative)</span>
<span id="cb68-26"><a href="finding-marker-genes-and-isoforms.html#cb68-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-27"><a href="finding-marker-genes-and-isoforms.html#cb68-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set identities in Seurat object</span></span>
<span id="cb68-28"><a href="finding-marker-genes-and-isoforms.html#cb68-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(seurat_obj) <span class="ot">&lt;-</span> cluster_res</span>
<span id="cb68-29"><a href="finding-marker-genes-and-isoforms.html#cb68-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-30"><a href="finding-marker-genes-and-isoforms.html#cb68-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge by cluster</span></span>
<span id="cb68-31"><a href="finding-marker-genes-and-isoforms.html#cb68-31" aria-hidden="true" tabindex="-1"></a>  cL_results <span class="ot">&lt;-</span> <span class="fu">do.call</span>(<span class="st">&quot;rbind&quot;</span>, <span class="fu">lapply</span>(<span class="fu">unique</span>(seurat_obj<span class="sc">@</span>meta.data[[cluster_res]]), <span class="cf">function</span>(cl) {</span>
<span id="cb68-32"><a href="finding-marker-genes-and-isoforms.html#cb68-32" aria-hidden="true" tabindex="-1"></a>    es.max.cl <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">rowSums</span>(es.max[, <span class="fu">rownames</span>(seurat_obj<span class="sc">@</span>meta.data[seurat_obj<span class="sc">@</span>meta.data[[cluster_res]] <span class="sc">==</span> cl, ])]), <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb68-33"><a href="finding-marker-genes-and-isoforms.html#cb68-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>(<span class="fu">data.frame</span>(<span class="at">cluster =</span> cl, <span class="at">type =</span> <span class="fu">names</span>(es.max.cl), <span class="at">scores =</span> es.max.cl, <span class="at">ncells =</span> <span class="fu">sum</span>(seurat_obj<span class="sc">@</span>meta.data[[cluster_res]] <span class="sc">==</span> cl)), <span class="dv">10</span>)</span>
<span id="cb68-34"><a href="finding-marker-genes-and-isoforms.html#cb68-34" aria-hidden="true" tabindex="-1"></a>  }))</span>
<span id="cb68-35"><a href="finding-marker-genes-and-isoforms.html#cb68-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-36"><a href="finding-marker-genes-and-isoforms.html#cb68-36" aria-hidden="true" tabindex="-1"></a>  sctype_scores <span class="ot">&lt;-</span> cL_results <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">1</span>, <span class="at">wt =</span> scores)</span>
<span id="cb68-37"><a href="finding-marker-genes-and-isoforms.html#cb68-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-38"><a href="finding-marker-genes-and-isoforms.html#cb68-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set low-confident clusters to &quot;Unknown&quot;</span></span>
<span id="cb68-39"><a href="finding-marker-genes-and-isoforms.html#cb68-39" aria-hidden="true" tabindex="-1"></a>  sctype_scores<span class="sc">$</span>scores <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(sctype_scores<span class="sc">$</span>scores)</span>
<span id="cb68-40"><a href="finding-marker-genes-and-isoforms.html#cb68-40" aria-hidden="true" tabindex="-1"></a>  sctype_scores<span class="sc">$</span>type[sctype_scores<span class="sc">$</span>scores <span class="sc">&lt;</span> sctype_scores<span class="sc">$</span>ncells <span class="sc">/</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Unknown&quot;</span></span>
<span id="cb68-41"><a href="finding-marker-genes-and-isoforms.html#cb68-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(sctype_scores[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>])</span>
<span id="cb68-42"><a href="finding-marker-genes-and-isoforms.html#cb68-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-43"><a href="finding-marker-genes-and-isoforms.html#cb68-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Overlay the labels</span></span>
<span id="cb68-44"><a href="finding-marker-genes-and-isoforms.html#cb68-44" aria-hidden="true" tabindex="-1"></a>  seurat_obj<span class="sc">@</span>meta.data[[metadat_col_prefix]] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb68-45"><a href="finding-marker-genes-and-isoforms.html#cb68-45" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="fu">unique</span>(sctype_scores<span class="sc">$</span>cluster)) {</span>
<span id="cb68-46"><a href="finding-marker-genes-and-isoforms.html#cb68-46" aria-hidden="true" tabindex="-1"></a>    cl_type <span class="ot">&lt;-</span> sctype_scores[sctype_scores<span class="sc">$</span>cluster <span class="sc">==</span> j,]</span>
<span id="cb68-47"><a href="finding-marker-genes-and-isoforms.html#cb68-47" aria-hidden="true" tabindex="-1"></a>    seurat_obj<span class="sc">@</span>meta.data[[metadat_col_prefix]][seurat_obj<span class="sc">@</span>meta.data[[cluster_res]] <span class="sc">==</span> j] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(cl_type<span class="sc">$</span>type[<span class="dv">1</span>])</span>
<span id="cb68-48"><a href="finding-marker-genes-and-isoforms.html#cb68-48" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb68-49"><a href="finding-marker-genes-and-isoforms.html#cb68-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-50"><a href="finding-marker-genes-and-isoforms.html#cb68-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plotting</span></span>
<span id="cb68-51"><a href="finding-marker-genes-and-isoforms.html#cb68-51" aria-hidden="true" tabindex="-1"></a>  pclass <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> reduction, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">group.by =</span> metadat_col_prefix)</span>
<span id="cb68-52"><a href="finding-marker-genes-and-isoforms.html#cb68-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pclass)</span>
<span id="cb68-53"><a href="finding-marker-genes-and-isoforms.html#cb68-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-54"><a href="finding-marker-genes-and-isoforms.html#cb68-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot to a PDF</span></span>
<span id="cb68-55"><a href="finding-marker-genes-and-isoforms.html#cb68-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="at">file =</span> <span class="fu">paste0</span>(figure_prefix, <span class="st">&quot;_&quot;</span>, metadat_col_prefix, <span class="st">&quot;_sctype_genes.pdf&quot;</span>), <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb68-56"><a href="finding-marker-genes-and-isoforms.html#cb68-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(pclass <span class="sc">+</span> <span class="fu">ggtitle</span>(figure_prefix))</span>
<span id="cb68-57"><a href="finding-marker-genes-and-isoforms.html#cb68-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb68-58"><a href="finding-marker-genes-and-isoforms.html#cb68-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-59"><a href="finding-marker-genes-and-isoforms.html#cb68-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the updated Seurat object to an RDS file</span></span>
<span id="cb68-60"><a href="finding-marker-genes-and-isoforms.html#cb68-60" aria-hidden="true" tabindex="-1"></a>  <span class="co">#if (output_file != &quot;&quot;) {</span></span>
<span id="cb68-61"><a href="finding-marker-genes-and-isoforms.html#cb68-61" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  saveRDS(seurat_obj, file = paste0(output_file, &quot;.rds&quot;))</span></span>
<span id="cb68-62"><a href="finding-marker-genes-and-isoforms.html#cb68-62" aria-hidden="true" tabindex="-1"></a>  <span class="co">#}</span></span>
<span id="cb68-63"><a href="finding-marker-genes-and-isoforms.html#cb68-63" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb68-64"><a href="finding-marker-genes-and-isoforms.html#cb68-64" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return the updated Seurat object</span></span>
<span id="cb68-65"><a href="finding-marker-genes-and-isoforms.html#cb68-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(seurat_obj)</span>
<span id="cb68-66"><a href="finding-marker-genes-and-isoforms.html#cb68-66" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb68-67"><a href="finding-marker-genes-and-isoforms.html#cb68-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-68"><a href="finding-marker-genes-and-isoforms.html#cb68-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-69"><a href="finding-marker-genes-and-isoforms.html#cb68-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Define variables</span></span>
<span id="cb68-70"><a href="finding-marker-genes-and-isoforms.html#cb68-70" aria-hidden="true" tabindex="-1"></a>db_ <span class="ot">=</span> <span class="st">&quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx&quot;</span>; <span class="co"># this is a defualt databse from sctype </span></span>
<span id="cb68-71"><a href="finding-marker-genes-and-isoforms.html#cb68-71" aria-hidden="true" tabindex="-1"></a>tissue <span class="ot">&lt;-</span> <span class="st">&quot;Brain&quot;</span></span>
<span id="cb68-72"><a href="finding-marker-genes-and-isoforms.html#cb68-72" aria-hidden="true" tabindex="-1"></a>gs_removal_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Tanycytes&quot;</span>) <span class="co"># list of cell types from the db to remove</span></span>
<span id="cb68-73"><a href="finding-marker-genes-and-isoforms.html#cb68-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-74"><a href="finding-marker-genes-and-isoforms.html#cb68-74" aria-hidden="true" tabindex="-1"></a>seu_obj <span class="ot">&lt;-</span> <span class="fu">perform_sctype_analysis</span>(seu_obj, db_, tissue, gs_removal_list, </span>
<span id="cb68-75"><a href="finding-marker-genes-and-isoforms.html#cb68-75" aria-hidden="true" tabindex="-1"></a>                        <span class="at">metadat_col_prefix =</span><span class="st">&quot;sctype_db&quot;</span>, <span class="at">figure_prefix =</span> <span class="st">&quot;Day_55&quot;</span>,</span>
<span id="cb68-76"><a href="finding-marker-genes-and-isoforms.html#cb68-76" aria-hidden="true" tabindex="-1"></a>                        <span class="at">output_file =</span> <span class="st">&quot;Day_55&quot;</span>, <span class="at">cluster_res =</span> <span class="st">&quot;RNA_snn_res.0.9&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
</details>
<pre><code>## # A tibble: 8 × 3
## # Groups:   cluster [8]
##   cluster type                      scores
##   &lt;fct&gt;   &lt;chr&gt;                      &lt;dbl&gt;
## 1 2       Immature neurons           81.3 
## 2 7       Myelinating Schwann cells   7.18
## 3 0       Mature neurons            108.  
## 4 6       Mature neurons             37.4 
## 5 3       Radial glial cells         85.1 
## 6 4       GABAergic neurons          82.8 
## 7 5       Radial glial cells         92.0 
## 8 1       Radial glial cells         94.0</code></pre>
<p><img src="bookdownproj_files/figure-html/sctype-1.png" width="672" /></p>
<p>Sctype gives us some indication of which cell types we have in our data. We can use the DE genes to get some more specific info. Glutatergic Neuronal Markers “SLC17A7”,“SLC17A6”,“GRIN1”,“GRIN2B” are all DE in cluster 0 - the mature neuron cluster. This suggests these cells are likely glutamatergic neurons.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="finding-marker-genes-and-isoforms.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># markers for </span></span>
<span id="cb70-2"><a href="finding-marker-genes-and-isoforms.html#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seu_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;SLC17A7&quot;</span>,<span class="st">&quot;SLC17A6&quot;</span>,<span class="st">&quot;GRIN1&quot;</span>,<span class="st">&quot;GRIN2B&quot;</span>))</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot%20marker%20genes-1.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="finding-marker-genes-and-isoforms.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gprofiler2)</span>
<span id="cb71-2"><a href="finding-marker-genes-and-isoforms.html#cb71-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-3"><a href="finding-marker-genes-and-isoforms.html#cb71-3" aria-hidden="true" tabindex="-1"></a>background_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(<span class="fu">GetAssayData</span>(seu_obj, <span class="at">assay =</span> <span class="st">&quot;RNA&quot;</span>, <span class="at">layer =</span> <span class="st">&quot;counts&quot;</span>))[</span>
<span id="cb71-4"><a href="finding-marker-genes-and-isoforms.html#cb71-4" aria-hidden="true" tabindex="-1"></a>  Matrix<span class="sc">::</span><span class="fu">rowSums</span>(<span class="fu">GetAssayData</span>(seu_obj, <span class="at">assay =</span> <span class="st">&quot;RNA&quot;</span>, <span class="at">layer =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb71-5"><a href="finding-marker-genes-and-isoforms.html#cb71-5" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb71-6"><a href="finding-marker-genes-and-isoforms.html#cb71-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-7"><a href="finding-marker-genes-and-isoforms.html#cb71-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for significant genes in the current cluster</span></span>
<span id="cb71-8"><a href="finding-marker-genes-and-isoforms.html#cb71-8" aria-hidden="true" tabindex="-1"></a>sig_genes <span class="ot">&lt;-</span> all_markers_gene_cluster <span class="sc">%&gt;%</span></span>
<span id="cb71-9"><a href="finding-marker-genes-and-isoforms.html#cb71-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(cluster <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span>  <span class="co"># Filter for cluster 0 and adjusted p-value &lt; 0.05</span></span>
<span id="cb71-10"><a href="finding-marker-genes-and-isoforms.html#cb71-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(gene)  <span class="co"># Extract gene names</span></span>
<span id="cb71-11"><a href="finding-marker-genes-and-isoforms.html#cb71-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 5: Run g:Profiler for pathway enrichment analysis</span></span>
<span id="cb71-12"><a href="finding-marker-genes-and-isoforms.html#cb71-12" aria-hidden="true" tabindex="-1"></a>  pathway_results <span class="ot">&lt;-</span> gprofiler2<span class="sc">::</span><span class="fu">gost</span>(</span>
<span id="cb71-13"><a href="finding-marker-genes-and-isoforms.html#cb71-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">query =</span> sig_genes,</span>
<span id="cb71-14"><a href="finding-marker-genes-and-isoforms.html#cb71-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">ordered_query =</span> <span class="cn">TRUE</span>,</span>
<span id="cb71-15"><a href="finding-marker-genes-and-isoforms.html#cb71-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">correction_method =</span> <span class="st">&#39;fdr&#39;</span>,</span>
<span id="cb71-16"><a href="finding-marker-genes-and-isoforms.html#cb71-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">custom_bg =</span> background_genes,</span>
<span id="cb71-17"><a href="finding-marker-genes-and-isoforms.html#cb71-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&quot;GO&quot;</span>, <span class="st">&quot;KEGG&quot;</span>, <span class="st">&quot;REACTOME&quot;</span>)</span>
<span id="cb71-18"><a href="finding-marker-genes-and-isoforms.html#cb71-18" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb71-19"><a href="finding-marker-genes-and-isoforms.html#cb71-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb71-20"><a href="finding-marker-genes-and-isoforms.html#cb71-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Prepare the data for plotting</span></span>
<span id="cb71-21"><a href="finding-marker-genes-and-isoforms.html#cb71-21" aria-hidden="true" tabindex="-1"></a>  df_path <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(pathway_results<span class="sc">$</span>result) <span class="sc">%&gt;%</span></span>
<span id="cb71-22"><a href="finding-marker-genes-and-isoforms.html#cb71-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(term_size <span class="sc">&lt;</span> <span class="dv">3000</span>, term_size <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-23"><a href="finding-marker-genes-and-isoforms.html#cb71-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>term_id <span class="sc">%in%</span> <span class="fu">unlist</span>(pathway_results<span class="sc">$</span>parents))</span>
<span id="cb71-24"><a href="finding-marker-genes-and-isoforms.html#cb71-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb71-25"><a href="finding-marker-genes-and-isoforms.html#cb71-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 6: Plot top 5 results per database</span></span>
<span id="cb71-26"><a href="finding-marker-genes-and-isoforms.html#cb71-26" aria-hidden="true" tabindex="-1"></a> df_path <span class="sc">%&gt;%</span></span>
<span id="cb71-27"><a href="finding-marker-genes-and-isoforms.html#cb71-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(source) <span class="sc">%&gt;%</span></span>
<span id="cb71-28"><a href="finding-marker-genes-and-isoforms.html#cb71-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_min</span>(p_value, <span class="at">n =</span> <span class="dv">5</span>, <span class="at">with_ties =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-29"><a href="finding-marker-genes-and-isoforms.html#cb71-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb71-30"><a href="finding-marker-genes-and-isoforms.html#cb71-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(term_name, <span class="sc">-</span>p_value), <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(p_value), <span class="at">fill =</span> source)) <span class="sc">+</span></span>
<span id="cb71-31"><a href="finding-marker-genes-and-isoforms.html#cb71-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&#39;identity&#39;</span>, <span class="at">position =</span> <span class="fu">position_identity</span>()) <span class="sc">+</span></span>
<span id="cb71-32"><a href="finding-marker-genes-and-isoforms.html#cb71-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb71-33"><a href="finding-marker-genes-and-isoforms.html#cb71-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb71-34"><a href="finding-marker-genes-and-isoforms.html#cb71-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-35"><a href="finding-marker-genes-and-isoforms.html#cb71-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(source <span class="sc">~</span> ., <span class="at">space =</span> <span class="st">&#39;free&#39;</span>, <span class="at">scales =</span> <span class="st">&#39;free&#39;</span>) <span class="sc">+</span></span>
<span id="cb71-36"><a href="finding-marker-genes-and-isoforms.html#cb71-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>,</span>
<span id="cb71-37"><a href="finding-marker-genes-and-isoforms.html#cb71-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">8</span>)) <span class="sc">+</span>  <span class="co"># Rotate and adjust y-axis text size</span></span>
<span id="cb71-38"><a href="finding-marker-genes-and-isoforms.html#cb71-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">&quot;Pathway Enrichment for cluster 0&quot;</span>))  <span class="co"># Add title with cluster name</span></span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/GO-1.png" width="672" /></p>
<p>Based on the enriched terms, we can confidently conclude that the cell type is neuronal. Both the KEGG and the <a href="GO:MF" class="uri">GO:MF</a> terms support the hypothesis that the cells in cluster 0 have glutamatergic synapses. This analysis can be done on all the clusters in the Seurat object.</p>
<p>We can now change the Mature neurons label to Glutamatergic neurons and plot the updated UMAP.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="finding-marker-genes-and-isoforms.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Change the names of ScType DF to Glutatertergic neurons in metadat</span></span>
<span id="cb72-2"><a href="finding-marker-genes-and-isoforms.html#cb72-2" aria-hidden="true" tabindex="-1"></a>seu_obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_db <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;Mature neurons&quot;</span>, <span class="st">&quot;Glutamatergic neurons&quot;</span>, seu_obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_db)</span>
<span id="cb72-3"><a href="finding-marker-genes-and-isoforms.html#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="finding-marker-genes-and-isoforms.html#cb72-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-5"><a href="finding-marker-genes-and-isoforms.html#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seu_obj, <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span>) <span class="sc">|</span> <span class="fu">DimPlot</span>(seu_obj, <span class="at">reduction =</span><span class="st">&quot;umap_iso&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot%20marker%20genes%20iso-1.png" width="960" /></p>
<p>Cell type identification is an iterative process and often one of the most challenging aspects of single-cell analysis. For this example, we will assume that our combined approach, using automated cell type identification, differential expression (DE) analysis based on clusters and Gene set enrichment, provides a good indication of the cell types present in our data. It is possible to explore the radial glial cells in more detail as there are likely many subtypes. For the purposes of the tutorial we will leave this annoation as is. Based on this we have 5 main cell types.</p>
<ol style="list-style-type: decimal">
<li><p>Radial glial cells (RG)</p></li>
<li><p>Immature neurons</p></li>
<li><p>Glutamatergic neurons</p></li>
<li><p>GABAergic neuorns</p></li>
<li><p>Myelinating Schwann cells</p></li>
</ol>
<p>We can use a very nice package called dittoSeq (<a href="https://bioconductor.org/packages/devel/bioc/vignettes/dittoSeq/inst/doc/dittoSeq.html" class="uri">https://bioconductor.org/packages/devel/bioc/vignettes/dittoSeq/inst/doc/dittoSeq.html</a>) to Visualise scRNA seq data directly from a seurat object.</p>
<p>We can plot the distrubtion of the 5 cell types in a few different ways using <code>dittoBarPlot</code></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="finding-marker-genes-and-isoforms.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dittoSeq)</span>
<span id="cb73-2"><a href="finding-marker-genes-and-isoforms.html#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoBarPlot</span>(seu_obj, <span class="st">&quot;orig.ident&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span>, <span class="at">scale =</span> <span class="st">&quot;count&quot;</span>) <span class="sc">|</span> <span class="fu">dittoBarPlot</span>(seu_obj, <span class="st">&quot;sctype_db&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;orig.ident&quot;</span>, <span class="at">scale =</span> <span class="st">&quot;percent&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/dito-1.png" width="672" /></p>
</div>
<div id="de-genes-and-isoforms-based-on-annotaed-cell-types." class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> DE genes and isoforms based on annotaed cell types.<a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotaed-cell-types." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>With this foundation, we can refine our DE analysis by focusing on cell types rather than clusters. This step is critical in nearly all transcriptomic analyses, offering a wide range of possibilities for downstream investigations.</p>
<p>A common downstream approach involves generating volcano plots to visualize DE genes and isoforms. performing gene set enrichment analysis, In the following sections, we will demonstrate how to perform these types of analyses and explore their potential applications.</p>
<p>The code chunk below provides an example of how to execute these analyses:</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="finding-marker-genes-and-isoforms.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Set identities based on cell type</span></span>
<span id="cb74-2"><a href="finding-marker-genes-and-isoforms.html#cb74-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-3"><a href="finding-marker-genes-and-isoforms.html#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(seu_obj) <span class="ot">&lt;-</span> <span class="st">&quot;sctype_db&quot;</span></span>
<span id="cb74-4"><a href="finding-marker-genes-and-isoforms.html#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="finding-marker-genes-and-isoforms.html#cb74-5" aria-hidden="true" tabindex="-1"></a>all_markers_gene_celltype <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb74-6"><a href="finding-marker-genes-and-isoforms.html#cb74-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">object =</span> seu_obj, </span>
<span id="cb74-7"><a href="finding-marker-genes-and-isoforms.html#cb74-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">&quot;RNA&quot;</span>, </span>
<span id="cb74-8"><a href="finding-marker-genes-and-isoforms.html#cb74-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span>,  <span class="co"># Replace with your metadata column name</span></span>
<span id="cb74-9"><a href="finding-marker-genes-and-isoforms.html#cb74-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>, </span>
<span id="cb74-10"><a href="finding-marker-genes-and-isoforms.html#cb74-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.20</span>, </span>
<span id="cb74-11"><a href="finding-marker-genes-and-isoforms.html#cb74-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> <span class="cn">FALSE</span> <span class="co"># changed this to false to get negatively DE genes to</span></span>
<span id="cb74-12"><a href="finding-marker-genes-and-isoforms.html#cb74-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb74-13"><a href="finding-marker-genes-and-isoforms.html#cb74-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-14"><a href="finding-marker-genes-and-isoforms.html#cb74-14" aria-hidden="true" tabindex="-1"></a>all_markers_iso_celltype <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb74-15"><a href="finding-marker-genes-and-isoforms.html#cb74-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">object =</span> seu_obj, </span>
<span id="cb74-16"><a href="finding-marker-genes-and-isoforms.html#cb74-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, </span>
<span id="cb74-17"><a href="finding-marker-genes-and-isoforms.html#cb74-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span>,  <span class="co"># Replace with your metadata column name</span></span>
<span id="cb74-18"><a href="finding-marker-genes-and-isoforms.html#cb74-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>, </span>
<span id="cb74-19"><a href="finding-marker-genes-and-isoforms.html#cb74-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">min.pct =</span> <span class="fl">0.20</span>, </span>
<span id="cb74-20"><a href="finding-marker-genes-and-isoforms.html#cb74-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">only.pos =</span> <span class="cn">FALSE</span> <span class="co"># changed this to false to get negatively DE genes to</span></span>
<span id="cb74-21"><a href="finding-marker-genes-and-isoforms.html#cb74-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb74-22"><a href="finding-marker-genes-and-isoforms.html#cb74-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-23"><a href="finding-marker-genes-and-isoforms.html#cb74-23" aria-hidden="true" tabindex="-1"></a><span class="co">#save the list of DE genes </span></span>
<span id="cb74-24"><a href="finding-marker-genes-and-isoforms.html#cb74-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(all_markers_gene_celltype, <span class="st">&quot;./output_files/DE/all_markers_one_gene_celltype.csv&quot;</span>)</span>
<span id="cb74-25"><a href="finding-marker-genes-and-isoforms.html#cb74-25" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(all_markers_iso_celltype, <span class="st">&quot;./output_files/DE/all_markers_one_iso_celltype.csv&quot;</span>)</span></code></pre></div>
</details>
<p>A basic way of exploring this data is to plot these markers on a heatmap. Seurat has a nice function to do so. Let us plot the top 5 marker gene and isoforms in each cell type</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="finding-marker-genes-and-isoforms.html#cb75-1" aria-hidden="true" tabindex="-1"></a>all_markers_gene_celltype <span class="sc">%&gt;%</span></span>
<span id="cb75-2"><a href="finding-marker-genes-and-isoforms.html#cb75-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb75-3"><a href="finding-marker-genes-and-isoforms.html#cb75-3" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb75-4"><a href="finding-marker-genes-and-isoforms.html#cb75-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb75-5"><a href="finding-marker-genes-and-isoforms.html#cb75-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="ot">-&gt;</span> G_top5</span>
<span id="cb75-6"><a href="finding-marker-genes-and-isoforms.html#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(seu_obj, <span class="at">features =</span> G_top5<span class="sc">$</span>gene, <span class="at">assay =</span> <span class="st">&quot;RNA&quot;</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span> </span>
<span id="cb75-7"><a href="finding-marker-genes-and-isoforms.html#cb75-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoLegend</span>() </span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/heatmap-1.png" width="960" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="finding-marker-genes-and-isoforms.html#cb76-1" aria-hidden="true" tabindex="-1"></a>all_markers_iso_celltype <span class="sc">%&gt;%</span></span>
<span id="cb76-2"><a href="finding-marker-genes-and-isoforms.html#cb76-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb76-3"><a href="finding-marker-genes-and-isoforms.html#cb76-3" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-4"><a href="finding-marker-genes-and-isoforms.html#cb76-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-5"><a href="finding-marker-genes-and-isoforms.html#cb76-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="ot">-&gt;</span> I_top5</span>
<span id="cb76-6"><a href="finding-marker-genes-and-isoforms.html#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(seu_obj, <span class="at">features =</span> I_top5<span class="sc">$</span>gene, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb76-7"><a href="finding-marker-genes-and-isoforms.html#cb76-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoLegend</span>() </span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/heatmap-2.png" width="960" /></p>
<p>Personally i find the same plots generated by <code>dittoHeatmap</code> much nicer. The function can also take a list of genes or isoforms of interest. Just remember to set the default assay accordingly.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="finding-marker-genes-and-isoforms.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co">#list_of_genes &lt;- c(&quot;VIM&quot;, &quot;MAPT&quot;, &quot;KLC1&quot;, &quot;RBFOX1&quot;, &quot;RBFOX3&quot;)</span></span>
<span id="cb77-2"><a href="finding-marker-genes-and-isoforms.html#cb77-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-3"><a href="finding-marker-genes-and-isoforms.html#cb77-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(seu_obj) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb77-4"><a href="finding-marker-genes-and-isoforms.html#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoHeatmap</span>(seu_obj, <span class="fu">head</span>(G_top5<span class="sc">$</span>gene, <span class="dv">25</span>),</span>
<span id="cb77-5"><a href="finding-marker-genes-and-isoforms.html#cb77-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">scaled.to.max =</span> <span class="cn">FALSE</span>,</span>
<span id="cb77-6"><a href="finding-marker-genes-and-isoforms.html#cb77-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">complex =</span> <span class="cn">FALSE</span>,</span>
<span id="cb77-7"><a href="finding-marker-genes-and-isoforms.html#cb77-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">heatmap.colors.max.scaled =</span> <span class="cn">FALSE</span>,</span>
<span id="cb77-8"><a href="finding-marker-genes-and-isoforms.html#cb77-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">annot.by =</span> <span class="fu">c</span>(<span class="st">&quot;sctype_db&quot;</span>))</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/heatma%20pdito-1.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="finding-marker-genes-and-isoforms.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(seu_obj) <span class="ot">&lt;-</span> <span class="st">&quot;iso&quot;</span></span>
<span id="cb78-2"><a href="finding-marker-genes-and-isoforms.html#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dittoHeatmap</span>(seu_obj, <span class="fu">head</span>(I_top5<span class="sc">$</span>gene, <span class="dv">25</span>),</span>
<span id="cb78-3"><a href="finding-marker-genes-and-isoforms.html#cb78-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">scaled.to.max =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-4"><a href="finding-marker-genes-and-isoforms.html#cb78-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">complex =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-5"><a href="finding-marker-genes-and-isoforms.html#cb78-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">heatmap.colors.max.scaled =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-6"><a href="finding-marker-genes-and-isoforms.html#cb78-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">annot.by =</span> <span class="fu">c</span>(<span class="st">&quot;sctype_db&quot;</span>))</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/heatma%20pdito-2.png" width="672" /></p>
</div>
<div id="volcano-plots" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Volcano plots<a href="finding-marker-genes-and-isoforms.html#volcano-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="findallmarkers-de" class="section level3 hasAnchor" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> FIndAllMarkers DE<a href="finding-marker-genes-and-isoforms.html#findallmarkers-de" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next we can explore this data by generating some volcano plots. This analysis can be useful to identify genes and isoforms that are DE and also have large fold changes. Often these types of features are the interesting for further analysis. This can be done for any of the cell types defined in our object. For the sake of brevity we will look a the Glutamatergic neurons.</p>
<p>When plotting the volcano plots, we observe that many genes exhibit both statistically significant p-values (p &lt; 0.05) and log2 fold changes -2&lt; or &gt;2. To highlight key findings, we have labeled some glutamatergic marker genes, demonstrating that the genes we expect to be upregulated in this cell type are indeed showing the expected pattern. Additionally, we have labeled VIM, a marker of radial glial and projenitor cells. As anticipated, VIM expression is downregulated in these neurons, which aligns with our expectations.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="finding-marker-genes-and-isoforms.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnhancedVolcano)</span>
<span id="cb79-2"><a href="finding-marker-genes-and-isoforms.html#cb79-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-3"><a href="finding-marker-genes-and-isoforms.html#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="co">#filter for the cell type of interest </span></span>
<span id="cb79-4"><a href="finding-marker-genes-and-isoforms.html#cb79-4" aria-hidden="true" tabindex="-1"></a>glut_DE_iso <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(all_markers_iso_celltype, cluster <span class="sc">==</span> <span class="st">&quot;Glutamatergic neurons&quot;</span>)</span>
<span id="cb79-5"><a href="finding-marker-genes-and-isoforms.html#cb79-5" aria-hidden="true" tabindex="-1"></a>glut_DE_gene <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(all_markers_gene_celltype, cluster <span class="sc">==</span> <span class="st">&quot;Glutamatergic neurons&quot;</span>) </span>
<span id="cb79-6"><a href="finding-marker-genes-and-isoforms.html#cb79-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-7"><a href="finding-marker-genes-and-isoforms.html#cb79-7" aria-hidden="true" tabindex="-1"></a><span class="co">#we can plot our volcano plot </span></span>
<span id="cb79-8"><a href="finding-marker-genes-and-isoforms.html#cb79-8" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(glut_DE_gene,</span>
<span id="cb79-9"><a href="finding-marker-genes-and-isoforms.html#cb79-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">lab=</span>glut_DE_gene<span class="sc">$</span>gene,</span>
<span id="cb79-10"><a href="finding-marker-genes-and-isoforms.html#cb79-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">x=</span><span class="st">&#39;avg_log2FC&#39;</span>, <span class="at">y=</span><span class="st">&#39;p_val_adj&#39;</span>, <span class="at">pCutoff=</span><span class="fl">0.05</span>, <span class="at">FCcutoff=</span><span class="dv">2</span>,</span>
<span id="cb79-11"><a href="finding-marker-genes-and-isoforms.html#cb79-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">boxedLabels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb79-12"><a href="finding-marker-genes-and-isoforms.html#cb79-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb79-13"><a href="finding-marker-genes-and-isoforms.html#cb79-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">selectLab=</span> <span class="fu">c</span>(<span class="st">&quot;SLC17A7&quot;</span>,<span class="st">&quot;SLC17A6&quot;</span>,<span class="st">&#39;GRIN1&#39;</span>,<span class="st">&quot;GRIN2B&quot;</span>,<span class="st">&#39;VIM&#39;</span>),</span>
<span id="cb79-14"><a href="finding-marker-genes-and-isoforms.html#cb79-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;Volcano Plot of Differentially Expressed Genes </span><span class="sc">\n</span><span class="st"> in the Glutamatergic Neurons&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/valocano1-1.png" width="768" /></p>
<p>Interestingly our long read data allows us to perform the same analysis but at the isoform level. This can be hard to interpret as there are many more features to plot on the volcano plot. for the sake a clarity we have just labelled TBR1 isoforms. This code bellow will allow users to plot all the isoforms from a given gene on the Volcano plot making interpretation of the data a bit cleaner. Here we can see two different isoforms of the TBR1 gene showing enrichment in this cell type</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="finding-marker-genes-and-isoforms.html#cb80-1" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="st">&quot;TBR1&quot;</span></span>
<span id="cb80-2"><a href="finding-marker-genes-and-isoforms.html#cb80-2" aria-hidden="true" tabindex="-1"></a>plot_features_list <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">&quot;(^|-|</span><span class="sc">\\</span><span class="st">b)&quot;</span>, gene, <span class="st">&quot;($|</span><span class="sc">\\</span><span class="st">b)&quot;</span>), features, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb80-3"><a href="finding-marker-genes-and-isoforms.html#cb80-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-4"><a href="finding-marker-genes-and-isoforms.html#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(glut_DE_iso,</span>
<span id="cb80-5"><a href="finding-marker-genes-and-isoforms.html#cb80-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">lab=</span>glut_DE_iso<span class="sc">$</span>gene,</span>
<span id="cb80-6"><a href="finding-marker-genes-and-isoforms.html#cb80-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">x=</span><span class="st">&#39;avg_log2FC&#39;</span>, <span class="at">y=</span><span class="st">&#39;p_val_adj&#39;</span>, <span class="at">pCutoff=</span><span class="fl">0.05</span>, <span class="at">FCcutoff=</span><span class="dv">2</span>,</span>
<span id="cb80-7"><a href="finding-marker-genes-and-isoforms.html#cb80-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">boxedLabels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb80-8"><a href="finding-marker-genes-and-isoforms.html#cb80-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb80-9"><a href="finding-marker-genes-and-isoforms.html#cb80-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">selectLab=</span> plot_features_list,</span>
<span id="cb80-10"><a href="finding-marker-genes-and-isoforms.html#cb80-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;Volcano Plot of Differentially Expressed Isoforms </span><span class="sc">\n</span><span class="st"> in the Glutamatergic Neurons&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/valocano2-1.png" width="768" /></p>
</div>
<div id="findmarkers-de" class="section level3 hasAnchor" number="5.4.2">
<h3><span class="header-section-number">5.4.2</span> FindMarkers DE<a href="finding-marker-genes-and-isoforms.html#findmarkers-de" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finding All Markers is just one type of differential expression (DE) analysis that can be performed. Seurat offers the <code>FindAllMarkers</code> function, which tests the cell type of interest against all other cells. While this approach is often sufficient for identifying marker genes, users may also want to test differences between two specific cell types. For instance, you might want to identify DE genes when comparing glutamatergic neurons to radial glial cells. Below, we demonstrate how to perform this type of analysis with the <code>FindMarkers</code> function.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="finding-marker-genes-and-isoforms.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(seu_obj) <span class="ot">&lt;-</span> <span class="st">&#39;RNA&#39;</span> <span class="co"># difeine the gne assay as default</span></span>
<span id="cb81-2"><a href="finding-marker-genes-and-isoforms.html#cb81-2" aria-hidden="true" tabindex="-1"></a>glu_RG_gene <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(seu_obj,</span>
<span id="cb81-3"><a href="finding-marker-genes-and-isoforms.html#cb81-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ident.1 =</span> <span class="st">&quot;Glutamatergic neurons&quot;</span>,</span>
<span id="cb81-4"><a href="finding-marker-genes-and-isoforms.html#cb81-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ident.2 =</span> <span class="st">&quot;Radial glial cells&quot;</span>,</span>
<span id="cb81-5"><a href="finding-marker-genes-and-isoforms.html#cb81-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>, <span class="at">min.pct =</span> <span class="fl">0.02</span>)</span>
<span id="cb81-6"><a href="finding-marker-genes-and-isoforms.html#cb81-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-7"><a href="finding-marker-genes-and-isoforms.html#cb81-7" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(seu_obj) <span class="ot">&lt;-</span> <span class="st">&#39;iso&#39;</span> <span class="co"># difeine the gne assay as default</span></span>
<span id="cb81-8"><a href="finding-marker-genes-and-isoforms.html#cb81-8" aria-hidden="true" tabindex="-1"></a>glu_RG_iso <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(seu_obj, <span class="at">ident.1 =</span> <span class="st">&quot;Glutamatergic neurons&quot;</span>,</span>
<span id="cb81-9"><a href="finding-marker-genes-and-isoforms.html#cb81-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ident.2 =</span> <span class="st">&quot;Radial glial cells&quot;</span>,</span>
<span id="cb81-10"><a href="finding-marker-genes-and-isoforms.html#cb81-10" aria-hidden="true" tabindex="-1"></a>                          <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>, <span class="at">min.pct =</span> <span class="fl">0.02</span>)</span>
<span id="cb81-11"><a href="finding-marker-genes-and-isoforms.html#cb81-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-12"><a href="finding-marker-genes-and-isoforms.html#cb81-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-13"><a href="finding-marker-genes-and-isoforms.html#cb81-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plots # to plot at gene level</span></span>
<span id="cb81-14"><a href="finding-marker-genes-and-isoforms.html#cb81-14" aria-hidden="true" tabindex="-1"></a><span class="co">#EnhancedVolcano(glu_RG_gene, lab=rownames(glu_RG_gene),</span></span>
<span id="cb81-15"><a href="finding-marker-genes-and-isoforms.html#cb81-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                x=&#39;avg_log2FC&#39;, y=&#39;p_val_adj&#39;,</span></span>
<span id="cb81-16"><a href="finding-marker-genes-and-isoforms.html#cb81-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                #selectLab= &quot;VIM&quot;,</span></span>
<span id="cb81-17"><a href="finding-marker-genes-and-isoforms.html#cb81-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                pCutoff=0.05, FCcutoff=2,</span></span>
<span id="cb81-18"><a href="finding-marker-genes-and-isoforms.html#cb81-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                title = &quot;Volcano Plot of Differentially Expressed Gene \n Glutamatergic Neurons vs Radial glial #Cells&quot;)</span></span>
<span id="cb81-19"><a href="finding-marker-genes-and-isoforms.html#cb81-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-20"><a href="finding-marker-genes-and-isoforms.html#cb81-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Volcano plots</span></span>
<span id="cb81-21"><a href="finding-marker-genes-and-isoforms.html#cb81-21" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(glu_RG_iso, <span class="at">lab=</span><span class="fu">rownames</span>(glu_RG_iso),</span>
<span id="cb81-22"><a href="finding-marker-genes-and-isoforms.html#cb81-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">x=</span><span class="st">&#39;avg_log2FC&#39;</span>, <span class="at">y=</span><span class="st">&#39;p_val_adj&#39;</span>,</span>
<span id="cb81-23"><a href="finding-marker-genes-and-isoforms.html#cb81-23" aria-hidden="true" tabindex="-1"></a>                <span class="co">#selectLab= &quot;VIM&quot;,</span></span>
<span id="cb81-24"><a href="finding-marker-genes-and-isoforms.html#cb81-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">pCutoff=</span><span class="fl">0.05</span>, <span class="at">FCcutoff=</span><span class="dv">2</span>,</span>
<span id="cb81-25"><a href="finding-marker-genes-and-isoforms.html#cb81-25" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;Volcano Plot of Differentially Expressed Isoforms </span><span class="sc">\n</span><span class="st"> Glutamatergic Neurons vs Radial glial Cells&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/valocano3-1.png" width="768" /></p>
<p>The volcano plot above is a useful tool for visualizing DE isoforms between two cell types. In this plot, red-labeled isoforms on the right-hand side indicate those that are unregulated in glutamatergic neurons compared to RG cells, while red dots on the left represent isoforms that are unregulated in radial glial cells compared to glutamatergic neurons. This analysis serves as an initial overview, and users can further explore the DE list (<code>glu_RG_iso</code>) to select specific isoforms of interest for more detailed analysis.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="add-isoform-counts-to-seurat-object.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="exploring-isoforms-of-interest.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/05_Finding_differentially_expressed_genes_and_isoforms.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/05_Finding_differentially_expressed_genes_and_isoforms.Rmd",
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
