<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Multisample analysis | FLAMES Single-Cell RNA-seq Tutorial</title>
  <meta name="description" content="Chapter 9 Multisample analysis | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Multisample analysis | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Multisample analysis | FLAMES Single-Cell RNA-seq Tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer Postdoctoral Research Fellow in Long Reads and Single-Cell Transcriptomics University of Melbourne" />


<meta name="date" content="2025-02-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="novel-isoforms.html"/>
<link rel="next" href="trajectory-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single cell RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualizing-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualizing OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>9</b> Multisample analysis</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>9.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="9.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>9.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>9.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multi-sample-integration"><i class="fa fa-check"></i><b>9.2</b> Multi-Sample integration</a></li>
<li class="chapter" data-level="9.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>9.3</b> Add isoform counts</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Trajectory analysis</strong></a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i><b>10.1</b> Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="10.2" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i><b>10.2</b> Multisample QC function</a></li>
<li class="chapter" data-level="10.3" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-count-matrix"><i class="fa fa-check"></i><b>10.3</b> Convert Oarfish files to count matrix</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FLAMES Single-Cell RNA-seq Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multisample-analysis" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Multisample analysis<a href="multisample-analysis.html#multisample-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The FLAMES pipeline supports the analysis of multiple samples simultaneously, allowing users to efficiently process complex experimental datasets. This is achieved using the <code>sc_long_multisample_pipeline</code> function. The output generated by this function is similar to the single-sample analysis outlined in previous chapters but offers additional capabilities when the experimental design includes sample replicates.</p>
<p>To take full advantage of these features we need to perform sample integration to generate a single object that we can use to integrate isoform expression across our conditions. First standard preprocessing as described in Chapters 3 and 4 is required. After integration we can perform the following key analysis:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Sample Integration:</strong> Combine multiple sample together and add isoform counts.</p></li>
<li><p><strong>Trajectory Analysis:</strong> Explore genes that change along pseudotime trajectories.</p></li>
<li><p><strong>Differential Expression Analysis:</strong> Identify differentially expressed isoforms across time points or sample conditions.</p></li>
<li><p><strong>Differential Transcript Usage (DTU) Analysis:</strong> Investigate changes in isoform proportions between conditions.</p></li>
</ol>
<p>Most preprocessing and QC steps remain the same. However, we will also demonstrate how to proceed directly from FLAMES output to initial QC without performing empty droplet removal or ambient RNA normalization. Although these steps are recommended, they may not always be necessary depending on the data.</p>
<hr />
<div id="dataset-information-1" class="section level3 hasAnchor" number="9.0.1">
<h3><span class="header-section-number">9.0.1</span> Dataset Information<a href="multisample-analysis.html#dataset-information-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For this analysis, we are using a dataset comprising eight samples spanning four time points from an excitatory neuronal differentiation protocol. The protocol is the same as the one used for the D55 sample in previous chapters but utilizes a different cell line. Two samples were collected from each of the following time points:</p>
<ul>
<li><p><strong>Stem cells</strong> (STC)</p></li>
<li><p><strong>Day 25</strong></p></li>
<li><p><strong>Day 55</strong></p>
<p><strong>Day 80</strong></p></li>
</ul>
<p>This comprehensive dataset will allow us to explore temporal changes in gene and isoform expression throughout neuronal differentiation.</p>
</div>
<div id="standard-pre-processing-and-quality-control" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Standard pre-processing and quality control<a href="multisample-analysis.html#standard-pre-processing-and-quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As before we will modify the count matrix for each sample so we use gene_symbol instead of ENSG id. The Data generated from this multi-sample experiment is quite large. This is especially the case with Oarfish isoform quantification as there are many more features to consider. This means much of the workflow bellow is time consuming and computationally intensive. Please keep this in mind when processing your samples. In the tutorial we provide code for the required steps but do not evaluate many of the sections bellow due to these computations constraints. The raw counts are available in the multisample data folder if users wish to run through the analysis.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="multisample-analysis.html#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert Gene_id to gene symbol for all counts</span></span>
<span id="cb114-2"><a href="multisample-analysis.html#cb114-2" aria-hidden="true" tabindex="-1"></a><span class="co">#run aloop to run this fucntion on all count files.</span></span>
<span id="cb114-3"><a href="multisample-analysis.html#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Directory with input files</span></span>
<span id="cb114-4"><a href="multisample-analysis.html#cb114-4" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">&lt;-</span> <span class="st">&quot;./data/muti_sample/&quot;</span></span>
<span id="cb114-5"><a href="multisample-analysis.html#cb114-5" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="st">&quot;./data/muti_sample/&quot;</span></span>
<span id="cb114-6"><a href="multisample-analysis.html#cb114-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-7"><a href="multisample-analysis.html#cb114-7" aria-hidden="true" tabindex="-1"></a><span class="co"># List all CSV files in the input directory</span></span>
<span id="cb114-8"><a href="multisample-analysis.html#cb114-8" aria-hidden="true" tabindex="-1"></a>input_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(input_dir, <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">count.csv$&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb114-9"><a href="multisample-analysis.html#cb114-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-10"><a href="multisample-analysis.html#cb114-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty list to store data frames</span></span>
<span id="cb114-11"><a href="multisample-analysis.html#cb114-11" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb114-12"><a href="multisample-analysis.html#cb114-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-13"><a href="multisample-analysis.html#cb114-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Process each file in the directory</span></span>
<span id="cb114-14"><a href="multisample-analysis.html#cb114-14" aria-hidden="true" tabindex="-1"></a>total_files <span class="ot">&lt;-</span> <span class="fu">length</span>(input_files)  <span class="co"># Total number of files</span></span>
<span id="cb114-15"><a href="multisample-analysis.html#cb114-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-16"><a href="multisample-analysis.html#cb114-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(input_files)) {</span>
<span id="cb114-17"><a href="multisample-analysis.html#cb114-17" aria-hidden="true" tabindex="-1"></a>  file_path <span class="ot">&lt;-</span> input_files[i]</span>
<span id="cb114-18"><a href="multisample-analysis.html#cb114-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-19"><a href="multisample-analysis.html#cb114-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract file name without extension</span></span>
<span id="cb114-20"><a href="multisample-analysis.html#cb114-20" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(<span class="fu">basename</span>(file_path))</span>
<span id="cb114-21"><a href="multisample-analysis.html#cb114-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-22"><a href="multisample-analysis.html#cb114-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct the output file name</span></span>
<span id="cb114-23"><a href="multisample-analysis.html#cb114-23" aria-hidden="true" tabindex="-1"></a>  output_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir, <span class="fu">paste0</span>(file_name, <span class="st">&quot;_gene_symbol.csv&quot;</span>))</span>
<span id="cb114-24"><a href="multisample-analysis.html#cb114-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-25"><a href="multisample-analysis.html#cb114-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Show progress to the user</span></span>
<span id="cb114-26"><a href="multisample-analysis.html#cb114-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Processing file %d of %d: %s&quot;</span>, i, total_files, file_name))</span>
<span id="cb114-27"><a href="multisample-analysis.html#cb114-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-28"><a href="multisample-analysis.html#cb114-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Call the function with return_df = TRUE to store the result</span></span>
<span id="cb114-29"><a href="multisample-analysis.html#cb114-29" aria-hidden="true" tabindex="-1"></a>  result_list[[file_name]] <span class="ot">&lt;-</span> <span class="fu">convert_ENSGID_to_geneSymbol</span>(</span>
<span id="cb114-30"><a href="multisample-analysis.html#cb114-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_count_matrix_path =</span> file_path,</span>
<span id="cb114-31"><a href="multisample-analysis.html#cb114-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_file =</span> output_file,</span>
<span id="cb114-32"><a href="multisample-analysis.html#cb114-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">return_df =</span> <span class="cn">TRUE</span></span>
<span id="cb114-33"><a href="multisample-analysis.html#cb114-33" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb114-34"><a href="multisample-analysis.html#cb114-34" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<div id="define-qc-function" class="section level3 hasAnchor" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Define QC function<a href="multisample-analysis.html#define-qc-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We have developed a multisample QC function @ref(Multisample QC function) for processing count matrices and constructing filtered Seurat objects, following best practices outlined in the Seurat tutorial. This function includes steps to identify and remove doublets, as well as generate key diagnostic plots for data inspection. We encourage users to customize this function to fit their specific needs, such as applying additional filtering criteria.</p>
<p>For the purposes of presenting concise and readable code we will apply the default filtering parameters to all 8 objects. If users wish to be more specific about filtering each sample independent this is possible and can be done by running the <code>perform_qc_filtering</code> function on each sample with desired parameters.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="multisample-analysis.html#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define sample names</span></span>
<span id="cb115-2"><a href="multisample-analysis.html#cb115-2" aria-hidden="true" tabindex="-1"></a>sample_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;C1_STC&quot;</span>, <span class="st">&quot;C4_STC&quot;</span>, <span class="st">&quot;C2_Day25&quot;</span>, <span class="st">&quot;C5_Day25&quot;</span>, <span class="st">&quot;C2_Day55&quot;</span>, <span class="st">&quot;C3_Day55&quot;</span>, <span class="st">&quot;C3_Day80&quot;</span>, <span class="st">&quot;C5_Day80&quot;</span>)</span>
<span id="cb115-3"><a href="multisample-analysis.html#cb115-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-4"><a href="multisample-analysis.html#cb115-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store UMAP objects</span></span>
<span id="cb115-5"><a href="multisample-analysis.html#cb115-5" aria-hidden="true" tabindex="-1"></a>umap_objects <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb115-6"><a href="multisample-analysis.html#cb115-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-7"><a href="multisample-analysis.html#cb115-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Start the PDF for output</span></span>
<span id="cb115-8"><a href="multisample-analysis.html#cb115-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="st">&quot;./output_files/mutli_sample/QC/multisample_QC.pdf&quot;</span>, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb115-9"><a href="multisample-analysis.html#cb115-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-10"><a href="multisample-analysis.html#cb115-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each sample and apply the QC filtering function</span></span>
<span id="cb115-11"><a href="multisample-analysis.html#cb115-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(sample_names)) {</span>
<span id="cb115-12"><a href="multisample-analysis.html#cb115-12" aria-hidden="true" tabindex="-1"></a>  sample_name <span class="ot">&lt;-</span> sample_names[i]</span>
<span id="cb115-13"><a href="multisample-analysis.html#cb115-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-14"><a href="multisample-analysis.html#cb115-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Print progress message</span></span>
<span id="cb115-15"><a href="multisample-analysis.html#cb115-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste0</span>(<span class="st">&quot;Processing sample &quot;</span>, i, <span class="st">&quot; of &quot;</span>, <span class="fu">length</span>(sample_names), <span class="st">&quot;: &quot;</span>, sample_name))</span>
<span id="cb115-16"><a href="multisample-analysis.html#cb115-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-17"><a href="multisample-analysis.html#cb115-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform QC filtering</span></span>
<span id="cb115-18"><a href="multisample-analysis.html#cb115-18" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">perform_qc_filtering</span>(</span>
<span id="cb115-19"><a href="multisample-analysis.html#cb115-19" aria-hidden="true" tabindex="-1"></a>    result_list[<span class="fu">paste0</span>(sample_name, <span class="st">&quot;_matched_reads_gene_count&quot;</span>)],</span>
<span id="cb115-20"><a href="multisample-analysis.html#cb115-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">fig_name =</span> sample_name,</span>
<span id="cb115-21"><a href="multisample-analysis.html#cb115-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">project =</span> sample_name</span>
<span id="cb115-22"><a href="multisample-analysis.html#cb115-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb115-23"><a href="multisample-analysis.html#cb115-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-24"><a href="multisample-analysis.html#cb115-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the UMAP object from the result</span></span>
<span id="cb115-25"><a href="multisample-analysis.html#cb115-25" aria-hidden="true" tabindex="-1"></a>  umap_objects[[sample_name]] <span class="ot">&lt;-</span> result[[<span class="dv">1</span>]]</span>
<span id="cb115-26"><a href="multisample-analysis.html#cb115-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb115-27"><a href="multisample-analysis.html#cb115-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-28"><a href="multisample-analysis.html#cb115-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the PDF device</span></span>
<span id="cb115-29"><a href="multisample-analysis.html#cb115-29" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb115-30"><a href="multisample-analysis.html#cb115-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-31"><a href="multisample-analysis.html#cb115-31" aria-hidden="true" tabindex="-1"></a><span class="co">#save the object</span></span>
<span id="cb115-32"><a href="multisample-analysis.html#cb115-32" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(umap_objects, file = &quot;./output_files/mutli_sample/umap_objects.rds&quot;)</span></span></code></pre></div>
</details>
</div>
</div>
<div id="multi-sample-integration" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Multi-Sample integration<a href="multisample-analysis.html#multi-sample-integration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First we will merge the seurat objects together and then perform integration. There are many options for sample integration and Seurat provides many wrappers for these functions. Here we will Harmony <span class="citation">(<a href="references.html#ref-korsunsky2019" role="doc-biblioref">Korsunsky et al., 2019</a>)</span> for fast and effective integration. As these processes are time consuming we will load our saved objects.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="multisample-analysis.html#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># intergrate samples  </span></span>
<span id="cb116-2"><a href="multisample-analysis.html#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="co">#from the fucntion above pull the filtered seurat object </span></span>
<span id="cb116-3"><a href="multisample-analysis.html#cb116-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-4"><a href="multisample-analysis.html#cb116-4" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb116-5"><a href="multisample-analysis.html#cb116-5" aria-hidden="true" tabindex="-1"></a>  umap_objects[[<span class="st">&quot;C1_STC&quot;</span>]],</span>
<span id="cb116-6"><a href="multisample-analysis.html#cb116-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">list</span>(</span>
<span id="cb116-7"><a href="multisample-analysis.html#cb116-7" aria-hidden="true" tabindex="-1"></a>    umap_objects[[<span class="st">&quot;C4_STC&quot;</span>]],</span>
<span id="cb116-8"><a href="multisample-analysis.html#cb116-8" aria-hidden="true" tabindex="-1"></a>    umap_objects[[<span class="st">&quot;C2_Day25&quot;</span>]],</span>
<span id="cb116-9"><a href="multisample-analysis.html#cb116-9" aria-hidden="true" tabindex="-1"></a>    umap_objects[[<span class="st">&quot;C5_Day25&quot;</span>]],</span>
<span id="cb116-10"><a href="multisample-analysis.html#cb116-10" aria-hidden="true" tabindex="-1"></a>    umap_objects[[<span class="st">&quot;C2_Day55&quot;</span>]],</span>
<span id="cb116-11"><a href="multisample-analysis.html#cb116-11" aria-hidden="true" tabindex="-1"></a>    umap_objects[[<span class="st">&quot;C3_Day55&quot;</span>]],</span>
<span id="cb116-12"><a href="multisample-analysis.html#cb116-12" aria-hidden="true" tabindex="-1"></a>    umap_objects[[<span class="st">&quot;C3_Day80&quot;</span>]],</span>
<span id="cb116-13"><a href="multisample-analysis.html#cb116-13" aria-hidden="true" tabindex="-1"></a>    umap_objects[[<span class="st">&quot;C5_Day80&quot;</span>]]</span>
<span id="cb116-14"><a href="multisample-analysis.html#cb116-14" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb116-15"><a href="multisample-analysis.html#cb116-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">add.cell.ids =</span> <span class="fu">c</span>(</span>
<span id="cb116-16"><a href="multisample-analysis.html#cb116-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;C1_STC&quot;</span>, <span class="st">&quot;C4_STC&quot;</span>, <span class="st">&quot;C2_Day25&quot;</span>, <span class="st">&quot;C5_Day25&quot;</span>,</span>
<span id="cb116-17"><a href="multisample-analysis.html#cb116-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;C2_Day55&quot;</span>, <span class="st">&quot;C3_Day55&quot;</span>, <span class="st">&quot;C3_Day80&quot;</span>, <span class="st">&quot;C5_Day80&quot;</span></span>
<span id="cb116-18"><a href="multisample-analysis.html#cb116-18" aria-hidden="true" tabindex="-1"></a>  ), </span>
<span id="cb116-19"><a href="multisample-analysis.html#cb116-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">&quot;Multi-sample_tutorial&quot;</span></span>
<span id="cb116-20"><a href="multisample-analysis.html#cb116-20" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb116-21"><a href="multisample-analysis.html#cb116-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-22"><a href="multisample-analysis.html#cb116-22" aria-hidden="true" tabindex="-1"></a><span class="co"># create a sample column</span></span>
<span id="cb116-23"><a href="multisample-analysis.html#cb116-23" aria-hidden="true" tabindex="-1"></a>merged_seurat<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(merged_seurat<span class="sc">@</span>meta.data)</span>
<span id="cb116-24"><a href="multisample-analysis.html#cb116-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-25"><a href="multisample-analysis.html#cb116-25" aria-hidden="true" tabindex="-1"></a><span class="do">## split sample column to makw a batch col</span></span>
<span id="cb116-26"><a href="multisample-analysis.html#cb116-26" aria-hidden="true" tabindex="-1"></a>merged_seurat<span class="sc">@</span>meta.data <span class="ot">&lt;-</span> <span class="fu">separate</span>(merged_seurat<span class="sc">@</span>meta.data, <span class="at">col =</span> <span class="st">&#39;sample&#39;</span>, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;batch&#39;</span>, <span class="st">&#39;Day&#39;</span>, <span class="st">&#39;Barcode&#39;</span>), </span>
<span id="cb116-27"><a href="multisample-analysis.html#cb116-27" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>)</span>
<span id="cb116-28"><a href="multisample-analysis.html#cb116-28" aria-hidden="true" tabindex="-1"></a><span class="co">#check some QC metrics </span></span>
<span id="cb116-29"><a href="multisample-analysis.html#cb116-29" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(merged_seurat, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">group.by =</span> <span class="st">&quot;Day&quot;</span>) <span class="sc">+</span> </span>
<span id="cb116-30"><a href="multisample-analysis.html#cb116-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;Vln plots grouped by Day&quot;</span>)</span>
<span id="cb116-31"><a href="multisample-analysis.html#cb116-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-32"><a href="multisample-analysis.html#cb116-32" aria-hidden="true" tabindex="-1"></a><span class="co">#can plot based on any metadat col</span></span>
<span id="cb116-33"><a href="multisample-analysis.html#cb116-33" aria-hidden="true" tabindex="-1"></a><span class="co">#VlnPlot(merged_seurat, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3, group.by = &quot;orig.ident&quot;)</span></span>
<span id="cb116-34"><a href="multisample-analysis.html#cb116-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-35"><a href="multisample-analysis.html#cb116-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of cells per group</span></span>
<span id="cb116-36"><a href="multisample-analysis.html#cb116-36" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged_seurat<span class="sc">$</span>orig.ident)</span>
<span id="cb116-37"><a href="multisample-analysis.html#cb116-37" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged_seurat<span class="sc">$</span>Day)</span>
<span id="cb116-38"><a href="multisample-analysis.html#cb116-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-39"><a href="multisample-analysis.html#cb116-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply normal preprocessing steps to merged file </span></span>
<span id="cb116-40"><a href="multisample-analysis.html#cb116-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-41"><a href="multisample-analysis.html#cb116-41" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(<span class="at">object =</span> merged_seurat)</span>
<span id="cb116-42"><a href="multisample-analysis.html#cb116-42" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(<span class="at">object =</span> merged_seurat)</span>
<span id="cb116-43"><a href="multisample-analysis.html#cb116-43" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(<span class="at">object =</span> merged_seurat) </span>
<span id="cb116-44"><a href="multisample-analysis.html#cb116-44" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(<span class="at">object =</span> merged_seurat)</span>
<span id="cb116-45"><a href="multisample-analysis.html#cb116-45" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(merged_seurat)</span>
<span id="cb116-46"><a href="multisample-analysis.html#cb116-46" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(<span class="at">object =</span> merged_seurat, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">16</span>)</span>
<span id="cb116-47"><a href="multisample-analysis.html#cb116-47" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(<span class="at">object =</span> merged_seurat, <span class="at">resolution =</span> <span class="fl">0.6</span>)</span>
<span id="cb116-48"><a href="multisample-analysis.html#cb116-48" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(<span class="at">object =</span> merged_seurat, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb116-49"><a href="multisample-analysis.html#cb116-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-50"><a href="multisample-analysis.html#cb116-50" aria-hidden="true" tabindex="-1"></a><span class="do">##save merged object</span></span>
<span id="cb116-51"><a href="multisample-analysis.html#cb116-51" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(merged_seurat, file = &quot;./output_files/mutli_sample/merged_seurat.rds&quot;)</span></span></code></pre></div>
</details>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="multisample-analysis.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">00003</span>)</span>
<span id="cb117-2"><a href="multisample-analysis.html#cb117-2" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./output_files/mutli_sample/merged_seurat.rds&quot;</span>)</span>
<span id="cb117-3"><a href="multisample-analysis.html#cb117-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-4"><a href="multisample-analysis.html#cb117-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-5"><a href="multisample-analysis.html#cb117-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Lets plot the merged file. </span></span>
<span id="cb117-6"><a href="multisample-analysis.html#cb117-6" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(merged_seurat, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">features =</span> <span class="st">&#39;nCount_RNA&#39;</span>)</span>
<span id="cb117-7"><a href="multisample-analysis.html#cb117-7" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(merged_seurat, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">features =</span> <span class="st">&#39;nFeature_RNA&#39;</span>)</span>
<span id="cb117-8"><a href="multisample-analysis.html#cb117-8" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(merged_seurat, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">group.by =</span> <span class="st">&#39;Day&#39;</span>)</span>
<span id="cb117-9"><a href="multisample-analysis.html#cb117-9" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(merged_seurat, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">group.by =</span> <span class="st">&#39;orig.ident&#39;</span>)</span>
<span id="cb117-10"><a href="multisample-analysis.html#cb117-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-11"><a href="multisample-analysis.html#cb117-11" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(p1, p2, p3, p4, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/multisample_integration_2-1.png" width="960" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="multisample-analysis.html#cb118-1" aria-hidden="true" tabindex="-1"></a>merged_seurat <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(<span class="at">object =</span> merged_seurat)</span>
<span id="cb118-2"><a href="multisample-analysis.html#cb118-2" aria-hidden="true" tabindex="-1"></a>merged_seurat[[<span class="st">&quot;RNA&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">split</span>(merged_seurat[[<span class="st">&quot;RNA&quot;</span>]], <span class="at">f =</span> merged_seurat<span class="sc">$</span>orig.ident)</span>
<span id="cb118-3"><a href="multisample-analysis.html#cb118-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-4"><a href="multisample-analysis.html#cb118-4" aria-hidden="true" tabindex="-1"></a><span class="co">#perform intergation with Harmony</span></span>
<span id="cb118-5"><a href="multisample-analysis.html#cb118-5" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">IntegrateLayers</span>(<span class="at">object =</span> merged_seurat,</span>
<span id="cb118-6"><a href="multisample-analysis.html#cb118-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> HarmonyIntegration,</span>
<span id="cb118-7"><a href="multisample-analysis.html#cb118-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">orig.reduction =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb118-8"><a href="multisample-analysis.html#cb118-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">new.reduction =</span> <span class="st">&#39;integrated.harm&#39;</span>,</span>
<span id="cb118-9"><a href="multisample-analysis.html#cb118-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">verbose =</span> <span class="cn">FALSE</span>,</span>
<span id="cb118-10"><a href="multisample-analysis.html#cb118-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</details>
<pre><code>## Warning in harmony::HarmonyMatrix(data_mat = Embeddings(object = orig), : HarmonyMatrix is deprecated and will be removed in the future from the API
## in the future</code></pre>
<pre><code>## Warning: Warning: The parameters do_pca and npcs are deprecated. They will be ignored for this function call and please remove parameters do_pca and npcs and pass to harmony cell_embeddings directly.
## This warning is displayed once per session.</code></pre>
<pre><code>## Warning: Warning: The parameter tau is deprecated. It will be ignored for this function call and please remove parameter tau in future function calls. Advanced users can set value of parameter tau by using parameter .options and function harmony_options().
## This warning is displayed once per session.</code></pre>
<pre><code>## Warning: Warning: The parameter block.size is deprecated. It will be ignored for this function call and please remove parameter block.size in future function calls. Advanced users can set value of parameter block.size by using parameter .options and function harmony_options().
## This warning is displayed once per session.</code></pre>
<pre><code>## Warning: Warning: The parameter max.iter.harmony is replaced with parameter max_iter. It will be ignored for this function call and please use parameter max_iter in future function calls.
## This warning is displayed once per session.</code></pre>
<pre><code>## Warning: Warning: The parameter max.iter.cluster is deprecated. It will be ignored for this function call and please remove parameter max.iter.cluster in future function calls. Advanced users can set value of parameter max.iter.cluster by using parameter .options and function harmony_options().
## This warning is displayed once per session.</code></pre>
<pre><code>## Warning: Warning: The parameter epsilon.cluster is deprecated. It will be ignored for this function call and please remove parameter epsilon.cluster in future function calls. Advanced users can set value of parameter epsilon.cluster by using parameter .options and function harmony_options().
## This warning is displayed once per session.</code></pre>
<pre><code>## Warning: Warning: The parameter epsilon.harmony is deprecated. It will be ignored for this function call and please remove parameter epsilon.harmony in future function calls. If users want to control if harmony would stop early or not, use parameter early_stop. Advanced users can set value of parameter epsilon.harmony by using parameter .options and function harmony_options().
## This warning is displayed once per session.</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="multisample-analysis.html#cb127-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(obj, <span class="at">reduction=</span><span class="st">&quot;integrated.harm&quot;</span>)</span>
<span id="cb127-2"><a href="multisample-analysis.html#cb127-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(obj, <span class="at">resolution=</span><span class="fl">0.4</span>, <span class="at">cluster.name=</span><span class="st">&quot;harm_cluster&quot;</span>)</span></code></pre></div>
</details>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 7216
## Number of edges: 233507
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.9338
## Number of communities: 10
## Elapsed time: 0 seconds</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="multisample-analysis.html#cb129-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">reduction=</span><span class="st">&quot;integrated.harm&quot;</span>, <span class="at">dims=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">reduction.name =</span> <span class="st">&quot;umap.harm&quot;</span>)</span>
<span id="cb129-2"><a href="multisample-analysis.html#cb129-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-3"><a href="multisample-analysis.html#cb129-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-4"><a href="multisample-analysis.html#cb129-4" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction =</span> <span class="st">&quot;umap.harm&quot;</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">&quot;Day&quot;</span>, <span class="st">&quot;orig.ident&quot;</span>, <span class="st">&quot;harm_cluster&quot;</span>), <span class="at">label =</span> T)  </span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/multisample_integration_2-2.png" width="960" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="multisample-analysis.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="do">### save object </span></span>
<span id="cb130-2"><a href="multisample-analysis.html#cb130-2" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(obj, file = &quot;./output_files/mutli_sample/integrated_harm_seurat.rds&quot;)</span></span></code></pre></div>
</details>
</div>
<div id="add-isoform-counts" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Add isoform counts<a href="multisample-analysis.html#add-isoform-counts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we can add isoform level information to the merged Seurat object. When doing this you may need to convert the oafish files into a count.csv file the row names in the ENSTID_gene_symbol ID form. The function to do so can be found here @ref(Convert Oarfish files to count matrix)</p>
<p>The workflow follows these steps.</p>
<ol style="list-style-type: decimal">
<li>Create a Seurat object per sample</li>
<li>Merge the objects together</li>
<li>Filter the isoform merged object based on cell types that are high quality (based on gene level filtering)</li>
<li>Add the isoform filtered and merged object to the gene level object as a new assay</li>
<li>Integration on the isoform assay is possible but not shown here.</li>
</ol>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="multisample-analysis.html#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="do">############ read in count matix and add isoform assay to Seurat object  #########</span></span>
<span id="cb131-2"><a href="multisample-analysis.html#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Define sample names</span></span>
<span id="cb131-3"><a href="multisample-analysis.html#cb131-3" aria-hidden="true" tabindex="-1"></a>sample_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;C1_STC&quot;</span>, <span class="st">&quot;C4_STC&quot;</span>, <span class="st">&quot;C2_Day25&quot;</span>, <span class="st">&quot;C5_Day25&quot;</span>, <span class="st">&quot;C2_Day55&quot;</span>, <span class="st">&quot;C3_Day55&quot;</span>, <span class="st">&quot;C3_Day80&quot;</span>, <span class="st">&quot;C5_Day80&quot;</span>)</span>
<span id="cb131-4"><a href="multisample-analysis.html#cb131-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-5"><a href="multisample-analysis.html#cb131-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Make Seurat objects not filtered</span></span>
<span id="cb131-6"><a href="multisample-analysis.html#cb131-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to read a CSV file and create a Seurat object</span></span>
<span id="cb131-7"><a href="multisample-analysis.html#cb131-7" aria-hidden="true" tabindex="-1"></a>create_seurat_object <span class="ot">&lt;-</span> <span class="cf">function</span>(sample_name, input_dir, project_name) {</span>
<span id="cb131-8"><a href="multisample-analysis.html#cb131-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct file path</span></span>
<span id="cb131-9"><a href="multisample-analysis.html#cb131-9" aria-hidden="true" tabindex="-1"></a>  file_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_dir, <span class="fu">paste0</span>(<span class="st">&quot;gene_symbol_oarfish_&quot;</span>, sample_name, <span class="st">&quot;_counts.csv&quot;</span>))</span>
<span id="cb131-10"><a href="multisample-analysis.html#cb131-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb131-11"><a href="multisample-analysis.html#cb131-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read the CSV file</span></span>
<span id="cb131-12"><a href="multisample-analysis.html#cb131-12" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">fread</span>(file_path)</span>
<span id="cb131-13"><a href="multisample-analysis.html#cb131-13" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts) </span>
<span id="cb131-14"><a href="multisample-analysis.html#cb131-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(counts) <span class="ot">&lt;-</span> counts[[<span class="dv">1</span>]]            <span class="co"># Set row names from the first column</span></span>
<span id="cb131-15"><a href="multisample-analysis.html#cb131-15" aria-hidden="true" tabindex="-1"></a>  counts[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="cn">NULL</span>     </span>
<span id="cb131-16"><a href="multisample-analysis.html#cb131-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb131-17"><a href="multisample-analysis.html#cb131-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the Seurat object</span></span>
<span id="cb131-18"><a href="multisample-analysis.html#cb131-18" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> counts, <span class="at">project =</span> project_name, <span class="at">min.cells =</span> <span class="dv">5</span>, <span class="at">min.features =</span> <span class="dv">500</span>)</span>
<span id="cb131-19"><a href="multisample-analysis.html#cb131-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb131-20"><a href="multisample-analysis.html#cb131-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(seurat_obj)</span>
<span id="cb131-21"><a href="multisample-analysis.html#cb131-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb131-22"><a href="multisample-analysis.html#cb131-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-23"><a href="multisample-analysis.html#cb131-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Directory where the CSV files are stored</span></span>
<span id="cb131-24"><a href="multisample-analysis.html#cb131-24" aria-hidden="true" tabindex="-1"></a>input_directory <span class="ot">&lt;-</span> <span class="st">&quot;./output_files/mutli_sample/oarfish_counts/&quot;</span></span>
<span id="cb131-25"><a href="multisample-analysis.html#cb131-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-26"><a href="multisample-analysis.html#cb131-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Create an empty list to store Seurat objects</span></span>
<span id="cb131-27"><a href="multisample-analysis.html#cb131-27" aria-hidden="true" tabindex="-1"></a>iso_seurat_objects <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb131-28"><a href="multisample-analysis.html#cb131-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-29"><a href="multisample-analysis.html#cb131-29" aria-hidden="true" tabindex="-1"></a>total_samples <span class="ot">&lt;-</span> <span class="fu">length</span>(sample_names)</span>
<span id="cb131-30"><a href="multisample-analysis.html#cb131-30" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(sample_names)) {</span>
<span id="cb131-31"><a href="multisample-analysis.html#cb131-31" aria-hidden="true" tabindex="-1"></a>  sample <span class="ot">&lt;-</span> sample_names[i]</span>
<span id="cb131-32"><a href="multisample-analysis.html#cb131-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb131-33"><a href="multisample-analysis.html#cb131-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">sprintf</span>(<span class="st">&quot;Processing sample %d of %d: %s&quot;</span>, i, total_samples, sample))</span>
<span id="cb131-34"><a href="multisample-analysis.html#cb131-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb131-35"><a href="multisample-analysis.html#cb131-35" aria-hidden="true" tabindex="-1"></a>  iso_seurat_objects[[sample]] <span class="ot">&lt;-</span> <span class="fu">create_seurat_object</span>(</span>
<span id="cb131-36"><a href="multisample-analysis.html#cb131-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">sample_name =</span> sample,</span>
<span id="cb131-37"><a href="multisample-analysis.html#cb131-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">input_dir =</span> input_directory,</span>
<span id="cb131-38"><a href="multisample-analysis.html#cb131-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">project_name =</span> sample</span>
<span id="cb131-39"><a href="multisample-analysis.html#cb131-39" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb131-40"><a href="multisample-analysis.html#cb131-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb131-41"><a href="multisample-analysis.html#cb131-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-42"><a href="multisample-analysis.html#cb131-42" aria-hidden="true" tabindex="-1"></a><span class="do">####### Merge the samples into one object ######</span></span>
<span id="cb131-43"><a href="multisample-analysis.html#cb131-43" aria-hidden="true" tabindex="-1"></a>iso.merged_seurat <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb131-44"><a href="multisample-analysis.html#cb131-44" aria-hidden="true" tabindex="-1"></a>  iso_seurat_objects[[<span class="st">&quot;C1_STC&quot;</span>]],</span>
<span id="cb131-45"><a href="multisample-analysis.html#cb131-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">list</span>(</span>
<span id="cb131-46"><a href="multisample-analysis.html#cb131-46" aria-hidden="true" tabindex="-1"></a>    iso_seurat_objects[[<span class="st">&quot;C4_STC&quot;</span>]],</span>
<span id="cb131-47"><a href="multisample-analysis.html#cb131-47" aria-hidden="true" tabindex="-1"></a>    iso_seurat_objects[[<span class="st">&quot;C2_Day25&quot;</span>]],</span>
<span id="cb131-48"><a href="multisample-analysis.html#cb131-48" aria-hidden="true" tabindex="-1"></a>    iso_seurat_objects[[<span class="st">&quot;C5_Day25&quot;</span>]],</span>
<span id="cb131-49"><a href="multisample-analysis.html#cb131-49" aria-hidden="true" tabindex="-1"></a>    iso_seurat_objects[[<span class="st">&quot;C2_Day55&quot;</span>]],</span>
<span id="cb131-50"><a href="multisample-analysis.html#cb131-50" aria-hidden="true" tabindex="-1"></a>    iso_seurat_objects[[<span class="st">&quot;C3_Day55&quot;</span>]],</span>
<span id="cb131-51"><a href="multisample-analysis.html#cb131-51" aria-hidden="true" tabindex="-1"></a>    iso_seurat_objects[[<span class="st">&quot;C3_Day80&quot;</span>]],</span>
<span id="cb131-52"><a href="multisample-analysis.html#cb131-52" aria-hidden="true" tabindex="-1"></a>    iso_seurat_objects[[<span class="st">&quot;C5_Day80&quot;</span>]]</span>
<span id="cb131-53"><a href="multisample-analysis.html#cb131-53" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb131-54"><a href="multisample-analysis.html#cb131-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">add.cell.ids =</span> <span class="fu">c</span>(</span>
<span id="cb131-55"><a href="multisample-analysis.html#cb131-55" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;C1_STC&quot;</span>, <span class="st">&quot;C4_STC&quot;</span>, <span class="st">&quot;C2_Day25&quot;</span>, <span class="st">&quot;C5_Day25&quot;</span>,</span>
<span id="cb131-56"><a href="multisample-analysis.html#cb131-56" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;C2_Day55&quot;</span>, <span class="st">&quot;C3_Day55&quot;</span>, <span class="st">&quot;C3_Day80&quot;</span>, <span class="st">&quot;C5_Day80&quot;</span></span>
<span id="cb131-57"><a href="multisample-analysis.html#cb131-57" aria-hidden="true" tabindex="-1"></a>  ), </span>
<span id="cb131-58"><a href="multisample-analysis.html#cb131-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">&quot;iso-Multi-sample_tutorial&quot;</span></span>
<span id="cb131-59"><a href="multisample-analysis.html#cb131-59" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb131-60"><a href="multisample-analysis.html#cb131-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-61"><a href="multisample-analysis.html#cb131-61" aria-hidden="true" tabindex="-1"></a><span class="co"># create a sample column</span></span>
<span id="cb131-62"><a href="multisample-analysis.html#cb131-62" aria-hidden="true" tabindex="-1"></a>iso.merged_seurat<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(iso.merged_seurat<span class="sc">@</span>meta.data)</span>
<span id="cb131-63"><a href="multisample-analysis.html#cb131-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-64"><a href="multisample-analysis.html#cb131-64" aria-hidden="true" tabindex="-1"></a><span class="co"># split sample column to makw a batch col</span></span>
<span id="cb131-65"><a href="multisample-analysis.html#cb131-65" aria-hidden="true" tabindex="-1"></a>iso.merged_seurat<span class="sc">@</span>meta.data <span class="ot">&lt;-</span> <span class="fu">separate</span>(iso.merged_seurat<span class="sc">@</span>meta.data, <span class="at">col =</span> <span class="st">&#39;sample&#39;</span>, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;batch&#39;</span>, <span class="st">&#39;Day&#39;</span>, <span class="st">&#39;Barcode&#39;</span>), </span>
<span id="cb131-66"><a href="multisample-analysis.html#cb131-66" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>)</span>
<span id="cb131-67"><a href="multisample-analysis.html#cb131-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-68"><a href="multisample-analysis.html#cb131-68" aria-hidden="true" tabindex="-1"></a><span class="do">## filter the data to iso and gene cells match</span></span>
<span id="cb131-69"><a href="multisample-analysis.html#cb131-69" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">subset</span>(iso.merged_seurat, <span class="at">cells =</span>obj<span class="sc">@</span>graphs[[<span class="st">&quot;RNA_nn&quot;</span>]]<span class="sc">@</span>Dimnames[[<span class="dv">1</span>]])</span>
<span id="cb131-70"><a href="multisample-analysis.html#cb131-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-71"><a href="multisample-analysis.html#cb131-71" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(merged_seurat_isoform_filtered, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb131-72"><a href="multisample-analysis.html#cb131-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-73"><a href="multisample-analysis.html#cb131-73" aria-hidden="true" tabindex="-1"></a><span class="co"># perform standard workflow steps</span></span>
<span id="cb131-74"><a href="multisample-analysis.html#cb131-74" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(<span class="at">object =</span> merged_seurat_isoform_filtered) <span class="co"># if using SCT dont run this</span></span>
<span id="cb131-75"><a href="multisample-analysis.html#cb131-75" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(<span class="at">object =</span> merged_seurat_isoform_filtered) <span class="co"># if using SCT dont run this</span></span>
<span id="cb131-76"><a href="multisample-analysis.html#cb131-76" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(<span class="at">object =</span> merged_seurat_isoform_filtered) <span class="co"># if using SCT dont run this</span></span>
<span id="cb131-77"><a href="multisample-analysis.html#cb131-77" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(<span class="at">object =</span> merged_seurat_isoform_filtered)</span>
<span id="cb131-78"><a href="multisample-analysis.html#cb131-78" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(merged_seurat_isoform_filtered)</span>
<span id="cb131-79"><a href="multisample-analysis.html#cb131-79" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(<span class="at">object =</span> merged_seurat_isoform_filtered, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb131-80"><a href="multisample-analysis.html#cb131-80" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(<span class="at">object =</span> merged_seurat_isoform_filtered, <span class="at">resolution =</span> <span class="fl">0.6</span>)</span>
<span id="cb131-81"><a href="multisample-analysis.html#cb131-81" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(<span class="at">object =</span> merged_seurat_isoform_filtered, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb131-82"><a href="multisample-analysis.html#cb131-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-83"><a href="multisample-analysis.html#cb131-83" aria-hidden="true" tabindex="-1"></a><span class="co">#Save file</span></span>
<span id="cb131-84"><a href="multisample-analysis.html#cb131-84" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(merged_seurat_isoform_filtered, file = &quot;./output_files/mutli_sample/merged_seurat_isoform_filtered.rds&quot;)</span></span></code></pre></div>
</details>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="multisample-analysis.html#cb132-1" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;./output_files/mutli_sample/merged_seurat_isoform_filtered.rds&quot;</span>)</span>
<span id="cb132-2"><a href="multisample-analysis.html#cb132-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-3"><a href="multisample-analysis.html#cb132-3" aria-hidden="true" tabindex="-1"></a><span class="co">#plots</span></span>
<span id="cb132-4"><a href="multisample-analysis.html#cb132-4" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(merged_seurat_isoform_filtered, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">group.by =</span> <span class="st">&#39;orig.ident&#39;</span>)</span>
<span id="cb132-5"><a href="multisample-analysis.html#cb132-5" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(merged_seurat_isoform_filtered, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">group.by =</span> <span class="st">&#39;Day&#39;</span>)</span>
<span id="cb132-6"><a href="multisample-analysis.html#cb132-6" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(merged_seurat_isoform_filtered, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">features =</span> <span class="st">&#39;nCount_RNA&#39;</span>)</span>
<span id="cb132-7"><a href="multisample-analysis.html#cb132-7" aria-hidden="true" tabindex="-1"></a>p8 <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(merged_seurat_isoform_filtered, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">features =</span> <span class="st">&#39;nFeature_RNA&#39;</span>)</span>
<span id="cb132-8"><a href="multisample-analysis.html#cb132-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-9"><a href="multisample-analysis.html#cb132-9" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(p5, p6, p7, p8, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/multisample_add_iso_counts_2-1.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="multisample-analysis.html#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="do">#####</span></span>
<span id="cb133-2"><a href="multisample-analysis.html#cb133-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-3"><a href="multisample-analysis.html#cb133-3" aria-hidden="true" tabindex="-1"></a>merged_seurat_isoform_filtered <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(merged_seurat_isoform_filtered)</span>
<span id="cb133-4"><a href="multisample-analysis.html#cb133-4" aria-hidden="true" tabindex="-1"></a>counts_table <span class="ot">&lt;-</span> merged_seurat_isoform_filtered[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">$</span>counts</span>
<span id="cb133-5"><a href="multisample-analysis.html#cb133-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-6"><a href="multisample-analysis.html#cb133-6" aria-hidden="true" tabindex="-1"></a>obj[[<span class="st">&quot;iso&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">CreateAssay5Object</span>(<span class="at">counts =</span> counts_table)</span>
<span id="cb133-7"><a href="multisample-analysis.html#cb133-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-8"><a href="multisample-analysis.html#cb133-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Normalize the new assay data</span></span>
<span id="cb133-9"><a href="multisample-analysis.html#cb133-9" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(obj, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>)</span>
<span id="cb133-10"><a href="multisample-analysis.html#cb133-10" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(obj, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>)</span>
<span id="cb133-11"><a href="multisample-analysis.html#cb133-11" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(obj, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>)</span>
<span id="cb133-12"><a href="multisample-analysis.html#cb133-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-13"><a href="multisample-analysis.html#cb133-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 4: Perform PCA</span></span>
<span id="cb133-14"><a href="multisample-analysis.html#cb133-14" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(obj, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">reduction.name =</span> <span class="st">&quot;pca_iso&quot;</span>)</span></code></pre></div>
</details>
<pre><code>## Warning: Key &#39;PC_&#39; taken, using &#39;pcaiso_&#39; instead</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="multisample-analysis.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 5: Run UMAP</span></span>
<span id="cb135-2"><a href="multisample-analysis.html#cb135-2" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(obj, <span class="at">reduction =</span> <span class="st">&quot;pca_iso&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">reduction.name =</span> <span class="st">&quot;umap_iso&quot;</span>)</span>
<span id="cb135-3"><a href="multisample-analysis.html#cb135-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-4"><a href="multisample-analysis.html#cb135-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize the UMAP</span></span>
<span id="cb135-5"><a href="multisample-analysis.html#cb135-5" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">reduction =</span> <span class="st">&quot;umap_iso&quot;</span>) <span class="sc">|</span> <span class="fu">DimPlot</span>(obj, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">reduction =</span> <span class="st">&quot;umap.harm&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/multisample_add_iso_counts_2-2.png" width="672" /></p>
<p>Here will annotate the cell types using the sctype and some prior knowledge.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="multisample-analysis.html#cb136-1" aria-hidden="true" tabindex="-1"></a>gs_removal_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Tanycytes&quot;</span>, <span class="st">&quot;Dopaminergic neurons&quot;</span>, <span class="st">&quot;Oligodendrocyte precursor cells&quot;</span>, <span class="st">&quot;Non myelinating Schwann cells&quot;</span>, <span class="st">&quot;Endothelial cells&quot;</span>) </span>
<span id="cb136-2"><a href="multisample-analysis.html#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="co"># list of cell types from the db to remove</span></span>
<span id="cb136-3"><a href="multisample-analysis.html#cb136-3" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">perform_sctype_analysis</span>(obj, db_, tissue, gs_removal_list, </span>
<span id="cb136-4"><a href="multisample-analysis.html#cb136-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">metadat_col_prefix =</span><span class="st">&quot;sctype_db&quot;</span>, <span class="at">figure_prefix =</span> <span class="st">&quot;multi&quot;</span>,</span>
<span id="cb136-5"><a href="multisample-analysis.html#cb136-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">output_file =</span> <span class="st">&quot;multi&quot;</span>, <span class="at">cluster_res =</span> <span class="st">&quot;harm_cluster&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap.harm&quot;</span>)</span></code></pre></div>
</details>
<pre><code>## # A tibble: 10 × 3
## # Groups:   cluster [10]
##    cluster type               scores
##    &lt;fct&gt;   &lt;chr&gt;               &lt;dbl&gt;
##  1 7       Cancer stem cells    713.
##  2 1       Cancer stem cells    819.
##  3 4       Radial glial cells  1136.
##  4 2       Radial glial cells   896.
##  5 9       Immature neurons     486.
##  6 3       Mature neurons       979.
##  7 0       GABAergic neurons   1268.
##  8 5       GABAergic neurons    792.
##  9 8       Radial glial cells  1506.
## 10 6       Microglial cells     408.</code></pre>
<p><img src="bookdownproj_files/figure-html/sctype_multi-1.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="multisample-analysis.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="co">#change name of cancer stem cell to stem cell</span></span>
<span id="cb138-2"><a href="multisample-analysis.html#cb138-2" aria-hidden="true" tabindex="-1"></a>obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_db <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;Cancer stem cells&quot;</span>, <span class="st">&quot;Stem cells&quot;</span>, obj<span class="sc">@</span>meta.data<span class="sc">$</span>sctype_db)</span>
<span id="cb138-3"><a href="multisample-analysis.html#cb138-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-4"><a href="multisample-analysis.html#cb138-4" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(obj, <span class="at">reduction=</span><span class="st">&quot;umap.harm&quot;</span>, <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span>, <span class="at">label =</span> T)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/sctype_multi-2.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="multisample-analysis.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(obj, file = &quot;./output_files/mutli_sample/multisample_seurat.intergrated_harm.isofrom.rds&quot;)</span></span></code></pre></div>
</details>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="novel-isoforms.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="trajectory-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/08_multi_sample_chapter_1.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/08_multi_sample_chapter_1.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
