<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Novel isofroms - Working | FLAMES Single-Cell RNA-seq Tutorial</title>
  <meta name="description" content="Chapter 7 Novel isofroms - Working | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Novel isofroms - Working | FLAMES Single-Cell RNA-seq Tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Novel isofroms - Working | FLAMES Single-Cell RNA-seq Tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer" />


<meta name="date" content="2025-01-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="exploring-isoforms-of-interest.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES single sample RNA-seq Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> Setup</a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> Removing sources of unwanted noise from the single cell dataset</a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> Add isoform counts to Seurat object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> Finding differentially expressed genes and isoforms</a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#de-genes-and-isofroms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isofroms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-differentially-expressed-genes-and-isoforms.html"><a href="finding-differentially-expressed-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> Exploring isoforms of interest</a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html"><i class="fa fa-check"></i><b>7</b> Novel isofroms - Working</a>
<ul>
<li class="chapter" data-level="7.1" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>7.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="7.2" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#add-analysis"><i class="fa fa-check"></i><b>7.2</b> add analysis</a></li>
<li class="chapter" data-level="7.3" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#isoform-proportion-analysis"><i class="fa fa-check"></i><b>7.3</b> <strong>Isoform Proportion Analysis</strong></a></li>
<li class="chapter" data-level="7.4" data-path="novel-isofroms---working.html"><a href="novel-isofroms---working.html#calculate-the-proportion-of-novel-isoforms-relative-to-total-gene-expression."><i class="fa fa-check"></i><b>7.4</b> Calculate the proportion of novel isoforms relative to total gene expression.</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>8</b> Session info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">FLAMES Single-Cell RNA-seq Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="novel-isofroms---working" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Novel isofroms - Working<a href="novel-isofroms---working.html#novel-isofroms---working" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>With Long read sequencing its possible to explore both known and novel isofroms. In FLAMES we use Bambu <span class="citation">(<a href="#ref-chen2023" role="doc-biblioref">Chen et al., 2023</a>)</span> Novel isoform can be interesting feature to explore and here we will provide some simple analysis that one can use to extract novel isoforms and explore them in some more detail.</p>
<div id="find-all-the-genes-that-express-at-least-1-novel-isoform" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Find all the genes that express at least 1 novel isoform<a href="novel-isofroms---working.html#find-all-the-genes-that-express-at-least-1-novel-isoform" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First lets get a list of genes with at least one novel isoform</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="novel-isofroms---working.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the genes that express at least one novel isoform and save them to a list</span></span>
<span id="cb137-2"><a href="novel-isofroms---working.html#cb137-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert row names to a data frame</span></span>
<span id="cb137-3"><a href="novel-isofroms---working.html#cb137-3" aria-hidden="true" tabindex="-1"></a>isoform_ids <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">row.names</span>(seu_obj<span class="sc">@</span>assays<span class="sc">$</span>iso<span class="sc">$</span>counts))</span>
<span id="cb137-4"><a href="novel-isofroms---working.html#cb137-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(isoform_ids) <span class="ot">&lt;-</span> <span class="st">&quot;IDs&quot;</span></span>
<span id="cb137-5"><a href="novel-isofroms---working.html#cb137-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-6"><a href="novel-isofroms---working.html#cb137-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Add in total expression  </span></span>
<span id="cb137-7"><a href="novel-isofroms---working.html#cb137-7" aria-hidden="true" tabindex="-1"></a>isoform_ids<span class="sc">$</span>Total_Expression <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(seu_obj<span class="sc">@</span>assays<span class="sc">$</span>iso<span class="sc">$</span>counts)</span>
<span id="cb137-8"><a href="novel-isofroms---working.html#cb137-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-9"><a href="novel-isofroms---working.html#cb137-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter rows where &#39;IDs&#39; contains the string &quot;Bambu&quot;</span></span>
<span id="cb137-10"><a href="novel-isofroms---working.html#cb137-10" aria-hidden="true" tabindex="-1"></a>isoform_ids <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(isoform_ids[<span class="fu">grepl</span>(<span class="st">&quot;Bambu&quot;</span>, isoform_ids<span class="sc">$</span>IDs), ])  <span class="co"># can comment this out if we want to do this for all isofroms </span></span>
<span id="cb137-11"><a href="novel-isofroms---working.html#cb137-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-12"><a href="novel-isofroms---working.html#cb137-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate the &#39;IDs&#39; column into two columns </span></span>
<span id="cb137-13"><a href="novel-isofroms---working.html#cb137-13" aria-hidden="true" tabindex="-1"></a>filtered_df <span class="ot">&lt;-</span> isoform_ids <span class="sc">%&gt;%</span> <span class="fu">separate</span>(IDs, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;transcript_id&quot;</span>, <span class="st">&quot;gene_id&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;-&quot;</span>,  <span class="at">extra =</span> <span class="st">&quot;merge&quot;</span>)</span>
<span id="cb137-14"><a href="novel-isofroms---working.html#cb137-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-15"><a href="novel-isofroms---working.html#cb137-15" aria-hidden="true" tabindex="-1"></a>isoform_counts <span class="ot">&lt;-</span> filtered_df <span class="sc">%&gt;%</span></span>
<span id="cb137-16"><a href="novel-isofroms---working.html#cb137-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb137-17"><a href="novel-isofroms---working.html#cb137-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb137-18"><a href="novel-isofroms---working.html#cb137-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">Isoform_Count =</span> <span class="fu">n</span>(),</span>
<span id="cb137-19"><a href="novel-isofroms---working.html#cb137-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">Total_Expression =</span> <span class="fu">sum</span>(Total_Expression)</span>
<span id="cb137-20"><a href="novel-isofroms---working.html#cb137-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb137-21"><a href="novel-isofroms---working.html#cb137-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Total_Expression))</span>
<span id="cb137-22"><a href="novel-isofroms---working.html#cb137-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-23"><a href="novel-isofroms---working.html#cb137-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-24"><a href="novel-isofroms---working.html#cb137-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Print or return the filtered data frame</span></span>
<span id="cb137-25"><a href="novel-isofroms---working.html#cb137-25" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(isoform_counts)</span></code></pre></div>
</details>
<pre><code>## # A tibble: 31 × 3
##    gene_id         Isoform_Count Total_Expression
##    &lt;chr&gt;                   &lt;int&gt;            &lt;dbl&gt;
##  1 OAZ2                        1            396. 
##  2 ENSG00000291214             1            159. 
##  3 BambuGene6113               1            141. 
##  4 ENSG00000305512             1            140. 
##  5 BambuGene19505              1            139. 
##  6 ENSG00000305069             1            132  
##  7 BambuGene18812              1             95.8
##  8 ZMAT5                       1             76.0
##  9 ENSG00000295459             1             68.7
## 10 ENSG00000231748             1             62  
## # ℹ 21 more rows</code></pre>
<p>In this table we can see that we have a total of 31 novel isofroms. This number is smaller than we might expect but this is because Bambu is quite conservative for single cell data. Users can change the sensitivity of Bambu when running FLAMES by setting the <code>NDR</code> parameter or using a different discovery method like stringtie2 <span class="citation">(<a href="#ref-kovaka2019" role="doc-biblioref">Kovaka et al., 2019</a>)</span>.</p>
<p>In the above table we have ordered the the genes that contain a novel isoform by total expression. OAZ2 is our top hit. We can look at the isoform structure using IsoViz as described in the section <a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures">6.5</a>.</p>
<p>Examining <strong>OAZ2</strong> in more detail reveals that the primary distinction between the novel isoform and the canonical isoform, <strong>OAZ2-201</strong>, is a ~65 bp variation in the 5’ UTR. This minor alteration does not affect the canonical open reading frame (ORF). The total expression of this gene by isoform, we observe that the vast majority of its expression is attributed to the <strong>BambuTx25</strong> isoform. However, it is challenging to determine whether this isoform is truly novel. The small change in the UTR might simply result from incomplete reads failing to capture the full 5’ end of the canonical isoform.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="novel-isofroms---working.html#cb139-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/IsoVis_ENSG00000180304.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:isoviz-plot2"></span>
<img src="images/IsoVis_ENSG00000180304.png" alt="IsoViz visualization of Novel OAZ2 isoform." width="1000px" />
<p class="caption">
Figure 7.1: IsoViz visualization of Novel OAZ2 isoform.
</p>
</div>
<p>We could explore this in some more detail and check to see if we can find any interesting regulatory elements in this missing section of the 5’UTR.</p>
</div>
<div id="add-analysis" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> add analysis<a href="novel-isofroms---working.html#add-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="novel-isofroms---working.html#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load necessary libraries</span></span></code></pre></div>
</details>
<p>We can further examine additional genes from our list. For instance, ZMAT5 has a novel isoform that utilizes a different, known splice junction in the 5’ UTR. Interestingly, this splice junction is also found in <strong>ENST00000489010</strong>, a transcript known to undergo nonsense-mediated decay (NMD) and which lacks a functional open reading frame (ORF). Notably, the novel isoform, <strong>BambuTx32</strong>, is the dominantly expressed isoform across all cell types, which is unusual since the canonical isoform typically accounts for the majority of reads.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="novel-isofroms---working.html#cb141-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/IsoVis_ENSG00000100319.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:isoviz-plot3"></span>
<img src="images/IsoVis_ENSG00000100319.png" alt="IsoViz visualization of Novel ZMAT5 isoform." width="1000px" />
<p class="caption">
Figure 7.2: IsoViz visualization of Novel ZMAT5 isoform.
</p>
</div>
<p>We have looked at two examples here and it is possible to explore these isofroms one at time. But in general an overview of the novel isoforms in our data could be helpful. Some useful tools to explore</p>
<p>Next we can look at the number of novel isofroms in each cluster</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="novel-isofroms---working.html#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of novel isofrm expressed in each clsuter </span></span>
<span id="cb142-2"><a href="novel-isofroms---working.html#cb142-2" aria-hidden="true" tabindex="-1"></a><span class="co"># cut of is at least 10% of cells need 5 counts or more</span></span>
<span id="cb142-3"><a href="novel-isofroms---working.html#cb142-3" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">GetAssayData</span>(seu_obj, <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>))</span>
<span id="cb142-4"><a href="novel-isofroms---working.html#cb142-4" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> seu_obj<span class="sc">$</span>sctype_db</span>
<span id="cb142-5"><a href="novel-isofroms---working.html#cb142-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-6"><a href="novel-isofroms---working.html#cb142-6" aria-hidden="true" tabindex="-1"></a>novel_isoforms <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Bambu&quot;</span>, <span class="fu">rownames</span>(counts), <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb142-7"><a href="novel-isofroms---working.html#cb142-7" aria-hidden="true" tabindex="-1"></a>novel_counts <span class="ot">&lt;-</span> counts[novel_isoforms, ]</span>
<span id="cb142-8"><a href="novel-isofroms---working.html#cb142-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-9"><a href="novel-isofroms---working.html#cb142-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list to store isoforms for each cluster</span></span>
<span id="cb142-10"><a href="novel-isofroms---working.html#cb142-10" aria-hidden="true" tabindex="-1"></a>isoforms_by_cluster <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb142-11"><a href="novel-isofroms---working.html#cb142-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-12"><a href="novel-isofroms---working.html#cb142-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over each cluster</span></span>
<span id="cb142-13"><a href="novel-isofroms---working.html#cb142-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cluster <span class="cf">in</span> <span class="fu">unique</span>(clusters)) {</span>
<span id="cb142-14"><a href="novel-isofroms---working.html#cb142-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset the cells belonging to the current cluster</span></span>
<span id="cb142-15"><a href="novel-isofroms---working.html#cb142-15" aria-hidden="true" tabindex="-1"></a>  cluster_cells <span class="ot">&lt;-</span> <span class="fu">which</span>(clusters <span class="sc">==</span> cluster)</span>
<span id="cb142-16"><a href="novel-isofroms---working.html#cb142-16" aria-hidden="true" tabindex="-1"></a>  cluster_counts <span class="ot">&lt;-</span> novel_counts[, cluster_cells]</span>
<span id="cb142-17"><a href="novel-isofroms---working.html#cb142-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb142-18"><a href="novel-isofroms---working.html#cb142-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check how many cells have counts &gt;= 5 for each isoform</span></span>
<span id="cb142-19"><a href="novel-isofroms---working.html#cb142-19" aria-hidden="true" tabindex="-1"></a>  expressed_in_cells <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(cluster_counts <span class="sc">&gt;=</span> <span class="dv">2</span>)</span>
<span id="cb142-20"><a href="novel-isofroms---working.html#cb142-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb142-21"><a href="novel-isofroms---working.html#cb142-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate the percentage of cells expressing each isoform</span></span>
<span id="cb142-22"><a href="novel-isofroms---working.html#cb142-22" aria-hidden="true" tabindex="-1"></a>  percent_cells_expressing <span class="ot">&lt;-</span> expressed_in_cells <span class="sc">/</span> <span class="fu">length</span>(cluster_cells) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb142-23"><a href="novel-isofroms---working.html#cb142-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb142-24"><a href="novel-isofroms---working.html#cb142-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select isoforms expressed in at least 10% of cells</span></span>
<span id="cb142-25"><a href="novel-isofroms---working.html#cb142-25" aria-hidden="true" tabindex="-1"></a>  expressed_isoforms <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cluster_counts)[percent_cells_expressing <span class="sc">&gt;=</span> <span class="dv">10</span>]</span>
<span id="cb142-26"><a href="novel-isofroms---working.html#cb142-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb142-27"><a href="novel-isofroms---working.html#cb142-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the list of expressed isoforms for this cluster</span></span>
<span id="cb142-28"><a href="novel-isofroms---working.html#cb142-28" aria-hidden="true" tabindex="-1"></a>  isoforms_by_cluster[[<span class="fu">paste0</span>(<span class="st">&quot;Cluster_&quot;</span>, cluster)]] <span class="ot">&lt;-</span> expressed_isoforms</span>
<span id="cb142-29"><a href="novel-isofroms---working.html#cb142-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb142-30"><a href="novel-isofroms---working.html#cb142-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-31"><a href="novel-isofroms---working.html#cb142-31" aria-hidden="true" tabindex="-1"></a><span class="co"># View the list of isoforms for each cluster</span></span>
<span id="cb142-32"><a href="novel-isofroms---working.html#cb142-32" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(isoforms_by_cluster)</span></code></pre></div>
</details>
<pre><code>## $`Cluster_Immature neurons`
## [1] &quot;BambuTx7-BambuGene6113&quot; &quot;BambuTx25-OAZ2&quot;        
## 
## $`Cluster_Myelinating Schwann cells`
## [1] &quot;BambuTx25-OAZ2&quot;
## 
## $`Cluster_Glutamatergic neurons`
## [1] &quot;BambuTx3-ENSG00000295459&quot;  &quot;BambuTx14-ENSG00000305069&quot; &quot;BambuTx25-OAZ2&quot;           
## 
## $`Cluster_Radial glial cells`
## [1] &quot;BambuTx6-ENSG00000305512&quot;  &quot;BambuTx7-BambuGene6113&quot;    &quot;BambuTx25-OAZ2&quot;            &quot;BambuTx26-ENSG00000291214&quot;
## [5] &quot;BambuTx31-BambuGene18812&quot;  &quot;BambuTx32-ZMAT5&quot;          
## 
## $`Cluster_GABAergic neurons`
## [1] &quot;BambuTx25-OAZ2&quot;            &quot;BambuTx26-ENSG00000291214&quot;</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="novel-isofroms---working.html#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace `seurat_obj` with the name of your Seurat object</span></span>
<span id="cb144-2"><a href="novel-isofroms---working.html#cb144-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique cell types</span></span>
<span id="cb144-3"><a href="novel-isofroms---working.html#cb144-3" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">unique</span>(seu_obj<span class="sc">$</span>sctype_db)</span>
<span id="cb144-4"><a href="novel-isofroms---working.html#cb144-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-5"><a href="novel-isofroms---working.html#cb144-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to store unique features for each cell type</span></span>
<span id="cb144-6"><a href="novel-isofroms---working.html#cb144-6" aria-hidden="true" tabindex="-1"></a>unique_features <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb144-7"><a href="novel-isofroms---working.html#cb144-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-8"><a href="novel-isofroms---working.html#cb144-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each cell type</span></span>
<span id="cb144-9"><a href="novel-isofroms---working.html#cb144-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cell_type <span class="cf">in</span> cell_types) {</span>
<span id="cb144-10"><a href="novel-isofroms---working.html#cb144-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset Seurat object to the current cell type</span></span>
<span id="cb144-11"><a href="novel-isofroms---working.html#cb144-11" aria-hidden="true" tabindex="-1"></a>  subset_obj <span class="ot">&lt;-</span> <span class="fu">subset</span>(seu_obj, <span class="at">idents =</span> cell_type)</span>
<span id="cb144-12"><a href="novel-isofroms---working.html#cb144-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-13"><a href="novel-isofroms---working.html#cb144-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the count matrix</span></span>
<span id="cb144-14"><a href="novel-isofroms---working.html#cb144-14" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">GetAssayData</span>(subset_obj, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb144-15"><a href="novel-isofroms---working.html#cb144-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-16"><a href="novel-isofroms---working.html#cb144-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify features that meet the threshold</span></span>
<span id="cb144-17"><a href="novel-isofroms---working.html#cb144-17" aria-hidden="true" tabindex="-1"></a>  feature_pass <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">&gt;=</span> (<span class="fl">0.10</span> <span class="sc">*</span> <span class="fu">ncol</span>(counts)) <span class="co"># can change these params </span></span>
<span id="cb144-18"><a href="novel-isofroms---working.html#cb144-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-19"><a href="novel-isofroms---working.html#cb144-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store features that pass the threshold</span></span>
<span id="cb144-20"><a href="novel-isofroms---working.html#cb144-20" aria-hidden="true" tabindex="-1"></a>  unique_features[[cell_type]] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts)[feature_pass]</span>
<span id="cb144-21"><a href="novel-isofroms---working.html#cb144-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb144-22"><a href="novel-isofroms---working.html#cb144-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-23"><a href="novel-isofroms---working.html#cb144-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Find unique features for each cell type</span></span>
<span id="cb144-24"><a href="novel-isofroms---working.html#cb144-24" aria-hidden="true" tabindex="-1"></a>final_unique_features <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(unique_features), <span class="cf">function</span>(cell_type) {</span>
<span id="cb144-25"><a href="novel-isofroms---working.html#cb144-25" aria-hidden="true" tabindex="-1"></a>  other_features <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">unlist</span>(unique_features[<span class="fu">names</span>(unique_features) <span class="sc">!=</span> cell_type]))</span>
<span id="cb144-26"><a href="novel-isofroms---working.html#cb144-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setdiff</span>(unique_features[[cell_type]], other_features)</span>
<span id="cb144-27"><a href="novel-isofroms---working.html#cb144-27" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb144-28"><a href="novel-isofroms---working.html#cb144-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-29"><a href="novel-isofroms---working.html#cb144-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results into a data frame</span></span>
<span id="cb144-30"><a href="novel-isofroms---working.html#cb144-30" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(final_unique_features) <span class="ot">&lt;-</span> cell_types</span>
<span id="cb144-31"><a href="novel-isofroms---working.html#cb144-31" aria-hidden="true" tabindex="-1"></a>unique_features_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(cell_types, <span class="cf">function</span>(cell_type) {</span>
<span id="cb144-32"><a href="novel-isofroms---working.html#cb144-32" aria-hidden="true" tabindex="-1"></a>  features <span class="ot">&lt;-</span> final_unique_features[[cell_type]]</span>
<span id="cb144-33"><a href="novel-isofroms---working.html#cb144-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-34"><a href="novel-isofroms---working.html#cb144-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If no features, return an empty data frame for this cell type</span></span>
<span id="cb144-35"><a href="novel-isofroms---working.html#cb144-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(features) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb144-36"><a href="novel-isofroms---working.html#cb144-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">CellType =</span> cell_type, <span class="at">Feature =</span> <span class="fu">character</span>(<span class="dv">0</span>)))</span>
<span id="cb144-37"><a href="novel-isofroms---working.html#cb144-37" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-38"><a href="novel-isofroms---working.html#cb144-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-39"><a href="novel-isofroms---working.html#cb144-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Otherwise, return the normal data frame</span></span>
<span id="cb144-40"><a href="novel-isofroms---working.html#cb144-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb144-41"><a href="novel-isofroms---working.html#cb144-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">CellType =</span> cell_type,</span>
<span id="cb144-42"><a href="novel-isofroms---working.html#cb144-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">Feature =</span> features</span>
<span id="cb144-43"><a href="novel-isofroms---working.html#cb144-43" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb144-44"><a href="novel-isofroms---working.html#cb144-44" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb144-45"><a href="novel-isofroms---working.html#cb144-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-46"><a href="novel-isofroms---working.html#cb144-46" aria-hidden="true" tabindex="-1"></a><span class="co"># View the unique features</span></span>
<span id="cb144-47"><a href="novel-isofroms---working.html#cb144-47" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(unique_features_df)</span></code></pre></div>
</details>
<pre><code>##           CellType                  Feature
## 1 Immature neurons ENST00000371824.7-ELAVL4
## 2 Immature neurons  ENST00000302101.6-NHLH1
## 3 Immature neurons  ENST00000366687.5-CCSAP
## 4 Immature neurons   ENST00000295025.12-REL
## 5 Immature neurons ENST00000324219.9-MAP4K4
## 6 Immature neurons ENST00000350878.9-MAP4K4</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="novel-isofroms---working.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="do">####### test plotting on iso assays ##</span></span>
<span id="cb146-2"><a href="novel-isofroms---working.html#cb146-2" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seu_obj<span class="sc">@</span>assays<span class="sc">$</span>iso<span class="sc">@</span>features)</span>
<span id="cb146-3"><a href="novel-isofroms---working.html#cb146-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-4"><a href="novel-isofroms---working.html#cb146-4" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="st">&quot;BLCAP&quot;</span></span>
<span id="cb146-5"><a href="novel-isofroms---working.html#cb146-5" aria-hidden="true" tabindex="-1"></a>plot_features_list <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">&quot;(^|-|</span><span class="sc">\\</span><span class="st">b)&quot;</span>, gene, <span class="st">&quot;($|</span><span class="sc">\\</span><span class="st">b)&quot;</span>), features, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb146-6"><a href="novel-isofroms---working.html#cb146-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-7"><a href="novel-isofroms---working.html#cb146-7" aria-hidden="true" tabindex="-1"></a><span class="co">#We can now plot our favorite gene and all its corresponding isoforms on either the gene or isoform UMAPs. To switch between them, simply change the reduction argument to the desired UMAP (either the gene or isoform reduction)</span></span>
<span id="cb146-8"><a href="novel-isofroms---working.html#cb146-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-9"><a href="novel-isofroms---working.html#cb146-9" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seu_obj, <span class="at">features =</span> plot_features_list, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>) </span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/working2-1.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="novel-isofroms---working.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co"># any other ideas?</span></span>
<span id="cb147-2"><a href="novel-isofroms---working.html#cb147-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-3"><a href="novel-isofroms---working.html#cb147-3" aria-hidden="true" tabindex="-1"></a><span class="co">#What are the number of novel isoforms per cluster?</span></span>
<span id="cb147-4"><a href="novel-isofroms---working.html#cb147-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-5"><a href="novel-isofroms---working.html#cb147-5" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">AggregateExpression</span>(</span>
<span id="cb147-6"><a href="novel-isofroms---working.html#cb147-6" aria-hidden="true" tabindex="-1"></a>  seu_obj, </span>
<span id="cb147-7"><a href="novel-isofroms---working.html#cb147-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">assays =</span> <span class="st">&quot;iso&quot;</span>, </span>
<span id="cb147-8"><a href="novel-isofroms---working.html#cb147-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">return.seurat =</span> <span class="cn">FALSE</span>,</span>
<span id="cb147-9"><a href="novel-isofroms---working.html#cb147-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span></span>
<span id="cb147-10"><a href="novel-isofroms---working.html#cb147-10" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb147-11"><a href="novel-isofroms---working.html#cb147-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-12"><a href="novel-isofroms---working.html#cb147-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for rows containing &quot;Bambu&quot; in the row names (novel isoforms)</span></span>
<span id="cb147-13"><a href="novel-isofroms---working.html#cb147-13" aria-hidden="true" tabindex="-1"></a>bambu_features <span class="ot">&lt;-</span> counts[<span class="fu">grepl</span>(<span class="st">&quot;Bambu&quot;</span>, <span class="fu">rownames</span>(counts)), ]</span>
<span id="cb147-14"><a href="novel-isofroms---working.html#cb147-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-15"><a href="novel-isofroms---working.html#cb147-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the expression filter (&gt;10) and calculate the number of novel isoforms per cluster</span></span>
<span id="cb147-16"><a href="novel-isofroms---working.html#cb147-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-17"><a href="novel-isofroms---working.html#cb147-17" aria-hidden="true" tabindex="-1"></a>novel_isoform_counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colSums</span>(bambu_features <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb147-18"><a href="novel-isofroms---working.html#cb147-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-19"><a href="novel-isofroms---working.html#cb147-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a bar plot</span></span>
<span id="cb147-20"><a href="novel-isofroms---working.html#cb147-20" aria-hidden="true" tabindex="-1"></a>P3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(novel_isoform_counts, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">row.names</span>(novel_isoform_counts), <span class="at">y =</span> novel_isoform_counts<span class="sc">$</span><span class="st">`</span><span class="at">colSums(bambu_features &gt; 0)</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb147-21"><a href="novel-isofroms---working.html#cb147-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb147-22"><a href="novel-isofroms---working.html#cb147-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Cluster&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of Novel Isoforms&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Number of Novel Isoforms (&gt; 1 count)&quot;</span>) <span class="sc">+</span></span>
<span id="cb147-23"><a href="novel-isofroms---working.html#cb147-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb147-24"><a href="novel-isofroms---working.html#cb147-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-25"><a href="novel-isofroms---working.html#cb147-25" aria-hidden="true" tabindex="-1"></a>P3</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/working3-1.png" width="672" /></p>
</div>
<div id="isoform-proportion-analysis" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> <strong>Isoform Proportion Analysis</strong><a href="novel-isofroms---working.html#isoform-proportion-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="calculate-the-proportion-of-novel-isoforms-relative-to-total-gene-expression." class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Calculate the proportion of novel isoforms relative to total gene expression.<a href="novel-isofroms---working.html#calculate-the-proportion-of-novel-isoforms-relative-to-total-gene-expression." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="novel-isofroms---working.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="do">#### number_uniq_isoform_per_cluster </span><span class="al">###</span></span>
<span id="cb148-2"><a href="novel-isofroms---working.html#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="co"># diferent method and im not sure how well compares to the upset plot method</span></span>
<span id="cb148-3"><a href="novel-isofroms---working.html#cb148-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-4"><a href="novel-isofroms---working.html#cb148-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the function with a prefix argument</span></span>
<span id="cb148-5"><a href="novel-isofroms---working.html#cb148-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 1: Aggregate expression based on &#39;broad_categories&#39;</span></span>
<span id="cb148-6"><a href="novel-isofroms---working.html#cb148-6" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">AggregateExpression</span>(</span>
<span id="cb148-7"><a href="novel-isofroms---working.html#cb148-7" aria-hidden="true" tabindex="-1"></a>    seu_obj, </span>
<span id="cb148-8"><a href="novel-isofroms---working.html#cb148-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">assays =</span> <span class="st">&quot;iso&quot;</span>, </span>
<span id="cb148-9"><a href="novel-isofroms---working.html#cb148-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">return.seurat =</span> <span class="cn">FALSE</span>,</span>
<span id="cb148-10"><a href="novel-isofroms---working.html#cb148-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">&quot;sctype_db&quot;</span> </span>
<span id="cb148-11"><a href="novel-isofroms---working.html#cb148-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb148-12"><a href="novel-isofroms---working.html#cb148-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb148-13"><a href="novel-isofroms---working.html#cb148-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 2: Convert the counts to a data frame</span></span>
<span id="cb148-14"><a href="novel-isofroms---working.html#cb148-14" aria-hidden="true" tabindex="-1"></a>  counts.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts)</span>
<span id="cb148-15"><a href="novel-isofroms---working.html#cb148-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb148-16"><a href="novel-isofroms---working.html#cb148-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 3: Transform data from wide to long format</span></span>
<span id="cb148-17"><a href="novel-isofroms---working.html#cb148-17" aria-hidden="true" tabindex="-1"></a>  long_df <span class="ot">&lt;-</span> counts.df <span class="sc">%&gt;%</span></span>
<span id="cb148-18"><a href="novel-isofroms---working.html#cb148-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;isoform&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-19"><a href="novel-isofroms---working.html#cb148-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>isoform, <span class="at">names_to =</span> <span class="st">&quot;cluster&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;expression&quot;</span>)</span>
<span id="cb148-20"><a href="novel-isofroms---working.html#cb148-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb148-21"><a href="novel-isofroms---working.html#cb148-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 4: Calculate total expression per isoform</span></span>
<span id="cb148-22"><a href="novel-isofroms---working.html#cb148-22" aria-hidden="true" tabindex="-1"></a>  isoform_total_expression <span class="ot">&lt;-</span> long_df <span class="sc">%&gt;%</span></span>
<span id="cb148-23"><a href="novel-isofroms---working.html#cb148-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(isoform) <span class="sc">%&gt;%</span></span>
<span id="cb148-24"><a href="novel-isofroms---working.html#cb148-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">total_expression =</span> <span class="fu">sum</span>(expression), <span class="at">.groups =</span> <span class="st">&#39;drop&#39;</span>)</span>
<span id="cb148-25"><a href="novel-isofroms---working.html#cb148-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb148-26"><a href="novel-isofroms---working.html#cb148-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 5: Calculate expression per cluster and filter based on percentage and total expression</span></span>
<span id="cb148-27"><a href="novel-isofroms---working.html#cb148-27" aria-hidden="true" tabindex="-1"></a>  isoform_summary <span class="ot">&lt;-</span> long_df <span class="sc">%&gt;%</span></span>
<span id="cb148-28"><a href="novel-isofroms---working.html#cb148-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(isoform, cluster) <span class="sc">%&gt;%</span></span>
<span id="cb148-29"><a href="novel-isofroms---working.html#cb148-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">cluster_expression =</span> <span class="fu">sum</span>(expression), <span class="at">.groups =</span> <span class="st">&#39;drop&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-30"><a href="novel-isofroms---working.html#cb148-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(isoform_total_expression, <span class="at">by =</span> <span class="st">&quot;isoform&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-31"><a href="novel-isofroms---working.html#cb148-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">percentage_expression =</span> cluster_expression <span class="sc">/</span> total_expression <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-32"><a href="novel-isofroms---working.html#cb148-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(percentage_expression <span class="sc">&gt;=</span> <span class="dv">90</span> <span class="sc">&amp;</span> total_expression <span class="sc">&gt;=</span> <span class="dv">10</span>)</span>
<span id="cb148-33"><a href="novel-isofroms---working.html#cb148-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb148-34"><a href="novel-isofroms---working.html#cb148-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 6: Count the number of unique isoforms per cluster</span></span>
<span id="cb148-35"><a href="novel-isofroms---working.html#cb148-35" aria-hidden="true" tabindex="-1"></a>  isoform_counts <span class="ot">&lt;-</span> isoform_summary <span class="sc">%&gt;%</span></span>
<span id="cb148-36"><a href="novel-isofroms---working.html#cb148-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb148-37"><a href="novel-isofroms---working.html#cb148-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">unique_isoform_count =</span> <span class="fu">n_distinct</span>(isoform), <span class="at">.groups =</span> <span class="st">&#39;drop&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-38"><a href="novel-isofroms---working.html#cb148-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(cluster)</span>
<span id="cb148-39"><a href="novel-isofroms---working.html#cb148-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb148-40"><a href="novel-isofroms---working.html#cb148-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Step 7: Create a bar plot (P1) showing the number of unique isoforms per cluster</span></span>
<span id="cb148-41"><a href="novel-isofroms---working.html#cb148-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(isoform_counts, <span class="fu">aes</span>(<span class="at">x =</span> cluster, <span class="at">y =</span> unique_isoform_count)) <span class="sc">+</span></span>
<span id="cb148-42"><a href="novel-isofroms---working.html#cb148-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-43"><a href="novel-isofroms---working.html#cb148-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Cluster&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Number of Unique Isoforms&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Number of Unique Isoforms per Cluster&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-44"><a href="novel-isofroms---working.html#cb148-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span>  </span>
<span id="cb148-45"><a href="novel-isofroms---working.html#cb148-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;number_uniq_isoform_per_cluster&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/uniq%20isofroms-1.png" width="672" /></p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" line-spacing="2">
<div id="ref-chen2023" class="csl-entry">
Chen, Y., Sim, A., Wan, Y. K., Yeo, K., Lee, J. J. X., Ling, M. H., Love, M. I., &amp; Göke, J. (2023). Context-aware transcript quantification from long-read RNA-seq data with Bambu. <em>Nature Methods</em>, <em>20</em>(8), 1187–1195. <a href="https://doi.org/10.1038/s41592-023-01908-w">https://doi.org/10.1038/s41592-023-01908-w</a>
</div>
<div id="ref-kovaka2019" class="csl-entry">
Kovaka, S., Zimin, A. V., Pertea, G. M., Razaghi, R., Salzberg, S. L., &amp; Pertea, M. (2019). Transcriptome assembly from long-read RNA-seq alignments with StringTie2. <em>Genome Biology</em>, <em>20</em>(1). <a href="https://doi.org/10.1186/s13059-019-1910-1">https://doi.org/10.1186/s13059-019-1910-1</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="exploring-isoforms-of-interest.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/07-Novel-isofroms.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/07-Novel-isofroms.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
