<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Novel isoforms | Long-read Single-Cell RNA-seq analysis tutorial</title>
  <meta name="description" content="Chapter 8 Novel isoforms | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Novel isoforms | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Novel isoforms | Long-read Single-Cell RNA-seq analysis tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer Postdoctoral Research Fellow in Long Reads and Single-Cell Transcriptomics University of Melbourne" />


<meta name="date" content="2025-06-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="isoform-classification.html"/>
<link rel="next" href="multisample-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotaed-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotaed cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FIndAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring MACF1 isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-mcaf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of MCAF1 isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualizing-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualizing OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>9</b> <strong>Multisample analysis</strong></a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>9.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="9.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>9.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>9.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multi-sample-integration"><i class="fa fa-check"></i><b>9.2</b> Multi-Sample integration</a></li>
<li class="chapter" data-level="9.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>9.3</b> Add isoform counts</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Trajectory analysis</strong></a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i><b>10.1</b> Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="10.2" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i><b>10.2</b> Multisample QC function</a></li>
<li class="chapter" data-level="10.3" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-count-matrix"><i class="fa fa-check"></i><b>10.3</b> Convert Oarfish files to count matrix</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="novel-isoforms" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> <strong>Novel isoforms</strong><a href="novel-isoforms.html#novel-isoforms" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>With long read sequencing its possible to explore both known and novel isoforms. Novel isoform can be interesting feature to explore and here we will provide some simple analysis that one can use to extract novel isoforms and explore them in some more detail. Although the bar plots above show that the vast majority of isoforms are full-splice-matches i.e they match the reference <code>GTF</code> there are a few we could extract.</p>
<div id="find-all-the-genes-that-express-at-least-1-novel-isoform" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Find all the genes that express at least 1 novel isoform<a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First lets get a list of genes with at least one novel isoform</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="novel-isoforms.html#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the genes that express at least one novel isoform and save them to a list</span></span>
<span id="cb102-2"><a href="novel-isoforms.html#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert row names to a data frame</span></span>
<span id="cb102-3"><a href="novel-isoforms.html#cb102-3" aria-hidden="true" tabindex="-1"></a>isoform_ids <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">row.names</span>(seu_obj<span class="sc">@</span>assays<span class="sc">$</span>iso<span class="sc">$</span>counts))</span>
<span id="cb102-4"><a href="novel-isoforms.html#cb102-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(isoform_ids) <span class="ot">&lt;-</span> <span class="st">&quot;IDs&quot;</span></span>
<span id="cb102-5"><a href="novel-isoforms.html#cb102-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-6"><a href="novel-isoforms.html#cb102-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Add in total expression  </span></span>
<span id="cb102-7"><a href="novel-isoforms.html#cb102-7" aria-hidden="true" tabindex="-1"></a>isoform_ids<span class="sc">$</span>Total_Expression <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(seu_obj<span class="sc">@</span>assays<span class="sc">$</span>iso<span class="sc">$</span>counts)</span>
<span id="cb102-8"><a href="novel-isoforms.html#cb102-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-9"><a href="novel-isoforms.html#cb102-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter rows where &#39;IDs&#39; contains the string &quot;Bambu&quot;</span></span>
<span id="cb102-10"><a href="novel-isoforms.html#cb102-10" aria-hidden="true" tabindex="-1"></a>isoform_ids <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(isoform_ids[<span class="fu">grepl</span>(<span class="st">&quot;Bambu&quot;</span>, isoform_ids<span class="sc">$</span>IDs), ])  <span class="co"># can comment this out if we want to do this for all isoforms </span></span>
<span id="cb102-11"><a href="novel-isoforms.html#cb102-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-12"><a href="novel-isoforms.html#cb102-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate the &#39;IDs&#39; column into two columns </span></span>
<span id="cb102-13"><a href="novel-isoforms.html#cb102-13" aria-hidden="true" tabindex="-1"></a>filtered_df <span class="ot">&lt;-</span> isoform_ids <span class="sc">%&gt;%</span> <span class="fu">separate</span>(IDs, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;transcript_id&quot;</span>, <span class="st">&quot;gene_id&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;-&quot;</span>,  <span class="at">extra =</span> <span class="st">&quot;merge&quot;</span>)</span>
<span id="cb102-14"><a href="novel-isoforms.html#cb102-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-15"><a href="novel-isoforms.html#cb102-15" aria-hidden="true" tabindex="-1"></a>isoform_counts <span class="ot">&lt;-</span> filtered_df <span class="sc">%&gt;%</span></span>
<span id="cb102-16"><a href="novel-isoforms.html#cb102-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb102-17"><a href="novel-isoforms.html#cb102-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb102-18"><a href="novel-isoforms.html#cb102-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">Isoform_Count =</span> <span class="fu">n</span>(),</span>
<span id="cb102-19"><a href="novel-isoforms.html#cb102-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">Total_Expression =</span> <span class="fu">sum</span>(Total_Expression)</span>
<span id="cb102-20"><a href="novel-isoforms.html#cb102-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb102-21"><a href="novel-isoforms.html#cb102-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Total_Expression))</span>
<span id="cb102-22"><a href="novel-isoforms.html#cb102-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-23"><a href="novel-isoforms.html#cb102-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-24"><a href="novel-isoforms.html#cb102-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Print or return the filtered data frame</span></span>
<span id="cb102-25"><a href="novel-isoforms.html#cb102-25" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(isoform_counts)</span></code></pre></div>
</details>
<pre><code>## # A tibble: 31 × 3
##    gene_id         Isoform_Count Total_Expression
##    &lt;chr&gt;                   &lt;int&gt;            &lt;dbl&gt;
##  1 OAZ2                        1            396. 
##  2 ENSG00000291214             1            159. 
##  3 BambuGene6113               1            141. 
##  4 ENSG00000305512             1            140. 
##  5 BambuGene19505              1            139. 
##  6 ENSG00000305069             1            132  
##  7 BambuGene18812              1             95.8
##  8 ZMAT5                       1             76.0
##  9 ENSG00000295459             1             68.7
## 10 ENSG00000231748             1             62  
## # ℹ 21 more rows</code></pre>
<p>In this table we can see that we have a total of 31 novel isoforms. This number is smaller than we might expect but this is because Bambu <span class="citation">(<a href="references.html#ref-chen2023" role="doc-biblioref">Chen et al., 2023</a>)</span> is quite conservative for single cell data and we haven’t sequenced these data very deeply. Users can change the sensitivity of Bambu when running FLAMES by setting the <code>NDR</code> parameter or using a different discovery method like stringtie2 <span class="citation">(<a href="references.html#ref-kovaka2019" role="doc-biblioref">Kovaka et al., 2019</a>)</span>.</p>
<p>In the above table we have ordered the the genes that contain a novel isoform by total expression. OAZ2 is our top hit. We can look at the isoform structure using IsoViz as described in the section <a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures">6.5</a>.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="novel-isoforms.html#cb104-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/IsoVis_ENSG00000180304.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:isoviz-plot2"></span>
<img src="images/IsoVis_ENSG00000180304.png" alt="IsoViz visualization of Novel OAZ2 isoform." width="1000px" />
<p class="caption">
Figure 8.1: IsoViz visualization of Novel OAZ2 isoform.
</p>
</div>
</div>
<div id="visualizing-oaz2-novel-isoform" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Visualizing OAZ2 novel isoform<a href="novel-isoforms.html#visualizing-oaz2-novel-isoform" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Visualizing <strong>OAZ2</strong> reveals that the primary distinction between the novel isoform and the canonical isoform, <strong>OAZ2-201</strong> seems to be a shorter 5’ UTR. This minor alteration at face value should not affect the canonical open reading frame (ORF). However, according to SQANTI, <strong>BambuTx25</strong> is classified as a <strong><em>novel_in_catalog</em></strong> isoform, meaning it is an isoform not present in the reference GTF and SQANTI suggests that <strong>BambuTx25</strong> is non-coding, which is quite surprising! Interestingly, SQANTI also indicates that the primary difference between <strong>BambuTx25</strong> and the canonical isoforms lies in intron retention (Specified in the <em>subcategory</em> column).</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="novel-isoforms.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load necessary libraries</span></span>
<span id="cb105-2"><a href="novel-isoforms.html#cb105-2" aria-hidden="true" tabindex="-1"></a>SQANTI[<span class="fu">grep</span>(<span class="st">&#39;BambuTx25&#39;</span>, SQANTI<span class="sc">$</span>isoform), ]</span></code></pre></div>
</details>
<pre><code>##         isoform chrom strand length exons structural_category    associated_gene associated_transcript ref_length ref_exons diff_to_TSS diff_to_TTS
## 21835 BambuTx25 chr15      -   1870     5    novel_in_catalog ENSG00000180304.14                 novel       1934         6          NA          NA
##       diff_to_gene_TSS diff_to_gene_TTS      subcategory RTS_stage all_canonical min_sample_cov min_cov min_cov_pos sd_cov FL n_indels n_indels_junc  bite iso_exp
## 21835               -2               -1 intron_retention     FALSE     canonical             NA      NA          NA     NA NA       NA            NA FALSE      NA
##       gene_exp ratio_exp FSM_class     coding ORF_length CDS_length CDS_start CDS_end CDS_genomic_start CDS_genomic_end predicted_NMD perc_A_downstream_TTS
## 21835       NA        NA         C non_coding         NA         NA        NA      NA                NA              NA            NA                    10
##       seq_A_downstream_TTS dist_to_CAGE_peak within_CAGE_peak dist_to_polyA_site within_polyA_site polyA_motif polyA_dist polyA_motif_found ORF_seq ratio_TSS
## 21835 GTGGTTGGTCTATTCTTTAT                NA               NA                 NA                NA          NA         NA                NA    &lt;NA&gt;        NA</code></pre>
<p>To better understand these SQANTI results lets zoom in on exon 2 of <em>OAZ2</em>. Here, we observe a small break within the exon. The novel <strong>BambuTx25</strong> does not contain this break so perhaps this change affects the resulting amino acid sequence.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="novel-isoforms.html#cb107-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/IsoVis_ENSG00000180304_zoom.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:isoviz-plot-zoom"></span>
<img src="images/IsoVis_ENSG00000180304_zoom.png" alt="IsoViz visualization of exon 2 of OAZ2." width="1000px" />
<p class="caption">
Figure 8.2: IsoViz visualization of exon 2 of OAZ2.
</p>
</div>
</div>
<div id="functional-impacts-of-novel-isoforms" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Functional impacts of novel isoforms<a href="novel-isoforms.html#functional-impacts-of-novel-isoforms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are many ways to investigate the translated sequence and compare the protein generated from conanocial isoform and the novel one. Simple approaches may involve extracting the fasta sequence and looking for the open reading frames (ORFs) in a online tool like ‘expasy translate’ which can be found here <a href="https://web.expasy.org/translate/" class="uri">https://web.expasy.org/translate/</a>.</p>
<p>Bellow we have written a function that will take a list of isoforms from a gene and plot the ORFs as defined by the package ORFik<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a> <span class="citation">(<a href="references.html#ref-tjeldnes2021" role="doc-biblioref">Tjeldnes et al., 2021</a>)</span>. We will visualize these ORFS in Gviz <span class="citation">(<a href="references.html#ref-hahne2016" role="doc-biblioref">Hahne &amp; Ivanek, 2016</a>)</span> an R package for visualizing genomic features. This analysis will help us determine if the ORF in the transcript <strong>BambuTx25</strong> has been impacted by the change in nucleotide sequence.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="novel-isoforms.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Global cache for TxDb</span></span>
<span id="cb108-2"><a href="novel-isoforms.html#cb108-2" aria-hidden="true" tabindex="-1"></a>txdb_cache <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb108-3"><a href="novel-isoforms.html#cb108-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-4"><a href="novel-isoforms.html#cb108-4" aria-hidden="true" tabindex="-1"></a>get_txdb <span class="ot">&lt;-</span> <span class="cf">function</span>(reference_gtf) {</span>
<span id="cb108-5"><a href="novel-isoforms.html#cb108-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if the TxDb for this file is already in cache</span></span>
<span id="cb108-6"><a href="novel-isoforms.html#cb108-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(txdb_cache[[reference_gtf]])) {</span>
<span id="cb108-7"><a href="novel-isoforms.html#cb108-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;Using cached TxDb...&quot;</span>)</span>
<span id="cb108-8"><a href="novel-isoforms.html#cb108-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(txdb_cache[[reference_gtf]])</span>
<span id="cb108-9"><a href="novel-isoforms.html#cb108-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb108-10"><a href="novel-isoforms.html#cb108-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-11"><a href="novel-isoforms.html#cb108-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create TxDb and cache it</span></span>
<span id="cb108-12"><a href="novel-isoforms.html#cb108-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Creating TxDb object from reference GTF. This may take time...&quot;</span>)</span>
<span id="cb108-13"><a href="novel-isoforms.html#cb108-13" aria-hidden="true" tabindex="-1"></a>  txdb <span class="ot">&lt;-</span> txdbmaker<span class="sc">::</span><span class="fu">makeTxDbFromGFF</span>(reference_gtf)</span>
<span id="cb108-14"><a href="novel-isoforms.html#cb108-14" aria-hidden="true" tabindex="-1"></a>  txdb_cache[[reference_gtf]] <span class="ot">&lt;&lt;-</span> txdb</span>
<span id="cb108-15"><a href="novel-isoforms.html#cb108-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(txdb)</span>
<span id="cb108-16"><a href="novel-isoforms.html#cb108-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb108-17"><a href="novel-isoforms.html#cb108-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-18"><a href="novel-isoforms.html#cb108-18" aria-hidden="true" tabindex="-1"></a>plot_isoform_ORFs <span class="ot">&lt;-</span> <span class="cf">function</span>(reference_gtf, target_gtf, isoforms_of_interest, </span>
<span id="cb108-19"><a href="novel-isoforms.html#cb108-19" aria-hidden="true" tabindex="-1"></a>                               chromosome, plot_start, plot_end) {</span>
<span id="cb108-20"><a href="novel-isoforms.html#cb108-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-21"><a href="novel-isoforms.html#cb108-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load genome reference</span></span>
<span id="cb108-22"><a href="novel-isoforms.html#cb108-22" aria-hidden="true" tabindex="-1"></a>  genomedb <span class="ot">&lt;-</span> BSgenome.Hsapiens.UCSC.hg38</span>
<span id="cb108-23"><a href="novel-isoforms.html#cb108-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-24"><a href="novel-isoforms.html#cb108-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Retrieve cached TxDb or create if not available</span></span>
<span id="cb108-25"><a href="novel-isoforms.html#cb108-25" aria-hidden="true" tabindex="-1"></a>  txdb <span class="ot">&lt;-</span> <span class="fu">get_txdb</span>(reference_gtf)</span>
<span id="cb108-26"><a href="novel-isoforms.html#cb108-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-27"><a href="novel-isoforms.html#cb108-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Import and filter GTF for specific isoforms</span></span>
<span id="cb108-28"><a href="novel-isoforms.html#cb108-28" aria-hidden="true" tabindex="-1"></a>  gtf_data <span class="ot">&lt;-</span> <span class="fu">import</span>(target_gtf)</span>
<span id="cb108-29"><a href="novel-isoforms.html#cb108-29" aria-hidden="true" tabindex="-1"></a>  gtf_filtered <span class="ot">&lt;-</span> gtf_data[gtf_data<span class="sc">$</span>transcript_id <span class="sc">%in%</span> isoforms_of_interest]</span>
<span id="cb108-30"><a href="novel-isoforms.html#cb108-30" aria-hidden="true" tabindex="-1"></a>  txdb_filtered <span class="ot">&lt;-</span> txdbmaker<span class="sc">::</span><span class="fu">makeTxDbFromGRanges</span>(gtf_filtered)</span>
<span id="cb108-31"><a href="novel-isoforms.html#cb108-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-32"><a href="novel-isoforms.html#cb108-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract exons and convert to GRangesList</span></span>
<span id="cb108-33"><a href="novel-isoforms.html#cb108-33" aria-hidden="true" tabindex="-1"></a>  txs <span class="ot">&lt;-</span> GenomicFeatures<span class="sc">::</span><span class="fu">exonsBy</span>(txdb_filtered, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;tx&quot;</span>, <span class="st">&quot;gene&quot;</span>), <span class="at">use.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-34"><a href="novel-isoforms.html#cb108-34" aria-hidden="true" tabindex="-1"></a>  txs_grl <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>(txs)</span>
<span id="cb108-35"><a href="novel-isoforms.html#cb108-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-36"><a href="novel-isoforms.html#cb108-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract transcript sequences and identify ORFs</span></span>
<span id="cb108-37"><a href="novel-isoforms.html#cb108-37" aria-hidden="true" tabindex="-1"></a>  tx_seqs <span class="ot">&lt;-</span> <span class="fu">extractTranscriptSeqs</span>(genomedb, txs_grl)</span>
<span id="cb108-38"><a href="novel-isoforms.html#cb108-38" aria-hidden="true" tabindex="-1"></a>  ORFs <span class="ot">&lt;-</span> <span class="fu">findMapORFs</span>(txs_grl, tx_seqs, <span class="at">groupByTx =</span> <span class="cn">FALSE</span>, </span>
<span id="cb108-39"><a href="novel-isoforms.html#cb108-39" aria-hidden="true" tabindex="-1"></a>                       <span class="at">longestORF =</span> <span class="cn">FALSE</span>, <span class="at">minimumLength =</span> <span class="dv">30</span>, </span>
<span id="cb108-40"><a href="novel-isoforms.html#cb108-40" aria-hidden="true" tabindex="-1"></a>                       <span class="at">startCodon =</span> <span class="st">&quot;ATG&quot;</span>, <span class="at">stopCodon =</span> <span class="fu">stopDefinition</span>(<span class="dv">1</span>))</span>
<span id="cb108-41"><a href="novel-isoforms.html#cb108-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-42"><a href="novel-isoforms.html#cb108-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Unlist and prepare ORF data for plotting</span></span>
<span id="cb108-43"><a href="novel-isoforms.html#cb108-43" aria-hidden="true" tabindex="-1"></a>  ORFs_unlisted <span class="ot">&lt;-</span> <span class="fu">unlist</span>(ORFs)</span>
<span id="cb108-44"><a href="novel-isoforms.html#cb108-44" aria-hidden="true" tabindex="-1"></a>  ORFs_unlisted<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="st">&quot;exon&quot;</span></span>
<span id="cb108-45"><a href="novel-isoforms.html#cb108-45" aria-hidden="true" tabindex="-1"></a>  ORFs_unlisted<span class="sc">$</span>transcript <span class="ot">&lt;-</span> ORFs_unlisted<span class="sc">$</span>names</span>
<span id="cb108-46"><a href="novel-isoforms.html#cb108-46" aria-hidden="true" tabindex="-1"></a>  ORFs_unlisted<span class="sc">$</span>transcript <span class="ot">&lt;-</span> <span class="fu">paste0</span>(ORFs_unlisted<span class="sc">$</span>transcript, <span class="st">&quot;_ORF&quot;</span>)</span>
<span id="cb108-47"><a href="novel-isoforms.html#cb108-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-48"><a href="novel-isoforms.html#cb108-48" aria-hidden="true" tabindex="-1"></a>  <span class="co">#could add in a filter for ORFs here </span></span>
<span id="cb108-49"><a href="novel-isoforms.html#cb108-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-50"><a href="novel-isoforms.html#cb108-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define start and stop codons</span></span>
<span id="cb108-51"><a href="novel-isoforms.html#cb108-51" aria-hidden="true" tabindex="-1"></a>  starts <span class="ot">&lt;-</span> <span class="fu">startCodons</span>(ORFs, <span class="at">is.sorted =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-52"><a href="novel-isoforms.html#cb108-52" aria-hidden="true" tabindex="-1"></a>  stops <span class="ot">&lt;-</span> <span class="fu">stopCodons</span>(ORFs, <span class="at">is.sorted =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-53"><a href="novel-isoforms.html#cb108-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-54"><a href="novel-isoforms.html#cb108-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Visualization Tracks</span></span>
<span id="cb108-55"><a href="novel-isoforms.html#cb108-55" aria-hidden="true" tabindex="-1"></a>  gtrack <span class="ot">&lt;-</span> <span class="fu">GenomeAxisTrack</span>()</span>
<span id="cb108-56"><a href="novel-isoforms.html#cb108-56" aria-hidden="true" tabindex="-1"></a>  itrack <span class="ot">&lt;-</span> <span class="fu">IdeogramTrack</span>(<span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>, <span class="at">chromosome =</span> chromosome)</span>
<span id="cb108-57"><a href="novel-isoforms.html#cb108-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-58"><a href="novel-isoforms.html#cb108-58" aria-hidden="true" tabindex="-1"></a>  ref_track <span class="ot">&lt;-</span> <span class="fu">GeneRegionTrack</span>(</span>
<span id="cb108-59"><a href="novel-isoforms.html#cb108-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">range =</span> txdb,</span>
<span id="cb108-60"><a href="novel-isoforms.html#cb108-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;gencodev47&quot;</span>,</span>
<span id="cb108-61"><a href="novel-isoforms.html#cb108-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb108-62"><a href="novel-isoforms.html#cb108-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">chromosome =</span> chromosome,</span>
<span id="cb108-63"><a href="novel-isoforms.html#cb108-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="st">&quot;darkblue&quot;</span>,</span>
<span id="cb108-64"><a href="novel-isoforms.html#cb108-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;lightblue&quot;</span>,</span>
<span id="cb108-65"><a href="novel-isoforms.html#cb108-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrowHead =</span> <span class="cn">FALSE</span></span>
<span id="cb108-66"><a href="novel-isoforms.html#cb108-66" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb108-67"><a href="novel-isoforms.html#cb108-67" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-68"><a href="novel-isoforms.html#cb108-68" aria-hidden="true" tabindex="-1"></a>  input_track <span class="ot">&lt;-</span> <span class="fu">GeneRegionTrack</span>(</span>
<span id="cb108-69"><a href="novel-isoforms.html#cb108-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">range =</span> txdb_filtered,</span>
<span id="cb108-70"><a href="novel-isoforms.html#cb108-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Isoforms (Exons)&quot;</span>,</span>
<span id="cb108-71"><a href="novel-isoforms.html#cb108-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb108-72"><a href="novel-isoforms.html#cb108-72" aria-hidden="true" tabindex="-1"></a>    <span class="at">chromosome =</span> chromosome,</span>
<span id="cb108-73"><a href="novel-isoforms.html#cb108-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="st">&quot;darkgreen&quot;</span>,</span>
<span id="cb108-74"><a href="novel-isoforms.html#cb108-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;lightgreen&quot;</span>,</span>
<span id="cb108-75"><a href="novel-isoforms.html#cb108-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">arrowHead =</span> <span class="cn">FALSE</span></span>
<span id="cb108-76"><a href="novel-isoforms.html#cb108-76" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb108-77"><a href="novel-isoforms.html#cb108-77" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-78"><a href="novel-isoforms.html#cb108-78" aria-hidden="true" tabindex="-1"></a>  orf_track <span class="ot">&lt;-</span> <span class="fu">GeneRegionTrack</span>(</span>
<span id="cb108-79"><a href="novel-isoforms.html#cb108-79" aria-hidden="true" tabindex="-1"></a>    ORFs_unlisted,</span>
<span id="cb108-80"><a href="novel-isoforms.html#cb108-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb108-81"><a href="novel-isoforms.html#cb108-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">chromosome =</span> <span class="fu">unique</span>(<span class="fu">seqnames</span>(ORFs_unlisted)),</span>
<span id="cb108-82"><a href="novel-isoforms.html#cb108-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">transcriptAnnotation =</span> <span class="st">&quot;transcript&quot;</span>,</span>
<span id="cb108-83"><a href="novel-isoforms.html#cb108-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;ORFs&quot;</span>,</span>
<span id="cb108-84"><a href="novel-isoforms.html#cb108-84" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="st">&quot;grey&quot;</span>,</span>
<span id="cb108-85"><a href="novel-isoforms.html#cb108-85" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;grey&quot;</span>,</span>
<span id="cb108-86"><a href="novel-isoforms.html#cb108-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">stacking =</span> <span class="st">&quot;squish&quot;</span>,</span>
<span id="cb108-87"><a href="novel-isoforms.html#cb108-87" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb108-88"><a href="novel-isoforms.html#cb108-88" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-89"><a href="novel-isoforms.html#cb108-89" aria-hidden="true" tabindex="-1"></a>  starts_track <span class="ot">&lt;-</span> <span class="fu">AnnotationTrack</span>(</span>
<span id="cb108-90"><a href="novel-isoforms.html#cb108-90" aria-hidden="true" tabindex="-1"></a>    <span class="at">range =</span> starts,</span>
<span id="cb108-91"><a href="novel-isoforms.html#cb108-91" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Starts&quot;</span>,</span>
<span id="cb108-92"><a href="novel-isoforms.html#cb108-92" aria-hidden="true" tabindex="-1"></a>    <span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb108-93"><a href="novel-isoforms.html#cb108-93" aria-hidden="true" tabindex="-1"></a>    <span class="at">chromosome =</span> <span class="fu">seqnames</span>(txs_grl[[<span class="dv">1</span>]])[<span class="dv">1</span>],</span>
<span id="cb108-94"><a href="novel-isoforms.html#cb108-94" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="st">&quot;green&quot;</span>,</span>
<span id="cb108-95"><a href="novel-isoforms.html#cb108-95" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;green&quot;</span>,</span>
<span id="cb108-96"><a href="novel-isoforms.html#cb108-96" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">&quot;box&quot;</span>, <span class="co"># Ensure no arrows by using box shape</span></span>
<span id="cb108-97"><a href="novel-isoforms.html#cb108-97" aria-hidden="true" tabindex="-1"></a>    <span class="at">stacking =</span> <span class="st">&quot;dense&quot;</span></span>
<span id="cb108-98"><a href="novel-isoforms.html#cb108-98" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb108-99"><a href="novel-isoforms.html#cb108-99" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-100"><a href="novel-isoforms.html#cb108-100" aria-hidden="true" tabindex="-1"></a>  stops_track <span class="ot">&lt;-</span> <span class="fu">AnnotationTrack</span>(</span>
<span id="cb108-101"><a href="novel-isoforms.html#cb108-101" aria-hidden="true" tabindex="-1"></a>    <span class="at">range =</span> stops,</span>
<span id="cb108-102"><a href="novel-isoforms.html#cb108-102" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;Stops&quot;</span>,</span>
<span id="cb108-103"><a href="novel-isoforms.html#cb108-103" aria-hidden="true" tabindex="-1"></a>    <span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb108-104"><a href="novel-isoforms.html#cb108-104" aria-hidden="true" tabindex="-1"></a>    <span class="at">chromosome =</span> <span class="fu">seqnames</span>(txs_grl[[<span class="dv">1</span>]])[<span class="dv">1</span>],</span>
<span id="cb108-105"><a href="novel-isoforms.html#cb108-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb108-106"><a href="novel-isoforms.html#cb108-106" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb108-107"><a href="novel-isoforms.html#cb108-107" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="st">&quot;box&quot;</span>, <span class="co"># Ensure no arrows by using box shape</span></span>
<span id="cb108-108"><a href="novel-isoforms.html#cb108-108" aria-hidden="true" tabindex="-1"></a>    <span class="at">stacking =</span> <span class="st">&quot;dense&quot;</span> </span>
<span id="cb108-109"><a href="novel-isoforms.html#cb108-109" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb108-110"><a href="novel-isoforms.html#cb108-110" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-111"><a href="novel-isoforms.html#cb108-111" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot tracks</span></span>
<span id="cb108-112"><a href="novel-isoforms.html#cb108-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotTracks</span>(<span class="fu">list</span>(itrack, gtrack, ref_track, input_track, </span>
<span id="cb108-113"><a href="novel-isoforms.html#cb108-113" aria-hidden="true" tabindex="-1"></a>                   orf_track, stops_track, starts_track), </span>
<span id="cb108-114"><a href="novel-isoforms.html#cb108-114" aria-hidden="true" tabindex="-1"></a>             <span class="at">from =</span> plot_start, <span class="at">to =</span> plot_end,  </span>
<span id="cb108-115"><a href="novel-isoforms.html#cb108-115" aria-hidden="true" tabindex="-1"></a>             <span class="at">transcriptAnnotation =</span> <span class="st">&quot;transcript&quot;</span>)</span>
<span id="cb108-116"><a href="novel-isoforms.html#cb108-116" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb108-117"><a href="novel-isoforms.html#cb108-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-118"><a href="novel-isoforms.html#cb108-118" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage</span></span>
<span id="cb108-119"><a href="novel-isoforms.html#cb108-119" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_isoform_ORFs</span>(</span>
<span id="cb108-120"><a href="novel-isoforms.html#cb108-120" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference_gtf =</span> <span class="st">&quot;data/gencode.v47.annotation.gtf&quot;</span>,</span>
<span id="cb108-121"><a href="novel-isoforms.html#cb108-121" aria-hidden="true" tabindex="-1"></a>  <span class="at">target_gtf =</span> <span class="st">&quot;data/FLAMES_out/isoform_annotated_unfiltered.gtf&quot;</span>,</span>
<span id="cb108-122"><a href="novel-isoforms.html#cb108-122" aria-hidden="true" tabindex="-1"></a>  <span class="at">isoforms_of_interest =</span> <span class="fu">c</span>(<span class="st">&quot;ENST00000326005.10&quot;</span>, <span class="st">&quot;BambuTx25&quot;</span>),</span>
<span id="cb108-123"><a href="novel-isoforms.html#cb108-123" aria-hidden="true" tabindex="-1"></a>  <span class="at">chromosome =</span> <span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb108-124"><a href="novel-isoforms.html#cb108-124" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot_start =</span> <span class="dv">64687015</span>,</span>
<span id="cb108-125"><a href="novel-isoforms.html#cb108-125" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot_end =</span> <span class="dv">64703412</span></span>
<span id="cb108-126"><a href="novel-isoforms.html#cb108-126" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ORFS-plot3-1.png" width="1152" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="novel-isoforms.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_isoform_ORFs</span>(</span>
<span id="cb109-2"><a href="novel-isoforms.html#cb109-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference_gtf =</span> <span class="st">&quot;data/gencode.v47.annotation.gtf&quot;</span>,</span>
<span id="cb109-3"><a href="novel-isoforms.html#cb109-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">target_gtf =</span> <span class="st">&quot;data/FLAMES_out/isoform_annotated_unfiltered.gtf&quot;</span>,</span>
<span id="cb109-4"><a href="novel-isoforms.html#cb109-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">isoforms_of_interest =</span> <span class="fu">c</span>(<span class="st">&quot;ENST00000326005.10&quot;</span>, <span class="st">&quot;BambuTx25&quot;</span>),</span>
<span id="cb109-5"><a href="novel-isoforms.html#cb109-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chromosome =</span> <span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb109-6"><a href="novel-isoforms.html#cb109-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot_start =</span> <span class="dv">64691416</span>,</span>
<span id="cb109-7"><a href="novel-isoforms.html#cb109-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot_end =</span> <span class="dv">64691685</span></span>
<span id="cb109-8"><a href="novel-isoforms.html#cb109-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/ORFS-plot3-2.png" width="1152" /></p>
<p>Visualizing ORFs using the above code chunck shows that the <strong>BambuTx25_1_ORF</strong> is incomplete. The additional sequence (retained intron as specified by SQANTI) results in a premature stop codon shown in red which likely triggers nonsense-mediated decay (NMD). The canonical ENST00000326005.10_3_ORF is complete.</p>
<p>We can examine additional genes from our list. For instance, <strong>ZMAT5</strong> has a novel isoform that utilizes a different, known splice junction in the 5’ UTR. Using the function above to examine ORFs, we can see that the ORF is unchanged compared to the canonical isoform. This may suggest some import regulatory role in the 5’ UTR yet the protein remains unchanged.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="novel-isoforms.html#cb110-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/IsoVis_ENSG00000100319.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:isoviz-plot3"></span>
<img src="images/IsoVis_ENSG00000100319.png" alt="IsoViz visualization of Novel ZMAT5 isoform." width="1000px" />
<p class="caption">
Figure 8.3: IsoViz visualization of Novel ZMAT5 isoform.
</p>
</div>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="novel-isoforms.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_isoform_ORFs</span>( <span class="at">reference_gtf =</span> <span class="st">&quot;../../../../resources/gencode.v47.annotation.gtf&quot;</span>, <span class="at">target_gtf =</span> <span class="st">&quot;/data/scratch/users/yairp/FLAMES_Day55/outs/isoform_annotated_unfiltered.gtf&quot;</span>, <span class="at">isoforms_of_interest =</span> <span class="fu">c</span>(<span class="st">&quot;BambuTx32&quot;</span>, <span class="st">&quot;ENST00000344318.4&quot;</span>), <span class="at">chromosome =</span> <span class="st">&quot;chr22&quot;</span>, <span class="at">plot_start =</span> <span class="dv">29728956</span>, <span class="at">plot_end =</span> <span class="dv">29769011</span> )</span></code></pre></div>
</details>
<pre><code>## Creating TxDb object from reference GTF. This may take time...</code></pre>
<pre><code>## Import genomic features from the file as a GRanges object ... OK
## Prepare the &#39;metadata&#39; data frame ... OK
## Make the TxDb object ... OK</code></pre>
<p><img src="bookdownproj_files/figure-html/ORFS-plot4-1.png" width="1152" /></p>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="5">
<li id="fn5"><p>ORFs found by the package ORFik may need to be filtered based on length or position when analysising differnt genes or isoforms.<a href="novel-isoforms.html#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="isoform-classification.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multisample-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/07-Isofrom-Classification.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/07-Isofrom-Classification.Rmd",
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
