<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Profiling Isoform Diversity | Long-read Single-Cell RNA-seq analysis tutorial</title>
  <meta name="description" content="Chapter 9 Profiling Isoform Diversity | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Profiling Isoform Diversity | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Profiling Isoform Diversity | Long-read Single-Cell RNA-seq analysis tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer Postdoctoral Research Fellow in Long Reads and Single-Cell Transcriptomics University of Melbourne" />


<meta name="date" content="2025-11-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="novel-isoforms.html"/>
<link rel="next" href="multisample-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> Load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotated-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotated cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FindAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring <em>MACF1</em> isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-macf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of <em>MACF1</em> isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualising-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualising OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html"><i class="fa fa-check"></i><b>9</b> <strong>Profiling Isoform Diversity</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#measuring-entropy-with-find_diversity-function"><i class="fa fa-check"></i><b>9.1</b> Measuring entropy with <code>find_diversity()</code> function</a></li>
<li class="chapter" data-level="9.2" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#example-of-a-low-entropy-gene"><i class="fa fa-check"></i><b>9.2</b> Example of a low entropy gene</a></li>
<li class="chapter" data-level="9.3" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#example-of-a-high-entropy-gene"><i class="fa fa-check"></i><b>9.3</b> Example of a high entropy gene</a></li>
<li class="chapter" data-level="9.4" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#broadening-entropy-analysis"><i class="fa fa-check"></i><b>9.4</b> Broadening Entropy Analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Multisample analysis</strong></a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>10.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="10.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>10.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>10.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multisample-integration"><i class="fa fa-check"></i><b>10.2</b> Multisample integration</a></li>
<li class="chapter" data-level="10.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>10.3</b> Add isoform counts</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>11</b> <strong>Trajectory analysis</strong></a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="11.1" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i><b>11.1</b> Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="11.2" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i><b>11.2</b> Multisample QC function</a></li>
<li class="chapter" data-level="11.3" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-a-count-matrix"><i class="fa fa-check"></i><b>11.3</b> Convert Oarfish files to a count matrix</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="profiling-isoform-diversity" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> <strong>Profiling Isoform Diversity</strong><a href="profiling-isoform-diversity.html#profiling-isoform-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>While traditional analyses—such as differential gene expression (DGE), differential isoform expression (DIE), or pseudotime trajectory inference—reveal important biological patterns, <strong>long-read single-cell data</strong> allow us to ask a different kind of question:</p>
<blockquote>
<p><strong>How is a gene’s expression distributed across its isoforms?</strong></p>
</blockquote>
<p>This concept is what we refer to as <strong>isoform diversity</strong>.<br />
Rather than simply asking <em>“Is isoform X upregulated in cluster A?”</em>, we can instead explore:</p>
<ul>
<li><p>Does a single cell rely on one dominant isoform for a given gene?</p></li>
<li><p>Or does it distribute expression more evenly across multiple isoforms?</p></li>
<li><p>And do cells of the same type agree on which isoforms they use—or is there variation even within a single population?</p></li>
</ul>
<p>In the <strong>FLAMESv2</strong> framework, we formalize this idea using <strong>Shannon’s entropy</strong>, implemented through the <code>find_diversity()</code> function, to quantify isoform diversity at single-cell resolution.</p>
<p>In simple terms, <strong>entropy</strong> provides a numerical score describing how evenly a gene’s expression is split across its isoforms:</p>
<ul>
<li><p><strong>Low entropy</strong> → the gene primarily expresses a single isoform in that cell.</p></li>
<li><p><strong>High entropy</strong> → the gene expresses multiple isoforms in roughly equal proportions.</p></li>
</ul>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="profiling-isoform-diversity.html#cb119-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/entropy_example.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-10"></span>
<img src="images/entropy_example.png" alt="Isoform diversity schematic. For each gene, we measure how transcript usage is distributed across isoforms within individual cells. Cells dominated by a single isoform show low isoform diversity (low entropy), while cells that express multiple isoforms at similar levels show high isoform diversity (high entropy)." width="1600px" />
<p class="caption">
Figure 9.1: Isoform diversity schematic. For each gene, we measure how transcript usage is distributed across isoforms within individual cells. Cells dominated by a single isoform show low isoform diversity (low entropy), while cells that express multiple isoforms at similar levels show high isoform diversity (high entropy).
</p>
</div>
<div id="measuring-entropy-with-find_diversity-function" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Measuring entropy with <code>find_diversity()</code> function<a href="profiling-isoform-diversity.html#measuring-entropy-with-find_diversity-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s focus on the Radial Glial cells and explore the diversity of gene expression within individual cells.</p>
<p>In the FLAMESv2 software, we provide a dedicated function called <code>find_diversity()</code>. This function offers a simple way to calculate Shannon’s entropy for every gene in every cell, providing a quantitative measure of isoform or gene expression diversity. In FLAMESv2, This function is designed to operate on SingleCellExperiment (SCE) objects; however, in this tutorial we use a Seurat object. Therefore, the function has been appropriately modified to accept a Seurat object as input.</p>
<p>For detailed explanations of the parameters and outputs, please refer to the FLAMES Bioconductor manual.</p>
<p>In the following section, we will examine the diversity metrics across approximately 100 Radial Glial cells to illustrate this functionality in action.</p>
<p>We can load and run the function. It may take a while if you are processing many genes or many isofroms.</p>
<p>We will subset the object and run the diversity function</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="profiling-isoform-diversity.html#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load diversity </span></span>
<span id="cb120-2"><a href="profiling-isoform-diversity.html#cb120-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-3"><a href="profiling-isoform-diversity.html#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;code/find_diversity.R&quot;</span>)</span>
<span id="cb120-4"><a href="profiling-isoform-diversity.html#cb120-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-5"><a href="profiling-isoform-diversity.html#cb120-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to RG</span></span>
<span id="cb120-6"><a href="profiling-isoform-diversity.html#cb120-6" aria-hidden="true" tabindex="-1"></a>RG_seu_obj <span class="ot">&lt;-</span> <span class="fu">subset</span>(seu_obj , <span class="at">subset =</span> sctype_db <span class="sc">==</span> <span class="st">&quot;Radial glial cells&quot;</span>)</span>
<span id="cb120-7"><a href="profiling-isoform-diversity.html#cb120-7" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(RG_seu_obj, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span>
<span id="cb120-8"><a href="profiling-isoform-diversity.html#cb120-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-9"><a href="profiling-isoform-diversity.html#cb120-9" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(RG_seu_obj) <span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb120-10"><a href="profiling-isoform-diversity.html#cb120-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-11"><a href="profiling-isoform-diversity.html#cb120-11" aria-hidden="true" tabindex="-1"></a><span class="co">#select genes we want to test. </span></span>
<span id="cb120-12"><a href="profiling-isoform-diversity.html#cb120-12" aria-hidden="true" tabindex="-1"></a><span class="co">#In this case all genes</span></span>
<span id="cb120-13"><a href="profiling-isoform-diversity.html#cb120-13" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(RG_seu_obj)</span>
<span id="cb120-14"><a href="profiling-isoform-diversity.html#cb120-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-15"><a href="profiling-isoform-diversity.html#cb120-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(progressr)</span>
<span id="cb120-16"><a href="profiling-isoform-diversity.html#cb120-16" aria-hidden="true" tabindex="-1"></a><span class="fu">handlers</span>(<span class="st">&quot;txtprogressbar&quot;</span>)</span>
<span id="cb120-17"><a href="profiling-isoform-diversity.html#cb120-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-18"><a href="profiling-isoform-diversity.html#cb120-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Use with_progress and create progressor inside</span></span>
<span id="cb120-19"><a href="profiling-isoform-diversity.html#cb120-19" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">find_diversity</span>(</span>
<span id="cb120-20"><a href="profiling-isoform-diversity.html#cb120-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">obj =</span> RG_seu_obj,</span>
<span id="cb120-21"><a href="profiling-isoform-diversity.html#cb120-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">min_counts_per_cell =</span> <span class="dv">10</span>,</span>
<span id="cb120-22"><a href="profiling-isoform-diversity.html#cb120-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>,</span>
<span id="cb120-23"><a href="profiling-isoform-diversity.html#cb120-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes =</span> genes</span>
<span id="cb120-24"><a href="profiling-isoform-diversity.html#cb120-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb120-25"><a href="profiling-isoform-diversity.html#cb120-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-26"><a href="profiling-isoform-diversity.html#cb120-26" aria-hidden="true" tabindex="-1"></a><span class="do">## save file </span></span>
<span id="cb120-27"><a href="profiling-isoform-diversity.html#cb120-27" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(res, &quot;output_files/res.rds&quot;)</span></span></code></pre></div>
</details>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="profiling-isoform-diversity.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load res files </span></span>
<span id="cb121-2"><a href="profiling-isoform-diversity.html#cb121-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;./output_files/res.rds&quot;</span>)</span>
<span id="cb121-3"><a href="profiling-isoform-diversity.html#cb121-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-4"><a href="profiling-isoform-diversity.html#cb121-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(res<span class="sc">$</span>raw_entropy)</span></code></pre></div>
</details>
<pre><code>## [1] 269 112</code></pre>
<p>After running the diversity function, we identified 269 genes for which we could calculate isoform diversity metrics. By default, these tend to be highly expressed genes that are detected in at least 20% of cells. These parameters can be changed by the user, but because single-cell data are sparse we recommend being conservative. This helps make sure that each gene has enough counts per cell to accurately estimate diversity.</p>
<p>We can summarise these results in a heatmap. In this plot, each column is a single cell and each row is a gene that passed filtering. Grey regions indicate cell–gene combinations where there were not enough counts to calculate entropy for that gene in that cell.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="profiling-isoform-diversity.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb123-2"><a href="profiling-isoform-diversity.html#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(res<span class="sc">$</span>normalized_entropy,</span>
<span id="cb123-3"><a href="profiling-isoform-diversity.html#cb123-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb123-4"><a href="profiling-isoform-diversity.html#cb123-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb123-5"><a href="profiling-isoform-diversity.html#cb123-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb123-6"><a href="profiling-isoform-diversity.html#cb123-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb123-7"><a href="profiling-isoform-diversity.html#cb123-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">&quot;Entropy heatmap&quot;</span>,</span>
<span id="cb123-8"><a href="profiling-isoform-diversity.html#cb123-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">na_col =</span> <span class="st">&quot;darkgrey&quot;</span>, )</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>We can summarise these data by examining the top and bottom entropy genes—those showing the highest and lowest levels of isoform diversity across Radial Glial cells.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="profiling-isoform-diversity.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb124-2"><a href="profiling-isoform-diversity.html#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Median entropy per gene</span></span>
<span id="cb124-3"><a href="profiling-isoform-diversity.html#cb124-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb124-4"><a href="profiling-isoform-diversity.html#cb124-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> <span class="fu">rownames</span>(res<span class="sc">$</span>normalized_entropy),</span>
<span id="cb124-5"><a href="profiling-isoform-diversity.html#cb124-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">median_entropy =</span> <span class="fu">apply</span>(res<span class="sc">$</span>normalized_entropy, <span class="dv">1</span>, median, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb124-6"><a href="profiling-isoform-diversity.html#cb124-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb124-7"><a href="profiling-isoform-diversity.html#cb124-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-8"><a href="profiling-isoform-diversity.html#cb124-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Global median</span></span>
<span id="cb124-9"><a href="profiling-isoform-diversity.html#cb124-9" aria-hidden="true" tabindex="-1"></a>global_median <span class="ot">&lt;-</span> <span class="fu">median</span>(df<span class="sc">$</span>median_entropy, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb124-10"><a href="profiling-isoform-diversity.html#cb124-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-11"><a href="profiling-isoform-diversity.html#cb124-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Select bottom 25 nonzero genes, always include VIM</span></span>
<span id="cb124-12"><a href="profiling-isoform-diversity.html#cb124-12" aria-hidden="true" tabindex="-1"></a>df_nonzero <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(median_entropy <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">|</span> gene <span class="sc">==</span> <span class="st">&quot;VIM&quot;</span>)</span>
<span id="cb124-13"><a href="profiling-isoform-diversity.html#cb124-13" aria-hidden="true" tabindex="-1"></a>df_low <span class="ot">&lt;-</span> df_nonzero <span class="sc">%&gt;%</span></span>
<span id="cb124-14"><a href="profiling-isoform-diversity.html#cb124-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">median_entropy =</span> <span class="fu">as.numeric</span>(median_entropy)) <span class="sc">%&gt;%</span></span>
<span id="cb124-15"><a href="profiling-isoform-diversity.html#cb124-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(median_entropy) <span class="sc">%&gt;%</span></span>
<span id="cb124-16"><a href="profiling-isoform-diversity.html#cb124-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>)</span>
<span id="cb124-17"><a href="profiling-isoform-diversity.html#cb124-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-18"><a href="profiling-isoform-diversity.html#cb124-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-19"><a href="profiling-isoform-diversity.html#cb124-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Top 25 by highest median entropy</span></span>
<span id="cb124-20"><a href="profiling-isoform-diversity.html#cb124-20" aria-hidden="true" tabindex="-1"></a>df_high <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb124-21"><a href="profiling-isoform-diversity.html#cb124-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">median_entropy =</span> <span class="fu">as.numeric</span>(median_entropy)) <span class="sc">%&gt;%</span>  <span class="co"># ensure it&#39;s numeric</span></span>
<span id="cb124-22"><a href="profiling-isoform-diversity.html#cb124-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(median_entropy)) <span class="sc">%&gt;%</span></span>
<span id="cb124-23"><a href="profiling-isoform-diversity.html#cb124-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>)</span>
<span id="cb124-24"><a href="profiling-isoform-diversity.html#cb124-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-25"><a href="profiling-isoform-diversity.html#cb124-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-26"><a href="profiling-isoform-diversity.html#cb124-26" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Low entropy plot (blue) with scale break ---</span></span>
<span id="cb124-27"><a href="profiling-isoform-diversity.html#cb124-27" aria-hidden="true" tabindex="-1"></a>p_low <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_low, <span class="fu">aes</span>(<span class="at">x =</span> median_entropy, <span class="at">y =</span> <span class="fu">reorder</span>(gene, median_entropy), <span class="at">fill =</span> median_entropy)) <span class="sc">+</span></span>
<span id="cb124-28"><a href="profiling-isoform-diversity.html#cb124-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb124-29"><a href="profiling-isoform-diversity.html#cb124-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> global_median, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb124-30"><a href="profiling-isoform-diversity.html#cb124-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;darkblue&quot;</span>, <span class="at">high =</span> <span class="st">&quot;skyblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb124-31"><a href="profiling-isoform-diversity.html#cb124-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Median Normalised Entropy&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Low Entropy Genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb124-32"><a href="profiling-isoform-diversity.html#cb124-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb124-33"><a href="profiling-isoform-diversity.html#cb124-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb124-34"><a href="profiling-isoform-diversity.html#cb124-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-35"><a href="profiling-isoform-diversity.html#cb124-35" aria-hidden="true" tabindex="-1"></a><span class="co"># --- High entropy plot (red) ---</span></span>
<span id="cb124-36"><a href="profiling-isoform-diversity.html#cb124-36" aria-hidden="true" tabindex="-1"></a>p_high <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_high, <span class="fu">aes</span>(<span class="at">x =</span> median_entropy, <span class="at">y =</span> <span class="fu">reorder</span>(gene, median_entropy), <span class="at">fill =</span> median_entropy)) <span class="sc">+</span></span>
<span id="cb124-37"><a href="profiling-isoform-diversity.html#cb124-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb124-38"><a href="profiling-isoform-diversity.html#cb124-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> global_median, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb124-39"><a href="profiling-isoform-diversity.html#cb124-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;mistyrose&quot;</span>, <span class="at">high =</span> <span class="st">&quot;firebrick&quot;</span>) <span class="sc">+</span></span>
<span id="cb124-40"><a href="profiling-isoform-diversity.html#cb124-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Median Normalised Entropy&quot;</span>, <span class="at">y =</span> <span class="st">&quot;High Entropy Genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb124-41"><a href="profiling-isoform-diversity.html#cb124-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb124-42"><a href="profiling-isoform-diversity.html#cb124-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb124-43"><a href="profiling-isoform-diversity.html#cb124-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-44"><a href="profiling-isoform-diversity.html#cb124-44" aria-hidden="true" tabindex="-1"></a>p_high <span class="sc">|</span> p_low</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
</div>
<div id="example-of-a-low-entropy-gene" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Example of a low entropy gene<a href="profiling-isoform-diversity.html#example-of-a-low-entropy-gene" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Lets choose two genes to have a closer look at. One with low entropy - VIM and one with high entropy NFIA. We can use some plotting functions to explore the expression of these isofroms for each gene.</p>
<p>First let’s look at VIM.</p>
<p>Vim is a simple in the sense that nearly all counts for these gene belong to the canonical isoform. We can plot this on a UMAP as shown bellow.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="profiling-isoform-diversity.html#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load Visualize fucntions </span></span>
<span id="cb125-2"><a href="profiling-isoform-diversity.html#cb125-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;code/plot_isoforms_exp_umap.R&quot;</span>)</span>
<span id="cb125-3"><a href="profiling-isoform-diversity.html#cb125-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;code/plot_isoform_entropy.R&quot;</span>) </span>
<span id="cb125-4"><a href="profiling-isoform-diversity.html#cb125-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;code/plotIsoformPieCells.R&quot;</span>) </span>
<span id="cb125-5"><a href="profiling-isoform-diversity.html#cb125-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-6"><a href="profiling-isoform-diversity.html#cb125-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_isoforms_exp_umap</span>(</span>
<span id="cb125-7"><a href="profiling-isoform-diversity.html#cb125-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">obj       =</span> RG_seu_obj,</span>
<span id="cb125-8"><a href="profiling-isoform-diversity.html#cb125-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene      =</span> <span class="st">&quot;VIM&quot;</span>,</span>
<span id="cb125-9"><a href="profiling-isoform-diversity.html#cb125-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb125-10"><a href="profiling-isoform-diversity.html#cb125-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">slot      =</span> <span class="st">&quot;data&quot;</span>,</span>
<span id="cb125-11"><a href="profiling-isoform-diversity.html#cb125-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>,</span>
<span id="cb125-12"><a href="profiling-isoform-diversity.html#cb125-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb125-13"><a href="profiling-isoform-diversity.html#cb125-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">n_features =</span> <span class="dv">10</span> <span class="co">#,  # Optional: limit to top 5 isoforms</span></span>
<span id="cb125-14"><a href="profiling-isoform-diversity.html#cb125-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#entropy_mat = res$normalized_entropy</span></span>
<span id="cb125-15"><a href="profiling-isoform-diversity.html#cb125-15" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</details>
<pre><code>##                                     Feature Expression
## ENST00000224237.9-VIM ENST00000224237.9-VIM 488.094333
## ENST00000487938.5-VIM ENST00000487938.5-VIM  11.304910
## ENST00000469543.5-VIM ENST00000469543.5-VIM   7.051489
## ENST00000485947.1-VIM ENST00000485947.1-VIM   3.992523</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>We can also look at the proportion of isoform expression for VIM in each cell. As clearly shown bellow nearly all cells use a single isofroms.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="profiling-isoform-diversity.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotIsoformPieCells</span>(<span class="at">seurat_obj =</span> RG_seu_obj,</span>
<span id="cb127-2"><a href="profiling-isoform-diversity.html#cb127-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gene =</span> <span class="st">&quot;VIM&quot;</span>,</span>
<span id="cb127-3"><a href="profiling-isoform-diversity.html#cb127-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>,</span>
<span id="cb127-4"><a href="profiling-isoform-diversity.html#cb127-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb127-5"><a href="profiling-isoform-diversity.html#cb127-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">top_n =</span> <span class="dv">3</span>,</span>
<span id="cb127-6"><a href="profiling-isoform-diversity.html#cb127-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n_cells =</span> <span class="dv">12</span>, <span class="co"># change this to mplot more cells </span></span>
<span id="cb127-7"><a href="profiling-isoform-diversity.html#cb127-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">plot_col=</span><span class="dv">4</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="example-of-a-high-entropy-gene" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Example of a high entropy gene<a href="profiling-isoform-diversity.html#example-of-a-high-entropy-gene" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s contrast this to a gene with high entropy. In this case we will look at NFIA.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="profiling-isoform-diversity.html#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_isoforms_exp_umap</span>(</span>
<span id="cb128-2"><a href="profiling-isoform-diversity.html#cb128-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">obj       =</span> RG_seu_obj,</span>
<span id="cb128-3"><a href="profiling-isoform-diversity.html#cb128-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene      =</span> <span class="st">&quot;NFIA&quot;</span>,</span>
<span id="cb128-4"><a href="profiling-isoform-diversity.html#cb128-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb128-5"><a href="profiling-isoform-diversity.html#cb128-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">slot      =</span> <span class="st">&quot;data&quot;</span>,</span>
<span id="cb128-6"><a href="profiling-isoform-diversity.html#cb128-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>,</span>
<span id="cb128-7"><a href="profiling-isoform-diversity.html#cb128-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb128-8"><a href="profiling-isoform-diversity.html#cb128-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">n_features =</span> <span class="dv">4</span> <span class="co">#,  # Optional: limit to top 4 isoforms</span></span>
<span id="cb128-9"><a href="profiling-isoform-diversity.html#cb128-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#entropy_mat = res$normalized_entropy</span></span>
<span id="cb128-10"><a href="profiling-isoform-diversity.html#cb128-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</details>
<pre><code>##                                       Feature Expression
## ENST00000403491.8-NFIA ENST00000403491.8-NFIA 161.913441
## ENST00000603233.2-NFIA ENST00000603233.2-NFIA 157.623083
## ENST00000664149.1-NFIA ENST00000664149.1-NFIA   7.102752
## ENST00000371187.7-NFIA ENST00000371187.7-NFIA   5.396720</code></pre>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="profiling-isoform-diversity.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotIsoformPieCells</span>(<span class="at">seurat_obj =</span> RG_seu_obj,</span>
<span id="cb130-2"><a href="profiling-isoform-diversity.html#cb130-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gene =</span> <span class="st">&quot;NFIA&quot;</span>,</span>
<span id="cb130-3"><a href="profiling-isoform-diversity.html#cb130-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">assay =</span> <span class="st">&quot;iso&quot;</span>,</span>
<span id="cb130-4"><a href="profiling-isoform-diversity.html#cb130-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb130-5"><a href="profiling-isoform-diversity.html#cb130-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">top_n =</span> <span class="dv">5</span>,</span>
<span id="cb130-6"><a href="profiling-isoform-diversity.html#cb130-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n_cells =</span> <span class="dv">12</span>,</span>
<span id="cb130-7"><a href="profiling-isoform-diversity.html#cb130-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">plot_col=</span><span class="dv">4</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p>These pie charts clearly show that, in contrast to VIM, cells expressing NFIA utilize at least two isoforms, often in nearly equal proportions. This pattern explains the high entropy observed for NFIA and suggests a more complex expression profile compared to a simpler, single-isoform gene like VIM.</p>
<p>Interestingly, there are numerous NFIA isoforms. The two most highly expressed isoforms shown here are quite distinct, as illustrated in the IsoVis plot below — one is protein-coding, containing critical functional domains, while the other represents a non-coding RNA variant.</p>
<p>Using IsoVis, we can further explore the structural differences between these isoforms and visualize how they contribute to transcriptomic complexity within individual cells.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="profiling-isoform-diversity.html#cb131-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/IsoVis_ENSG00000162599.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-19"></span>
<img src="images/IsoVis_ENSG00000162599.png" alt="Isoform diversity in NFIA (ENSG00000162599). Structural comparison of NFIA isoforms highlights the diversity of transcript usage detected in single-cell long-read data. The canonical and user-discovered isoforms differ markedly in exon structure and domain composition. One isoform encodes a full-length protein containing key functional domains (Nuclear factor, MH1, and CTF/NF-I), while another represents a non-coding or truncated transcript lacking these regions." width="1600px" />
<p class="caption">
Figure 9.2: Isoform diversity in NFIA (ENSG00000162599). Structural comparison of NFIA isoforms highlights the diversity of transcript usage detected in single-cell long-read data. The canonical and user-discovered isoforms differ markedly in exon structure and domain composition. One isoform encodes a full-length protein containing key functional domains (Nuclear factor, MH1, and CTF/NF-I), while another represents a non-coding or truncated transcript lacking these regions.
</p>
</div>
<p>By quantifying entropy at the single-cell level, we can distinguish simple genes - those dominated by one isoform, from complex genes that express multiple isoforms with variable expression across cells.</p>
<p>In the case of Radial Glial cells, this approach reveals a mixture of both expression modes: some genes like VIM exhibit low isoform diversity, consistent with stable structural roles, whereas others like NFIA show high entropy, reflecting more dynamic or context-dependent regulation.</p>
</div>
<div id="broadening-entropy-analysis" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Broadening Entropy Analysis<a href="profiling-isoform-diversity.html#broadening-entropy-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For simplicity, in this example we explored isoform entropy in only a small subset of cells. However, this analysis can easily be expanded to include broader cell populations or even the entire dataset. Doing so enables more comprehensive, cell type–specific investigations for example, assessing whether certain genes show different entropy levels across cell types Such comparisons can reveal important biological insights: some cell types may display low entropy, consistent with a simple and stable transcriptional program, while others—particularly more mature or specialized lineages may exhibit higher entropy, reflecting more complex isoform regulation and splicing diversity.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="novel-isoforms.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multisample-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/08_diversity.rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/08_diversity.rmd",
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
