<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Removing sources of unwanted noise from the single cell dataset | Long-read Single-Cell RNA-seq analysis tutorial</title>
  <meta name="description" content="Chapter 3 Removing sources of unwanted noise from the single cell dataset | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Removing sources of unwanted noise from the single cell dataset | Long-read Single-Cell RNA-seq analysis tutorial" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Removing sources of unwanted noise from the single cell dataset | Long-read Single-Cell RNA-seq analysis tutorial" />
  
  
  

<meta name="author" content="Sefi Prawer Postdoctoral Research Fellow in Long Reads and Single-Cell Transcriptomics University of Melbourne" />


<meta name="date" content="2025-11-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="setup.html"/>
<link rel="next" href="add-isoform-counts-to-seurat-object.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/codefolding-lua-1.1/codefolding-lua.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.1</b> Prerequisites</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#getting-started-with-the-data"><i class="fa fa-check"></i><b>1.2</b> Getting Started with the Data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#dataset-information"><i class="fa fa-check"></i><b>1.3</b> Dataset Information</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.4</b> Citation</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>2</b> <strong>Setup</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="setup.html"><a href="setup.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> Load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="setup.html"><a href="setup.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="setup.html"><a href="setup.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> <strong>Removing sources of unwanted noise from the single cell dataset</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.2</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.3</b> Standard gene QC to remove low quality cells</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html"><i class="fa fa-check"></i><b>4</b> <strong>Add isoform counts to Seurat object</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#create-a-seurat-object-with-isoform-expression-data"><i class="fa fa-check"></i><b>4.1</b> Create a Seurat object with isoform expression data</a></li>
<li class="chapter" data-level="4.2" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#filter-the-new-seurat-object-based-on-gene-level-information"><i class="fa fa-check"></i><b>4.2</b> Filter the new Seurat object based on gene level information</a></li>
<li class="chapter" data-level="4.3" data-path="add-isoform-counts-to-seurat-object.html"><a href="add-isoform-counts-to-seurat-object.html#add-the-isoform-assay-to-the-seurat-object"><i class="fa fa-check"></i><b>4.3</b> Add the isoform assay to the Seurat object</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html"><i class="fa fa-check"></i><b>5</b> <strong>Finding marker genes and isoforms</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#differentially-expressed-genes-by-cluster-identity"><i class="fa fa-check"></i><b>5.1</b> Differentially expressed genes by cluster identity</a></li>
<li class="chapter" data-level="5.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#identifying-cell-types"><i class="fa fa-check"></i><b>5.2</b> Identifying cell types</a></li>
<li class="chapter" data-level="5.3" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#de-genes-and-isoforms-based-on-annotated-cell-types."><i class="fa fa-check"></i><b>5.3</b> DE genes and isoforms based on annotated cell types.</a></li>
<li class="chapter" data-level="5.4" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#volcano-plots"><i class="fa fa-check"></i><b>5.4</b> Volcano plots</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findallmarkers-de"><i class="fa fa-check"></i><b>5.4.1</b> FindAllMarkers DE</a></li>
<li class="chapter" data-level="5.4.2" data-path="finding-marker-genes-and-isoforms.html"><a href="finding-marker-genes-and-isoforms.html#findmarkers-de"><i class="fa fa-check"></i><b>5.4.2</b> FindMarkers DE</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html"><i class="fa fa-check"></i><b>6</b> <strong>Exploring isoforms of interest</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#isoforms-expressed-per-gene"><i class="fa fa-check"></i><b>6.1</b> Isoforms expressed per gene</a></li>
<li class="chapter" data-level="6.2" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#top-10-genes-with-most-isoforms"><i class="fa fa-check"></i><b>6.2</b> Top 10 Genes with Most Isoforms</a></li>
<li class="chapter" data-level="6.3" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#exploring-macf1-isoforms"><i class="fa fa-check"></i><b>6.3</b> Exploring <em>MACF1</em> isoforms</a></li>
<li class="chapter" data-level="6.4" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#expression-of-macf1-isoforms-across-cell-types"><i class="fa fa-check"></i><b>6.4</b> Expression of <em>MACF1</em> isoforms Across Cell Types</a></li>
<li class="chapter" data-level="6.5" data-path="exploring-isoforms-of-interest.html"><a href="exploring-isoforms-of-interest.html#visualization-of-isoform-structures"><i class="fa fa-check"></i><b>6.5</b> Visualization of Isoform Structures</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="isoform-classification.html"><a href="isoform-classification.html"><i class="fa fa-check"></i><b>7</b> <strong>Isoform Classification</strong></a>
<ul>
<li class="chapter" data-level="7.1" data-path="isoform-classification.html"><a href="isoform-classification.html#classification-with-sqanti"><i class="fa fa-check"></i><b>7.1</b> Classification with SQANTI</a></li>
<li class="chapter" data-level="7.2" data-path="isoform-classification.html"><a href="isoform-classification.html#cell-type-specific-isoforms"><i class="fa fa-check"></i><b>7.2</b> Cell type specific isoforms</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="novel-isoforms.html"><a href="novel-isoforms.html"><i class="fa fa-check"></i><b>8</b> <strong>Novel isoforms</strong></a>
<ul>
<li class="chapter" data-level="8.1" data-path="novel-isoforms.html"><a href="novel-isoforms.html#find-all-the-genes-that-express-at-least-1-novel-isoform"><i class="fa fa-check"></i><b>8.1</b> Find all the genes that express at least 1 novel isoform</a></li>
<li class="chapter" data-level="8.2" data-path="novel-isoforms.html"><a href="novel-isoforms.html#visualising-oaz2-novel-isoform"><i class="fa fa-check"></i><b>8.2</b> Visualising OAZ2 novel isoform</a></li>
<li class="chapter" data-level="8.3" data-path="novel-isoforms.html"><a href="novel-isoforms.html#functional-impacts-of-novel-isoforms"><i class="fa fa-check"></i><b>8.3</b> Functional impacts of novel isoforms</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html"><i class="fa fa-check"></i><b>9</b> <strong>Profiling Isoform Diversity</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#measuring-entropy-with-find_diversity-function"><i class="fa fa-check"></i><b>9.1</b> Measuring entropy with <code>find_diversity()</code> function</a></li>
<li class="chapter" data-level="9.2" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#example-of-a-low-entropy-gene"><i class="fa fa-check"></i><b>9.2</b> Example of a low entropy gene</a></li>
<li class="chapter" data-level="9.3" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#example-of-a-high-entropy-gene"><i class="fa fa-check"></i><b>9.3</b> Example of a high entropy gene</a></li>
<li class="chapter" data-level="9.4" data-path="profiling-isoform-diversity.html"><a href="profiling-isoform-diversity.html#broadening-entropy-analysis"><i class="fa fa-check"></i><b>9.4</b> Broadening Entropy Analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="multisample-analysis.html"><a href="multisample-analysis.html"><i class="fa fa-check"></i><b>10</b> <strong>Multisample analysis</strong></a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#dataset-information-1"><i class="fa fa-check"></i><b>10.0.1</b> Dataset Information</a></li>
<li class="chapter" data-level="10.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#standard-pre-processing-and-quality-control"><i class="fa fa-check"></i><b>10.1</b> Standard pre-processing and quality control</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="multisample-analysis.html"><a href="multisample-analysis.html#define-qc-function"><i class="fa fa-check"></i><b>10.1.1</b> Define QC function</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="multisample-analysis.html"><a href="multisample-analysis.html#multisample-integration"><i class="fa fa-check"></i><b>10.2</b> Multisample integration</a></li>
<li class="chapter" data-level="10.3" data-path="multisample-analysis.html"><a href="multisample-analysis.html#add-isoform-counts"><i class="fa fa-check"></i><b>10.3</b> Add isoform counts</a></li>
<li class="chapter" data-level="10.4" data-path="multisample-analysis.html"><a href="multisample-analysis.html#identify-cell-types"><i class="fa fa-check"></i><b>10.4</b> Identify cell types</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html"><i class="fa fa-check"></i><b>11</b> <strong>Trajectory analysis</strong></a>
<ul>
<li class="chapter" data-level="11.1" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#running-pseudotime-analysis-on-genes"><i class="fa fa-check"></i><b>11.1</b> Running pseudotime analysis on genes</a></li>
<li class="chapter" data-level="11.2" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#running-pseudotime-analysis-on-isoforms"><i class="fa fa-check"></i><b>11.2</b> Running pseudotime analysis on isoforms</a></li>
<li class="chapter" data-level="11.3" data-path="trajectory-analysis.html"><a href="trajectory-analysis.html#summary-and-downstream-applications"><i class="fa fa-check"></i><b>11.3</b> Summary and downstream applications</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><strong>Conclusion</strong></a>
<ul>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="conclusion.html"><a href="conclusion.html#session-info"><i class="fa fa-check"></i>Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><strong>References</strong></a></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><strong>Appendix</strong></a>
<ul>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html#calculating-the-ambient-rna-profile"><i class="fa fa-check"></i>Calculating the ambient RNA profile</a></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html#multisample-qc-function"><i class="fa fa-check"></i>Multisample QC function</a></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html#convert-oarfish-files-to-a-count-matrix"><i class="fa fa-check"></i>Convert Oarfish files to a count matrix</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Long-read Single-Cell RNA-seq analysis tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="removing-sources-of-unwanted-noise-from-the-single-cell-dataset" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> <strong>Removing sources of unwanted noise from the single cell dataset</strong><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-sources-of-unwanted-noise-from-the-single-cell-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<hr />
<p>Now we will do some initial preprocessing of single cell data to ensure we have some high quality data. This will involve 3 main steps</p>
<ol style="list-style-type: decimal">
<li><p>Removing empty droplets - droplets that do not contain true cells.</p></li>
<li><p>Removing ambient RNA contamination - optional</p></li>
<li><p>Removing low quality cells</p></li>
</ol>
<div id="empty-droplets" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Empty droplets<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This function removes empty droplets, a critical step to ensure that only true cells are retained for analysis. In short-read analysis using CellRanger, this process is automated, and empty droplets are removed by the software. However, FLAMES does not perform this step automatically, so it must be done manually. The function provided here not only removes empty droplets but also generates general QC metrics, enabling users to assess the reasonableness of the number of cells removed.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">### notes: This function should be refactored and cleaned up. It&#39;s very long and complex. </span></span>
<span id="cb8-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-2" aria-hidden="true" tabindex="-1"></a>perform_empty_drops_analysis <span class="ot">&lt;-</span> <span class="cf">function</span>(output_path, gene_count_file, empty_drops_file, output_seurat_file, <span class="at">fdr_threshold =</span> <span class="fl">0.001</span>, <span class="at">lower =</span> <span class="dv">100</span>) {</span>
<span id="cb8-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load required libraries</span></span>
<span id="cb8-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read in data</span></span>
<span id="cb8-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(gene_count_file, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb8-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-6" aria-hidden="true" tabindex="-1"></a>  df_emptydrops <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(empty_drops_file, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb8-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine the dataframes by row names</span></span>
<span id="cb8-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-9" aria-hidden="true" tabindex="-1"></a>  combined_df <span class="ot">&lt;-</span> <span class="fu">merge</span>(df, df_emptydrops, <span class="at">by =</span> <span class="st">&quot;row.names&quot;</span>, <span class="at">all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(combined_df) <span class="ot">&lt;-</span> combined_df[, <span class="dv">1</span>]</span>
<span id="cb8-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-11" aria-hidden="true" tabindex="-1"></a>  combined_df[, <span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb8-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-12" aria-hidden="true" tabindex="-1"></a>  combined_df[<span class="fu">is.na</span>(combined_df)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb8-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-14"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform standard pre-processing before empty drops analysis</span></span>
<span id="cb8-15"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-15" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> df, <span class="at">project =</span> <span class="st">&quot;Day_55&quot;</span>, <span class="at">min.features =</span> <span class="dv">20</span>)</span>
<span id="cb8-16"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-16" aria-hidden="true" tabindex="-1"></a>  seurat_obj[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(seurat_obj, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb8-17"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb8-18"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="co">#seurat_obj &lt;- subset(seurat_obj, subset = nFeature_RNA &gt; 10 &amp; nFeature_RNA &lt; 100000 &amp; percent.mt &lt; 100)</span></span>
<span id="cb8-19"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-19" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(seurat_obj, <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="at">scale.factor =</span> <span class="dv">10000</span>)</span>
<span id="cb8-20"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-20" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(seurat_obj, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb8-21"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-21" aria-hidden="true" tabindex="-1"></a>  all.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seurat_obj)</span>
<span id="cb8-22"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-22" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seurat_obj, <span class="at">features =</span> all.genes)</span>
<span id="cb8-23"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-23" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_obj, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> seurat_obj))</span>
<span id="cb8-24"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ElbowPlot</span>(seurat_obj)</span>
<span id="cb8-25"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-25" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb8-26"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-26" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(seurat_obj, <span class="at">resolution =</span> <span class="fl">0.5</span>)</span>
<span id="cb8-27"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-27" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb8-28"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span>
<span id="cb8-29"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-30"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define function to make dgCMatrix from combined counts</span></span>
<span id="cb8-31"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-31" aria-hidden="true" tabindex="-1"></a>  makedgcmatrix <span class="ot">&lt;-</span> <span class="cf">function</span>(count.matrix) {</span>
<span id="cb8-32"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-32" aria-hidden="true" tabindex="-1"></a>    seurat_object <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> count.matrix, <span class="at">project =</span> <span class="st">&quot;singlecell&quot;</span>)</span>
<span id="cb8-33"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(seurat_object[[<span class="st">&quot;RNA&quot;</span>]]<span class="sc">$</span>counts)</span>
<span id="cb8-34"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-34" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-35"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-36"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Empty Drops Analysis</span></span>
<span id="cb8-37"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-37" aria-hidden="true" tabindex="-1"></a>  combined_df[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(combined_df, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x)))</span>
<span id="cb8-38"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-38" aria-hidden="true" tabindex="-1"></a>  outs.ddcmatrix <span class="ot">&lt;-</span> <span class="fu">makedgcmatrix</span>(combined_df)[[<span class="dv">1</span>]]</span>
<span id="cb8-39"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-39" aria-hidden="true" tabindex="-1"></a>  br.out <span class="ot">&lt;-</span> DropletUtils<span class="sc">::</span><span class="fu">barcodeRanks</span>(outs.ddcmatrix)</span>
<span id="cb8-40"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-41"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-41" aria-hidden="true" tabindex="-1"></a>  e.out <span class="ot">&lt;-</span> <span class="fu">emptyDrops</span>(outs.ddcmatrix, <span class="at">lower =</span> lower, <span class="at">niters =</span> <span class="dv">10000</span>, <span class="at">test.ambient =</span> <span class="cn">TRUE</span>, <span class="at">BPPARAM =</span> <span class="fu">SerialParam</span>())</span>
<span id="cb8-42"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-42" aria-hidden="true" tabindex="-1"></a>  is.cell <span class="ot">&lt;-</span> e.out<span class="sc">$</span>FDR <span class="sc">&lt;</span> fdr_threshold</span>
<span id="cb8-43"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-44"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a dataframe with FDR of TRUE cells</span></span>
<span id="cb8-45"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-45" aria-hidden="true" tabindex="-1"></a>  is.true.cell_CR <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(e.out<span class="sc">@</span>listData[[<span class="st">&quot;FDR&quot;</span>]], e.out<span class="sc">@</span>rownames)</span>
<span id="cb8-46"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-46" aria-hidden="true" tabindex="-1"></a>  is.true.cell_CR <span class="ot">&lt;-</span> is.true.cell_CR <span class="sc">%&gt;%</span> <span class="fu">filter</span>(is.true.cell_CR<span class="sc">$</span><span class="st">`</span><span class="at">e.out@listData[[&quot;FDR&quot;]]</span><span class="st">`</span> <span class="sc">&lt;=</span> fdr_threshold)</span>
<span id="cb8-47"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-47" aria-hidden="true" tabindex="-1"></a>  is.true.cell_CR <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(is.true.cell_CR, <span class="st">&quot;cell_id&quot;</span>)</span>
<span id="cb8-48"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-49"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Function for retrieving the Seurat cells and cluster in dataframe</span></span>
<span id="cb8-50"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-50" aria-hidden="true" tabindex="-1"></a>  overlap_true_cell <span class="ot">&lt;-</span> <span class="cf">function</span>(seurat_object) {</span>
<span id="cb8-51"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-51" aria-hidden="true" tabindex="-1"></a>    seurat_cluster.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(seurat_object<span class="sc">$</span>seurat_clusters)</span>
<span id="cb8-52"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-52" aria-hidden="true" tabindex="-1"></a>    seurat_cluster.df <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(seurat_cluster.df, <span class="st">&quot;cell_id&quot;</span>)</span>
<span id="cb8-53"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-53" aria-hidden="true" tabindex="-1"></a>    seurat_cluster.df</span>
<span id="cb8-54"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-54" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-55"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-56"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-56" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Obtain cluster dataframe from Seurat object</span></span>
<span id="cb8-57"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-57" aria-hidden="true" tabindex="-1"></a>  overlap_CR <span class="ot">&lt;-</span> <span class="fu">overlap_true_cell</span>(seurat_obj)</span>
<span id="cb8-58"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-59"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check overlaps between Seurat object and true cells</span></span>
<span id="cb8-60"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(overlap_CR<span class="sc">$</span>cell_id <span class="sc">%in%</span> is.true.cell_CR<span class="sc">$</span>cell_id)</span>
<span id="cb8-61"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-62"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Function to add metadata to Seurat object</span></span>
<span id="cb8-63"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-63" aria-hidden="true" tabindex="-1"></a>  True.cells <span class="ot">&lt;-</span> <span class="cf">function</span>(e.out) {</span>
<span id="cb8-64"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-64" aria-hidden="true" tabindex="-1"></a>    cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(e.out<span class="sc">@</span>rownames)</span>
<span id="cb8-65"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-65" aria-hidden="true" tabindex="-1"></a>    fdr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(e.out<span class="sc">$</span>FDR)</span>
<span id="cb8-66"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-66" aria-hidden="true" tabindex="-1"></a>    T.F.cells <span class="ot">&lt;-</span> <span class="fu">cbind</span>(cells, fdr)</span>
<span id="cb8-67"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-67" aria-hidden="true" tabindex="-1"></a>    T.F.cells <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(T.F.cells[,<span class="sc">-</span><span class="dv">1</span>], <span class="at">row.names =</span> T.F.cells[,<span class="dv">1</span>])</span>
<span id="cb8-68"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setnames</span>(T.F.cells, <span class="fu">c</span>(<span class="st">&#39;FDR&#39;</span>))</span>
<span id="cb8-69"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-69" aria-hidden="true" tabindex="-1"></a>    T.F.cells <span class="sc">%&gt;%</span></span>
<span id="cb8-70"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-70" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">FDR =</span> <span class="fu">case_when</span>(FDR <span class="sc">&lt;</span> fdr_threshold <span class="sc">~</span> <span class="st">&quot;Cells&quot;</span>, FDR <span class="sc">&gt;</span> fdr_threshold <span class="sc">~</span> <span class="st">&quot;Empty_drops&quot;</span>))</span>
<span id="cb8-71"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-71" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-72"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-73"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-73" aria-hidden="true" tabindex="-1"></a>  cells_CR <span class="ot">&lt;-</span> <span class="fu">True.cells</span>(e.out)</span>
<span id="cb8-74"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-74" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(seurat_obj, <span class="at">metadata =</span> cells_CR, <span class="at">col.name =</span> <span class="st">&#39;is.cell&#39;</span>)</span>
<span id="cb8-75"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-76"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-76" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot Empty drops on Gene UMAP</span></span>
<span id="cb8-77"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-77" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create a ggplot object</span></span>
<span id="cb8-78"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-78" aria-hidden="true" tabindex="-1"></a>  rankplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(br.out, <span class="fu">aes</span>(<span class="at">x =</span> rank, <span class="at">y =</span> total)) <span class="sc">+</span></span>
<span id="cb8-79"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb8-80"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-80" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb8-81"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb8-82"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-82" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Rank&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Total&quot;</span>) <span class="sc">+</span></span>
<span id="cb8-83"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> fitted), <span class="at">color =</span> <span class="st">&quot;red&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;solid&quot;</span>) <span class="sc">+</span></span>
<span id="cb8-84"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-84" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">metadata</span>(br.out)<span class="sc">$</span>knee, <span class="at">color =</span> <span class="st">&quot;dodgerblue&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb8-85"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-85" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">metadata</span>(br.out)<span class="sc">$</span>inflection, <span class="at">color =</span> <span class="st">&quot;forestgreen&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb8-86"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb8-87"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-87" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">&quot;bottomleft&quot;</span></span>
<span id="cb8-88"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-88" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb8-89"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">linetype =</span> <span class="fu">c</span>(<span class="st">&quot;dashed&quot;</span>, <span class="st">&quot;dashed&quot;</span>)))) <span class="sc">+</span></span>
<span id="cb8-90"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="cn">Inf</span>, <span class="at">y =</span> <span class="fu">metadata</span>(br.out)<span class="sc">$</span>knee, <span class="at">label =</span> <span class="st">&quot;knee&quot;</span>, <span class="at">color =</span> <span class="st">&quot;dodgerblue&quot;</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb8-91"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="cn">Inf</span>, <span class="at">y =</span> <span class="fu">metadata</span>(br.out)<span class="sc">$</span>inflection, <span class="at">label =</span> <span class="st">&quot;inflection&quot;</span>, <span class="at">color =</span> <span class="st">&quot;forestgreen&quot;</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)</span>
<span id="cb8-92"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-92" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-93"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Summary table -&gt; may want to add a bunch of other summary metrics </span></span>
<span id="cb8-94"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-94" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract counts with checks for NULL</span></span>
<span id="cb8-95"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-95" aria-hidden="true" tabindex="-1"></a>  cell_counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>is.cell))</span>
<span id="cb8-96"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-96" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-97"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-97" aria-hidden="true" tabindex="-1"></a>  count_true_cells <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">length</span>(cell_counts<span class="sc">$</span>Freq[cell_counts<span class="sc">$</span>Var1 <span class="sc">==</span> <span class="st">&quot;Cells&quot;</span>]) <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb8-98"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-98" aria-hidden="true" tabindex="-1"></a>                             cell_counts<span class="sc">$</span>Freq[cell_counts<span class="sc">$</span>Var1 <span class="sc">==</span> <span class="st">&quot;Cells&quot;</span>], <span class="dv">0</span>)</span>
<span id="cb8-99"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-99" aria-hidden="true" tabindex="-1"></a>  count_empty_drops <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">length</span>(cell_counts<span class="sc">$</span>Freq[cell_counts<span class="sc">$</span>Var1 <span class="sc">==</span> <span class="st">&quot;Empty_drops&quot;</span>]) <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb8-100"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-100" aria-hidden="true" tabindex="-1"></a>                              cell_counts<span class="sc">$</span>Freq[cell_counts<span class="sc">$</span>Var1 <span class="sc">==</span> <span class="st">&quot;Empty_drops&quot;</span>], <span class="dv">0</span>)</span>
<span id="cb8-101"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-102"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-102" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the summary table</span></span>
<span id="cb8-103"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-103" aria-hidden="true" tabindex="-1"></a>  summary_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb8-104"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-104" aria-hidden="true" tabindex="-1"></a>    <span class="at">Description =</span> <span class="fu">c</span>(<span class="st">&#39;fdr&#39;</span>, <span class="st">&#39;lower Counts&#39;</span>, <span class="st">&#39;number of true cells&#39;</span>, <span class="st">&#39;number of empty drops&#39;</span>),</span>
<span id="cb8-105"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">Value =</span> <span class="fu">c</span>(fdr_threshold, lower, count_true_cells, count_empty_drops)</span>
<span id="cb8-106"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-106" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb8-107"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-107" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-108"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-108" aria-hidden="true" tabindex="-1"></a>  summary_grob <span class="ot">&lt;-</span> <span class="fu">tableGrob</span>(summary_table, <span class="at">rows =</span> <span class="cn">NULL</span>, <span class="at">cols =</span> <span class="cn">NULL</span>)</span>
<span id="cb8-109"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-109" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-110"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-110" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the combined plot</span></span>
<span id="cb8-111"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-111" aria-hidden="true" tabindex="-1"></a>  plot1 <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(</span>
<span id="cb8-112"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-112" aria-hidden="true" tabindex="-1"></a>    rankplot,</span>
<span id="cb8-113"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-113" aria-hidden="true" tabindex="-1"></a>    <span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">group.by =</span> <span class="st">&#39;is.cell&#39;</span>) <span class="sc">+</span> </span>
<span id="cb8-114"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-114" aria-hidden="true" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">&quot;is.cell&quot;</span>, <span class="at">title =</span> <span class="st">&#39;Seurat Object&#39;</span>) <span class="sc">+</span> </span>
<span id="cb8-115"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-115" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>), <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;white&quot;</span>)),</span>
<span id="cb8-116"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-116" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> <span class="st">&quot;nCount_RNA&quot;</span>) <span class="sc">+</span> </span>
<span id="cb8-117"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-117" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;white&quot;</span>)),</span>
<span id="cb8-118"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-118" aria-hidden="true" tabindex="-1"></a>    <span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="sc">+</span> </span>
<span id="cb8-119"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-119" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;white&quot;</span>)),</span>
<span id="cb8-120"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-120" aria-hidden="true" tabindex="-1"></a>    summary_grob,</span>
<span id="cb8-121"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-121" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb8-122"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-122" aria-hidden="true" tabindex="-1"></a>    <span class="at">top =</span> <span class="fu">textGrob</span>(<span class="st">&#39;Empty drops vs real cells&#39;</span>)</span>
<span id="cb8-123"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-123" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb8-124"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-124" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-125"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-125" aria-hidden="true" tabindex="-1"></a>  <span class="co">#output the plot and summary stats</span></span>
<span id="cb8-126"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="at">file =</span> <span class="fu">file.path</span>(output_path, <span class="fu">paste0</span>(output_seurat_file, <span class="st">&quot;_plots.pdf&quot;</span>)), <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">bg =</span> <span class="st">&quot;white&quot;</span>)</span>
<span id="cb8-127"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(plot1)</span>
<span id="cb8-128"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb8-129"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-129" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-130"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-130" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset the Seurat object to remove cells marked as empty drops</span></span>
<span id="cb8-131"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-131" aria-hidden="true" tabindex="-1"></a>  seurat_obj_rm_empty <span class="ot">&lt;-</span> <span class="fu">subset</span>(seurat_obj, <span class="at">subset =</span> is.cell <span class="sc">==</span> <span class="st">&#39;Cells&#39;</span>)</span>
<span id="cb8-132"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-132" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-133"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-133" aria-hidden="true" tabindex="-1"></a>  <span class="co">#save the seurat objects</span></span>
<span id="cb8-134"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(seurat_obj, <span class="at">file =</span> <span class="fu">file.path</span>(output_path, <span class="fu">paste0</span>(<span class="st">&quot;with_empty_&quot;</span>, output_seurat_file, <span class="st">&quot;.rds&quot;</span>)))</span>
<span id="cb8-135"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(seurat_obj_rm_empty, <span class="at">file =</span> <span class="fu">file.path</span>(output_path, <span class="fu">paste0</span>(<span class="st">&quot;removed_empty_&quot;</span>, output_seurat_file, <span class="st">&quot;.rds&quot;</span>)))</span>
<span id="cb8-136"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-136" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-137"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-138"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-138" aria-hidden="true" tabindex="-1"></a><span class="do">#####################</span></span>
<span id="cb8-139"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-139" aria-hidden="true" tabindex="-1"></a><span class="co"># usage</span></span>
<span id="cb8-140"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-140" aria-hidden="true" tabindex="-1"></a><span class="fu">perform_empty_drops_analysis</span>(</span>
<span id="cb8-141"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-141" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_count_file =</span> <span class="st">&quot;./output_files/counts/geneSymbol_gene_count.csv&quot;</span>,</span>
<span id="cb8-142"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-142" aria-hidden="true" tabindex="-1"></a>  <span class="at">empty_drops_file =</span> <span class="st">&quot;./output_files/counts/background_geneSymbol_gene_count.csv&quot;</span>,</span>
<span id="cb8-143"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-143" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_path =</span> <span class="st">&quot;./output_files/empty_drops/&quot;</span>, </span>
<span id="cb8-144"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-144" aria-hidden="true" tabindex="-1"></a>  <span class="at">output_seurat_file =</span> <span class="st">&quot;Day55&quot;</span>,</span>
<span id="cb8-145"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-145" aria-hidden="true" tabindex="-1"></a>  <span class="at">fdr_threshold =</span> <span class="fl">0.001</span>,  <span class="co"># see droplet utils if you want to adjust these params</span></span>
<span id="cb8-146"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-146" aria-hidden="true" tabindex="-1"></a>  <span class="at">lower =</span> <span class="dv">500</span> <span class="co"># see droplet utils if you want to adjust these params</span></span>
<span id="cb8-147"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb8-147" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</details>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 344
## Number of edges: 7209
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8510
## Number of communities: 7
## Elapsed time: 0 seconds</code></pre>
<p><img src="bookdownproj_files/figure-html/perform_empty_drops_analysis-1.png" width="576" /></p>
</div>
<div id="removing-ambient-rna-contamination" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Removing ambient RNA contamination<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we will remove ambient RNA contamination using decontX (or SoupX). This step is optional, and its up to the user to decide whether its necessary. In this case, the barcode rank plot (Figure <a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#fig:barcode-rank-plot">3.1</a>) shows a clear distinction between true barcodes and background barcodes, suggesting that ambient RNA contamination may not significantly contribute to noise in this dataset. However, well demonstrate how to run this step in case your barcode rank plot is noisier and contamination is a concern. if users wish to skip this step take the seurat_obj_rm_empty object and proceed to standard gene QC step.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb10-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;images/knee_plot.png&quot;</span>)</span></code></pre></div>
</details>
<div class="figure"><span style="display:block;" id="fig:barcode-rank-plot"></span>
<img src="images/knee_plot.png" alt="Barcode rank plot produced by FLAMES." width="600px" />
<p class="caption">
Figure 3.1: Barcode rank plot produced by FLAMES.
</p>
</div>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to run decontX on a single Seurat object</span></span>
<span id="cb11-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-2" aria-hidden="true" tabindex="-1"></a>run_decontX <span class="ot">&lt;-</span> <span class="cf">function</span>(seurat_obj_path, background_counts_path, sample_id) {</span>
<span id="cb11-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load the Seurat object</span></span>
<span id="cb11-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-4" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(seurat_obj_path)</span>
<span id="cb11-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-5" aria-hidden="true" tabindex="-1"></a>  filtered_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">GetAssayData</span>(seurat_obj, <span class="at">layer =</span> <span class="st">&quot;counts&quot;</span>))</span>
<span id="cb11-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read background counts</span></span>
<span id="cb11-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-8" aria-hidden="true" tabindex="-1"></a>  raw_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.csv</span>(background_counts_path, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb11-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get cluster info from Seurat object</span></span>
<span id="cb11-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-11" aria-hidden="true" tabindex="-1"></a>  cluster_info <span class="ot">&lt;-</span> <span class="fu">setNames</span>(seurat_obj<span class="sc">$</span>seurat_clusters, <span class="fu">colnames</span>(seurat_obj))</span>
<span id="cb11-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find common genes</span></span>
<span id="cb11-14"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-14" aria-hidden="true" tabindex="-1"></a>  common_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(filtered_counts), <span class="fu">rownames</span>(raw_counts))</span>
<span id="cb11-15"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-15" aria-hidden="true" tabindex="-1"></a>  raw_counts <span class="ot">&lt;-</span> raw_counts[common_genes, ]</span>
<span id="cb11-16"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-16" aria-hidden="true" tabindex="-1"></a>  filtered_counts <span class="ot">&lt;-</span> filtered_counts[common_genes, ]</span>
<span id="cb11-17"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-18"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create SingleCellExperiment objects</span></span>
<span id="cb11-19"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-19" aria-hidden="true" tabindex="-1"></a>  sce_raw <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(<span class="fu">list</span>(<span class="at">counts =</span> raw_counts))</span>
<span id="cb11-20"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-20" aria-hidden="true" tabindex="-1"></a>  sce_object <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(<span class="fu">list</span>(<span class="at">counts =</span> filtered_counts))</span>
<span id="cb11-21"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-22"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run decontX with background</span></span>
<span id="cb11-23"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-23" aria-hidden="true" tabindex="-1"></a>  sce <span class="ot">&lt;-</span> <span class="fu">decontX</span>(sce_object, <span class="at">z =</span> cluster_info, <span class="at">background =</span> sce_raw)</span>
<span id="cb11-24"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-25"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Summarize contamination levels</span></span>
<span id="cb11-26"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-26" aria-hidden="true" tabindex="-1"></a>  contamination_summary <span class="ot">&lt;-</span> <span class="fu">as.array</span>(<span class="fu">summary</span>(sce<span class="sc">$</span>decontX_contamination))</span>
<span id="cb11-27"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(contamination_summary)</span>
<span id="cb11-28"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-29"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add contamination levels to Seurat object metadata</span></span>
<span id="cb11-30"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-30" aria-hidden="true" tabindex="-1"></a>  contamination <span class="ot">&lt;-</span> <span class="fu">colData</span>(sce)<span class="sc">$</span>decontX_contamination</span>
<span id="cb11-31"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-31" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(seurat_obj, <span class="at">metadata =</span> contamination, <span class="at">col.name =</span> <span class="st">&quot;decontX_contamination&quot;</span>)</span>
<span id="cb11-32"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-33"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract decontaminated counts from SCE object</span></span>
<span id="cb11-34"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-34" aria-hidden="true" tabindex="-1"></a>  decontaminated_counts <span class="ot">&lt;-</span> <span class="fu">assay</span>(sce, <span class="st">&quot;decontXcounts&quot;</span>)</span>
<span id="cb11-35"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-35" aria-hidden="true" tabindex="-1"></a>  decontaminated_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(decontaminated_counts)</span>
<span id="cb11-36"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-37"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a new assay with decontaminated counts and add it to Seurat object</span></span>
<span id="cb11-38"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-38" aria-hidden="true" tabindex="-1"></a>  new_assay <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> decontaminated_counts)</span>
<span id="cb11-39"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-39" aria-hidden="true" tabindex="-1"></a>  seurat_obj[[<span class="st">&quot;decontaminated&quot;</span>]] <span class="ot">&lt;-</span> new_assay</span>
<span id="cb11-40"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-41"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-41" aria-hidden="true" tabindex="-1"></a>  clusters_umap_orig <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(</span>
<span id="cb11-42"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> seurat_obj,</span>
<span id="cb11-43"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>,</span>
<span id="cb11-44"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb11-45"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-46"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.size =</span> <span class="fl">0.5</span></span>
<span id="cb11-47"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-47" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;UMAP with Clusters&quot;</span>)</span>
<span id="cb11-48"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-49"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot UMAP with contamination levels</span></span>
<span id="cb11-50"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-50" aria-hidden="true" tabindex="-1"></a>  contamination_umap <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(</span>
<span id="cb11-51"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> seurat_obj, </span>
<span id="cb11-52"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">features =</span> <span class="st">&quot;decontX_contamination&quot;</span>, </span>
<span id="cb11-53"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span></span>
<span id="cb11-54"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-54" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;decontX contamination value&quot;</span>)</span>
<span id="cb11-55"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-56"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DefaultAssay</span>(seurat_obj) <span class="ot">&lt;-</span> <span class="st">&quot;decontaminated&quot;</span></span>
<span id="cb11-57"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-58"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Normalization, variable feature selection, and scaling</span></span>
<span id="cb11-59"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-59" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(seurat_obj)</span>
<span id="cb11-60"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-60" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(seurat_obj)</span>
<span id="cb11-61"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-61" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seurat_obj)</span>
<span id="cb11-62"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-63"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># PCA and clustering</span></span>
<span id="cb11-64"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-64" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_obj)</span>
<span id="cb11-65"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-65" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb11-66"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-66" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(seurat_obj, <span class="at">resolution =</span> <span class="fl">0.7</span>)</span>
<span id="cb11-67"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-67" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-68"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-68" aria-hidden="true" tabindex="-1"></a>  <span class="co"># UMAP</span></span>
<span id="cb11-69"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-69" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb11-70"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-70" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-71"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot UMAP with updated clusters</span></span>
<span id="cb11-72"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-72" aria-hidden="true" tabindex="-1"></a>  clusters_umap <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(</span>
<span id="cb11-73"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> seurat_obj,</span>
<span id="cb11-74"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>,</span>
<span id="cb11-75"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb11-76"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-77"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.size =</span> <span class="fl">0.5</span></span>
<span id="cb11-78"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-78" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;UMAP with Corrected Clusters&quot;</span>)</span>
<span id="cb11-79"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-80"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-80" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine plots</span></span>
<span id="cb11-81"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-81" aria-hidden="true" tabindex="-1"></a>  combined_umap <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(clusters_umap_orig, contamination_umap, clusters_umap, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb11-82"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Making plots</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb11-83"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-83" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the combined plot as a PDF</span></span>
<span id="cb11-84"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-84" aria-hidden="true" tabindex="-1"></a>  <span class="co">#pdf(file = paste0(sample_id, &quot;_decontx_plots.pdf&quot;), width = 18, height = 6)</span></span>
<span id="cb11-85"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(combined_umap)</span>
<span id="cb11-86"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-86" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dev.off()</span></span>
<span id="cb11-87"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-88"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Saving seurat obj</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb11-89"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-89" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the Seurat object</span></span>
<span id="cb11-90"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(seurat_obj, <span class="at">file =</span> <span class="fu">paste0</span>(sample_id, <span class="st">&quot;_decontx_seurat_obj.rds&quot;</span>))</span>
<span id="cb11-91"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-92"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-92" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save decontaminated counts and contamination summary</span></span>
<span id="cb11-93"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Saving decontx counts</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb11-94"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(decontaminated_counts, <span class="fu">paste0</span>(sample_id, <span class="st">&quot;_decontx_counts.csv&quot;</span>))</span>
<span id="cb11-95"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-95" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-96"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-96" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Print a message indicating that the contamination summary is being saved</span></span>
<span id="cb11-97"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Saving contamination summary</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb11-98"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-98" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-99"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-99" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure contamination_summary is a data frame</span></span>
<span id="cb11-100"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-100" aria-hidden="true" tabindex="-1"></a>  contamination_summary_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(contamination_summary)</span>
<span id="cb11-101"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(contamination_summary_df, <span class="at">file =</span> <span class="fu">paste0</span>(sample_id, <span class="st">&quot;_contamination_summary.txt&quot;</span>))</span>
<span id="cb11-102"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-103"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-103" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Optionally return the results</span></span>
<span id="cb11-104"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">seurat_obj =</span> seurat_obj, <span class="at">decontaminated_counts =</span> decontaminated_counts, <span class="at">contamination_summary =</span> contamination_summary))</span>
<span id="cb11-105"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-105" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-106"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-107"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-107" aria-hidden="true" tabindex="-1"></a><span class="co"># Run decontX on the input Seurat object and background counts file</span></span>
<span id="cb11-108"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb11-108" aria-hidden="true" tabindex="-1"></a>deconx_results <span class="ot">&lt;-</span> <span class="fu">run_decontX</span>(<span class="st">&quot;./output_files/empty_drops/removed_empty_Day55.rds&quot;</span>, <span class="st">&quot;./output_files/counts/background_geneSymbol_gene_count.csv&quot;</span>, <span class="st">&quot;./output_files/decontx//Day55&quot;</span>)</span></code></pre></div>
</details>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
## 0.001336 0.040294 0.065590 0.095857 0.113042 0.810218</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 339
## Number of edges: 7342
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8177
## Number of communities: 8
## Elapsed time: 0 seconds</code></pre>
<pre><code>## Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
## To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
## This message will be shown once per session</code></pre>
<pre><code>## Making plots</code></pre>
<p><img src="bookdownproj_files/figure-html/decontX-1.png" width="672" /></p>
<pre><code>## Saving seurat obj
## Saving decontx counts
## Saving contamination summary</code></pre>
</div>
<div id="standard-gene-qc-to-remove-low-quality-cells" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Standard gene QC to remove low quality cells<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have removed empty drops and ambient RNA we will perform standard QC as described in the Seurat tutorial (<a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial.html">found here</a>). First we will determine what our filtering criteria should be using some basic QC plots.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># standard QC filtering and also remove doublets </span></span>
<span id="cb17-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co">#define sample name</span></span>
<span id="cb17-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-3" aria-hidden="true" tabindex="-1"></a>sample_id <span class="ot">=</span> <span class="st">&#39;Day55_tutorial&#39;</span></span>
<span id="cb17-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create Seurat object</span></span>
<span id="cb17-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-6" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> deconx_results<span class="sc">$</span>decontaminated_counts, <span class="at">project =</span> sample_id)</span>
<span id="cb17-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relationship between reads and unique genes per cell</span></span>
<span id="cb17-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-9" aria-hidden="true" tabindex="-1"></a>plot_scatter1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(seurat_object, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb17-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span></span>
<span id="cb17-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Reads vs Unique Genes per Cell BEFORE Filtering&quot;</span>)</span>
<span id="cb17-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(plot_scatter1)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/gene%20QC-1.png" width="576" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add mitochondrial percentage</span></span>
<span id="cb18-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb18-2" aria-hidden="true" tabindex="-1"></a>seurat_object[[<span class="st">&quot;joined&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(seurat_object[[<span class="st">&quot;RNA&quot;</span>]])</span>
<span id="cb18-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb18-3" aria-hidden="true" tabindex="-1"></a>seurat_object[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(seurat_object, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb18-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb18-5" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(seurat_object, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>))</span>
<span id="cb18-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb18-6" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;QC plots (gene level) BEFORE Filtering&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/gene%20QC-2.png" width="576" /></p>
<p>Based on these QC plots we will filter the data with the following values listed in the bellow code chunk. QC paramaters might vary based on your own data so please make sure you filter your data accordingly. Here we will also remove doublets - droplets that contain two or more cells - using the package <a href="https://github.com/chris-mcginnis-ucsf/DoubletFinder">doubletfinder</a>.</p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter cells based on feature and count thresholds</span></span>
<span id="cb19-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="do">## define the filtering params - (change these based on your data)</span></span>
<span id="cb19-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-3" aria-hidden="true" tabindex="-1"></a>max.features <span class="ot">=</span> <span class="dv">10000</span></span>
<span id="cb19-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-4" aria-hidden="true" tabindex="-1"></a>min.features <span class="ot">=</span> <span class="dv">1000</span></span>
<span id="cb19-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-5" aria-hidden="true" tabindex="-1"></a>min.counts <span class="ot">=</span> <span class="dv">800</span></span>
<span id="cb19-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-6" aria-hidden="true" tabindex="-1"></a>max.counts <span class="ot">=</span> <span class="dv">100000</span></span>
<span id="cb19-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-7" aria-hidden="true" tabindex="-1"></a>MT <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb19-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-8" aria-hidden="true" tabindex="-1"></a>npc <span class="ot">=</span> <span class="dv">15</span></span>
<span id="cb19-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-9" aria-hidden="true" tabindex="-1"></a>doublet_rate <span class="ot">=</span> <span class="fl">0.039</span></span>
<span id="cb19-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-10" aria-hidden="true" tabindex="-1"></a>cluster_res <span class="ot">=</span> <span class="fl">0.9</span></span>
<span id="cb19-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">#now we filter the seurat object based on the QC params listed above </span></span>
<span id="cb19-14"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-14" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">subset</span>(seurat_object, <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> min.features <span class="sc">&amp;</span> nFeature_RNA <span class="sc">&lt;</span> max.features <span class="sc">&amp;</span> percent.mt <span class="sc">&lt;</span> MT <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> max.counts <span class="sc">&amp;</span> nCount_RNA <span class="sc">&gt;</span> min.counts)</span>
<span id="cb19-15"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-16"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot quality metrics after filtering</span></span>
<span id="cb19-17"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-17" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(filt_seurat_object, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>))</span>
<span id="cb19-18"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb19-18" aria-hidden="true" tabindex="-1"></a>p2 <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;QC metrics gene level AFTER Filtering&quot;</span>)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot-1.png" width="1152" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize data</span></span>
<span id="cb20-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-2" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(filt_seurat_object, <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="at">scale.factor =</span> <span class="dv">10000</span>)</span>
<span id="cb20-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify highly variable features</span></span>
<span id="cb20-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-5" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(filt_seurat_object, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb20-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply linear transformation</span></span>
<span id="cb20-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-8" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(filt_seurat_object)</span>
<span id="cb20-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-9" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(filt_seurat_object, <span class="at">features =</span> all_genes)</span>
<span id="cb20-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA</span></span>
<span id="cb20-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-12" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(filt_seurat_object, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> filt_seurat_object))</span>
<span id="cb20-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-14"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Cluster cells</span></span>
<span id="cb20-15"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-15" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(filt_seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>npc)</span>
<span id="cb20-16"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb20-16" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(filt_seurat_object, <span class="at">resolution =</span> cluster_res)</span></code></pre></div>
</details>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 319
## Number of edges: 6816
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.7804
## Number of communities: 8
## Elapsed time: 0 seconds</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform UMAP</span></span>
<span id="cb22-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb22-2" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(filt_seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>npc)</span>
<span id="cb22-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="do">### Filter out doublets (remember to modify doublet rate if samples have variable target cells)</span></span>
<span id="cb22-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="do">## pK Identification (no ground-truth) </span></span>
<span id="cb22-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb22-6" aria-hidden="true" tabindex="-1"></a>sweep.res.list_pbmc <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(filt_seurat_object, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span>npc, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-1" aria-hidden="true" tabindex="-1"></a>sweep.stats_pbmc <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_pbmc, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb42-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-2" aria-hidden="true" tabindex="-1"></a>bcmvn_pbmc <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_pbmc)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot-2.png" width="1152" /></p>
<pre><code>## NULL</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="do">#####</span></span>
<span id="cb44-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-2" aria-hidden="true" tabindex="-1"></a>pK <span class="ot">&lt;-</span> bcmvn_pbmc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(BCmetric <span class="sc">==</span> <span class="fu">max</span>(BCmetric)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(pK) </span>
<span id="cb44-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-3" aria-hidden="true" tabindex="-1"></a>pK <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(pK[[<span class="dv">1</span>]]))</span>
<span id="cb44-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb44-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Homotypic Doublet Proportion Estimate </span></span>
<span id="cb44-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-6" aria-hidden="true" tabindex="-1"></a>annotations <span class="ot">&lt;-</span> filt_seurat_object<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb44-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-7" aria-hidden="true" tabindex="-1"></a>homotypic.prop <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations)</span>
<span id="cb44-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-8" aria-hidden="true" tabindex="-1"></a>nExp_poi <span class="ot">&lt;-</span> <span class="fu">round</span>(doublet_rate <span class="sc">*</span> <span class="fu">nrow</span>(filt_seurat_object<span class="sc">@</span>meta.data))</span>
<span id="cb44-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-9" aria-hidden="true" tabindex="-1"></a>nExp_poi.adj <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> homotypic.prop))</span>
<span id="cb44-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb44-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run doubletFinder </span></span>
<span id="cb44-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-12" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(filt_seurat_object, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span>npc, <span class="at">pN =</span> <span class="fl">0.25</span>, <span class="at">pK =</span> pK, <span class="at">nExp =</span> nExp_poi.adj, <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
<pre><code>## [1] &quot;Creating 106 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(filt_seurat_object<span class="sc">@</span>meta.data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;DF.classifications_.*$&quot;</span>, <span class="st">&quot;DF.classifications&quot;</span>, <span class="fu">colnames</span>(filt_seurat_object<span class="sc">@</span>meta.data))</span>
<span id="cb49-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary doublets</span></span>
<span id="cb49-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-4" aria-hidden="true" tabindex="-1"></a>statsDoublets <span class="ot">&lt;-</span> filt_seurat_object<span class="sc">@</span>meta.data <span class="sc">%&gt;%</span></span>
<span id="cb49-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(DF.classifications) <span class="sc">%&gt;%</span></span>
<span id="cb49-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Median_nCount_RNA =</span> <span class="fu">median</span>(nCount_RNA), <span class="at">Median_nFeature_RNA =</span> <span class="fu">median</span>(nFeature_RNA), <span class="at">Count =</span> <span class="fu">n</span>())</span>
<span id="cb49-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-9" aria-hidden="true" tabindex="-1"></a><span class="do">###Save the seurat object with doublets listed </span></span>
<span id="cb49-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-10" aria-hidden="true" tabindex="-1"></a>filt_seurat_object_doublets <span class="ot">&lt;-</span> filt_seurat_object</span>
<span id="cb49-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-11" aria-hidden="true" tabindex="-1"></a>filt_seurat_object <span class="ot">&lt;-</span> <span class="fu">subset</span>(filt_seurat_object, <span class="at">subset =</span> DF.classifications <span class="sc">==</span> <span class="st">&#39;Singlet&#39;</span>)</span>
<span id="cb49-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-13" aria-hidden="true" tabindex="-1"></a><span class="co"># figures</span></span>
<span id="cb49-14"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-14" aria-hidden="true" tabindex="-1"></a>  ggplot_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb49-15"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ElbowPlot</span>(filt_seurat_object) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&#39;SD explained by each PC&#39;</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb49-16"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-17"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeatureScatter</span>(filt_seurat_object, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb49-18"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Association between reads and </span><span class="sc">\n</span><span class="st">unique genes per cell AFTER filtering&quot;</span>),</span>
<span id="cb49-19"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-20"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DimPlot</span>(filt_seurat_object, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>) <span class="sc">+</span> </span>
<span id="cb49-21"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">&quot;Cluster </span><span class="sc">\n</span><span class="st">(from PCA)&quot;</span>, <span class="at">title =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span> </span>
<span id="cb49-22"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb49-23"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-24"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">features =</span> <span class="st">&#39;nCount_RNA&#39;</span>) <span class="sc">+</span> </span>
<span id="cb49-25"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">&quot;UMI count&quot;</span>, <span class="at">title =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span> </span>
<span id="cb49-26"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb49-27"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-28"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(filt_seurat_object, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">features =</span> <span class="st">&#39;nFeature_RNA&#39;</span>) <span class="sc">+</span> </span>
<span id="cb49-29"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color =</span> <span class="fu">str_wrap</span>(<span class="st">&quot;Feature count (gene)&quot;</span>, <span class="dv">15</span>), <span class="at">title =</span> <span class="st">&#39;&#39;</span>) <span class="sc">+</span> </span>
<span id="cb49-30"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)),</span>
<span id="cb49-31"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-32"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-32" aria-hidden="true" tabindex="-1"></a>  p2</span>
<span id="cb49-33"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-33" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-34"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-35"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-35" aria-hidden="true" tabindex="-1"></a>  combined_plots <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(<span class="at">plotlist =</span> ggplot_list, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb49-36"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb49-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(combined_plots)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot-3.png" width="1152" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb50-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">DimPlot</span>(filt_seurat_object_doublets, <span class="at">reduction =</span> <span class="st">&#39;umap&#39;</span>, <span class="at">group.by =</span> <span class="st">&quot;DF.classifications&quot;</span>))</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot-4.png" width="1152" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb51-1" aria-hidden="true" tabindex="-1"></a>  tbl_sts1 <span class="ot">&lt;-</span> <span class="fu">tableGrob</span>(statsDoublets)</span>
<span id="cb51-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb51-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.newpage</span>()</span>
<span id="cb51-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb51-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.draw</span>(tbl_sts1)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot-5.png" width="1152" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-1" aria-hidden="true" tabindex="-1"></a>  stats_sumary <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="st">&quot;Sample ID&quot;</span> <span class="ot">=</span> sample_id,</span>
<span id="cb52-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-2" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Cells_before_filter&quot;</span> <span class="ot">=</span> <span class="fu">dim</span>(seurat_object)[<span class="dv">2</span>],</span>
<span id="cb52-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-3" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Cells_after_filter&quot;</span> <span class="ot">=</span> <span class="fu">dim</span>(filt_seurat_object)[<span class="dv">2</span>],</span>
<span id="cb52-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-4" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Feature per Cell before filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nFeature_RNA),</span>
<span id="cb52-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-5" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Reads per Gene before filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb52-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-6" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Feature per Cell&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(filt_seurat_object<span class="sc">$</span>nFeature_RNA),</span>
<span id="cb52-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-7" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Reads per Gene&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(filt_seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb52-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-8" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Max Features&quot;</span> <span class="ot">=</span> max.features,</span>
<span id="cb52-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-9" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Min Features&quot;</span> <span class="ot">=</span> min.features,</span>
<span id="cb52-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-10" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Min Counts&quot;</span> <span class="ot">=</span> min.counts,</span>
<span id="cb52-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-11" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Max Counts&quot;</span> <span class="ot">=</span> max.counts,</span>
<span id="cb52-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-12" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;MT Percentage&quot;</span> <span class="ot">=</span> MT,</span>
<span id="cb52-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-13" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;NPCs&quot;</span> <span class="ot">=</span> npc,</span>
<span id="cb52-14"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-14" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Percent MT before Filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(seurat_object<span class="sc">@</span>meta.data[[<span class="st">&quot;percent.mt&quot;</span>]]),</span>
<span id="cb52-15"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-15" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Median Percent MT after Filter&quot;</span> <span class="ot">=</span> <span class="fu">median</span>(filt_seurat_object<span class="sc">@</span>meta.data[[<span class="st">&quot;percent.mt&quot;</span>]])</span>
<span id="cb52-16"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-16" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb52-17"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-18"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-18" aria-hidden="true" tabindex="-1"></a>  tbl_sts2 <span class="ot">&lt;-</span> <span class="fu">tableGrob</span>(stats_sumary)</span>
<span id="cb52-19"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.newpage</span>()</span>
<span id="cb52-20"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb52-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.draw</span>(tbl_sts2)</span></code></pre></div>
</details>
<p><img src="bookdownproj_files/figure-html/plot-6.png" width="1152" /></p>
<details class=chunk-details open><summary class=chunk-summary><span class=chunk-summary-text>Code</span></summary>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb53-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#save files</span></span>
<span id="cb53-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(filt_seurat_object, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">&quot;./output_files/QC/&quot;</span>, sample_id, <span class="st">&quot;_umap_object.rds&quot;</span>))</span>
<span id="cb53-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(filt_seurat_object_doublets, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">&quot;./output_files/QC/&quot;</span>,sample_id, <span class="st">&quot;_with_doublets_umap_object.rds&quot;</span>))</span>
<span id="cb53-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb53-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(stats_sumary, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">&quot;./output_files/QC/&quot;</span>, sample_id, <span class="st">&quot;_stats.csv&quot;</span>)) </span></code></pre></div>
</details>
<p>Now that we have filtered the object to ensure we retain high quality data we are going to add in the isoform level information.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="setup.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="add-isoform-counts-to-seurat-object.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/03-Cleaning-the-data-and-making-a-Seurat-object.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/03-Cleaning-the-data-and-making-a-Seurat-object.Rmd",
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
