<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Removing sources of unwanted noise from the single cell dataset | Adding Isoform level information to your object</title>
  <meta name="description" content="Chapter 3 Removing sources of unwanted noise from the single cell dataset | Adding Isoform level information to your object" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Removing sources of unwanted noise from the single cell dataset | Adding Isoform level information to your object" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Removing sources of unwanted noise from the single cell dataset | Adding Isoform level information to your object" />
  
  
  

<meta name="author" content="Sefi" />


<meta name="date" content="2024-12-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">FLAMES_tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Down the rabbit-hole</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#load-in-required-packages"><i class="fa fa-check"></i><b>2.1</b> load in required packages</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#creating-resource-files."><i class="fa fa-check"></i><b>2.2</b> Creating resource files.</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#convert-count-matrices-from-gene-id-to-gene-symbol"><i class="fa fa-check"></i><b>2.3</b> Convert count matrices from Gene ID to gene Symbol</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><i class="fa fa-check"></i><b>3</b> Removing sources of unwanted noise from the single cell dataset</a>
<ul>
<li class="chapter" data-level="3.0.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets"><i class="fa fa-check"></i><b>3.0.1</b> Empty droplets</a></li>
<li class="chapter" data-level="3.1" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination"><i class="fa fa-check"></i><b>3.1</b> Removing ambient RNA contamination</a></li>
<li class="chapter" data-level="3.2" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells"><i class="fa fa-check"></i><b>3.2</b> Standard gene QC to remove low quality cells</a></li>
<li class="chapter" data-level="3.3" data-path="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#add-isoform-counts-to-seurat-object"><i class="fa fa-check"></i><b>3.3</b> Add isoform counts to Seurat object</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Adding Isoform level information to your object</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="removing-sources-of-unwanted-noise-from-the-single-cell-dataset" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Removing sources of unwanted noise from the single cell dataset<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-sources-of-unwanted-noise-from-the-single-cell-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Now we will do some initial prepossessing of single cell data to ensure we have some high quality data.
This will involve 3 main steps</p>
<ol style="list-style-type: decimal">
<li><p>Removing empty droplets - droplets that do not contain true cells.</p></li>
<li><p>Removing ambient RNA contamination - optional</p></li>
<li><p>Removing low quality cells that</p></li>
</ol>
<div id="empty-droplets" class="section level3 hasAnchor" number="3.0.1">
<h3><span class="header-section-number">3.0.1</span> Empty droplets<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#empty-droplets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This function removes empty droplets, a critical step to ensure that only true cells are retained for analysis.
In short-read analysis using CellRanger, this process is automated, and empty droplets are removed by the software.
However, FLAMES does not perform this step automatically, so it must be done manually.
The function provided here not only removes empty droplets but also generates general QC metrics, enabling users to assess the reasonableness of the number of cells removed.</p>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 344
## Number of edges: 7209
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8510
## Number of communities: 7
## Elapsed time: 0 seconds</code></pre>
<p><img src="bookdownproj_files/figure-html/perform_empty_drops_analysis-1.png" width="672" /></p>
</div>
<div id="removing-ambient-rna-contamination" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Removing ambient RNA contamination<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-ambient-rna-contamination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we will remove ambient RNA contamination using decontX (or SoupX).
This step is optional, and it’s up to the user to decide whether it’s necessary.
In this case, the barcode rank plot (<u>found here…</u>) shows a clear distinction between true barcodes and background barcodes, suggesting that ambient RNA contamination may not significantly contribute to noise in this dataset.
However, we’ll demonstrate how to run this step in case your barcode rank plot is noisier and contamination is a concern.
if users wish to skip this step take the ‘seurat_obj_rm_empty’ object and proceed to standard gene QC step.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to run decontX on a single Seurat object</span></span>
<span id="cb6-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-2" aria-hidden="true" tabindex="-1"></a>run_decontX <span class="ot">&lt;-</span> <span class="cf">function</span>(seurat_obj_path, background_counts_path, sample_id) {</span>
<span id="cb6-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load the Seurat object</span></span>
<span id="cb6-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-4" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(seurat_obj_path)</span>
<span id="cb6-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-5" aria-hidden="true" tabindex="-1"></a>  filtered_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">GetAssayData</span>(seurat_obj, <span class="at">layer =</span> <span class="st">&quot;counts&quot;</span>))</span>
<span id="cb6-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-7"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read background counts</span></span>
<span id="cb6-8"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-8" aria-hidden="true" tabindex="-1"></a>  raw_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.csv</span>(background_counts_path, <span class="at">row.names =</span> <span class="dv">1</span>))</span>
<span id="cb6-9"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-10"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get cluster info from Seurat object</span></span>
<span id="cb6-11"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-11" aria-hidden="true" tabindex="-1"></a>  cluster_info <span class="ot">&lt;-</span> <span class="fu">setNames</span>(seurat_obj<span class="sc">$</span>seurat_clusters, <span class="fu">colnames</span>(seurat_obj))</span>
<span id="cb6-12"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-13"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find common genes</span></span>
<span id="cb6-14"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-14" aria-hidden="true" tabindex="-1"></a>  common_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(filtered_counts), <span class="fu">rownames</span>(raw_counts))</span>
<span id="cb6-15"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-15" aria-hidden="true" tabindex="-1"></a>  raw_counts <span class="ot">&lt;-</span> raw_counts[common_genes, ]</span>
<span id="cb6-16"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-16" aria-hidden="true" tabindex="-1"></a>  filtered_counts <span class="ot">&lt;-</span> filtered_counts[common_genes, ]</span>
<span id="cb6-17"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-18"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create SingleCellExperiment objects</span></span>
<span id="cb6-19"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-19" aria-hidden="true" tabindex="-1"></a>  sce_raw <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(<span class="fu">list</span>(<span class="at">counts =</span> raw_counts))</span>
<span id="cb6-20"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-20" aria-hidden="true" tabindex="-1"></a>  sce_object <span class="ot">&lt;-</span> <span class="fu">SingleCellExperiment</span>(<span class="fu">list</span>(<span class="at">counts =</span> filtered_counts))</span>
<span id="cb6-21"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-22"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run decontX with background</span></span>
<span id="cb6-23"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-23" aria-hidden="true" tabindex="-1"></a>  sce <span class="ot">&lt;-</span> <span class="fu">decontX</span>(sce_object, <span class="at">z =</span> cluster_info, <span class="at">background =</span> sce_raw)</span>
<span id="cb6-24"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-25"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Summarize contamination levels</span></span>
<span id="cb6-26"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-26" aria-hidden="true" tabindex="-1"></a>  contamination_summary <span class="ot">&lt;-</span> <span class="fu">as.array</span>(<span class="fu">summary</span>(sce<span class="sc">$</span>decontX_contamination))</span>
<span id="cb6-27"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(contamination_summary)</span>
<span id="cb6-28"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-29"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add contamination levels to Seurat object metadata</span></span>
<span id="cb6-30"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-30" aria-hidden="true" tabindex="-1"></a>  contamination <span class="ot">&lt;-</span> <span class="fu">colData</span>(sce)<span class="sc">$</span>decontX_contamination</span>
<span id="cb6-31"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-31" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(seurat_obj, <span class="at">metadata =</span> contamination, <span class="at">col.name =</span> <span class="st">&quot;decontX_contamination&quot;</span>)</span>
<span id="cb6-32"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-33"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract decontaminated counts from SCE object</span></span>
<span id="cb6-34"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-34" aria-hidden="true" tabindex="-1"></a>  decontaminated_counts <span class="ot">&lt;-</span> <span class="fu">assay</span>(sce, <span class="st">&quot;decontXcounts&quot;</span>)</span>
<span id="cb6-35"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-35" aria-hidden="true" tabindex="-1"></a>  decontaminated_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(decontaminated_counts)</span>
<span id="cb6-36"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-37"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a new assay with decontaminated counts and add it to Seurat object</span></span>
<span id="cb6-38"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-38" aria-hidden="true" tabindex="-1"></a>  new_assay <span class="ot">&lt;-</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> decontaminated_counts)</span>
<span id="cb6-39"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-39" aria-hidden="true" tabindex="-1"></a>  seurat_obj[[<span class="st">&quot;decontaminated&quot;</span>]] <span class="ot">&lt;-</span> new_assay</span>
<span id="cb6-40"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-41"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-41" aria-hidden="true" tabindex="-1"></a>  clusters_umap_orig <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(</span>
<span id="cb6-42"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> seurat_obj,</span>
<span id="cb6-43"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>,</span>
<span id="cb6-44"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb6-45"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-46"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.size =</span> <span class="fl">0.5</span></span>
<span id="cb6-47"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-47" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Old UMAP with Clusters&quot;</span>)</span>
<span id="cb6-48"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-49"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot UMAP with contamination levels</span></span>
<span id="cb6-50"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-50" aria-hidden="true" tabindex="-1"></a>  contamination_umap <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(</span>
<span id="cb6-51"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> seurat_obj, </span>
<span id="cb6-52"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">features =</span> <span class="st">&quot;decontX_contamination&quot;</span>, </span>
<span id="cb6-53"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span></span>
<span id="cb6-54"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-54" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;UMAP Colored by decontX_contamination&quot;</span>)</span>
<span id="cb6-55"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-56"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DefaultAssay</span>(seurat_obj) <span class="ot">&lt;-</span> <span class="st">&quot;decontaminated&quot;</span></span>
<span id="cb6-57"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-58"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Normalization, variable feature selection, and scaling</span></span>
<span id="cb6-59"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-59" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(seurat_obj)</span>
<span id="cb6-60"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-60" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(seurat_obj)</span>
<span id="cb6-61"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-61" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seurat_obj)</span>
<span id="cb6-62"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-63"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># PCA and clustering</span></span>
<span id="cb6-64"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-64" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_obj)</span>
<span id="cb6-65"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-65" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb6-66"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-66" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(seurat_obj, <span class="at">resolution =</span> <span class="fl">0.7</span>)</span>
<span id="cb6-67"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-67" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-68"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-68" aria-hidden="true" tabindex="-1"></a>  <span class="co"># UMAP</span></span>
<span id="cb6-69"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-69" aria-hidden="true" tabindex="-1"></a>  seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb6-70"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-70" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-71"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot UMAP with updated clusters</span></span>
<span id="cb6-72"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-72" aria-hidden="true" tabindex="-1"></a>  clusters_umap <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(</span>
<span id="cb6-73"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> seurat_obj,</span>
<span id="cb6-74"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">&quot;seurat_clusters&quot;</span>,</span>
<span id="cb6-75"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb6-76"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-77"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.size =</span> <span class="fl">0.5</span></span>
<span id="cb6-78"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-78" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;UMAP with Corrected Clusters&quot;</span>)</span>
<span id="cb6-79"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-80"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-80" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine plots</span></span>
<span id="cb6-81"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-81" aria-hidden="true" tabindex="-1"></a>  combined_umap <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(clusters_umap_orig, contamination_umap, clusters_umap, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb6-82"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Making plots</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb6-83"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-83" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the combined plot as a PDF</span></span>
<span id="cb6-84"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-84" aria-hidden="true" tabindex="-1"></a>  <span class="co">#pdf(file = paste0(sample_id, &quot;_decontx_plots.pdf&quot;), width = 18, height = 6)</span></span>
<span id="cb6-85"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(combined_umap)</span>
<span id="cb6-86"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-86" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dev.off()</span></span>
<span id="cb6-87"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-88"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Saving seurat obj</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb6-89"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-89" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the Seurat object</span></span>
<span id="cb6-90"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(seurat_obj, <span class="at">file =</span> <span class="fu">paste0</span>(sample_id, <span class="st">&quot;_decontx_seurat_obj.rds&quot;</span>))</span>
<span id="cb6-91"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-92"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-92" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save decontaminated counts and contamination summary</span></span>
<span id="cb6-93"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Saving decontx counts</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb6-94"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.csv</span>(decontaminated_counts, <span class="fu">paste0</span>(sample_id, <span class="st">&quot;_decontx_counts.csv&quot;</span>))</span>
<span id="cb6-95"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-95" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-96"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-96" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Print a message indicating that the contamination summary is being saved</span></span>
<span id="cb6-97"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;Saving contamination summary</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb6-98"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-98" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-99"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-99" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure contamination_summary is a data frame</span></span>
<span id="cb6-100"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-100" aria-hidden="true" tabindex="-1"></a>  contamination_summary_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(contamination_summary)</span>
<span id="cb6-101"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(contamination_summary_df, <span class="at">file =</span> <span class="fu">paste0</span>(sample_id, <span class="st">&quot;_contamination_summary.txt&quot;</span>))</span>
<span id="cb6-102"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-103"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-103" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Optionally return the results</span></span>
<span id="cb6-104"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">seurat_obj =</span> seurat_obj, <span class="at">decontaminated_counts =</span> decontaminated_counts, <span class="at">contamination_summary =</span> contamination_summary))</span>
<span id="cb6-105"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-105" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-106"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-107"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-107" aria-hidden="true" tabindex="-1"></a><span class="co"># Run decontX on the input Seurat object and background counts file</span></span>
<span id="cb6-108"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb6-108" aria-hidden="true" tabindex="-1"></a>deconx_results <span class="ot">&lt;-</span> <span class="fu">run_decontX</span>(<span class="st">&quot;./output_files/empty_drops/removed_empty_Day55.rds&quot;</span>, <span class="st">&quot;./output_files/counts/background_geneSymbol_gene_count.csv&quot;</span>, <span class="st">&quot;./output_files/decontx//Day55&quot;</span>)</span></code></pre></div>
<pre><code>## --------------------------------------------------</code></pre>
<pre><code>## Starting DecontX</code></pre>
<pre><code>## --------------------------------------------------</code></pre>
<pre><code>## Tue Dec 10 19:43:52 2024 .. Analyzing all cells</code></pre>
<pre><code>## Tue Dec 10 19:43:52 2024 .... Converting to sparse matrix</code></pre>
<pre><code>## Tue Dec 10 19:43:53 2024 .... Generating UMAP</code></pre>
<pre><code>## Tue Dec 10 19:43:58 2024 .... Estimating contamination</code></pre>
<pre><code>## Tue Dec 10 19:43:58 2024 ...... Completed iteration: 10 | converge: 0.02483</code></pre>
<pre><code>## Tue Dec 10 19:43:58 2024 ...... Completed iteration: 20 | converge: 0.009197</code></pre>
<pre><code>## Tue Dec 10 19:43:58 2024 ...... Completed iteration: 30 | converge: 0.005009</code></pre>
<pre><code>## Tue Dec 10 19:43:59 2024 ...... Completed iteration: 40 | converge: 0.003423</code></pre>
<pre><code>## Tue Dec 10 19:43:59 2024 ...... Completed iteration: 50 | converge: 0.00254</code></pre>
<pre><code>## Tue Dec 10 19:43:59 2024 ...... Completed iteration: 60 | converge: 0.002116</code></pre>
<pre><code>## Tue Dec 10 19:43:59 2024 ...... Completed iteration: 70 | converge: 0.001571</code></pre>
<pre><code>## Tue Dec 10 19:44:00 2024 ...... Completed iteration: 80 | converge: 0.00118</code></pre>
<pre><code>## Tue Dec 10 19:44:00 2024 ...... Completed iteration: 87 | converge: 0.0009944</code></pre>
<pre><code>## Tue Dec 10 19:44:00 2024 .. Calculating final decontaminated matrix</code></pre>
<pre><code>## --------------------------------------------------</code></pre>
<pre><code>## Completed DecontX. Total time: 7.631166 secs</code></pre>
<pre><code>## --------------------------------------------------</code></pre>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
## 0.001336 0.040294 0.065590 0.095857 0.113042 0.810218</code></pre>
<pre><code>## Centering and scaling data matrix</code></pre>
<pre><code>## PC_ 1 
## Positive:  VIM, ADGRV1, GLI3, DOK5, CACHD1, DACH1, SMC4, SHROOM3, RCN1, PARD3 
##     FBXL7, PHGDH, HMGA2, HMGN2, ID3, ENSG00000286757, ID4, LRIG1, SOX2, NBPF14 
##     LTBP1, YAP1, HMGB2, ZFP36L1, SLC35F1, TRIM59-IFT80, SMC2, TUBA1B, COL4A5, LEF1 
## Negative:  NRXN1, PTPRD, MYT1L, LRRC7, ANK3, MAPT, DCC, UNC79, PTPRN2, SLC4A10 
##     KCNH7, ENOX1, ANKS1B, EPHA5, RALYL, PHACTR3, CNTNAP2, RUNX1T1, NBEA, CACNA1E 
##     THSD7A, CACNA1A, GRIA2, NSG2, LINC01122, NELL2, RBFOX1, CELF4, NFASC, NEUROD6 
## PC_ 2 
## Positive:  HECW1, LHX2, MPPED2, IVNS1ABP, LEF1, SEZ6L, SFRP1, ENSG00000251680, CADPS, ROBO2 
##     ADAMTS19, B3GAT2, LRRC3B, FUT9, LMNB1, SNCAIP, TMTC1, NKAIN3, LTBP1, FOXP1 
##     NRG1, STOX1, SV2C, PANTR1, PREX2, SEMA3C, ITGA2, GLI3, UHRF1, ENSG00000301688 
## Negative:  SLIT3, RSPO2, RBM47, DYNLRB2-AS1, PRTG, LMX1A, PCP4, WLS, ENSG00000287949, MAF 
##     GJA1, FYB2, RSPO1, PLS3, MIR4435-2HG, NR3C2, RBM20, ZIC1, KRT8, LGI1 
##     EYA4, KRT18, ENSG00000241231, TPBG, TTR, OTX2, WNT2B, PTCHD1-AS, DLK1, IFITM1 
## PC_ 3 
## Positive:  NHLH1, EBF2, SLC22A23, DISP3, ENSG00000285755, CECR2, LINC01830, FRMD4B, ARL4D, BHLHE22 
##     ENC1, RASGEF1B, ENSG00000310169, BHLHE22-AS1, PPP1R17, CRABP1, CHD7, FAM149A, RBFOX3, NFIA-AS2 
##     HYCC1, LMO1, NPTX1, HUNK, UNC5A, SERINC5, PGAP1, P2RX3, PRDM8, ENSG00000231252 
## Negative:  OPCML, NAV3, FRMD4A, GRIN2B, XKR4, KIRREL3, NALF1, KCNMA1, CNTN1, NAV2 
##     GRIK3, CALN1, LRP1B, SMARCA2, DGKI, NRG3, SYN3, PRKG1, MACROD2, GAREM1 
##     SLCO3A1, MDGA2, SYT13, ATXN1, ULK4, GRID2, GABBR2, ZC3H12B, LMO3, RIMS1 
## PC_ 4 
## Positive:  ZFPM2, ZFPM2-AS1, ZEB2, MYO5B, CSRP2, ZBTB18, PLXDC2, ST18, NEUROD6, CAP2 
##     NELL2, ENSG00000307341, PDE1A, GAREM1, ADAMTS3, FLRT2, DLGAP1, EPHA7, NEUROD2, GRIA3 
##     CTTNBP2, SLC24A2, PDGFC, DMD, SYNDIG1, CORO2B, CENPE, KCNH7, SDK1, THSD7A 
## Negative:  COPG2, PBX3, GAD2, COPG2IT1, VGF, ENSG00000259460, KCNMB2, MEIS2, KCNMB2-AS1, LINC01414 
##     SIX3, ENSG00000270953, MEST, ENSG00000298285, SPRY4-AS1, PCDH7, DLX1, DLX6-AS1, SORCS3, SIX3-AS1 
##     GRM7, KCNH1, CDH8, DLX2, SGCD, MDGA2, SLC35F3, NRXN3, GAD1, MIR325HG 
## PC_ 5 
## Positive:  CDC20, CCNB1, PLK1, UBE2S, DLGAP5, KNSTRN, CCNB2, ARL6IP1, TPX2, AURKA 
##     NEK2, PTTG1, CDKN3, TROAP, GTSE1, UBE2C, BUB1, SGO2, KIF20A, CEP55 
##     PRR11, CENPE, GAS2L3, CDCA8, ASPM, PIMREG, ARHGAP11A, BIRC5, NDE1, DEPDC1 
## Negative:  HECW1, DPP10, GPC6, LINC03051, BMPR1B, SLC1A3, SPIDR, DMD, ZFPM2, TMTC1 
##     SDK1, ADAMTS6, ADAMTS19, ENSG00000301688, HDAC9, MEGF11, PVT1, MCM4, LRP4, RHBDL3 
##     FREM2, TAFA1, STOX1, LAMA1, RORA, NCKAP5, HELLS, PDGFC, STOX2, KHDRBS2</code></pre>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 339
## Number of edges: 7342
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8177
## Number of communities: 8
## Elapsed time: 0 seconds</code></pre>
<pre><code>## 19:44:01 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>## 19:44:01 Read 339 rows and found 20 numeric columns</code></pre>
<pre><code>## 19:44:01 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>## 19:44:01 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|
## 19:44:01 Writing NN index file to temp file /tmp/RtmpHn6FSX/file1b55b1b67d9e
## 19:44:01 Searching Annoy index using 1 thread, search_k = 3000
## 19:44:01 Annoy recall = 100%
## 19:44:02 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
## 19:44:04 Initializing from normalized Laplacian + noise (using RSpectra)
## 19:44:04 Commencing optimization for 500 epochs, with 11376 positive edges
## 19:44:05 Optimization finished</code></pre>
<pre><code>## Making plots</code></pre>
<p><img src="bookdownproj_files/figure-html/decontX-1.png" width="672" /></p>
<pre><code>## Saving seurat obj
## Saving decontx counts
## Saving contamination summary</code></pre>
</div>
<div id="standard-gene-qc-to-remove-low-quality-cells" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Standard gene QC to remove low quality cells<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#standard-gene-qc-to-remove-low-quality-cells" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have removed empty drops and ambient RNA we will perform standard QC as described in the Seurat tutorial (<a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial.html">found here</a>).
First we will determine what our filtering criteria should be using some basic QC plots.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># standard QC filtering and also remove doublets </span></span>
<span id="cb42-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co">#define sample name</span></span>
<span id="cb42-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-3" aria-hidden="true" tabindex="-1"></a>sample_id <span class="ot">=</span> <span class="st">&#39;Day55_tutorial&#39;</span></span>
<span id="cb42-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create Seurat object</span></span>
<span id="cb42-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb42-6" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> deconx_results<span class="sc">$</span>decontaminated_counts, <span class="at">project =</span> sample_id)</span></code></pre></div>
<pre><code>## Warning: Data is of class matrix. Coercing to dgCMatrix.</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relationship between reads and unique genes per cell</span></span>
<span id="cb44-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-2" aria-hidden="true" tabindex="-1"></a>plot_scatter1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(seurat_object, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span></span>
<span id="cb44-4"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Reads vs Unique Genes per Cell BEFORE Filtering&quot;</span>)</span>
<span id="cb44-5"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-6"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb44-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(plot_scatter1)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="bookdownproj_files/figure-html/gene%20QC-1.png" width="672" /></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add mitochondrial percentage</span></span>
<span id="cb46-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb46-2" aria-hidden="true" tabindex="-1"></a>seurat_object[[<span class="st">&quot;joined&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(seurat_object[[<span class="st">&quot;RNA&quot;</span>]])</span></code></pre></div>
<pre><code>## Warning: Key &#39;rna_&#39; taken, using &#39;joined_&#39; instead</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb48-1" aria-hidden="true" tabindex="-1"></a>seurat_object[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(seurat_object, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb48-2"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb48-3" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(seurat_object, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>))</span></code></pre></div>
<pre><code>## Warning: Default search for &quot;data&quot; layer in &quot;RNA&quot; assay yielded no results; utilizing &quot;counts&quot; layer instead.</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#cb50-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">&quot;QC plots (gene level) BEFORE Filtering&quot;</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/gene%20QC-2.png" width="672" /></p>
<p>Based on these QC plots we will filter the data with the following values listed in the bellow code chunk.
QC paramaters might vary based on your own data so please make sure you filter your data accordingly.
Here we will also remove doublets - droplets that contain two or more cells - using the package ‘<a href="https://github.com/chris-mcginnis-ucsf/DoubletFinder">doubletfinder</a>’.</p>
<pre><code>## Warning: Default search for &quot;data&quot; layer in &quot;RNA&quot; assay yielded no results; utilizing &quot;counts&quot; layer instead.</code></pre>
<p><img src="bookdownproj_files/figure-html/plot-1.png" width="672" /></p>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 319
## Number of edges: 6816
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.7804
## Number of communities: 8
## Elapsed time: 0 seconds</code></pre>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/plot-2.png" width="672" /></p>
<pre><code>## NULL</code></pre>
<pre><code>## [1] &quot;Creating 106 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<p><img src="bookdownproj_files/figure-html/plot-3.png" width="672" /><img src="bookdownproj_files/figure-html/plot-4.png" width="672" /><img src="bookdownproj_files/figure-html/plot-5.png" width="672" /><img src="bookdownproj_files/figure-html/plot-6.png" width="672" /></p>
<p>Now that we have filtered the object to ensure we retain high quality data we are going to add in the isoform level information. so lets see if we have a live update</p>

</div>
<div id="add-isoform-counts-to-seurat-object" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Add isoform counts to Seurat object<a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#add-isoform-counts-to-seurat-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have 309 high-quality cells for our downstream analysis, let’s incorporate isoform-level information into our Seurat object. The first step is to read in the outputs from FLAMES, which provides isoform-level counts from Oarfish. We’ll create a new Seurat object for this data, specifically for the isoform counts, and add it as a new assay to our existing object. This setup will allow us to explore both gene-level and isoform-level counts within a single Seurat object, giving us a more comprehensive view of our data.</p>
<pre><code>## Processed sample: oarfish 
## Output saved to: ./output_files/counts//gene_symbol_oarfish_counts.csv</code></pre>
<p><img src="bookdownproj_files/figure-html/process_oarfish_file-1.png" width="672" /></p>
<p>We have now created a Seurat object with isoform-level count data. Users can take this object and follow similar processing steps as at the gene level <a href="removing-sources-of-unwanted-noise-from-the-single-cell-dataset.html#removing-sources-of-unwanted-noise-from-the-single-cell-dataset">3</a>, which might include filtering out low-quality cells and removing ambient RNA counts. In our case, we evaluated these steps and found they don’t add much value as low quality cells are already removed at the gene level and we can leverage this information rather than processing the isoform counts in the same way. Additionally, tools like DeconX and SoupX haven’t been tested on isoform count data, so we decided to exclude this analysis. Instead, we believe the most valuable approach is to append the isoform data to our gene-level Seurat object, resulting in two assays: one for gene-level and one for isoform-level data. If users have short-read data, they could similarly add it as a third assay.</p>
<p>Since we’ve already filtered out low-quality cells based on our gene-level data, we’ll use these high-quality cells to filter our isoform data.</p>
<pre><code>## An object of class Seurat 
## 93557 features across 309 samples within 3 assays 
## Active assay: RNA (14906 features, 2000 variable features)
##  3 layers present: counts, data, scale.data
##  2 other assays present: joined, iso
##  4 dimensional reductions calculated: pca, umap, pca_iso, umap_iso</code></pre>
<p><img src="bookdownproj_files/figure-html/isoform_assay-1.png" width="672" /></p>
<p>Great! We now have an object containing both assays, so we can start by plotting some of our favorite genes and isoforms. This setup gives us the flexibility to visualize gene expression on isoform UMAPs and vice versa, allowing us to integrate and explore the expression of both gene and isoform expression within single cells on the same dimensional reduction.</p>
<p>Let’s begin by plotting VIM and TBR1. VIM is a marker of progenitor cells, while TBR1 is a marker of deep layer neurons.</p>
<pre><code>## Warning: No layers found matching search pattern provided
## Warning: No layers found matching search pattern provided
## Warning: No layers found matching search pattern provided
## Warning: No layers found matching search pattern provided</code></pre>
<pre><code>## Warning: Could not find ENST00000224237.9-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000487938.5-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000485947.1-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000469543.5-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<p><img src="bookdownproj_files/figure-html/ploting_features-1.png" width="672" /><img src="bookdownproj_files/figure-html/ploting_features-2.png" width="672" /></p>
<pre><code>## Warning: No layers found matching search pattern provided</code></pre>
<pre><code>## Warning: No layers found matching search pattern provided
## Warning: No layers found matching search pattern provided
## Warning: No layers found matching search pattern provided</code></pre>
<pre><code>## Warning: Could not find ENST00000224237.9-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000487938.5-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000485947.1-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000469543.5-VIM in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<p><img src="bookdownproj_files/figure-html/ploting_features-3.png" width="672" /></p>
<pre><code>## Warning: No layers found matching search pattern provided</code></pre>
<pre><code>## Warning: No layers found matching search pattern provided
## Warning: No layers found matching search pattern provided</code></pre>
<pre><code>## Warning: Could not find ENST00000389554.8-TBR1 in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000463544.1-TBR1 in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000410035.1-TBR1 in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<p><img src="bookdownproj_files/figure-html/ploting_features-4.png" width="672" /><img src="bookdownproj_files/figure-html/ploting_features-5.png" width="672" /></p>
<pre><code>## Warning: No layers found matching search pattern provided</code></pre>
<pre><code>## Warning: No layers found matching search pattern provided
## Warning: No layers found matching search pattern provided</code></pre>
<pre><code>## Warning: Could not find ENST00000389554.8-TBR1 in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000463544.1-TBR1 in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<pre><code>## Warning: Could not find ENST00000410035.1-TBR1 in the default search locations, found in &#39;iso&#39; assay instead</code></pre>
<p><img src="bookdownproj_files/figure-html/ploting_features-6.png" width="672" /></p>
<p>We observe that the expression of VIM and TBR genes is localized to two distinct clusters, which aligns with our expectations. This suggests that some cells remain in the progenitor/radial glial stage, while others have already begun differentiating into neurons. Furthermore, our data reveals isoform-level expression for each of our genes, including TBR1, which shows distinct isoform expression patterns across different cell populations. To plot genes and expressed isoforms from that gene simply change the ‘gene’ value to your favorite gene and run the code chunk above.</p>
<p>Now we have a count matrix with two assays, gene and isoform. There are many avenues for analysis. We will begin with finding DE genes and isoforms. Then we identify cell types and perform trajectory analysis. finally we look into interesting isoforms unique to each cluster</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/edit/main/02-Cleaning-the-data-and-making-a-Seurat-object.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/sefi196/FLAMESv2_LR_sc_tutorial/blob/main/02-Cleaning-the-data-and-making-a-Seurat-object.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
